// reactjs-tiptap-editor@1.0.2/lib/Callout.js
import { N as D, m as H } from "./index-Dx-Dh7KS.js";
import { N as M, c as x, a as B } from "./clsx-_bq4MQQ7.js";
import { h as V, u as P, d as E, D as R, n as W, A as T, o as j, p as F, L as u, S as U, q as Y, r as _, s as $, t as q, k as v, v as z, B as C } from "./index-C52cuQ_Y.js";
import { jsx as o, jsxs as l, Fragment as K } from "react/jsx-runtime";
import { useState as s } from "react";
import "./theme.js";
import { Info as G, Lightbulb as J, AlertCircle as Q, TriangleAlert as X, OctagonAlert as Z } from "lucide-react";
import { u as tt } from "./index-DtyAGMbq.js";
const y = [
  {
    value: "note",
    label: "Note",
    icon: G,
    color: "#1f6feb",
    background: "#1f6feb1f"
  },
  {
    value: "tip",
    label: "Tip",
    icon: J,
    color: "#238636",
    background: "#2386361f"
  },
  {
    value: "important",
    label: "Important",
    icon: Q,
    color: "#ab7df8",
    background: "#ab7df81f"
  },
  {
    value: "warning",
    label: "Warning",
    icon: X,
    color: "#d29922",
    background: "#d299221f"
  },
  {
    value: "caution",
    label: "Caution",
    icon: Z,
    color: "#f85149",
    background: "#f851491f"
  }
];
function et({ node: t }) {
  const { type: e = "note", title: a = "", body: n = "" } = t.attrs, c = (y.find((d) => d.value === e) || y[0]).icon;
  return /* @__PURE__ */ o(M, { children: /* @__PURE__ */ l(
    "div",
    {
      className: x(
        "richtext-relative richtext-my-4 richtext-rounded-lg richtext-border richtext-p-4",
        {
          "richtext-border-[#1f6feb] richtext-bg-[#1f6feb1f]": e === "note",
          "richtext-border-[#238636] richtext-bg-[#2386361f]": e === "tip",
          "richtext-border-[#ab7df8] richtext-bg-[#ab7df81f]": e === "important",
          "richtext-border-[#d29922] richtext-bg-[#d299221f]": e === "warning",
          "richtext-border-[#f85149] richtext-bg-[#f851491f]": e === "caution"
        }
      ),
      children: [
        /* @__PURE__ */ l(
          "div",
          {
            className: x("richtext-mb-2 richtext-flex richtext-items-center richtext-gap-2", {
              "richtext-text-[#1f6feb]": e === "note",
              "richtext-text-[#238636]": e === "tip",
              "richtext-text-[#ab7df8]": e === "important",
              "richtext-text-[#d29922]": e === "warning",
              "richtext-text-[#f85149]": e === "caution"
            }),
            children: [
              /* @__PURE__ */ o(c, { className: "richtext-size-5" }),
              /* @__PURE__ */ o("span", { className: "richtext-font-semibold", children: a })
            ]
          }
        ),
        n && /* @__PURE__ */ o("p", { className: "richtext-pl-[28px]", children: n })
      ]
    }
  ) });
}
const ot = [
  { value: "note", label: "Note", icon: "Info" },
  { value: "tip", label: "Tip", icon: "Lightbulb" },
  { value: "important", label: "Important", icon: "AlertCircle" },
  { value: "warning", label: "Warning", icon: "TriangleAlert" },
  { value: "caution", label: "Caution", icon: "OctagonAlert" }
];
function pt() {
  const { t } = tt(), e = V(), a = P(lt.name), [n, i] = s(!1), [c, d] = s("note"), [h, b] = s(""), [g, f] = s(""), {
    icon: A = void 0,
    tooltip: N = void 0,
    shortcutKeys: L = void 0,
    tooltipOptions: S = {},
    isActive: k = void 0
  } = (a == null ? void 0 : a.componentProps) ?? {}, { dataState: I, disabled: m } = E(k), w = () => {
    e && (e.chain().focus().setCallout({ type: c, title: h, body: g }).run(), i(!1), d("note"), b(""), f(""));
  }, O = () => {
    m || i(!0);
  };
  return a ? /* @__PURE__ */ l(R, { onOpenChange: i, open: n, children: [
    /* @__PURE__ */ o(W, { asChild: !0, children: /* @__PURE__ */ o(
      T,
      {
        action: O,
        dataState: I,
        disabled: m,
        icon: A,
        shortcutKeys: L,
        tooltip: N,
        tooltipOptions: S
      }
    ) }),
    /* @__PURE__ */ l(j, { children: [
      /* @__PURE__ */ o(F, { children: t("editor.callout.dialog.title") }),
      /* @__PURE__ */ l("div", { className: "richtext-space-y-4 richtext-py-4", children: [
        /* @__PURE__ */ l("div", { className: "richtext-space-y-2", children: [
          /* @__PURE__ */ o(u, { children: t("editor.callout.dialog.type") }),
          /* @__PURE__ */ l(U, { onValueChange: d, value: c, children: [
            /* @__PURE__ */ o(Y, { children: /* @__PURE__ */ o(_, { placeholder: t("editor.callout.dialog.type.placeholder") }) }),
            /* @__PURE__ */ o($, { children: ot.map((r) => /* @__PURE__ */ o(q, { value: r.value, children: t(`editor.callout.type.${r.value}`) }, r.value)) })
          ] })
        ] }),
        /* @__PURE__ */ l("div", { className: "richtext-space-y-2", children: [
          /* @__PURE__ */ o(u, { children: t("editor.callout.dialog.title.label") }),
          /* @__PURE__ */ o(
            v,
            {
              onChange: (r) => b(r.target.value),
              placeholder: t("editor.callout.dialog.title.placeholder"),
              value: h
            }
          )
        ] }),
        /* @__PURE__ */ l("div", { className: "richtext-space-y-2", children: [
          /* @__PURE__ */ o(u, { children: t("editor.callout.dialog.body.label") }),
          /* @__PURE__ */ o(
            v,
            {
              onChange: (r) => f(r.target.value),
              placeholder: t("editor.callout.dialog.body.placeholder"),
              value: g
            }
          )
        ] })
      ] }),
      /* @__PURE__ */ l(z, { children: [
        /* @__PURE__ */ o(C, { onClick: () => i(!1), variant: "outline", children: t("editor.callout.dialog.button.cancel") }),
        /* @__PURE__ */ o(C, { onClick: w, children: t("editor.callout.dialog.button.apply") })
      ] })
    ] })
  ] }) : /* @__PURE__ */ o(K, {});
}
function p(t) {
  return (e) => e.getAttribute(t);
}
const lt = /* @__PURE__ */ D.create({
  name: "callout",
  group: "block",
  selectable: !0,
  atom: !0,
  draggable: !0,
  inline: !1,
  //@ts-expect-error
  addOptions() {
    var t;
    return {
      ...(t = this.parent) == null ? void 0 : t.call(this),
      HTMLAttributes: {
        class: "callout"
      },
      button: ({ editor: e, t: a }) => ({
        component: T,
        componentProps: {
          action: () => !0,
          isActive: () => e.isActive("callout"),
          disabled: !1,
          icon: "Callout",
          tooltip: a("editor.callout.tooltip")
        }
      })
    };
  },
  parseHTML() {
    return [{ tag: "div.callout" }];
  },
  renderHTML({ HTMLAttributes: t }) {
    return [
      "div",
      H(this.options && this.options.HTMLAttributes || {}, t)
    ];
  },
  addAttributes() {
    return {
      type: {
        default: "",
        parseHTML: p("type")
      },
      title: {
        default: "",
        parseHTML: p("title")
      },
      body: {
        default: "",
        parseHTML: p("body")
      }
    };
  },
  addCommands() {
    return {
      setCallout: (t) => ({ chain: e }) => e().insertContent({
        type: this.name,
        attrs: t
      }).run()
    };
  },
  addNodeView() {
    return B(et);
  }
});
export {
  lt as Callout,
  pt as RichTextCallout
};
