// reactjs-tiptap-editor@1.0.2/lib/clsx-_bq4MQQ7.js
import A, { createContext as U, useContext as G, createRef as de, memo as fe, createElement as he, version as k, useState as pe, useDebugValue as me, useLayoutEffect as Ee, useEffect as ye, forwardRef as ve, useMemo as z } from "react";
import Se, { flushSync as _e } from "react-dom";
import { jsx as b, jsxs as ge, Fragment as ie } from "react/jsx-runtime";
import { $ as Oe, a0 as we, a as Ae } from "./index-Dx-Dh7KS.js";
var D = { exports: {} }, P = {};
/**
 * @license React
 * use-sync-external-store-shim.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var H;
function be() {
  if (H) return P;
  H = 1;
  var e = A;
  function t(p, v) {
    return p === v && (p !== 0 || 1 / p === 1 / v) || p !== p && v !== v;
  }
  var r = typeof Object.is == "function" ? Object.is : t, n = e.useState, o = e.useEffect, u = e.useLayoutEffect, i = e.useDebugValue;
  function s(p, v) {
    var l = v(), f = n({ inst: { value: l, getSnapshot: v } }), m = f[0].inst, a = f[1];
    return u(
      function() {
        m.value = l, m.getSnapshot = v, y(m) && a({ inst: m });
      },
      [p, l, v]
    ), o(
      function() {
        return y(m) && a({ inst: m }), p(function() {
          y(m) && a({ inst: m });
        });
      },
      [p]
    ), i(l), l;
  }
  function y(p) {
    var v = p.getSnapshot;
    p = p.value;
    try {
      var l = v();
      return !r(p, l);
    } catch {
      return !0;
    }
  }
  function c(p, v) {
    return v();
  }
  var h = typeof window > "u" || typeof window.document > "u" || typeof window.document.createElement > "u" ? c : s;
  return P.useSyncExternalStore = e.useSyncExternalStore !== void 0 ? e.useSyncExternalStore : h, P;
}
var V = {};
/**
 * @license React
 * use-sync-external-store-shim.development.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var K;
function Re() {
  return K || (K = 1, process.env.NODE_ENV !== "production" && (function() {
    function e(l, f) {
      return l === f && (l !== 0 || 1 / l === 1 / f) || l !== l && f !== f;
    }
    function t(l, f) {
      h || o.startTransition === void 0 || (h = !0, console.error(
        "You are using an outdated, pre-release alpha of React 18 that does not support useSyncExternalStore. The use-sync-external-store shim will not work correctly. Upgrade to a newer pre-release."
      ));
      var m = f();
      if (!p) {
        var a = f();
        u(m, a) || (console.error(
          "The result of getSnapshot should be cached to avoid an infinite loop"
        ), p = !0);
      }
      a = i({
        inst: { value: m, getSnapshot: f }
      });
      var d = a[0].inst, E = a[1];
      return y(
        function() {
          d.value = m, d.getSnapshot = f, r(d) && E({ inst: d });
        },
        [l, m, f]
      ), s(
        function() {
          return r(d) && E({ inst: d }), l(function() {
            r(d) && E({ inst: d });
          });
        },
        [l]
      ), c(m), m;
    }
    function r(l) {
      var f = l.getSnapshot;
      l = l.value;
      try {
        var m = f();
        return !u(l, m);
      } catch {
        return !0;
      }
    }
    function n(l, f) {
      return f();
    }
    typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ < "u" && typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStart == "function" && __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStart(Error());
    var o = A, u = typeof Object.is == "function" ? Object.is : e, i = o.useState, s = o.useEffect, y = o.useLayoutEffect, c = o.useDebugValue, h = !1, p = !1, v = typeof window > "u" || typeof window.document > "u" || typeof window.document.createElement > "u" ? n : t;
    V.useSyncExternalStore = o.useSyncExternalStore !== void 0 ? o.useSyncExternalStore : v, typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ < "u" && typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStop == "function" && __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStop(Error());
  })()), V;
}
var F;
function W() {
  return F || (F = 1, process.env.NODE_ENV === "production" ? D.exports = be() : D.exports = Re()), D.exports;
}
var Te = W();
const { getOwnPropertyNames: Ce, getOwnPropertySymbols: Ne } = Object, { hasOwnProperty: De } = Object.prototype;
function M(e, t) {
  return function(n, o, u) {
    return e(n, o, u) && t(n, o, u);
  };
}
function j(e) {
  return function(r, n, o) {
    if (!r || !n || typeof r != "object" || typeof n != "object")
      return e(r, n, o);
    const { cache: u } = o, i = u.get(r), s = u.get(n);
    if (i && s)
      return i === n && s === r;
    u.set(r, n), u.set(n, r);
    const y = e(r, n, o);
    return u.delete(r), u.delete(n), y;
  };
}
function je(e) {
  return e != null ? e[Symbol.toStringTag] : void 0;
}
function Y(e) {
  return Ce(e).concat(Ne(e));
}
const Le = (
  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
  Object.hasOwn || ((e, t) => De.call(e, t))
);
function T(e, t) {
  return e === t || !e && !t && e !== e && t !== t;
}
const xe = "__v", Pe = "__o", Ve = "_owner", { getOwnPropertyDescriptor: J, keys: X } = Object;
function Me(e, t) {
  return e.byteLength === t.byteLength && x(new Uint8Array(e), new Uint8Array(t));
}
function Ie(e, t, r) {
  let n = e.length;
  if (t.length !== n)
    return !1;
  for (; n-- > 0; )
    if (!r.equals(e[n], t[n], n, n, e, t, r))
      return !1;
  return !0;
}
function qe(e, t) {
  return e.byteLength === t.byteLength && x(new Uint8Array(e.buffer, e.byteOffset, e.byteLength), new Uint8Array(t.buffer, t.byteOffset, t.byteLength));
}
function Ue(e, t) {
  return T(e.getTime(), t.getTime());
}
function Ge(e, t) {
  return e.name === t.name && e.message === t.message && e.cause === t.cause && e.stack === t.stack;
}
function We(e, t) {
  return e === t;
}
function Z(e, t, r) {
  const n = e.size;
  if (n !== t.size)
    return !1;
  if (!n)
    return !0;
  const o = new Array(n), u = e.entries();
  let i, s, y = 0;
  for (; (i = u.next()) && !i.done; ) {
    const c = t.entries();
    let h = !1, p = 0;
    for (; (s = c.next()) && !s.done; ) {
      if (o[p]) {
        p++;
        continue;
      }
      const v = i.value, l = s.value;
      if (r.equals(v[0], l[0], y, p, e, t, r) && r.equals(v[1], l[1], v[0], l[0], e, t, r)) {
        h = o[p] = !0;
        break;
      }
      p++;
    }
    if (!h)
      return !1;
    y++;
  }
  return !0;
}
const Be = T;
function $e(e, t, r) {
  const n = X(e);
  let o = n.length;
  if (X(t).length !== o)
    return !1;
  for (; o-- > 0; )
    if (!se(e, t, r, n[o]))
      return !1;
  return !0;
}
function N(e, t, r) {
  const n = Y(e);
  let o = n.length;
  if (Y(t).length !== o)
    return !1;
  let u, i, s;
  for (; o-- > 0; )
    if (u = n[o], !se(e, t, r, u) || (i = J(e, u), s = J(t, u), (i || s) && (!i || !s || i.configurable !== s.configurable || i.enumerable !== s.enumerable || i.writable !== s.writable)))
      return !1;
  return !0;
}
function ke(e, t) {
  return T(e.valueOf(), t.valueOf());
}
function ze(e, t) {
  return e.source === t.source && e.flags === t.flags;
}
function Q(e, t, r) {
  const n = e.size;
  if (n !== t.size)
    return !1;
  if (!n)
    return !0;
  const o = new Array(n), u = e.values();
  let i, s;
  for (; (i = u.next()) && !i.done; ) {
    const y = t.values();
    let c = !1, h = 0;
    for (; (s = y.next()) && !s.done; ) {
      if (!o[h] && r.equals(i.value, s.value, i.value, s.value, e, t, r)) {
        c = o[h] = !0;
        break;
      }
      h++;
    }
    if (!c)
      return !1;
  }
  return !0;
}
function x(e, t) {
  let r = e.byteLength;
  if (t.byteLength !== r || e.byteOffset !== t.byteOffset)
    return !1;
  for (; r-- > 0; )
    if (e[r] !== t[r])
      return !1;
  return !0;
}
function He(e, t) {
  return e.hostname === t.hostname && e.pathname === t.pathname && e.protocol === t.protocol && e.port === t.port && e.hash === t.hash && e.username === t.username && e.password === t.password;
}
function se(e, t, r, n) {
  return (n === Ve || n === Pe || n === xe) && (e.$$typeof || t.$$typeof) ? !0 : Le(t, n) && r.equals(e[n], t[n], n, n, e, t, r);
}
const Ke = "[object ArrayBuffer]", Fe = "[object Arguments]", Ye = "[object Boolean]", Je = "[object DataView]", Xe = "[object Date]", Ze = "[object Error]", Qe = "[object Map]", et = "[object Number]", tt = "[object Object]", rt = "[object RegExp]", nt = "[object Set]", ot = "[object String]", it = {
  "[object Int8Array]": !0,
  "[object Uint8Array]": !0,
  "[object Uint8ClampedArray]": !0,
  "[object Int16Array]": !0,
  "[object Uint16Array]": !0,
  "[object Int32Array]": !0,
  "[object Uint32Array]": !0,
  "[object Float16Array]": !0,
  "[object Float32Array]": !0,
  "[object Float64Array]": !0,
  "[object BigInt64Array]": !0,
  "[object BigUint64Array]": !0
}, st = "[object URL]", at = Object.prototype.toString;
function ut({ areArrayBuffersEqual: e, areArraysEqual: t, areDataViewsEqual: r, areDatesEqual: n, areErrorsEqual: o, areFunctionsEqual: u, areMapsEqual: i, areNumbersEqual: s, areObjectsEqual: y, arePrimitiveWrappersEqual: c, areRegExpsEqual: h, areSetsEqual: p, areTypedArraysEqual: v, areUrlsEqual: l, unknownTagComparators: f }) {
  return function(a, d, E) {
    if (a === d)
      return !0;
    if (a == null || d == null)
      return !1;
    const w = typeof a;
    if (w !== typeof d)
      return !1;
    if (w !== "object")
      return w === "number" ? s(a, d, E) : w === "function" ? u(a, d, E) : !1;
    const g = a.constructor;
    if (g !== d.constructor)
      return !1;
    if (g === Object)
      return y(a, d, E);
    if (Array.isArray(a))
      return t(a, d, E);
    if (g === Date)
      return n(a, d, E);
    if (g === RegExp)
      return h(a, d, E);
    if (g === Map)
      return i(a, d, E);
    if (g === Set)
      return p(a, d, E);
    const _ = at.call(a);
    if (_ === Xe)
      return n(a, d, E);
    if (_ === rt)
      return h(a, d, E);
    if (_ === Qe)
      return i(a, d, E);
    if (_ === nt)
      return p(a, d, E);
    if (_ === tt)
      return typeof a.then != "function" && typeof d.then != "function" && y(a, d, E);
    if (_ === st)
      return l(a, d, E);
    if (_ === Ze)
      return o(a, d, E);
    if (_ === Fe)
      return y(a, d, E);
    if (it[_])
      return v(a, d, E);
    if (_ === Ke)
      return e(a, d, E);
    if (_ === Je)
      return r(a, d, E);
    if (_ === Ye || _ === et || _ === ot)
      return c(a, d, E);
    if (f) {
      let S = f[_];
      if (!S) {
        const O = je(a);
        O && (S = f[O]);
      }
      if (S)
        return S(a, d, E);
    }
    return !1;
  };
}
function ct({ circular: e, createCustomConfig: t, strict: r }) {
  let n = {
    areArrayBuffersEqual: Me,
    areArraysEqual: r ? N : Ie,
    areDataViewsEqual: qe,
    areDatesEqual: Ue,
    areErrorsEqual: Ge,
    areFunctionsEqual: We,
    areMapsEqual: r ? M(Z, N) : Z,
    areNumbersEqual: Be,
    areObjectsEqual: r ? N : $e,
    arePrimitiveWrappersEqual: ke,
    areRegExpsEqual: ze,
    areSetsEqual: r ? M(Q, N) : Q,
    areTypedArraysEqual: r ? M(x, N) : x,
    areUrlsEqual: He,
    unknownTagComparators: void 0
  };
  if (t && (n = Object.assign({}, n, t(n))), e) {
    const o = j(n.areArraysEqual), u = j(n.areMapsEqual), i = j(n.areObjectsEqual), s = j(n.areSetsEqual);
    n = Object.assign({}, n, {
      areArraysEqual: o,
      areMapsEqual: u,
      areObjectsEqual: i,
      areSetsEqual: s
    });
  }
  return n;
}
function lt(e) {
  return function(t, r, n, o, u, i, s) {
    return e(t, r, s);
  };
}
function dt({ circular: e, comparator: t, createState: r, equals: n, strict: o }) {
  if (r)
    return function(s, y) {
      const { cache: c = e ? /* @__PURE__ */ new WeakMap() : void 0, meta: h } = r();
      return t(s, y, {
        cache: c,
        equals: n,
        meta: h,
        strict: o
      });
    };
  if (e)
    return function(s, y) {
      return t(s, y, {
        cache: /* @__PURE__ */ new WeakMap(),
        equals: n,
        meta: void 0,
        strict: o
      });
    };
  const u = {
    cache: void 0,
    equals: n,
    meta: void 0,
    strict: o
  };
  return function(s, y) {
    return t(s, y, u);
  };
}
const ft = R();
R({ strict: !0 });
R({ circular: !0 });
R({
  circular: !0,
  strict: !0
});
R({
  createInternalComparator: () => T
});
R({
  strict: !0,
  createInternalComparator: () => T
});
R({
  circular: !0,
  createInternalComparator: () => T
});
R({
  circular: !0,
  createInternalComparator: () => T,
  strict: !0
});
function R(e = {}) {
  const { circular: t = !1, createInternalComparator: r, createState: n, strict: o = !1 } = e, u = ct(e), i = ut(u), s = r ? r(i) : lt(i);
  return dt({ circular: t, comparator: i, createState: n, equals: s, strict: o });
}
var L = { exports: {} }, I = {};
/**
 * @license React
 * use-sync-external-store-shim/with-selector.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var ee;
function ht() {
  if (ee) return I;
  ee = 1;
  var e = A, t = W();
  function r(c, h) {
    return c === h && (c !== 0 || 1 / c === 1 / h) || c !== c && h !== h;
  }
  var n = typeof Object.is == "function" ? Object.is : r, o = t.useSyncExternalStore, u = e.useRef, i = e.useEffect, s = e.useMemo, y = e.useDebugValue;
  return I.useSyncExternalStoreWithSelector = function(c, h, p, v, l) {
    var f = u(null);
    if (f.current === null) {
      var m = { hasValue: !1, value: null };
      f.current = m;
    } else m = f.current;
    f = s(
      function() {
        function d(S) {
          if (!E) {
            if (E = !0, w = S, S = v(S), l !== void 0 && m.hasValue) {
              var O = m.value;
              if (l(O, S))
                return g = O;
            }
            return g = S;
          }
          if (O = g, n(w, S)) return O;
          var C = v(S);
          return l !== void 0 && l(O, C) ? (w = S, O) : (w = S, g = C);
        }
        var E = !1, w, g, _ = p === void 0 ? null : p;
        return [
          function() {
            return d(h());
          },
          _ === null ? void 0 : function() {
            return d(_());
          }
        ];
      },
      [h, p, v, l]
    );
    var a = o(c, f[0], f[1]);
    return i(
      function() {
        m.hasValue = !0, m.value = a;
      },
      [a]
    ), y(a), a;
  }, I;
}
var q = {};
/**
 * @license React
 * use-sync-external-store-shim/with-selector.development.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var te;
function pt() {
  return te || (te = 1, process.env.NODE_ENV !== "production" && (function() {
    function e(c, h) {
      return c === h && (c !== 0 || 1 / c === 1 / h) || c !== c && h !== h;
    }
    typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ < "u" && typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStart == "function" && __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStart(Error());
    var t = A, r = W(), n = typeof Object.is == "function" ? Object.is : e, o = r.useSyncExternalStore, u = t.useRef, i = t.useEffect, s = t.useMemo, y = t.useDebugValue;
    q.useSyncExternalStoreWithSelector = function(c, h, p, v, l) {
      var f = u(null);
      if (f.current === null) {
        var m = { hasValue: !1, value: null };
        f.current = m;
      } else m = f.current;
      f = s(
        function() {
          function d(S) {
            if (!E) {
              if (E = !0, w = S, S = v(S), l !== void 0 && m.hasValue) {
                var O = m.value;
                if (l(O, S))
                  return g = O;
              }
              return g = S;
            }
            if (O = g, n(w, S))
              return O;
            var C = v(S);
            return l !== void 0 && l(O, C) ? (w = S, O) : (w = S, g = C);
          }
          var E = !1, w, g, _ = p === void 0 ? null : p;
          return [
            function() {
              return d(h());
            },
            _ === null ? void 0 : function() {
              return d(_());
            }
          ];
        },
        [h, p, v, l]
      );
      var a = o(c, f[0], f[1]);
      return i(
        function() {
          m.hasValue = !0, m.value = a;
        },
        [a]
      ), y(a), a;
    }, typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ < "u" && typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStop == "function" && __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStop(Error());
  })()), q;
}
var re;
function mt() {
  return re || (re = 1, process.env.NODE_ENV === "production" ? L.exports = ht() : L.exports = pt()), L.exports;
}
var Et = mt(), yt = (...e) => (t) => {
  e.forEach((r) => {
    typeof r == "function" ? r(t) : r && (r.current = t);
  });
}, vt = ({ contentComponent: e }) => {
  const t = Te.useSyncExternalStore(
    e.subscribe,
    e.getSnapshot,
    e.getServerSnapshot
  );
  return /* @__PURE__ */ b(ie, { children: Object.values(t) });
};
function St() {
  const e = /* @__PURE__ */ new Set();
  let t = {};
  return {
    /**
     * Subscribe to the editor instance's changes.
     */
    subscribe(r) {
      return e.add(r), () => {
        e.delete(r);
      };
    },
    getSnapshot() {
      return t;
    },
    getServerSnapshot() {
      return t;
    },
    /**
     * Adds a new NodeView Renderer to the editor.
     */
    setRenderer(r, n) {
      t = {
        ...t,
        [r]: Se.createPortal(n.reactElement, n.element, r)
      }, e.forEach((o) => o());
    },
    /**
     * Removes a NodeView Renderer from the editor.
     */
    removeRenderer(r) {
      const n = { ...t };
      delete n[r], t = n, e.forEach((o) => o());
    }
  };
}
var _t = class extends A.Component {
  constructor(e) {
    super(e), this.editorContentRef = A.createRef();
  }
  componentDidMount() {
    this.init();
  }
  componentDidUpdate() {
    this.init();
  }
  init() {
    var e;
    const t = this.props.editor;
    if (t && !t.isDestroyed && ((e = t.view.dom) != null && e.parentNode)) {
      if (t.contentComponent)
        return;
      const r = this.editorContentRef.current;
      r.append(...t.view.dom.parentNode.childNodes), t.setOptions({
        element: r
      }), t.contentComponent = St(), t.createNodeViews(), t.isEditorContentInitialized = !0, this.forceUpdate();
    }
  }
  componentWillUnmount() {
    var e;
    const t = this.props.editor;
    if (t) {
      t.isEditorContentInitialized = !1, t.isDestroyed || t.view.setProps({
        nodeViews: {}
      }), t.contentComponent = null;
      try {
        if (!((e = t.view.dom) != null && e.parentNode))
          return;
        const r = document.createElement("div");
        r.append(...t.view.dom.parentNode.childNodes), t.setOptions({
          element: r
        });
      } catch {
      }
    }
  }
  render() {
    const { editor: e, innerRef: t, ...r } = this.props;
    return /* @__PURE__ */ ge(ie, { children: [
      /* @__PURE__ */ b("div", { ref: yt(t, this.editorContentRef), ...r }),
      (e == null ? void 0 : e.contentComponent) && /* @__PURE__ */ b(vt, { contentComponent: e.contentComponent })
    ] });
  }
}, gt = ve(
  (e, t) => {
    const r = A.useMemo(() => Math.floor(Math.random() * 4294967295).toString(), [e.editor]);
    return A.createElement(_t, {
      key: r,
      innerRef: t,
      ...e
    });
  }
), Ot = A.memo(gt), wt = typeof window < "u" ? Ee : ye, At = class {
  constructor(e) {
    this.transactionNumber = 0, this.lastTransactionNumber = 0, this.subscribers = /* @__PURE__ */ new Set(), this.editor = e, this.lastSnapshot = { editor: e, transactionNumber: 0 }, this.getSnapshot = this.getSnapshot.bind(this), this.getServerSnapshot = this.getServerSnapshot.bind(this), this.watch = this.watch.bind(this), this.subscribe = this.subscribe.bind(this);
  }
  /**
   * Get the current editor instance.
   */
  getSnapshot() {
    return this.transactionNumber === this.lastTransactionNumber ? this.lastSnapshot : (this.lastTransactionNumber = this.transactionNumber, this.lastSnapshot = { editor: this.editor, transactionNumber: this.transactionNumber }, this.lastSnapshot);
  }
  /**
   * Always disable the editor on the server-side.
   */
  getServerSnapshot() {
    return { editor: null, transactionNumber: 0 };
  }
  /**
   * Subscribe to the editor instance's changes.
   */
  subscribe(e) {
    return this.subscribers.add(e), () => {
      this.subscribers.delete(e);
    };
  }
  /**
   * Watch the editor instance for changes.
   */
  watch(e) {
    if (this.editor = e, this.editor) {
      const t = () => {
        this.transactionNumber += 1, this.subscribers.forEach((n) => n());
      }, r = this.editor;
      return r.on("transaction", t), () => {
        r.off("transaction", t);
      };
    }
  }
};
function Mt(e) {
  var t;
  const [r] = pe(() => new At(e.editor)), n = Et.useSyncExternalStoreWithSelector(
    r.subscribe,
    r.getSnapshot,
    r.getServerSnapshot,
    e.selector,
    (t = e.equalityFn) != null ? t : ft
  );
  return wt(() => r.watch(e.editor), [e.editor, r]), me(n), n;
}
process.env.NODE_ENV;
var B = U({
  editor: null
});
B.Consumer;
var It = () => G(B), ae = U({
  onDragStart: () => {
  },
  nodeViewContentChildren: void 0,
  nodeViewContentRef: () => {
  }
}), bt = () => G(ae), qt = A.forwardRef((e, t) => {
  const { onDragStart: r } = bt(), n = e.as || "div";
  return (
    // @ts-ignore
    /* @__PURE__ */ b(
      n,
      {
        ...e,
        ref: t,
        "data-node-view-wrapper": "",
        onDragStart: r,
        style: {
          whiteSpace: "normal",
          ...e.style
        }
      }
    )
  );
});
function ne(e) {
  return !!(typeof e == "function" && e.prototype && e.prototype.isReactComponent);
}
function oe(e) {
  return !!(typeof e == "object" && e.$$typeof && (e.$$typeof.toString() === "Symbol(react.forward_ref)" || e.$$typeof.description === "react.forward_ref"));
}
function Rt(e) {
  return !!(typeof e == "object" && e.$$typeof && (e.$$typeof.toString() === "Symbol(react.memo)" || e.$$typeof.description === "react.memo"));
}
function Tt(e) {
  if (ne(e) || oe(e))
    return !0;
  if (Rt(e)) {
    const t = e.type;
    if (t)
      return ne(t) || oe(t);
  }
  return !1;
}
function Ct() {
  try {
    if (k)
      return parseInt(k.split(".")[0], 10) >= 19;
  } catch {
  }
  return !1;
}
var Nt = class {
  /**
   * Immediately creates element and renders the provided React component.
   */
  constructor(e, { editor: t, props: r = {}, as: n = "div", className: o = "" }) {
    this.ref = null, this.destroyed = !1, this.id = Math.floor(Math.random() * 4294967295).toString(), this.component = e, this.editor = t, this.props = r, this.element = document.createElement(n), this.element.classList.add("react-renderer"), o && this.element.classList.add(...o.split(" ")), this.editor.isEditorContentInitialized ? _e(() => {
      this.render();
    }) : queueMicrotask(() => {
      this.destroyed || this.render();
    });
  }
  /**
   * Render the React component.
   */
  render() {
    var e;
    if (this.destroyed)
      return;
    const t = this.component, r = this.props, n = this.editor, o = Ct(), u = Tt(t), i = { ...r };
    i.ref && !(o || u) && delete i.ref, !i.ref && (o || u) && (i.ref = (s) => {
      this.ref = s;
    }), this.reactElement = /* @__PURE__ */ b(t, { ...i }), (e = n == null ? void 0 : n.contentComponent) == null || e.setRenderer(this.id, this);
  }
  /**
   * Re-renders the React component with new props.
   * Skips the render if none of the supplied props actually changed,
   * to avoid unnecessary portal re-creation.
   */
  updateProps(e = {}) {
    if (this.destroyed)
      return;
    let t = !1;
    const r = Object.keys(e);
    for (let n = 0; n < r.length; n += 1) {
      const o = r[n];
      if (e[o] !== this.props[o]) {
        t = !0;
        break;
      }
    }
    t && (this.props = {
      ...this.props,
      ...e
    }, this.render());
  }
  /**
   * Destroy the React component.
   */
  destroy() {
    var e;
    this.destroyed = !0;
    const t = this.editor;
    (e = t == null ? void 0 : t.contentComponent) == null || e.removeRenderer(this.id);
    try {
      this.element && this.element.parentNode && this.element.parentNode.removeChild(this.element);
    } catch {
    }
  }
  /**
   * Update the attributes of the element that holds the React component.
   */
  updateAttributes(e) {
    Object.keys(e).forEach((t) => {
      this.element.setAttribute(t, e[t]);
    });
  }
};
A.createContext({
  markViewContentRef: () => {
  }
});
var Dt = class extends Oe {
  constructor(e, t, r) {
    if (super(e, t, r), this.selectionRafId = null, this.handlePositionUpdate = () => {
      const n = this.getPos();
      typeof n != "number" || n === this.currentPos || (this.currentPos = n, this.renderer.updateProps({ getPos: () => this.getPos() }), typeof this.options.attrs == "function" && this.updateElementAttributes());
    }, this.cachedExtensionWithSyncedStorage = null, !this.node.isLeaf) {
      this.options.contentDOMElementTag ? this.contentDOMElement = document.createElement(this.options.contentDOMElementTag) : this.contentDOMElement = document.createElement(this.node.isInline ? "span" : "div"), this.contentDOMElement.dataset.nodeViewContentReact = "", this.contentDOMElement.dataset.nodeViewWrapper = "", this.contentDOMElement.style.whiteSpace = "inherit";
      const n = this.dom.querySelector("[data-node-view-content]");
      if (!n)
        return;
      n.appendChild(this.contentDOMElement);
    }
    this.options.trackNodeViewPosition && this.editor.on("update", this.handlePositionUpdate);
  }
  /**
   * Returns a proxy of the extension that redirects storage access to the editor's mutable storage.
   * This preserves the original prototype chain (instanceof checks, methods like configure/extend work).
   * Cached to avoid proxy creation on every update.
   */
  get extensionWithSyncedStorage() {
    if (!this.cachedExtensionWithSyncedStorage) {
      const e = this.editor, t = this.extension;
      this.cachedExtensionWithSyncedStorage = new Proxy(t, {
        get(r, n, o) {
          var u;
          return n === "storage" ? (u = e.storage[t.name]) != null ? u : {} : Reflect.get(r, n, o);
        }
      });
    }
    return this.cachedExtensionWithSyncedStorage;
  }
  /**
   * Setup the React component.
   * Called on initialization.
   */
  mount() {
    const t = {
      editor: this.editor,
      node: this.node,
      decorations: this.decorations,
      innerDecorations: this.innerDecorations,
      view: this.view,
      selected: !1,
      extension: this.extensionWithSyncedStorage,
      HTMLAttributes: this.HTMLAttributes,
      getPos: () => this.getPos(),
      updateAttributes: (c = {}) => this.updateAttributes(c),
      deleteNode: () => this.deleteNode(),
      ref: de()
    };
    if (!this.component.displayName) {
      const c = (h) => h.charAt(0).toUpperCase() + h.substring(1);
      this.component.displayName = c(this.extension.name);
    }
    const o = { onDragStart: this.onDragStart.bind(this), nodeViewContentRef: (c) => {
      c && this.contentDOMElement && c.firstChild !== this.contentDOMElement && (c.hasAttribute("data-node-view-wrapper") && c.removeAttribute("data-node-view-wrapper"), c.appendChild(this.contentDOMElement));
    } }, u = this.component, i = fe(
      (c) => /* @__PURE__ */ b(ae.Provider, { value: o, children: he(u, c) })
    );
    i.displayName = "ReactNodeView";
    let s = this.node.isInline ? "span" : "div";
    this.options.as && (s = this.options.as);
    const { className: y = "" } = this.options;
    this.handleSelectionUpdate = this.handleSelectionUpdate.bind(this), this.renderer = new Nt(i, {
      editor: this.editor,
      props: t,
      as: s,
      className: `node-${this.node.type.name} ${y}`.trim()
    }), this.editor.on("selectionUpdate", this.handleSelectionUpdate), this.updateElementAttributes(), this.currentPos = this.getPos();
  }
  /**
   * Return the DOM element.
   * This is the element that will be used to display the node view.
   */
  get dom() {
    var e;
    if (this.renderer.element.firstElementChild && !((e = this.renderer.element.firstElementChild) != null && e.hasAttribute("data-node-view-wrapper")))
      throw Error("Please use the NodeViewWrapper component for your node view.");
    return this.renderer.element;
  }
  /**
   * Return the content DOM element.
   * This is the element that will be used to display the rich-text content of the node.
   */
  get contentDOM() {
    return this.node.isLeaf ? null : this.contentDOMElement;
  }
  /**
   * On editor selection update, check if the node is selected.
   * If it is, call `selectNode`, otherwise call `deselectNode`.
   */
  handleSelectionUpdate() {
    this.selectionRafId && (cancelAnimationFrame(this.selectionRafId), this.selectionRafId = null), this.selectionRafId = requestAnimationFrame(() => {
      this.selectionRafId = null;
      const e = this.currentPos;
      if (typeof e != "number")
        return;
      if (we({
        selection: this.editor.state.selection,
        pos: e,
        nodeSize: this.node.nodeSize,
        selectedOnTextSelection: this.options.selectedOnTextSelection
      })) {
        if (this.renderer.props.selected)
          return;
        this.selectNode();
      } else {
        if (!this.renderer.props.selected)
          return;
        this.deselectNode();
      }
    });
  }
  /**
   * On update, update the React component.
   * To prevent unnecessary updates, the `update` option can be used.
   */
  update(e, t, r) {
    const n = (s) => {
      this.renderer.updateProps(s), typeof this.options.attrs == "function" && this.updateElementAttributes();
    };
    if (e.type !== this.node.type)
      return !1;
    if (typeof this.options.update == "function") {
      const s = this.node, y = this.decorations, c = this.innerDecorations;
      return this.node = e, this.decorations = t, this.innerDecorations = r, this.currentPos = this.getPos(), this.options.update({
        oldNode: s,
        oldDecorations: y,
        newNode: e,
        newDecorations: t,
        oldInnerDecorations: c,
        innerDecorations: r,
        updateProps: () => n({
          node: e,
          decorations: t,
          innerDecorations: r,
          extension: this.extensionWithSyncedStorage
        })
      });
    }
    if (!(e !== this.node))
      return this.node = e, this.decorations = t, this.innerDecorations = r, !0;
    const u = this.getPos();
    this.node = e, this.decorations = t, this.innerDecorations = r, this.currentPos = u;
    const i = {
      node: e,
      decorations: t,
      innerDecorations: r,
      extension: this.extensionWithSyncedStorage
    };
    return this.options.trackNodeViewPosition && (i.getPos = () => this.getPos()), n(i), !0;
  }
  /**
   * Select the node.
   * Add the `selected` prop and the `ProseMirror-selectednode` class.
   */
  selectNode() {
    this.renderer.updateProps({
      selected: !0
    }), this.renderer.element.classList.add("ProseMirror-selectednode");
  }
  /**
   * Deselect the node.
   * Remove the `selected` prop and the `ProseMirror-selectednode` class.
   */
  deselectNode() {
    this.renderer.updateProps({
      selected: !1
    }), this.renderer.element.classList.remove("ProseMirror-selectednode");
  }
  /**
   * Destroy the React component instance.
   */
  destroy() {
    this.renderer.destroy(), this.editor.off("selectionUpdate", this.handleSelectionUpdate), this.options.trackNodeViewPosition && this.editor.off("update", this.handlePositionUpdate), this.contentDOMElement = null, this.selectionRafId && (cancelAnimationFrame(this.selectionRafId), this.selectionRafId = null);
  }
  /**
   * Update the attributes of the top-level element that holds the React component.
   * Applying the attributes defined in the `attrs` option.
   */
  updateElementAttributes() {
    if (this.options.attrs) {
      let e = {};
      if (typeof this.options.attrs == "function") {
        const t = this.editor.extensionManager.attributes, r = Ae(this.node, t);
        e = this.options.attrs({ node: this.node, HTMLAttributes: r });
      } else
        e = this.options.attrs;
      this.renderer.updateAttributes(e);
    }
  }
};
function Ut(e, t) {
  return (r) => r.editor.contentComponent ? new Dt(e, r, t) : {};
}
var $ = U({
  get editor() {
    throw new Error("useTiptap must be used within a <Tiptap> provider");
  }
});
$.displayName = "TiptapContext";
var jt = () => G($);
function ue({ editor: e, instance: t, children: r }) {
  const n = e ?? t;
  if (!n)
    throw new Error("Tiptap: An editor instance is required. Pass a non-null `editor` prop.");
  const o = z(
    () => ({ editor: n }),
    [n]
  ), u = z(() => ({ editor: n }), [n]);
  return /* @__PURE__ */ b(B.Provider, { value: u, children: /* @__PURE__ */ b($.Provider, { value: o, children: r }) });
}
ue.displayName = "Tiptap";
function ce({ ...e }) {
  const { editor: t } = jt();
  return /* @__PURE__ */ b(Ot, { editor: t, ...e });
}
ce.displayName = "Tiptap.Content";
Object.assign(ue, {
  /**
   * The Tiptap Content component that renders the EditorContent with the editor instance from the context.
   * @see TiptapContent
   */
  Content: ce
});
function le(e) {
  var t, r, n = "";
  if (typeof e == "string" || typeof e == "number") n += e;
  else if (typeof e == "object") if (Array.isArray(e)) {
    var o = e.length;
    for (t = 0; t < o; t++) e[t] && (r = le(e[t])) && (n && (n += " "), n += r);
  } else for (r in e) e[r] && (n && (n += " "), n += r);
  return n;
}
function Gt() {
  for (var e, t, r = 0, n = "", o = arguments.length; r < o; r++) (e = arguments[r]) && (t = le(e)) && (n && (n += " "), n += t);
  return n;
}
export {
  B as E,
  qt as N,
  Nt as R,
  Ut as a,
  Mt as b,
  Gt as c,
  Te as s,
  It as u
};
