// reactjs-tiptap-editor@1.0.2/lib/Code.js
import { M as m, h as l, j as h, m as x } from "./index-Dx-Dh7KS.js";
import { jsx as d, Fragment as g } from "react/jsx-runtime";
import { u as M, d as f, A as v } from "./index-C52cuQ_Y.js";
import "react";
import "./theme.js";
var C = (e) => {
  const t = /`([^`]+)`(?!`)$/.exec(e);
  return !t || t.index > 0 && e[t.index - 1] === "`" ? null : {
    index: t.index,
    text: t[0],
    replaceWith: t[1]
  };
}, k = (e) => {
  const t = /`([^`]+)`(?!`)/g, o = [];
  let n;
  for (; (n = t.exec(e)) !== null; )
    n.index > 0 && e[n.index - 1] === "`" || o.push({
      index: n.index,
      text: n[0],
      replaceWith: n[1]
    });
  return o;
}, y = m.create({
  name: "code",
  addOptions() {
    return {
      HTMLAttributes: {}
    };
  },
  excludes: "_",
  code: !0,
  exitable: !0,
  parseHTML() {
    return [{ tag: "code" }];
  },
  renderHTML({ HTMLAttributes: e }) {
    return ["code", x(this.options.HTMLAttributes, e), 0];
  },
  markdownTokenName: "codespan",
  parseMarkdown: (e, t) => t.applyMark("code", [{ type: "text", text: e.text || "" }]),
  renderMarkdown: (e, t) => e.content ? `\`${t.renderChildren(e.content)}\`` : "",
  addCommands() {
    return {
      setCode: () => ({ commands: e }) => e.setMark(this.name),
      toggleCode: () => ({ commands: e }) => e.toggleMark(this.name),
      unsetCode: () => ({ commands: e }) => e.unsetMark(this.name)
    };
  },
  addKeyboardShortcuts() {
    return {
      "Mod-e": () => this.editor.commands.toggleCode()
    };
  },
  addInputRules() {
    return [
      h({
        find: C,
        type: this.type
      })
    ];
  },
  addPasteRules() {
    return [
      l({
        find: k,
        type: this.type
      })
    ];
  }
});
function K() {
  const e = M(A.name), {
    icon: t = void 0,
    tooltip: o = void 0,
    shortcutKeys: n = void 0,
    tooltipOptions: s = {},
    action: r = void 0,
    isActive: a = void 0
  } = (e == null ? void 0 : e.componentProps) ?? {}, { dataState: c, disabled: i, update: u } = f(a), p = () => {
    i || r && (r(), u());
  };
  return e ? /* @__PURE__ */ d(
    v,
    {
      action: p,
      dataState: c,
      disabled: i,
      icon: t,
      shortcutKeys: n,
      tooltip: o,
      tooltipOptions: s
    }
  ) : /* @__PURE__ */ d(g, {});
}
const A = /* @__PURE__ */ y.extend({
  //@ts-expect-error
  addOptions() {
    var e;
    return {
      ...(e = this.parent) == null ? void 0 : e.call(this),
      button: ({ editor: t, t: o, extension: n }) => ({
        componentProps: {
          action: () => t.commands.toggleCode(),
          isActive: () => t.isActive("code"),
          disabled: !t.can().toggleCode(),
          icon: "Code",
          shortcutKeys: n.options.shortcutKeys ?? ["mod", "E"],
          tooltip: o("editor.code.tooltip")
        }
      })
    };
  }
});
export {
  A as Code,
  K as RichTextCode
};
