// reactjs-tiptap-editor@1.0.2/lib/CodeBlock.js
import { N as Qt, t as Ke, m as en, K as Oe } from "./index-Dx-Dh7KS.js";
import { Plugin as Ve, PluginKey as qe, TextSelection as Fe, Selection as tn } from "@tiptap/pm/state";
import { Decoration as nn, DecorationSet as rn } from "@tiptap/pm/view";
import { g as sn } from "./_commonjsHelpers-B4e78b8K.js";
import { jsx as Xe, Fragment as on } from "react/jsx-runtime";
import { u as an, d as cn, A as ln } from "./index-C52cuQ_Y.js";
import "react";
import "./theme.js";
var ye = 4, un = /^```([a-z]+)?[\s\n]$/, fn = /^~~~([a-z]+)?[\s\n]$/, gn = Qt.create({
  name: "codeBlock",
  addOptions() {
    return {
      languageClassPrefix: "language-",
      exitOnTripleEnter: !0,
      exitOnArrowDown: !0,
      defaultLanguage: null,
      enableTabIndentation: !1,
      tabSize: ye,
      HTMLAttributes: {}
    };
  },
  content: "text*",
  marks: "",
  group: "block",
  code: !0,
  defining: !0,
  addAttributes() {
    return {
      language: {
        default: this.options.defaultLanguage,
        parseHTML: (i) => {
          var f;
          const { languageClassPrefix: h } = this.options;
          if (!h)
            return null;
          const x = [...((f = i.firstElementChild) == null ? void 0 : f.classList) || []].filter((M) => M.startsWith(h)).map((M) => M.replace(h, ""))[0];
          return x || null;
        },
        rendered: !1
      }
    };
  },
  parseHTML() {
    return [
      {
        tag: "pre",
        preserveWhitespace: "full"
      }
    ];
  },
  renderHTML({ node: i, HTMLAttributes: f }) {
    return [
      "pre",
      en(this.options.HTMLAttributes, f),
      [
        "code",
        {
          class: i.attrs.language ? this.options.languageClassPrefix + i.attrs.language : null
        },
        0
      ]
    ];
  },
  markdownTokenName: "code",
  parseMarkdown: (i, f) => {
    var h, u;
    return ((h = i.raw) == null ? void 0 : h.startsWith("```")) === !1 && ((u = i.raw) == null ? void 0 : u.startsWith("~~~")) === !1 && i.codeBlockStyle !== "indented" ? [] : f.createNode(
      "codeBlock",
      { language: i.lang || null },
      i.text ? [f.createTextNode(i.text)] : []
    );
  },
  renderMarkdown: (i, f) => {
    var h;
    let u = "";
    const E = ((h = i.attrs) == null ? void 0 : h.language) || "";
    return i.content ? u = [`\`\`\`${E}`, f.renderChildren(i.content), "```"].join(`
`) : u = `\`\`\`${E}

\`\`\``, u;
  },
  addCommands() {
    return {
      setCodeBlock: (i) => ({ commands: f }) => f.setNode(this.name, i),
      toggleCodeBlock: (i) => ({ commands: f }) => f.toggleNode(this.name, "paragraph", i)
    };
  },
  addKeyboardShortcuts() {
    return {
      "Mod-Alt-c": () => this.editor.commands.toggleCodeBlock(),
      // remove code block when at start of document or code block is empty
      Backspace: () => {
        const { empty: i, $anchor: f } = this.editor.state.selection, h = f.pos === 1;
        return !i || f.parent.type.name !== this.name ? !1 : h || !f.parent.textContent.length ? this.editor.commands.clearNodes() : !1;
      },
      // handle tab indentation
      Tab: ({ editor: i }) => {
        var f;
        if (!this.options.enableTabIndentation)
          return !1;
        const h = (f = this.options.tabSize) != null ? f : ye, { state: u } = i, { selection: E } = u, { $from: x, empty: M } = E;
        if (x.parent.type !== this.type)
          return !1;
        const y = " ".repeat(h);
        return M ? i.commands.insertContent(y) : i.commands.command(({ tr: A }) => {
          const { from: v, to: P } = E, C = u.doc.textBetween(v, P, `
`, `
`).split(`
`).map((W) => y + W).join(`
`);
          return A.replaceWith(v, P, u.schema.text(C)), !0;
        });
      },
      // handle shift+tab reverse indentation
      "Shift-Tab": ({ editor: i }) => {
        var f;
        if (!this.options.enableTabIndentation)
          return !1;
        const h = (f = this.options.tabSize) != null ? f : ye, { state: u } = i, { selection: E } = u, { $from: x, empty: M } = E;
        return x.parent.type !== this.type ? !1 : M ? i.commands.command(({ tr: y }) => {
          var A;
          const { pos: v } = x, P = x.start(), D = x.end(), C = u.doc.textBetween(P, D, `
`, `
`).split(`
`);
          let W = 0, j = 0;
          const Q = v - P;
          for (let $ = 0; $ < C.length; $ += 1) {
            if (j + C[$].length >= Q) {
              W = $;
              break;
            }
            j += C[$].length + 1;
          }
          const re = ((A = C[W].match(/^ */)) == null ? void 0 : A[0]) || "", ee = Math.min(re.length, h);
          if (ee === 0)
            return !0;
          let Y = P;
          for (let $ = 0; $ < W; $ += 1)
            Y += C[$].length + 1;
          return y.delete(Y, Y + ee), v - Y <= ee && y.setSelection(Fe.create(y.doc, Y)), !0;
        }) : i.commands.command(({ tr: y }) => {
          const { from: A, to: v } = E, m = u.doc.textBetween(A, v, `
`, `
`).split(`
`).map((C) => {
            var W;
            const j = ((W = C.match(/^ */)) == null ? void 0 : W[0]) || "", Q = Math.min(j.length, h);
            return C.slice(Q);
          }).join(`
`);
          return y.replaceWith(A, v, u.schema.text(m)), !0;
        });
      },
      // exit node on triple enter
      Enter: ({ editor: i }) => {
        if (!this.options.exitOnTripleEnter)
          return !1;
        const { state: f } = i, { selection: h } = f, { $from: u, empty: E } = h;
        if (!E || u.parent.type !== this.type)
          return !1;
        const x = u.parentOffset === u.parent.nodeSize - 2, M = u.parent.textContent.endsWith(`

`);
        return !x || !M ? !1 : i.chain().command(({ tr: y }) => (y.delete(u.pos - 2, u.pos), !0)).exitCode().run();
      },
      // exit node on arrow down
      ArrowDown: ({ editor: i }) => {
        if (!this.options.exitOnArrowDown)
          return !1;
        const { state: f } = i, { selection: h, doc: u } = f, { $from: E, empty: x } = h;
        if (!x || E.parent.type !== this.type || !(E.parentOffset === E.parent.nodeSize - 2))
          return !1;
        const y = E.after();
        return y === void 0 ? !1 : u.nodeAt(y) ? i.commands.command(({ tr: v }) => (v.setSelection(tn.near(u.resolve(y))), !0)) : i.commands.exitCode();
      }
    };
  },
  addInputRules() {
    return [
      Ke({
        find: un,
        type: this.type,
        getAttributes: (i) => ({
          language: i[1]
        })
      }),
      Ke({
        find: fn,
        type: this.type,
        getAttributes: (i) => ({
          language: i[1]
        })
      })
    ];
  },
  addProseMirrorPlugins() {
    return [
      // this plugin creates a code block for pasted content from VS Code
      // we can also detect the copied code language
      new Ve({
        key: new qe("codeBlockVSCodeHandler"),
        props: {
          handlePaste: (i, f) => {
            if (!f.clipboardData || this.editor.isActive(this.type.name))
              return !1;
            const h = f.clipboardData.getData("text/plain"), u = f.clipboardData.getData("vscode-editor-data"), E = u ? JSON.parse(u) : void 0, x = E == null ? void 0 : E.mode;
            if (!h || !x)
              return !1;
            const { tr: M, schema: y } = i.state, A = y.text(h.replace(/\r\n?/g, `
`));
            return M.replaceSelectionWith(this.type.create({ language: x }, A)), M.selection.$from.parent.type !== this.type && M.setSelection(
              Fe.near(M.doc.resolve(Math.max(0, M.selection.from - 2)))
            ), M.setMeta("paste", !0), i.dispatch(M), !0;
          }
        }
      })
    ];
  }
}), Se, Ye;
function hn() {
  if (Ye) return Se;
  Ye = 1;
  function i(e) {
    return e instanceof Map ? e.clear = e.delete = e.set = function() {
      throw new Error("map is read-only");
    } : e instanceof Set && (e.add = e.clear = e.delete = function() {
      throw new Error("set is read-only");
    }), Object.freeze(e), Object.getOwnPropertyNames(e).forEach((t) => {
      const r = e[t], d = typeof r;
      (d === "object" || d === "function") && !Object.isFrozen(r) && i(r);
    }), e;
  }
  class f {
    /**
     * @param {CompiledMode} mode
     */
    constructor(t) {
      t.data === void 0 && (t.data = {}), this.data = t.data, this.isMatchIgnored = !1;
    }
    ignoreMatch() {
      this.isMatchIgnored = !0;
    }
  }
  function h(e) {
    return e.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&#x27;");
  }
  function u(e, ...t) {
    const r = /* @__PURE__ */ Object.create(null);
    for (const d in e)
      r[d] = e[d];
    return t.forEach(function(d) {
      for (const N in d)
        r[N] = d[N];
    }), /** @type {T} */
    r;
  }
  const E = "</span>", x = (e) => !!e.scope, M = (e, { prefix: t }) => {
    if (e.startsWith("language:"))
      return e.replace("language:", "language-");
    if (e.includes(".")) {
      const r = e.split(".");
      return [
        `${t}${r.shift()}`,
        ...r.map((d, N) => `${d}${"_".repeat(N + 1)}`)
      ].join(" ");
    }
    return `${t}${e}`;
  };
  class y {
    /**
     * Creates a new HTMLRenderer
     *
     * @param {Tree} parseTree - the parse tree (must support `walk` API)
     * @param {{classPrefix: string}} options
     */
    constructor(t, r) {
      this.buffer = "", this.classPrefix = r.classPrefix, t.walk(this);
    }
    /**
     * Adds texts to the output stream
     *
     * @param {string} text */
    addText(t) {
      this.buffer += h(t);
    }
    /**
     * Adds a node open to the output stream (if needed)
     *
     * @param {Node} node */
    openNode(t) {
      if (!x(t)) return;
      const r = M(
        t.scope,
        { prefix: this.classPrefix }
      );
      this.span(r);
    }
    /**
     * Adds a node close to the output stream (if needed)
     *
     * @param {Node} node */
    closeNode(t) {
      x(t) && (this.buffer += E);
    }
    /**
     * returns the accumulated buffer
    */
    value() {
      return this.buffer;
    }
    // helpers
    /**
     * Builds a span element
     *
     * @param {string} className */
    span(t) {
      this.buffer += `<span class="${t}">`;
    }
  }
  const A = (e = {}) => {
    const t = { children: [] };
    return Object.assign(t, e), t;
  };
  class v {
    constructor() {
      this.rootNode = A(), this.stack = [this.rootNode];
    }
    get top() {
      return this.stack[this.stack.length - 1];
    }
    get root() {
      return this.rootNode;
    }
    /** @param {Node} node */
    add(t) {
      this.top.children.push(t);
    }
    /** @param {string} scope */
    openNode(t) {
      const r = A({ scope: t });
      this.add(r), this.stack.push(r);
    }
    closeNode() {
      if (this.stack.length > 1)
        return this.stack.pop();
    }
    closeAllNodes() {
      for (; this.closeNode(); ) ;
    }
    toJSON() {
      return JSON.stringify(this.rootNode, null, 4);
    }
    /**
     * @typedef { import("./html_renderer").Renderer } Renderer
     * @param {Renderer} builder
     */
    walk(t) {
      return this.constructor._walk(t, this.rootNode);
    }
    /**
     * @param {Renderer} builder
     * @param {Node} node
     */
    static _walk(t, r) {
      return typeof r == "string" ? t.addText(r) : r.children && (t.openNode(r), r.children.forEach((d) => this._walk(t, d)), t.closeNode(r)), t;
    }
    /**
     * @param {Node} node
     */
    static _collapse(t) {
      typeof t != "string" && t.children && (t.children.every((r) => typeof r == "string") ? t.children = [t.children.join("")] : t.children.forEach((r) => {
        v._collapse(r);
      }));
    }
  }
  class P extends v {
    /**
     * @param {*} options
     */
    constructor(t) {
      super(), this.options = t;
    }
    /**
     * @param {string} text
     */
    addText(t) {
      t !== "" && this.add(t);
    }
    /** @param {string} scope */
    startScope(t) {
      this.openNode(t);
    }
    endScope() {
      this.closeNode();
    }
    /**
     * @param {Emitter & {root: DataNode}} emitter
     * @param {string} name
     */
    __addSublanguage(t, r) {
      const d = t.root;
      r && (d.scope = `language:${r}`), this.add(d);
    }
    toHTML() {
      return new y(this, this.options).value();
    }
    finalize() {
      return this.closeAllNodes(), !0;
    }
  }
  function D(e) {
    return e ? typeof e == "string" ? e : e.source : null;
  }
  function m(e) {
    return j("(?=", e, ")");
  }
  function C(e) {
    return j("(?:", e, ")*");
  }
  function W(e) {
    return j("(?:", e, ")?");
  }
  function j(...e) {
    return e.map((r) => D(r)).join("");
  }
  function Q(e) {
    const t = e[e.length - 1];
    return typeof t == "object" && t.constructor === Object ? (e.splice(e.length - 1, 1), t) : {};
  }
  function ie(...e) {
    return "(" + (Q(e).capture ? "" : "?:") + e.map((d) => D(d)).join("|") + ")";
  }
  function re(e) {
    return new RegExp(e.toString() + "|").exec("").length - 1;
  }
  function ee(e, t) {
    const r = e && e.exec(t);
    return r && r.index === 0;
  }
  const Y = /\[(?:[^\\\]]|\\.)*\]|\(\??|\\([1-9][0-9]*)|\\./;
  function se(e, { joinWith: t }) {
    let r = 0;
    return e.map((d) => {
      r += 1;
      const N = r;
      let R = D(d), a = "";
      for (; R.length > 0; ) {
        const o = Y.exec(R);
        if (!o) {
          a += R;
          break;
        }
        a += R.substring(0, o.index), R = R.substring(o.index + o[0].length), o[0][0] === "\\" && o[1] ? a += "\\" + String(Number(o[1]) + N) : (a += o[0], o[0] === "(" && r++);
      }
      return a;
    }).map((d) => `(${d})`).join(t);
  }
  const $ = /\b\B/, Ae = "[a-zA-Z]\\w*", de = "[a-zA-Z_]\\w*", Ne = "\\b\\d+(\\.\\d+)?", Re = "(-?)(\\b0[xX][a-fA-F0-9]+|(\\b\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)", ve = "\\b(0b[01]+)", et = "!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|-|-=|/=|/|:|;|<<|<<=|<=|<|===|==|=|>>>=|>>=|>=|>>>|>>|>|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~", tt = (e = {}) => {
    const t = /^#![ ]*\//;
    return e.binary && (e.begin = j(
      t,
      /.*\b/,
      e.binary,
      /\b.*/
    )), u({
      scope: "meta",
      begin: t,
      end: /$/,
      relevance: 0,
      /** @type {ModeCallback} */
      "on:begin": (r, d) => {
        r.index !== 0 && d.ignoreMatch();
      }
    }, e);
  }, te = {
    begin: "\\\\[\\s\\S]",
    relevance: 0
  }, nt = {
    scope: "string",
    begin: "'",
    end: "'",
    illegal: "\\n",
    contains: [te]
  }, it = {
    scope: "string",
    begin: '"',
    end: '"',
    illegal: "\\n",
    contains: [te]
  }, rt = {
    begin: /\b(a|an|the|are|I'm|isn't|don't|doesn't|won't|but|just|should|pretty|simply|enough|gonna|going|wtf|so|such|will|you|your|they|like|more)\b/
  }, oe = function(e, t, r = {}) {
    const d = u(
      {
        scope: "comment",
        begin: e,
        end: t,
        contains: []
      },
      r
    );
    d.contains.push({
      scope: "doctag",
      // hack to avoid the space from being included. the space is necessary to
      // match here to prevent the plain text rule below from gobbling up doctags
      begin: "[ ]*(?=(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):)",
      end: /(TODO|FIXME|NOTE|BUG|OPTIMIZE|HACK|XXX):/,
      excludeBegin: !0,
      relevance: 0
    });
    const N = ie(
      // list of common 1 and 2 letter words in English
      "I",
      "a",
      "is",
      "so",
      "us",
      "to",
      "at",
      "if",
      "in",
      "it",
      "on",
      // note: this is not an exhaustive list of contractions, just popular ones
      /[A-Za-z]+['](d|ve|re|ll|t|s|n)/,
      // contractions - can't we'd they're let's, etc
      /[A-Za-z]+[-][a-z]+/,
      // `no-way`, etc.
      /[A-Za-z][a-z]{2,}/
      // allow capitalized words at beginning of sentences
    );
    return d.contains.push(
      {
        // TODO: how to include ", (, ) without breaking grammars that use these for
        // comment delimiters?
        // begin: /[ ]+([()"]?([A-Za-z'-]{3,}|is|a|I|so|us|[tT][oO]|at|if|in|it|on)[.]?[()":]?([.][ ]|[ ]|\))){3}/
        // ---
        // this tries to find sequences of 3 english words in a row (without any
        // "programming" type syntax) this gives us a strong signal that we've
        // TRULY found a comment - vs perhaps scanning with the wrong language.
        // It's possible to find something that LOOKS like the start of the
        // comment - but then if there is no readable text - good chance it is a
        // false match and not a comment.
        //
        // for a visual example please see:
        // https://github.com/highlightjs/highlight.js/issues/2827
        begin: j(
          /[ ]+/,
          // necessary to prevent us gobbling up doctags like /* @author Bob Mcgill */
          "(",
          N,
          /[.]?[:]?([.][ ]|[ ])/,
          "){3}"
        )
        // look for 3 words in a row
      }
    ), d;
  }, st = oe("//", "$"), ot = oe("/\\*", "\\*/"), at = oe("#", "$"), ct = {
    scope: "number",
    begin: Ne,
    relevance: 0
  }, lt = {
    scope: "number",
    begin: Re,
    relevance: 0
  }, ut = {
    scope: "number",
    begin: ve,
    relevance: 0
  }, ft = {
    scope: "regexp",
    begin: /\/(?=[^/\n]*\/)/,
    end: /\/[gimuy]*/,
    contains: [
      te,
      {
        begin: /\[/,
        end: /\]/,
        relevance: 0,
        contains: [te]
      }
    ]
  }, gt = {
    scope: "title",
    begin: Ae,
    relevance: 0
  }, ht = {
    scope: "title",
    begin: de,
    relevance: 0
  }, dt = {
    // excludes method names from keyword processing
    begin: "\\.\\s*" + de,
    relevance: 0
  };
  var ae = /* @__PURE__ */ Object.freeze({
    __proto__: null,
    APOS_STRING_MODE: nt,
    BACKSLASH_ESCAPE: te,
    BINARY_NUMBER_MODE: ut,
    BINARY_NUMBER_RE: ve,
    COMMENT: oe,
    C_BLOCK_COMMENT_MODE: ot,
    C_LINE_COMMENT_MODE: st,
    C_NUMBER_MODE: lt,
    C_NUMBER_RE: Re,
    END_SAME_AS_BEGIN: function(e) {
      return Object.assign(
        e,
        {
          /** @type {ModeCallback} */
          "on:begin": (t, r) => {
            r.data._beginMatch = t[1];
          },
          /** @type {ModeCallback} */
          "on:end": (t, r) => {
            r.data._beginMatch !== t[1] && r.ignoreMatch();
          }
        }
      );
    },
    HASH_COMMENT_MODE: at,
    IDENT_RE: Ae,
    MATCH_NOTHING_RE: $,
    METHOD_GUARD: dt,
    NUMBER_MODE: ct,
    NUMBER_RE: Ne,
    PHRASAL_WORDS_MODE: rt,
    QUOTE_STRING_MODE: it,
    REGEXP_MODE: ft,
    RE_STARTERS_RE: et,
    SHEBANG: tt,
    TITLE_MODE: gt,
    UNDERSCORE_IDENT_RE: de,
    UNDERSCORE_TITLE_MODE: ht
  });
  function pt(e, t) {
    e.input[e.index - 1] === "." && t.ignoreMatch();
  }
  function bt(e, t) {
    e.className !== void 0 && (e.scope = e.className, delete e.className);
  }
  function Et(e, t) {
    t && e.beginKeywords && (e.begin = "\\b(" + e.beginKeywords.split(" ").join("|") + ")(?!\\.)(?=\\b|\\s)", e.__beforeBegin = pt, e.keywords = e.keywords || e.beginKeywords, delete e.beginKeywords, e.relevance === void 0 && (e.relevance = 0));
  }
  function _t(e, t) {
    Array.isArray(e.illegal) && (e.illegal = ie(...e.illegal));
  }
  function xt(e, t) {
    if (e.match) {
      if (e.begin || e.end) throw new Error("begin & end are not supported with match");
      e.begin = e.match, delete e.match;
    }
  }
  function wt(e, t) {
    e.relevance === void 0 && (e.relevance = 1);
  }
  const Mt = (e, t) => {
    if (!e.beforeMatch) return;
    if (e.starts) throw new Error("beforeMatch cannot be used with starts");
    const r = Object.assign({}, e);
    Object.keys(e).forEach((d) => {
      delete e[d];
    }), e.keywords = r.keywords, e.begin = j(r.beforeMatch, m(r.begin)), e.starts = {
      relevance: 0,
      contains: [
        Object.assign(r, { endsParent: !0 })
      ]
    }, e.relevance = 0, delete r.beforeMatch;
  }, yt = [
    "of",
    "and",
    "for",
    "in",
    "not",
    "or",
    "if",
    "then",
    "parent",
    // common variable name
    "list",
    // common variable name
    "value"
    // common variable name
  ], St = "keyword";
  function Te(e, t, r = St) {
    const d = /* @__PURE__ */ Object.create(null);
    return typeof e == "string" ? N(r, e.split(" ")) : Array.isArray(e) ? N(r, e) : Object.keys(e).forEach(function(R) {
      Object.assign(
        d,
        Te(e[R], t, R)
      );
    }), d;
    function N(R, a) {
      t && (a = a.map((o) => o.toLowerCase())), a.forEach(function(o) {
        const g = o.split("|");
        d[g[0]] = [R, Ot(g[0], g[1])];
      });
    }
  }
  function Ot(e, t) {
    return t ? Number(t) : At(e) ? 0 : 1;
  }
  function At(e) {
    return yt.includes(e.toLowerCase());
  }
  const ke = {}, Z = (e) => {
    console.error(e);
  }, me = (e, ...t) => {
    console.log(`WARN: ${e}`, ...t);
  }, V = (e, t) => {
    ke[`${e}/${t}`] || (console.log(`Deprecated as of ${e}. ${t}`), ke[`${e}/${t}`] = !0);
  }, ce = new Error();
  function Ce(e, t, { key: r }) {
    let d = 0;
    const N = e[r], R = {}, a = {};
    for (let o = 1; o <= t.length; o++)
      a[o + d] = N[o], R[o + d] = !0, d += re(t[o - 1]);
    e[r] = a, e[r]._emit = R, e[r]._multi = !0;
  }
  function Nt(e) {
    if (Array.isArray(e.begin)) {
      if (e.skip || e.excludeBegin || e.returnBegin)
        throw Z("skip, excludeBegin, returnBegin not compatible with beginScope: {}"), ce;
      if (typeof e.beginScope != "object" || e.beginScope === null)
        throw Z("beginScope must be object"), ce;
      Ce(e, e.begin, { key: "beginScope" }), e.begin = se(e.begin, { joinWith: "" });
    }
  }
  function Rt(e) {
    if (Array.isArray(e.end)) {
      if (e.skip || e.excludeEnd || e.returnEnd)
        throw Z("skip, excludeEnd, returnEnd not compatible with endScope: {}"), ce;
      if (typeof e.endScope != "object" || e.endScope === null)
        throw Z("endScope must be object"), ce;
      Ce(e, e.end, { key: "endScope" }), e.end = se(e.end, { joinWith: "" });
    }
  }
  function vt(e) {
    e.scope && typeof e.scope == "object" && e.scope !== null && (e.beginScope = e.scope, delete e.scope);
  }
  function Tt(e) {
    vt(e), typeof e.beginScope == "string" && (e.beginScope = { _wrap: e.beginScope }), typeof e.endScope == "string" && (e.endScope = { _wrap: e.endScope }), Nt(e), Rt(e);
  }
  function kt(e) {
    function t(a, o) {
      return new RegExp(
        D(a),
        "m" + (e.case_insensitive ? "i" : "") + (e.unicodeRegex ? "u" : "") + (o ? "g" : "")
      );
    }
    class r {
      constructor() {
        this.matchIndexes = {}, this.regexes = [], this.matchAt = 1, this.position = 0;
      }
      // @ts-ignore
      addRule(o, g) {
        g.position = this.position++, this.matchIndexes[this.matchAt] = g, this.regexes.push([g, o]), this.matchAt += re(o) + 1;
      }
      compile() {
        this.regexes.length === 0 && (this.exec = () => null);
        const o = this.regexes.map((g) => g[1]);
        this.matcherRe = t(se(o, { joinWith: "|" }), !0), this.lastIndex = 0;
      }
      /** @param {string} s */
      exec(o) {
        this.matcherRe.lastIndex = this.lastIndex;
        const g = this.matcherRe.exec(o);
        if (!g)
          return null;
        const B = g.findIndex((ne, be) => be > 0 && ne !== void 0), T = this.matchIndexes[B];
        return g.splice(0, B), Object.assign(g, T);
      }
    }
    class d {
      constructor() {
        this.rules = [], this.multiRegexes = [], this.count = 0, this.lastIndex = 0, this.regexIndex = 0;
      }
      // @ts-ignore
      getMatcher(o) {
        if (this.multiRegexes[o]) return this.multiRegexes[o];
        const g = new r();
        return this.rules.slice(o).forEach(([B, T]) => g.addRule(B, T)), g.compile(), this.multiRegexes[o] = g, g;
      }
      resumingScanAtSamePosition() {
        return this.regexIndex !== 0;
      }
      considerAll() {
        this.regexIndex = 0;
      }
      // @ts-ignore
      addRule(o, g) {
        this.rules.push([o, g]), g.type === "begin" && this.count++;
      }
      /** @param {string} s */
      exec(o) {
        const g = this.getMatcher(this.regexIndex);
        g.lastIndex = this.lastIndex;
        let B = g.exec(o);
        if (this.resumingScanAtSamePosition() && !(B && B.index === this.lastIndex)) {
          const T = this.getMatcher(0);
          T.lastIndex = this.lastIndex + 1, B = T.exec(o);
        }
        return B && (this.regexIndex += B.position + 1, this.regexIndex === this.count && this.considerAll()), B;
      }
    }
    function N(a) {
      const o = new d();
      return a.contains.forEach((g) => o.addRule(g.begin, { rule: g, type: "begin" })), a.terminatorEnd && o.addRule(a.terminatorEnd, { type: "end" }), a.illegal && o.addRule(a.illegal, { type: "illegal" }), o;
    }
    function R(a, o) {
      const g = (
        /** @type CompiledMode */
        a
      );
      if (a.isCompiled) return g;
      [
        bt,
        // do this early so compiler extensions generally don't have to worry about
        // the distinction between match/begin
        xt,
        Tt,
        Mt
      ].forEach((T) => T(a, o)), e.compilerExtensions.forEach((T) => T(a, o)), a.__beforeBegin = null, [
        Et,
        // do this later so compiler extensions that come earlier have access to the
        // raw array if they wanted to perhaps manipulate it, etc.
        _t,
        // default to 1 relevance if not specified
        wt
      ].forEach((T) => T(a, o)), a.isCompiled = !0;
      let B = null;
      return typeof a.keywords == "object" && a.keywords.$pattern && (a.keywords = Object.assign({}, a.keywords), B = a.keywords.$pattern, delete a.keywords.$pattern), B = B || /\w+/, a.keywords && (a.keywords = Te(a.keywords, e.case_insensitive)), g.keywordPatternRe = t(B, !0), o && (a.begin || (a.begin = /\B|\b/), g.beginRe = t(g.begin), !a.end && !a.endsWithParent && (a.end = /\B|\b/), a.end && (g.endRe = t(g.end)), g.terminatorEnd = D(g.end) || "", a.endsWithParent && o.terminatorEnd && (g.terminatorEnd += (a.end ? "|" : "") + o.terminatorEnd)), a.illegal && (g.illegalRe = t(
        /** @type {RegExp | string} */
        a.illegal
      )), a.contains || (a.contains = []), a.contains = [].concat(...a.contains.map(function(T) {
        return mt(T === "self" ? a : T);
      })), a.contains.forEach(function(T) {
        R(
          /** @type Mode */
          T,
          g
        );
      }), a.starts && R(a.starts, o), g.matcher = N(g), g;
    }
    if (e.compilerExtensions || (e.compilerExtensions = []), e.contains && e.contains.includes("self"))
      throw new Error("ERR: contains `self` is not supported at the top-level of a language.  See documentation.");
    return e.classNameAliases = u(e.classNameAliases || {}), R(
      /** @type Mode */
      e
    );
  }
  function Be(e) {
    return e ? e.endsWithParent || Be(e.starts) : !1;
  }
  function mt(e) {
    return e.variants && !e.cachedVariants && (e.cachedVariants = e.variants.map(function(t) {
      return u(e, { variants: null }, t);
    })), e.cachedVariants ? e.cachedVariants : Be(e) ? u(e, { starts: e.starts ? u(e.starts) : null }) : Object.isFrozen(e) ? u(e) : e;
  }
  var Ct = "11.11.1";
  class Bt extends Error {
    constructor(t, r) {
      super(t), this.name = "HTMLInjectionError", this.html = r;
    }
  }
  const pe = h, Ie = u, Le = Symbol("nomatch"), It = 7, De = function(e) {
    const t = /* @__PURE__ */ Object.create(null), r = /* @__PURE__ */ Object.create(null), d = [];
    let N = !0;
    const R = "Could not find the language '{}', did you forget to load/include a language module?", a = { disableAutodetect: !0, name: "Plain text", contains: [] };
    let o = {
      ignoreUnescapedHTML: !1,
      throwUnescapedHTML: !1,
      noHighlightRe: /^(no-?highlight)$/i,
      languageDetectRe: /\blang(?:uage)?-([\w-]+)\b/i,
      classPrefix: "hljs-",
      cssSelector: "pre code",
      languages: null,
      // beta configuration options, subject to change, welcome to discuss
      // https://github.com/highlightjs/highlight.js/issues/1086
      __emitter: P
    };
    function g(n) {
      return o.noHighlightRe.test(n);
    }
    function B(n) {
      let l = n.className + " ";
      l += n.parentNode ? n.parentNode.className : "";
      const _ = o.languageDetectRe.exec(l);
      if (_) {
        const S = F(_[1]);
        return S || (me(R.replace("{}", _[1])), me("Falling back to no-highlight mode for this block.", n)), S ? _[1] : "no-highlight";
      }
      return l.split(/\s+/).find((S) => g(S) || F(S));
    }
    function T(n, l, _) {
      let S = "", k = "";
      typeof l == "object" ? (S = n, _ = l.ignoreIllegals, k = l.language) : (V("10.7.0", "highlight(lang, code, ...args) has been deprecated."), V("10.7.0", `Please use highlight(code, options) instead.
https://github.com/highlightjs/highlight.js/issues/2277`), k = n, S = l), _ === void 0 && (_ = !0);
      const U = {
        code: S,
        language: k
      };
      ue("before:highlight", U);
      const X = U.result ? U.result : ne(U.language, U.code, _);
      return X.code = U.code, ue("after:highlight", X), X;
    }
    function ne(n, l, _, S) {
      const k = /* @__PURE__ */ Object.create(null);
      function U(s, c) {
        return s.keywords[c];
      }
      function X() {
        if (!p.keywords) {
          I.addText(O);
          return;
        }
        let s = 0;
        p.keywordPatternRe.lastIndex = 0;
        let c = p.keywordPatternRe.exec(O), b = "";
        for (; c; ) {
          b += O.substring(s, c.index);
          const w = G.case_insensitive ? c[0].toLowerCase() : c[0], L = U(p, w);
          if (L) {
            const [K, Vt] = L;
            if (I.addText(b), b = "", k[w] = (k[w] || 0) + 1, k[w] <= It && (he += Vt), K.startsWith("_"))
              b += c[0];
            else {
              const qt = G.classNameAliases[K] || K;
              z(c[0], qt);
            }
          } else
            b += c[0];
          s = p.keywordPatternRe.lastIndex, c = p.keywordPatternRe.exec(O);
        }
        b += O.substring(s), I.addText(b);
      }
      function fe() {
        if (O === "") return;
        let s = null;
        if (typeof p.subLanguage == "string") {
          if (!t[p.subLanguage]) {
            I.addText(O);
            return;
          }
          s = ne(p.subLanguage, O, !0, Ge[p.subLanguage]), Ge[p.subLanguage] = /** @type {CompiledMode} */
          s._top;
        } else
          s = Ee(O, p.subLanguage.length ? p.subLanguage : null);
        p.relevance > 0 && (he += s.relevance), I.__addSublanguage(s._emitter, s.language);
      }
      function H() {
        p.subLanguage != null ? fe() : X(), O = "";
      }
      function z(s, c) {
        s !== "" && (I.startScope(c), I.addText(s), I.endScope());
      }
      function $e(s, c) {
        let b = 1;
        const w = c.length - 1;
        for (; b <= w; ) {
          if (!s._emit[b]) {
            b++;
            continue;
          }
          const L = G.classNameAliases[s[b]] || s[b], K = c[b];
          L ? z(K, L) : (O = K, X(), O = ""), b++;
        }
      }
      function Ue(s, c) {
        return s.scope && typeof s.scope == "string" && I.openNode(G.classNameAliases[s.scope] || s.scope), s.beginScope && (s.beginScope._wrap ? (z(O, G.classNameAliases[s.beginScope._wrap] || s.beginScope._wrap), O = "") : s.beginScope._multi && ($e(s.beginScope, c), O = "")), p = Object.create(s, { parent: { value: p } }), p;
      }
      function We(s, c, b) {
        let w = ee(s.endRe, b);
        if (w) {
          if (s["on:end"]) {
            const L = new f(s);
            s["on:end"](c, L), L.isMatchIgnored && (w = !1);
          }
          if (w) {
            for (; s.endsParent && s.parent; )
              s = s.parent;
            return s;
          }
        }
        if (s.endsWithParent)
          return We(s.parent, c, b);
      }
      function Ft(s) {
        return p.matcher.regexIndex === 0 ? (O += s[0], 1) : (Me = !0, 0);
      }
      function Xt(s) {
        const c = s[0], b = s.rule, w = new f(b), L = [b.__beforeBegin, b["on:begin"]];
        for (const K of L)
          if (K && (K(s, w), w.isMatchIgnored))
            return Ft(c);
        return b.skip ? O += c : (b.excludeBegin && (O += c), H(), !b.returnBegin && !b.excludeBegin && (O = c)), Ue(b, s), b.returnBegin ? 0 : c.length;
      }
      function Yt(s) {
        const c = s[0], b = l.substring(s.index), w = We(p, s, b);
        if (!w)
          return Le;
        const L = p;
        p.endScope && p.endScope._wrap ? (H(), z(c, p.endScope._wrap)) : p.endScope && p.endScope._multi ? (H(), $e(p.endScope, s)) : L.skip ? O += c : (L.returnEnd || L.excludeEnd || (O += c), H(), L.excludeEnd && (O = c));
        do
          p.scope && I.closeNode(), !p.skip && !p.subLanguage && (he += p.relevance), p = p.parent;
        while (p !== w.parent);
        return w.starts && Ue(w.starts, s), L.returnEnd ? 0 : c.length;
      }
      function Zt() {
        const s = [];
        for (let c = p; c !== G; c = c.parent)
          c.scope && s.unshift(c.scope);
        s.forEach((c) => I.openNode(c));
      }
      let ge = {};
      function ze(s, c) {
        const b = c && c[0];
        if (O += s, b == null)
          return H(), 0;
        if (ge.type === "begin" && c.type === "end" && ge.index === c.index && b === "") {
          if (O += l.slice(c.index, c.index + 1), !N) {
            const w = new Error(`0 width match regex (${n})`);
            throw w.languageName = n, w.badRule = ge.rule, w;
          }
          return 1;
        }
        if (ge = c, c.type === "begin")
          return Xt(c);
        if (c.type === "illegal" && !_) {
          const w = new Error('Illegal lexeme "' + b + '" for mode "' + (p.scope || "<unnamed>") + '"');
          throw w.mode = p, w;
        } else if (c.type === "end") {
          const w = Yt(c);
          if (w !== Le)
            return w;
        }
        if (c.type === "illegal" && b === "")
          return O += `
`, 1;
        if (we > 1e5 && we > c.index * 3)
          throw new Error("potential infinite loop, way more iterations than matches");
        return O += b, b.length;
      }
      const G = F(n);
      if (!G)
        throw Z(R.replace("{}", n)), new Error('Unknown language: "' + n + '"');
      const Jt = kt(G);
      let xe = "", p = S || Jt;
      const Ge = {}, I = new o.__emitter(o);
      Zt();
      let O = "", he = 0, J = 0, we = 0, Me = !1;
      try {
        if (G.__emitTokens)
          G.__emitTokens(l, I);
        else {
          for (p.matcher.considerAll(); ; ) {
            we++, Me ? Me = !1 : p.matcher.considerAll(), p.matcher.lastIndex = J;
            const s = p.matcher.exec(l);
            if (!s) break;
            const c = l.substring(J, s.index), b = ze(c, s);
            J = s.index + b;
          }
          ze(l.substring(J));
        }
        return I.finalize(), xe = I.toHTML(), {
          language: n,
          value: xe,
          relevance: he,
          illegal: !1,
          _emitter: I,
          _top: p
        };
      } catch (s) {
        if (s.message && s.message.includes("Illegal"))
          return {
            language: n,
            value: pe(l),
            illegal: !0,
            relevance: 0,
            _illegalBy: {
              message: s.message,
              index: J,
              context: l.slice(J - 100, J + 100),
              mode: s.mode,
              resultSoFar: xe
            },
            _emitter: I
          };
        if (N)
          return {
            language: n,
            value: pe(l),
            illegal: !1,
            relevance: 0,
            errorRaised: s,
            _emitter: I,
            _top: p
          };
        throw s;
      }
    }
    function be(n) {
      const l = {
        value: pe(n),
        illegal: !1,
        relevance: 0,
        _top: a,
        _emitter: new o.__emitter(o)
      };
      return l._emitter.addText(n), l;
    }
    function Ee(n, l) {
      l = l || o.languages || Object.keys(t);
      const _ = be(n), S = l.filter(F).filter(je).map(
        (H) => ne(H, n, !1)
      );
      S.unshift(_);
      const k = S.sort((H, z) => {
        if (H.relevance !== z.relevance) return z.relevance - H.relevance;
        if (H.language && z.language) {
          if (F(H.language).supersetOf === z.language)
            return 1;
          if (F(z.language).supersetOf === H.language)
            return -1;
        }
        return 0;
      }), [U, X] = k, fe = U;
      return fe.secondBest = X, fe;
    }
    function Lt(n, l, _) {
      const S = l && r[l] || _;
      n.classList.add("hljs"), n.classList.add(`language-${S}`);
    }
    function _e(n) {
      let l = null;
      const _ = B(n);
      if (g(_)) return;
      if (ue(
        "before:highlightElement",
        { el: n, language: _ }
      ), n.dataset.highlighted) {
        console.log("Element previously highlighted. To highlight again, first unset `dataset.highlighted`.", n);
        return;
      }
      if (n.children.length > 0 && (o.ignoreUnescapedHTML || (console.warn("One of your code blocks includes unescaped HTML. This is a potentially serious security risk."), console.warn("https://github.com/highlightjs/highlight.js/wiki/security"), console.warn("The element with unescaped HTML:"), console.warn(n)), o.throwUnescapedHTML))
        throw new Bt(
          "One of your code blocks includes unescaped HTML.",
          n.innerHTML
        );
      l = n;
      const S = l.textContent, k = _ ? T(S, { language: _, ignoreIllegals: !0 }) : Ee(S);
      n.innerHTML = k.value, n.dataset.highlighted = "yes", Lt(n, _, k.language), n.result = {
        language: k.language,
        // TODO: remove with version 11.0
        re: k.relevance,
        relevance: k.relevance
      }, k.secondBest && (n.secondBest = {
        language: k.secondBest.language,
        relevance: k.secondBest.relevance
      }), ue("after:highlightElement", { el: n, result: k, text: S });
    }
    function Dt(n) {
      o = Ie(o, n);
    }
    const Pt = () => {
      le(), V("10.6.0", "initHighlighting() deprecated.  Use highlightAll() now.");
    };
    function Ht() {
      le(), V("10.6.0", "initHighlightingOnLoad() deprecated.  Use highlightAll() now.");
    }
    let Pe = !1;
    function le() {
      function n() {
        le();
      }
      if (document.readyState === "loading") {
        Pe || window.addEventListener("DOMContentLoaded", n, !1), Pe = !0;
        return;
      }
      document.querySelectorAll(o.cssSelector).forEach(_e);
    }
    function jt(n, l) {
      let _ = null;
      try {
        _ = l(e);
      } catch (S) {
        if (Z("Language definition for '{}' could not be registered.".replace("{}", n)), N)
          Z(S);
        else
          throw S;
        _ = a;
      }
      _.name || (_.name = n), t[n] = _, _.rawDefinition = l.bind(null, e), _.aliases && He(_.aliases, { languageName: n });
    }
    function $t(n) {
      delete t[n];
      for (const l of Object.keys(r))
        r[l] === n && delete r[l];
    }
    function Ut() {
      return Object.keys(t);
    }
    function F(n) {
      return n = (n || "").toLowerCase(), t[n] || t[r[n]];
    }
    function He(n, { languageName: l }) {
      typeof n == "string" && (n = [n]), n.forEach((_) => {
        r[_.toLowerCase()] = l;
      });
    }
    function je(n) {
      const l = F(n);
      return l && !l.disableAutodetect;
    }
    function Wt(n) {
      n["before:highlightBlock"] && !n["before:highlightElement"] && (n["before:highlightElement"] = (l) => {
        n["before:highlightBlock"](
          Object.assign({ block: l.el }, l)
        );
      }), n["after:highlightBlock"] && !n["after:highlightElement"] && (n["after:highlightElement"] = (l) => {
        n["after:highlightBlock"](
          Object.assign({ block: l.el }, l)
        );
      });
    }
    function zt(n) {
      Wt(n), d.push(n);
    }
    function Gt(n) {
      const l = d.indexOf(n);
      l !== -1 && d.splice(l, 1);
    }
    function ue(n, l) {
      const _ = n;
      d.forEach(function(S) {
        S[_] && S[_](l);
      });
    }
    function Kt(n) {
      return V("10.7.0", "highlightBlock will be removed entirely in v12.0"), V("10.7.0", "Please use highlightElement now."), _e(n);
    }
    Object.assign(e, {
      highlight: T,
      highlightAuto: Ee,
      highlightAll: le,
      highlightElement: _e,
      // TODO: Remove with v12 API
      highlightBlock: Kt,
      configure: Dt,
      initHighlighting: Pt,
      initHighlightingOnLoad: Ht,
      registerLanguage: jt,
      unregisterLanguage: $t,
      listLanguages: Ut,
      getLanguage: F,
      registerAliases: He,
      autoDetection: je,
      inherit: Ie,
      addPlugin: zt,
      removePlugin: Gt
    }), e.debugMode = function() {
      N = !1;
    }, e.safeMode = function() {
      N = !0;
    }, e.versionString = Ct, e.regex = {
      concat: j,
      lookahead: m,
      either: ie,
      optional: W,
      anyNumberOfTimes: C
    };
    for (const n in ae)
      typeof ae[n] == "object" && i(ae[n]);
    return Object.assign(e, ae), e;
  }, q = De({});
  return q.newInstance = () => De({}), Se = q, q.HighlightJS = q, q.default = q, Se;
}
var dn = /* @__PURE__ */ hn();
const pn = /* @__PURE__ */ sn(dn);
function Qe(i, f = []) {
  return i.flatMap((h) => {
    const u = [...f, ...h.properties ? h.properties.className : []];
    return h.children ? Qe(h.children, u) : {
      text: h.value,
      classes: u
    };
  });
}
function Ze(i) {
  return i.value || i.children || [];
}
function bn(i) {
  return !!pn.getLanguage(i);
}
function Je({
  doc: i,
  name: f,
  lowlight: h,
  defaultLanguage: u
}) {
  const E = [];
  return Oe(i, (x) => x.type.name === f).forEach((x) => {
    var M;
    let y = x.pos + 1;
    const A = x.node.attrs.language || u, v = h.listLanguages(), P = A && (v.includes(A) || bn(A) || (M = h.registered) != null && M.call(h, A)) ? Ze(h.highlight(A, x.node.textContent)) : Ze(h.highlightAuto(x.node.textContent));
    Qe(P).forEach((D) => {
      const m = y + D.text.length;
      if (D.classes.length) {
        const C = nn.inline(y, m, {
          class: D.classes.join(" ")
        });
        E.push(C);
      }
      y = m;
    });
  }), rn.create(i, E);
}
function En(i) {
  return typeof i == "function";
}
function _n({
  name: i,
  lowlight: f,
  defaultLanguage: h
}) {
  if (!["highlight", "highlightAuto", "listLanguages"].every((E) => En(f[E])))
    throw Error(
      "You should provide an instance of lowlight to use the code-block-lowlight extension"
    );
  const u = new Ve({
    key: new qe("lowlight"),
    state: {
      init: (E, { doc: x }) => Je({
        doc: x,
        name: i,
        lowlight: f,
        defaultLanguage: h
      }),
      apply: (E, x, M, y) => {
        const A = M.selection.$head.parent.type.name, v = y.selection.$head.parent.type.name, P = Oe(M.doc, (m) => m.type.name === i), D = Oe(y.doc, (m) => m.type.name === i);
        return E.docChanged && // Apply decorations if:
        // selection includes named node,
        ([A, v].includes(i) || // OR transaction adds/removes named node,
        D.length !== P.length || // OR transaction has changes that completely encapsulte a node
        // (for example, a transaction that affects the entire document).
        // Such transactions can happen during collab syncing via y-prosemirror, for example.
        E.steps.some((m) => (
          // @ts-ignore
          m.from !== void 0 && // @ts-ignore
          m.to !== void 0 && P.some((C) => (
            // @ts-ignore
            C.pos >= m.from && // @ts-ignore
            C.pos + C.node.nodeSize <= m.to
          ))
        ))) ? Je({
          doc: E.doc,
          name: i,
          lowlight: f,
          defaultLanguage: h
        }) : x.map(E.mapping, E.doc);
      }
    },
    props: {
      decorations(E) {
        return u.getState(E);
      }
    }
  });
  return u;
}
var xn = gn.extend({
  addOptions() {
    var i;
    return {
      ...(i = this.parent) == null ? void 0 : i.call(this),
      lowlight: {},
      languageClassPrefix: "language-",
      exitOnTripleEnter: !0,
      exitOnArrowDown: !0,
      defaultLanguage: null,
      enableTabIndentation: !1,
      tabSize: 4,
      HTMLAttributes: {}
    };
  },
  addProseMirrorPlugins() {
    var i;
    return [
      ...((i = this.parent) == null ? void 0 : i.call(this)) || [],
      _n({
        name: this.name,
        lowlight: this.options.lowlight,
        defaultLanguage: this.options.defaultLanguage
      })
    ];
  }
}), wn = xn;
function mn() {
  const i = an(Mn.name), {
    icon: f = void 0,
    tooltip: h = void 0,
    tooltipOptions: u = {},
    action: E = void 0,
    isActive: x = void 0
  } = (i == null ? void 0 : i.componentProps) ?? {}, { dataState: M, disabled: y, update: A } = cn(x), v = () => {
    y || E && (E(), A());
  };
  return i ? /* @__PURE__ */ Xe(
    ln,
    {
      action: v,
      dataState: M,
      disabled: y,
      icon: f,
      tooltip: h,
      tooltipOptions: u
    }
  ) : /* @__PURE__ */ Xe(on, {});
}
const Mn = wn.extend({
  //@ts-expect-error
  addOptions() {
    var i;
    return {
      ...(i = this.parent) == null ? void 0 : i.call(this),
      button: ({ editor: f, t: h }) => ({
        componentProps: {
          action: () => f.chain().focus().setCodeBlock({ language: "plaintext" }).run(),
          isActive: () => f.isActive("codeBlock"),
          disabled: !1,
          icon: "Code2",
          tooltip: h("editor.codeblock.tooltip")
        }
      })
    };
  }
});
export {
  Mn as CodeBlock,
  mn as RichTextCodeBlock
};
