// reactjs-tiptap-editor@1.0.2/lib/FormatPainter.js
import { E as M } from "./index-Dx-Dh7KS.js";
import { PluginKey as k, Plugin as h } from "@tiptap/pm/state";
import { h as S, j as v, u as x, A as d } from "./index-C52cuQ_Y.js";
import { jsx as p, Fragment as A } from "react/jsx-runtime";
import { useState as K, useEffect as b } from "react";
import "./theme.js";
function T() {
  const t = S(), e = v(), n = x(j.name), {
    icon: r = void 0,
    tooltip: s = void 0,
    shortcutKeys: o = void 0,
    tooltipOptions: y = {},
    action: c = void 0,
    isActive: u = void 0
  } = (n == null ? void 0 : n.componentProps) ?? {}, [P, g] = K(() => !!(u != null && u()));
  b(() => {
    if (!t || !u) return;
    const i = () => {
      g(!!u());
    };
    return i(), t.on("selectionUpdate", i), t.on("transaction", i), () => {
      t.off("selectionUpdate", i), t.off("transaction", i);
    };
  }, [t, u]);
  const l = !e || !t, E = () => {
    l || c == null || c();
  };
  return n ? /* @__PURE__ */ p(
    d,
    {
      action: E,
      dataState: P,
      disabled: l,
      icon: r,
      shortcutKeys: o,
      tooltip: s,
      tooltipOptions: y
    }
  ) : /* @__PURE__ */ p(A, {});
}
const f = /* @__PURE__ */ new Set(["link"]), a = new k("format-painter");
function m(t) {
  return a.getState(t) ?? [];
}
function F(t) {
  const { selection: e, storedMarks: n } = t, r = [];
  return n != null && n.length || e.empty ? ((n ?? e.$from.marks()).forEach((o) => {
    f.has(o.type.name) || r.push(o);
  }), r) : (t.doc.nodesBetween(e.from, e.to, (s) => !s.isText || r.length > 0 ? !0 : (s.marks.forEach((o) => {
    f.has(o.type.name) || r.push(o);
  }), !1)), r);
}
function O(t, e, n) {
  const { from: r, to: s } = e.selection;
  return Object.values(e.schema.marks).forEach((o) => {
    f.has(o.name) || (t = t.removeMark(r, s, o));
  }), n.forEach((o) => {
    t = t.addMark(r, s, o);
  }), t;
}
const j = /* @__PURE__ */ M.create({
  name: "painter",
  //@ts-expect-error
  addOptions() {
    var t;
    return {
      ...(t = this.parent) == null ? void 0 : t.call(this),
      button: ({ editor: e, t: n }) => ({
        component: d,
        componentProps: {
          action: () => {
            if (m(e.state).length > 0) {
              e.commands.unsetPainter();
              return;
            }
            e.commands.setPainter();
          },
          isActive: () => m(e.state).length > 0,
          icon: "PaintRoller",
          tooltip: n("editor.format")
        }
      })
    };
  },
  addCommands() {
    return {
      setPainter: () => ({ dispatch: t, state: e }) => {
        const n = F(e);
        return n.length ? (t == null || t(
          e.tr.setMeta(a, {
            type: "start",
            marks: n
          })
        ), !0) : (t == null || t(e.tr.setMeta(a, { type: "end" })), !1);
      },
      unsetPainter: () => ({ dispatch: t, state: e }) => (t == null || t(e.tr.setMeta(a, { type: "end" })), !0)
    };
  },
  addProseMirrorPlugins() {
    return [
      new h({
        key: a,
        state: {
          init: () => [],
          apply: (t, e) => {
            const n = t.getMeta(a);
            return (n == null ? void 0 : n.type) === "start" ? n.marks : (n == null ? void 0 : n.type) === "end" ? [] : e;
          }
        },
        props: {
          handleDOMEvents: {
            keydown: (t, e) => e.key !== "Escape" || m(t.state).length === 0 ? !1 : (t.dispatch(t.state.tr.setMeta(a, { type: "end" })), !0),
            mousedown: (t) => {
              const e = m(t.state);
              if (!e.length) return !1;
              const n = () => {
                document.removeEventListener("mouseup", n);
                const { state: r } = t, { selection: s } = r, o = s.empty ? r.tr : O(r.tr, r, e);
                t.dispatch(o.setMeta(a, { type: "end" }));
              };
              return document.addEventListener("mouseup", n), !1;
            }
          }
        }
      })
    ];
  }
});
export {
  j as FormatPainter,
  T as RichTextFormatPainter,
  a as formatPainterPluginKey
};
