// reactjs-tiptap-editor@1.0.2/lib/History.js
import { O as L, x as ce, F as _, v as U, T as Y, u as Z, S as x, Q as J, R as de, E as P, U as ue, V as pe, W as fe, o as he, X as B } from "./index-Dx-Dh7KS.js";
import { Plugin as A, PluginKey as S } from "@tiptap/pm/state";
import { DecorationSet as O, Decoration as V } from "@tiptap/pm/view";
import { k as me, D as ge, a as ve } from "./index-DQy9iBzV.js";
import { jsx as R, Fragment as Q } from "react/jsx-runtime";
import { u as $, e as q, A as ee, i as we } from "./index-C52cuQ_Y.js";
import "react";
import "./theme.js";
function ye(n = {}) {
  return new L({
    view(e) {
      return new Ce(e, n);
    }
  });
}
class Ce {
  constructor(e, t) {
    var o;
    this.editorView = e, this.cursorPos = null, this.element = null, this.timeout = -1, this.width = (o = t.width) !== null && o !== void 0 ? o : 1, this.color = t.color === !1 ? void 0 : t.color || "black", this.class = t.class, this.handlers = ["dragover", "dragend", "drop", "dragleave"].map((r) => {
      let i = (s) => {
        this[r](s);
      };
      return e.dom.addEventListener(r, i), { name: r, handler: i };
    });
  }
  destroy() {
    this.handlers.forEach(({ name: e, handler: t }) => this.editorView.dom.removeEventListener(e, t));
  }
  update(e, t) {
    this.cursorPos != null && t.doc != e.state.doc && (this.cursorPos > e.state.doc.content.size ? this.setCursor(null) : this.updateOverlay());
  }
  setCursor(e) {
    e != this.cursorPos && (this.cursorPos = e, e == null ? (this.element.parentNode.removeChild(this.element), this.element = null) : this.updateOverlay());
  }
  updateOverlay() {
    let e = this.editorView.state.doc.resolve(this.cursorPos), t = !e.parent.inlineContent, o, r = this.editorView.dom, i = r.getBoundingClientRect(), s = i.width / r.offsetWidth, l = i.height / r.offsetHeight;
    if (t) {
      let u = e.nodeBefore, p = e.nodeAfter;
      if (u || p) {
        let f = this.editorView.nodeDOM(this.cursorPos - (u ? u.nodeSize : 0));
        if (f) {
          let h = f.getBoundingClientRect(), v = u ? h.bottom : h.top;
          u && p && (v = (v + this.editorView.nodeDOM(this.cursorPos).getBoundingClientRect().top) / 2);
          let y = this.width / 2 * l;
          o = { left: h.left, right: h.right, top: v - y, bottom: v + y };
        }
      }
    }
    if (!o) {
      let u = this.editorView.coordsAtPos(this.cursorPos), p = this.width / 2 * s;
      o = { left: u.left - p, right: u.left + p, top: u.top, bottom: u.bottom };
    }
    let a = this.editorView.dom.offsetParent;
    this.element || (this.element = a.appendChild(document.createElement("div")), this.class && (this.element.className = this.class), this.element.style.cssText = "position: absolute; z-index: 50; pointer-events: none;", this.color && (this.element.style.backgroundColor = this.color)), this.element.classList.toggle("prosemirror-dropcursor-block", t), this.element.classList.toggle("prosemirror-dropcursor-inline", !t);
    let c, d;
    if (!a || a == document.body && getComputedStyle(a).position == "static")
      c = -pageXOffset, d = -pageYOffset;
    else {
      let u = a.getBoundingClientRect(), p = u.width / a.offsetWidth, f = u.height / a.offsetHeight;
      c = u.left - a.scrollLeft * p, d = u.top - a.scrollTop * f;
    }
    this.element.style.left = (o.left - c) / s + "px", this.element.style.top = (o.top - d) / l + "px", this.element.style.width = (o.right - o.left) / s + "px", this.element.style.height = (o.bottom - o.top) / l + "px";
  }
  scheduleRemoval(e) {
    clearTimeout(this.timeout), this.timeout = setTimeout(() => this.setCursor(null), e);
  }
  dragover(e) {
    if (!this.editorView.editable)
      return;
    let t = this.editorView.posAtCoords({ left: e.clientX, top: e.clientY }), o = t && t.inside >= 0 && this.editorView.state.doc.nodeAt(t.inside), r = o && o.type.spec.disableDropCursor, i = typeof r == "function" ? r(this.editorView, t, e) : r;
    if (t && !i) {
      let s = t.pos;
      if (this.editorView.dragging && this.editorView.dragging.slice) {
        let l = ce(this.editorView.state.doc, s, this.editorView.dragging.slice);
        l != null && (s = l);
      }
      this.setCursor(s), this.scheduleRemoval(5e3);
    }
  }
  dragend() {
    this.scheduleRemoval(20);
  }
  drop() {
    this.scheduleRemoval(20);
  }
  dragleave(e) {
    this.editorView.dom.contains(e.relatedTarget) || this.setCursor(null);
  }
}
class m extends x {
  /**
  Create a gap cursor.
  */
  constructor(e) {
    super(e, e);
  }
  map(e, t) {
    let o = e.resolve(t.map(this.head));
    return m.valid(o) ? new m(o) : x.near(o);
  }
  content() {
    return U.empty;
  }
  eq(e) {
    return e instanceof m && e.head == this.head;
  }
  toJSON() {
    return { type: "gapcursor", pos: this.head };
  }
  /**
  @internal
  */
  static fromJSON(e, t) {
    if (typeof t.pos != "number")
      throw new RangeError("Invalid input for GapCursor.fromJSON");
    return new m(e.resolve(t.pos));
  }
  /**
  @internal
  */
  getBookmark() {
    return new F(this.anchor);
  }
  /**
  @internal
  */
  static valid(e) {
    let t = e.parent;
    if (t.inlineContent || !be(e) || !Pe(e))
      return !1;
    let o = t.type.spec.allowGapCursor;
    if (o != null)
      return o;
    let r = t.contentMatchAt(e.index()).defaultType;
    return r && r.isTextblock;
  }
  /**
  @internal
  */
  static findGapCursorFrom(e, t, o = !1) {
    e: for (; ; ) {
      if (!o && m.valid(e))
        return e;
      let r = e.pos, i = null;
      for (let s = e.depth; ; s--) {
        let l = e.node(s);
        if (t > 0 ? e.indexAfter(s) < l.childCount : e.index(s) > 0) {
          i = l.child(t > 0 ? e.indexAfter(s) : e.index(s) - 1);
          break;
        } else if (s == 0)
          return null;
        r += t;
        let a = e.doc.resolve(r);
        if (m.valid(a))
          return a;
      }
      for (; ; ) {
        let s = t > 0 ? i.firstChild : i.lastChild;
        if (!s) {
          if (i.isAtom && !i.isText && !Z.isSelectable(i)) {
            e = e.doc.resolve(r + i.nodeSize * t), o = !1;
            continue e;
          }
          break;
        }
        i = s, r += t;
        let l = e.doc.resolve(r);
        if (m.valid(l))
          return l;
      }
      return null;
    }
  }
}
m.prototype.visible = !1;
m.findFrom = m.findGapCursorFrom;
x.jsonID("gapcursor", m);
class F {
  constructor(e) {
    this.pos = e;
  }
  map(e) {
    return new F(e.map(this.pos));
  }
  resolve(e) {
    let t = e.resolve(this.pos);
    return m.valid(t) ? new m(t) : x.near(t);
  }
}
function te(n) {
  return n.isAtom || n.spec.isolating || n.spec.createGapCursor;
}
function be(n) {
  for (let e = n.depth; e >= 0; e--) {
    let t = n.index(e), o = n.node(e);
    if (t == 0) {
      if (o.type.spec.isolating)
        return !0;
      continue;
    }
    for (let r = o.child(t - 1); ; r = r.lastChild) {
      if (r.childCount == 0 && !r.inlineContent || te(r.type))
        return !0;
      if (r.inlineContent)
        return !1;
    }
  }
  return !0;
}
function Pe(n) {
  for (let e = n.depth; e >= 0; e--) {
    let t = n.indexAfter(e), o = n.node(e);
    if (t == o.childCount) {
      if (o.type.spec.isolating)
        return !0;
      continue;
    }
    for (let r = o.child(t); ; r = r.firstChild) {
      if (r.childCount == 0 && !r.inlineContent || te(r.type))
        return !0;
      if (r.inlineContent)
        return !1;
    }
  }
  return !0;
}
function Ee() {
  return new L({
    props: {
      decorations: Te,
      createSelectionBetween(n, e, t) {
        return e.pos == t.pos && m.valid(t) ? new m(t) : null;
      },
      handleClick: Ae,
      handleKeyDown: Me,
      handleDOMEvents: { beforeinput: Se }
    }
  });
}
const Me = me({
  ArrowLeft: T("horiz", -1),
  ArrowRight: T("horiz", 1),
  ArrowUp: T("vert", -1),
  ArrowDown: T("vert", 1)
});
function T(n, e) {
  const t = n == "vert" ? e > 0 ? "down" : "up" : e > 0 ? "right" : "left";
  return function(o, r, i) {
    let s = o.selection, l = e > 0 ? s.$to : s.$from, a = s.empty;
    if (s instanceof Y) {
      if (!i.endOfTextblock(t) || l.depth == 0)
        return !1;
      a = !1, l = o.doc.resolve(e > 0 ? l.after() : l.before());
    }
    let c = m.findGapCursorFrom(l, e, a);
    return c ? (r && r(o.tr.setSelection(new m(c))), !0) : !1;
  };
}
function Ae(n, e, t) {
  if (!n || !n.editable)
    return !1;
  let o = n.state.doc.resolve(e);
  if (!m.valid(o))
    return !1;
  let r = n.posAtCoords({ left: t.clientX, top: t.clientY });
  return r && r.inside > -1 && Z.isSelectable(n.state.doc.nodeAt(r.inside)) ? !1 : (n.dispatch(n.state.tr.setSelection(new m(o))), !0);
}
function Se(n, e) {
  if (e.inputType != "insertCompositionText" || !(n.state.selection instanceof m))
    return !1;
  let { $from: t } = n.state.selection, o = t.parent.contentMatchAt(t.index()).findWrapping(n.state.schema.nodes.text);
  if (!o)
    return !1;
  let r = _.empty;
  for (let s = o.length - 1; s >= 0; s--)
    r = _.from(o[s].createAndFill(null, r));
  let i = n.state.tr.replace(t.pos, t.pos, new U(r, 0, 0));
  return i.setSelection(Y.near(i.doc.resolve(t.pos + 1))), n.dispatch(i), !1;
}
function Te(n) {
  if (!(n.selection instanceof m))
    return null;
  let e = document.createElement("div");
  return e.className = "ProseMirror-gapcursor", ge.create(n.doc, [ve.widget(n.selection.head, e, { key: "gapcursor" })]);
}
var D = 200, g = function() {
};
g.prototype.append = function(e) {
  return e.length ? (e = g.from(e), !this.length && e || e.length < D && this.leafAppend(e) || this.length < D && e.leafPrepend(this) || this.appendInner(e)) : this;
};
g.prototype.prepend = function(e) {
  return e.length ? g.from(e).append(this) : this;
};
g.prototype.appendInner = function(e) {
  return new Ie(this, e);
};
g.prototype.slice = function(e, t) {
  return e === void 0 && (e = 0), t === void 0 && (t = this.length), e >= t ? g.empty : this.sliceInner(Math.max(0, e), Math.min(this.length, t));
};
g.prototype.get = function(e) {
  if (!(e < 0 || e >= this.length))
    return this.getInner(e);
};
g.prototype.forEach = function(e, t, o) {
  t === void 0 && (t = 0), o === void 0 && (o = this.length), t <= o ? this.forEachInner(e, t, o, 0) : this.forEachInvertedInner(e, t, o, 0);
};
g.prototype.map = function(e, t, o) {
  t === void 0 && (t = 0), o === void 0 && (o = this.length);
  var r = [];
  return this.forEach(function(i, s) {
    return r.push(e(i, s));
  }, t, o), r;
};
g.from = function(e) {
  return e instanceof g ? e : e && e.length ? new ne(e) : g.empty;
};
var ne = /* @__PURE__ */ (function(n) {
  function e(o) {
    n.call(this), this.values = o;
  }
  n && (e.__proto__ = n), e.prototype = Object.create(n && n.prototype), e.prototype.constructor = e;
  var t = { length: { configurable: !0 }, depth: { configurable: !0 } };
  return e.prototype.flatten = function() {
    return this.values;
  }, e.prototype.sliceInner = function(r, i) {
    return r == 0 && i == this.length ? this : new e(this.values.slice(r, i));
  }, e.prototype.getInner = function(r) {
    return this.values[r];
  }, e.prototype.forEachInner = function(r, i, s, l) {
    for (var a = i; a < s; a++)
      if (r(this.values[a], l + a) === !1)
        return !1;
  }, e.prototype.forEachInvertedInner = function(r, i, s, l) {
    for (var a = i - 1; a >= s; a--)
      if (r(this.values[a], l + a) === !1)
        return !1;
  }, e.prototype.leafAppend = function(r) {
    if (this.length + r.length <= D)
      return new e(this.values.concat(r.flatten()));
  }, e.prototype.leafPrepend = function(r) {
    if (this.length + r.length <= D)
      return new e(r.flatten().concat(this.values));
  }, t.length.get = function() {
    return this.values.length;
  }, t.depth.get = function() {
    return 0;
  }, Object.defineProperties(e.prototype, t), e;
})(g);
g.empty = new ne([]);
var Ie = /* @__PURE__ */ (function(n) {
  function e(t, o) {
    n.call(this), this.left = t, this.right = o, this.length = t.length + o.length, this.depth = Math.max(t.depth, o.depth) + 1;
  }
  return n && (e.__proto__ = n), e.prototype = Object.create(n && n.prototype), e.prototype.constructor = e, e.prototype.flatten = function() {
    return this.left.flatten().concat(this.right.flatten());
  }, e.prototype.getInner = function(o) {
    return o < this.left.length ? this.left.get(o) : this.right.get(o - this.left.length);
  }, e.prototype.forEachInner = function(o, r, i, s) {
    var l = this.left.length;
    if (r < l && this.left.forEachInner(o, r, Math.min(i, l), s) === !1 || i > l && this.right.forEachInner(o, Math.max(r - l, 0), Math.min(this.length, i) - l, s + l) === !1)
      return !1;
  }, e.prototype.forEachInvertedInner = function(o, r, i, s) {
    var l = this.left.length;
    if (r > l && this.right.forEachInvertedInner(o, r - l, Math.max(i, l) - l, s + l) === !1 || i < l && this.left.forEachInvertedInner(o, Math.min(r, l), i, s) === !1)
      return !1;
  }, e.prototype.sliceInner = function(o, r) {
    if (o == 0 && r == this.length)
      return this;
    var i = this.left.length;
    return r <= i ? this.left.slice(o, r) : o >= i ? this.right.slice(o - i, r - i) : this.left.slice(o, i).append(this.right.slice(0, r - i));
  }, e.prototype.leafAppend = function(o) {
    var r = this.right.leafAppend(o);
    if (r)
      return new e(this.left, r);
  }, e.prototype.leafPrepend = function(o) {
    var r = this.left.leafPrepend(o);
    if (r)
      return new e(r, this.right);
  }, e.prototype.appendInner = function(o) {
    return this.left.depth >= Math.max(this.right.depth, o.depth) + 1 ? new e(this.left, new e(this.right, o)) : new e(this, o);
  }, e;
})(g);
const xe = 500;
class w {
  constructor(e, t) {
    this.items = e, this.eventCount = t;
  }
  // Pop the latest event off the branch's history and apply it
  // to a document transform.
  popEvent(e, t) {
    if (this.eventCount == 0)
      return null;
    let o = this.items.length;
    for (; ; o--)
      if (this.items.get(o - 1).selection) {
        --o;
        break;
      }
    let r, i;
    t && (r = this.remapping(o, this.items.length), i = r.maps.length);
    let s = e.tr, l, a, c = [], d = [];
    return this.items.forEach((u, p) => {
      if (!u.step) {
        r || (r = this.remapping(o, p + 1), i = r.maps.length), i--, d.push(u);
        return;
      }
      if (r) {
        d.push(new C(u.map));
        let f = u.step.map(r.slice(i)), h;
        f && s.maybeStep(f).doc && (h = s.mapping.maps[s.mapping.maps.length - 1], c.push(new C(h, void 0, void 0, c.length + d.length))), i--, h && r.appendMap(h, i);
      } else
        s.maybeStep(u.step);
      if (u.selection)
        return l = r ? u.selection.map(r.slice(i)) : u.selection, a = new w(this.items.slice(0, o).append(d.reverse().concat(c)), this.eventCount - 1), !1;
    }, this.items.length, 0), { remaining: a, transform: s, selection: l };
  }
  // Create a new branch with the given transform added.
  addTransform(e, t, o, r) {
    let i = [], s = this.eventCount, l = this.items, a = !r && l.length ? l.get(l.length - 1) : null;
    for (let d = 0; d < e.steps.length; d++) {
      let u = e.steps[d].invert(e.docs[d]), p = new C(e.mapping.maps[d], u, t), f;
      (f = a && a.merge(p)) && (p = f, d ? i.pop() : l = l.slice(0, l.length - 1)), i.push(p), t && (s++, t = void 0), r || (a = p);
    }
    let c = s - o.depth;
    return c > Re && (l = Oe(l, c), s -= c), new w(l.append(i), s);
  }
  remapping(e, t) {
    let o = new de();
    return this.items.forEach((r, i) => {
      let s = r.mirrorOffset != null && i - r.mirrorOffset >= e ? o.maps.length - r.mirrorOffset : void 0;
      o.appendMap(r.map, s);
    }, e, t), o;
  }
  addMaps(e) {
    return this.eventCount == 0 ? this : new w(this.items.append(e.map((t) => new C(t))), this.eventCount);
  }
  // When the collab module receives remote changes, the history has
  // to know about those, so that it can adjust the steps that were
  // rebased on top of the remote changes, and include the position
  // maps for the remote changes in its array of items.
  rebased(e, t) {
    if (!this.eventCount)
      return this;
    let o = [], r = Math.max(0, this.items.length - t), i = e.mapping, s = e.steps.length, l = this.eventCount;
    this.items.forEach((p) => {
      p.selection && l--;
    }, r);
    let a = t;
    this.items.forEach((p) => {
      let f = i.getMirror(--a);
      if (f == null)
        return;
      s = Math.min(s, f);
      let h = i.maps[f];
      if (p.step) {
        let v = e.steps[f].invert(e.docs[f]), y = p.selection && p.selection.map(i.slice(a + 1, f));
        y && l++, o.push(new C(h, v, y));
      } else
        o.push(new C(h));
    }, r);
    let c = [];
    for (let p = t; p < s; p++)
      c.push(new C(i.maps[p]));
    let d = this.items.slice(0, r).append(c).append(o), u = new w(d, l);
    return u.emptyItemCount() > xe && (u = u.compress(this.items.length - o.length)), u;
  }
  emptyItemCount() {
    let e = 0;
    return this.items.forEach((t) => {
      t.step || e++;
    }), e;
  }
  // Compressing a branch means rewriting it to push the air (map-only
  // items) out. During collaboration, these naturally accumulate
  // because each remote change adds one. The `upto` argument is used
  // to ensure that only the items below a given level are compressed,
  // because `rebased` relies on a clean, untouched set of items in
  // order to associate old items with rebased steps.
  compress(e = this.items.length) {
    let t = this.remapping(0, e), o = t.maps.length, r = [], i = 0;
    return this.items.forEach((s, l) => {
      if (l >= e)
        r.push(s), s.selection && i++;
      else if (s.step) {
        let a = s.step.map(t.slice(o)), c = a && a.getMap();
        if (o--, c && t.appendMap(c, o), a) {
          let d = s.selection && s.selection.map(t.slice(o));
          d && i++;
          let u = new C(c.invert(), a, d), p, f = r.length - 1;
          (p = r.length && r[f].merge(u)) ? r[f] = p : r.push(u);
        }
      } else s.map && o--;
    }, this.items.length, 0), new w(g.from(r.reverse()), i);
  }
}
w.empty = new w(g.empty, 0);
function Oe(n, e) {
  let t;
  return n.forEach((o, r) => {
    if (o.selection && e-- == 0)
      return t = r, !1;
  }), n.slice(t);
}
class C {
  constructor(e, t, o, r) {
    this.map = e, this.step = t, this.selection = o, this.mirrorOffset = r;
  }
  merge(e) {
    if (this.step && e.step && !e.selection) {
      let t = e.step.merge(this.step);
      if (t)
        return new C(t.getMap().invert(), t, this.selection);
    }
  }
}
class b {
  constructor(e, t, o, r, i) {
    this.done = e, this.undone = t, this.prevRanges = o, this.prevTime = r, this.prevComposition = i;
  }
}
const Re = 20;
function De(n, e, t, o) {
  let r = t.getMeta(E), i;
  if (r)
    return r.historyState;
  t.getMeta(ze) && (n = new b(n.done, n.undone, null, 0, -1));
  let s = t.getMeta("appendedTransaction");
  if (t.steps.length == 0)
    return n;
  if (s && s.getMeta(E))
    return s.getMeta(E).redo ? new b(n.done.addTransform(t, void 0, o, I(e)), n.undone, K(t.mapping.maps), n.prevTime, n.prevComposition) : new b(n.done, n.undone.addTransform(t, void 0, o, I(e)), null, n.prevTime, n.prevComposition);
  if (t.getMeta("addToHistory") !== !1 && !(s && s.getMeta("addToHistory") === !1)) {
    let l = t.getMeta("composition"), a = n.prevTime == 0 || !s && n.prevComposition != l && (n.prevTime < (t.time || 0) - o.newGroupDelay || !Ne(t, n.prevRanges)), c = s ? k(n.prevRanges, t.mapping) : K(t.mapping.maps);
    return new b(n.done.addTransform(t, a ? e.selection.getBookmark() : void 0, o, I(e)), w.empty, c, t.time, l ?? n.prevComposition);
  } else return (i = t.getMeta("rebased")) ? new b(n.done.rebased(t, i), n.undone.rebased(t, i), k(n.prevRanges, t.mapping), n.prevTime, n.prevComposition) : new b(n.done.addMaps(t.mapping.maps), n.undone.addMaps(t.mapping.maps), k(n.prevRanges, t.mapping), n.prevTime, n.prevComposition);
}
function Ne(n, e) {
  if (!e)
    return !1;
  if (!n.docChanged)
    return !0;
  let t = !1;
  return n.mapping.maps[0].forEach((o, r) => {
    for (let i = 0; i < e.length; i += 2)
      o <= e[i + 1] && r >= e[i] && (t = !0);
  }), t;
}
function K(n) {
  let e = [];
  for (let t = n.length - 1; t >= 0 && e.length == 0; t--)
    n[t].forEach((o, r, i, s) => e.push(i, s));
  return e;
}
function k(n, e) {
  if (!n)
    return null;
  let t = [];
  for (let o = 0; o < n.length; o += 2) {
    let r = e.map(n[o], 1), i = e.map(n[o + 1], -1);
    r <= i && t.push(r, i);
  }
  return t;
}
function ke(n, e, t) {
  let o = I(e), r = E.get(e).spec.config, i = (t ? n.undone : n.done).popEvent(e, o);
  if (!i)
    return null;
  let s = i.selection.resolve(i.transform.doc), l = (t ? n.done : n.undone).addTransform(i.transform, e.selection.getBookmark(), r, o), a = new b(t ? l : i.remaining, t ? i.remaining : l, null, 0, -1);
  return i.transform.setSelection(s).setMeta(E, { redo: t, historyState: a });
}
let z = !1, H = null;
function I(n) {
  let e = n.plugins;
  if (H != e) {
    z = !1, H = e;
    for (let t = 0; t < e.length; t++)
      if (e[t].spec.historyPreserveItems) {
        z = !0;
        break;
      }
  }
  return z;
}
const E = new J("history"), ze = new J("closeHistory");
function Le(n = {}) {
  return n = {
    depth: n.depth || 100,
    newGroupDelay: n.newGroupDelay || 500
  }, new L({
    key: E,
    state: {
      init() {
        return new b(w.empty, w.empty, null, 0, -1);
      },
      apply(e, t, o) {
        return De(t, o, e, n);
      }
    },
    config: n,
    props: {
      handleDOMEvents: {
        beforeinput(e, t) {
          let o = t.inputType, r = o == "historyUndo" ? re : o == "historyRedo" ? ie : null;
          return !r || !e.editable ? !1 : (t.preventDefault(), r(e.state, e.dispatch));
        }
      }
    }
  });
}
function oe(n, e) {
  return (t, o) => {
    let r = E.getState(t);
    if (!r || (n ? r.undone : r.done).eventCount == 0)
      return !1;
    if (o) {
      let i = ke(r, t, n);
      i && o(e ? i.scrollIntoView() : i);
    }
    return !0;
  };
}
const re = oe(!1, !0), ie = oe(!0, !0);
P.create({
  name: "characterCount",
  addOptions() {
    return {
      limit: null,
      autoTrim: !0,
      mode: "textSize",
      textCounter: (n) => n.length,
      wordCounter: (n) => n.split(" ").filter((e) => e !== "").length
    };
  },
  addStorage() {
    return {
      characters: () => 0,
      words: () => 0
    };
  },
  onBeforeCreate() {
    this.storage.characters = (n) => {
      const e = (n == null ? void 0 : n.node) || this.editor.state.doc;
      if (((n == null ? void 0 : n.mode) || this.options.mode) === "textSize") {
        const o = e.textBetween(0, e.content.size, void 0, " ");
        return this.options.textCounter(o);
      }
      return e.nodeSize;
    }, this.storage.words = (n) => {
      const e = (n == null ? void 0 : n.node) || this.editor.state.doc, t = e.textBetween(0, e.content.size, " ", " ");
      return this.options.wordCounter(t);
    };
  },
  addProseMirrorPlugins() {
    let n = !1;
    return [
      new A({
        key: new S("characterCount"),
        appendTransaction: (e, t, o) => {
          if (n)
            return;
          const r = this.options.limit, i = this.options.autoTrim;
          if (r == null || r === 0 || i === !1) {
            n = !0;
            return;
          }
          const s = this.storage.characters({ node: o.doc });
          if (s > r) {
            const l = s - r, a = 0, c = l;
            console.warn(
              `[CharacterCount] Initial content exceeded limit of ${r} characters. Content was automatically trimmed.`
            );
            const d = o.tr.deleteRange(a, c);
            return n = !0, d;
          }
          n = !0;
        },
        filterTransaction: (e, t) => {
          const o = this.options.limit;
          if (!e.docChanged || o === 0 || o === null || o === void 0)
            return !0;
          const r = this.storage.characters({ node: t.doc }), i = this.storage.characters({ node: e.doc });
          if (i <= o || r > o && i > o && i <= r)
            return !0;
          if (r > o && i > o && i > r || !e.getMeta("paste"))
            return !1;
          const l = e.selection.$head.pos, a = i - o, c = l - a, d = l;
          return e.deleteRange(c, d), !(this.storage.characters({ node: e.doc }) > o);
        }
      })
    ];
  }
});
P.create({
  name: "dropCursor",
  addOptions() {
    return {
      color: "currentColor",
      width: 1,
      class: void 0
    };
  },
  addProseMirrorPlugins() {
    return [ye(this.options)];
  }
});
P.create({
  name: "focus",
  addOptions() {
    return {
      className: "has-focus",
      mode: "all"
    };
  },
  addProseMirrorPlugins() {
    return [
      new A({
        key: new S("focus"),
        props: {
          decorations: ({ doc: n, selection: e }) => {
            const { isEditable: t, isFocused: o } = this.editor, { anchor: r } = e, i = [];
            if (!t || !o)
              return O.create(n, []);
            let s = 0;
            this.options.mode === "deepest" && n.descendants((a, c) => {
              if (a.isText)
                return;
              if (!(r >= c && r <= c + a.nodeSize - 1))
                return !1;
              s += 1;
            });
            let l = 0;
            return n.descendants((a, c) => {
              if (a.isText || !(r >= c && r <= c + a.nodeSize - 1))
                return !1;
              if (l += 1, this.options.mode === "deepest" && s - l > 0 || this.options.mode === "shallowest" && l > 1)
                return this.options.mode === "deepest";
              i.push(
                V.node(c, c + a.nodeSize, {
                  class: this.options.className
                })
              );
            }), O.create(n, i);
          }
        }
      })
    ];
  }
});
P.create({
  name: "gapCursor",
  addProseMirrorPlugins() {
    return [Ee()];
  },
  extendNodeSchema(n) {
    var e;
    const t = {
      name: n.name,
      options: n.options,
      storage: n.storage
    };
    return {
      allowGapCursor: (e = ue(pe(n, "allowGapCursor", t))) != null ? e : null
    };
  }
});
var se = "placeholder", M = new S("tiptap__placeholder"), G = 200;
function W(n) {
  const {
    editor: e,
    placeholder: t,
    dataAttribute: o,
    pos: r,
    node: i,
    isEmptyDoc: s,
    hasAnchor: l,
    classes: { emptyNode: a, emptyEditor: c }
  } = n, d = [a];
  return s && d.push(c), V.node(r, r + i.nodeSize, {
    class: d.join(" "),
    [o]: typeof t == "function" ? t({
      editor: e,
      node: i,
      pos: r,
      hasAnchor: l
    }) : t
  });
}
function X(n, e) {
  return typeof n == "function" ? n(e) : n;
}
function Ve({
  editor: n,
  options: e,
  dataAttribute: t,
  doc: o,
  selection: r
}) {
  var i, s;
  if (!(n.isEditable || !e.showOnlyWhenEditable))
    return null;
  const { anchor: a } = r, c = [], d = n.isEmpty;
  if (e.showOnlyCurrent && !e.includeChildren) {
    const p = o.resolve(a), f = p.depth > 0 ? p.node(1) : p.nodeAfter, h = p.depth > 0 ? p.before(1) : a;
    if (f && f.type.isTextblock && B(f)) {
      const v = a >= h && a <= h + f.nodeSize;
      c.push(
        W({
          editor: n,
          isEmptyDoc: d,
          dataAttribute: t,
          hasAnchor: v,
          placeholder: e.placeholder,
          classes: {
            emptyEditor: e.emptyEditorClass,
            emptyNode: X(e.emptyNodeClass, {
              editor: n,
              node: f,
              pos: h,
              hasAnchor: v
            })
          },
          node: f,
          pos: h
        })
      );
    }
  } else {
    const p = M.getState(n.state), f = (i = p == null ? void 0 : p.topPos) != null ? i : 0, h = (s = p == null ? void 0 : p.bottomPos) != null ? s : o.content.size;
    o.nodesBetween(f, h, (v, y) => {
      const N = a >= y && a <= y + v.nodeSize, ae = !v.isLeaf && B(v);
      return v.type.isTextblock && (N || !e.showOnlyCurrent) && ae && c.push(
        W({
          editor: n,
          isEmptyDoc: d,
          dataAttribute: t,
          hasAnchor: N,
          placeholder: e.placeholder,
          classes: {
            emptyEditor: e.emptyEditorClass,
            emptyNode: X(e.emptyNodeClass, {
              editor: n,
              node: v,
              pos: y,
              hasAnchor: N
            })
          },
          node: v,
          pos: y
        })
      ), e.includeChildren;
    });
  }
  return O.create(o, c);
}
function Fe(n) {
  return n.replace(/\s+/g, "-").replace(/[^a-zA-Z0-9-]/g, "").replace(/^[0-9-]+/, "").replace(/^-+/, "").toLowerCase();
}
function _e(n) {
  const e = getComputedStyle(n), t = `${e.overflow} ${e.overflowY} ${e.overflowX}`;
  return /auto|scroll|overlay/.test(t);
}
function Be(n) {
  let e = n;
  for (; e; ) {
    if (_e(e))
      return e;
    const t = e.parentElement;
    if (!t) {
      const o = e.getRootNode();
      if (o instanceof ShadowRoot) {
        e = o.host;
        continue;
      }
      return window;
    }
    e = t;
  }
  return window;
}
function Ke(n) {
  return n === window ? { top: 0, bottom: window.innerHeight } : n.getBoundingClientRect();
}
function He({
  view: n,
  scrollContainer: e
}) {
  const t = n.dom.getBoundingClientRect();
  if (t.width <= 0 || t.height <= 0)
    return null;
  const o = e ? Ke(e) : { top: 0, bottom: window.innerHeight }, r = Math.max(t.top, o.top) - G, i = Math.min(t.bottom, o.bottom) + G;
  if (r >= i)
    return null;
  const s = t.left + 1, l = t.right - 1;
  if (s > l)
    return null;
  const c = getComputedStyle(n.dom).direction === "rtl" ? t.right - 2 : t.left + 2, d = Math.min(Math.max(c, s), l), u = Math.max(r + 2, t.top + 1), p = Math.min(i - 2, t.bottom - 1);
  if (u > p)
    return null;
  const f = n.posAtCoords({ left: d, top: u }), h = n.posAtCoords({ left: d, top: p });
  return !f || !h ? null : { top: f.pos, bottom: h.pos };
}
var Ge = {
  /**
   * Initialises the viewport state with no known positions.
   * @returns The initial viewport state.
   */
  init() {
    return { topPos: null, bottomPos: null };
  },
  /**
   * Updates the viewport state from incoming transactions.
   * @param tr - The transaction being applied.
   * @param prev - The previous viewport state.
   * @returns The next viewport state.
   */
  apply(n, e) {
    const t = n.getMeta(M);
    return t != null && t.positions ? { topPos: t.positions.top, bottomPos: t.positions.bottom } : n.docChanged ? {
      topPos: e.topPos !== null ? n.mapping.map(e.topPos) : null,
      bottomPos: e.bottomPos !== null ? n.mapping.map(e.bottomPos) : null
    } : e;
  }
};
function We(n) {
  const e = Be(n.dom), t = () => {
    const c = He({
      view: n,
      scrollContainer: e
    });
    if (c === null)
      return;
    const d = M.getState(n.state);
    if ((d == null ? void 0 : d.topPos) === c.top && (d == null ? void 0 : d.bottomPos) === c.bottom)
      return;
    const u = n.state.tr.setMeta(M, { positions: c });
    n.dispatch(u);
  };
  let o = null, r = 0;
  const i = 150, s = () => {
    o === null && (o = requestAnimationFrame(() => {
      o = null;
      const c = performance.now();
      c - r >= i ? (r = c, t()) : s();
    }));
  };
  e.addEventListener("scroll", s, { passive: !0 });
  const l = typeof ResizeObserver < "u" ? new ResizeObserver(s) : null;
  l == null || l.observe(n.dom);
  const a = typeof IntersectionObserver < "u" ? new IntersectionObserver(s) : null;
  return a == null || a.observe(n.dom), n.dom.addEventListener("focus", s), t(), {
    update(c, d) {
      n.state.doc.content.size !== d.doc.content.size && s();
    },
    destroy: () => {
      o !== null && cancelAnimationFrame(o), e.removeEventListener("scroll", s), l == null || l.disconnect(), a == null || a.disconnect(), n.dom.removeEventListener("focus", s);
    }
  };
}
function Xe({ editor: n, options: e }) {
  const t = e.dataAttribute ? `data-${Fe(e.dataAttribute)}` : `data-${se}`;
  return new A({
    key: M,
    state: Ge,
    view: We,
    props: {
      decorations: ({ doc: o, selection: r }) => Ve({ editor: n, options: e, dataAttribute: t, doc: o, selection: r })
    }
  });
}
P.create({
  name: "placeholder",
  addOptions() {
    return {
      emptyEditorClass: "is-editor-empty",
      emptyNodeClass: "is-empty",
      dataAttribute: se,
      placeholder: "Write something …",
      showOnlyWhenEditable: !0,
      showOnlyCurrent: !0,
      includeChildren: !1
    };
  },
  addProseMirrorPlugins() {
    return [Xe({ editor: this.editor, options: this.options })];
  }
});
var je = `.ProseMirror:not(.ProseMirror-focused) *::selection {
  background: transparent;
}

.ProseMirror:not(.ProseMirror-focused) *::-moz-selection {
  background: transparent;
}`;
P.create({
  name: "selection",
  addOptions() {
    return {
      className: "selection"
    };
  },
  addProseMirrorPlugins() {
    const { editor: n, options: e } = this;
    return n.options.injectCSS && typeof document < "u" && fe(je, n.options.injectNonce, "selection"), [
      new A({
        key: new S("selection"),
        props: {
          decorations(t) {
            return t.selection.empty || n.isFocused || !n.isEditable || he(t.selection) || n.view.dragging ? null : O.create(t.doc, [
              V.inline(t.selection.from, t.selection.to, {
                class: e.className
              })
            ]);
          }
        }
      })
    ];
  }
});
var Ue = "skipTrailingNode";
function j({
  types: n,
  node: e
}) {
  return e && Array.isArray(n) && n.includes(e.type) || (e == null ? void 0 : e.type) === n;
}
P.create({
  name: "trailingNode",
  addOptions() {
    return {
      node: void 0,
      notAfter: []
    };
  },
  addProseMirrorPlugins() {
    var n;
    const e = new S(this.name), t = this.options.node || ((n = this.editor.schema.topNodeType.contentMatch.defaultType) == null ? void 0 : n.name) || "paragraph", o = Object.entries(this.editor.schema.nodes).map(([, r]) => r).filter((r) => (this.options.notAfter || []).concat(t).includes(r.name));
    return [
      new A({
        key: e,
        appendTransaction: (r, i, s) => {
          const { doc: l, tr: a, schema: c } = s, d = e.getState(s), u = l.content.size, p = c.nodes[t];
          if (!r.some((f) => f.getMeta(Ue)) && d)
            return a.insert(u, p.create());
        },
        state: {
          init: (r, i) => {
            const s = i.tr.doc.lastChild;
            return !j({ node: s, types: o });
          },
          apply: (r, i) => {
            if (!r.docChanged || r.getMeta("__uniqueIDTransaction"))
              return i;
            const s = r.doc.lastChild;
            return !j({ node: s, types: o });
          }
        }
      })
    ];
  }
});
var Ye = P.create({
  name: "undoRedo",
  addOptions() {
    return {
      depth: 100,
      newGroupDelay: 500
    };
  },
  addCommands() {
    return {
      undo: () => ({ state: n, dispatch: e }) => re(n, e),
      redo: () => ({ state: n, dispatch: e }) => ie(n, e)
    };
  },
  addProseMirrorPlugins() {
    return [Le(this.options)];
  },
  addKeyboardShortcuts() {
    return {
      "Mod-z": () => this.editor.commands.undo(),
      "Shift-Mod-z": () => this.editor.commands.redo(),
      "Mod-y": () => this.editor.commands.redo(),
      // Russian keyboard layouts
      "Mod-я": () => this.editor.commands.undo(),
      "Shift-Mod-я": () => this.editor.commands.redo()
    };
  }
});
function ot() {
  var d;
  const n = $(le.name), {
    icon: e = void 0,
    tooltip: t = void 0,
    shortcutKeys: o = void 0,
    tooltipOptions: r = {},
    action: i = void 0,
    isActive: s = void 0
  } = ((d = n == null ? void 0 : n.componentProps) == null ? void 0 : d.undo) ?? {}, { disabled: l } = q(s), a = we[e], c = () => {
    l || i && i();
  };
  return !n || !a ? /* @__PURE__ */ R(Q, {}) : /* @__PURE__ */ R(
    ee,
    {
      action: c,
      disabled: l,
      icon: e,
      shortcutKeys: o,
      tooltip: t,
      tooltipOptions: r
    }
  );
}
function rt() {
  var c;
  const n = $(le.name), {
    icon: e = void 0,
    tooltip: t = void 0,
    shortcutKeys: o = void 0,
    tooltipOptions: r = {},
    action: i = void 0,
    isActive: s = void 0
  } = ((c = n == null ? void 0 : n.componentProps) == null ? void 0 : c.redo) ?? {}, { disabled: l } = q(s), a = () => {
    l || i && i();
  };
  return n ? /* @__PURE__ */ R(
    ee,
    {
      action: a,
      disabled: l,
      icon: e,
      shortcutKeys: o,
      tooltip: t,
      tooltipOptions: r
    }
  ) : /* @__PURE__ */ R(Q, {});
}
const le = /* @__PURE__ */ Ye.extend({
  //@ts-expect-error
  addOptions() {
    var n;
    return {
      ...(n = this.parent) == null ? void 0 : n.call(this),
      depth: 100,
      newGroupDelay: 500,
      button: ({ editor: e, t, extension: o }) => {
        var r, i;
        return {
          componentProps: {
            undo: {
              action: () => {
                e.chain().focus().undo().run();
              },
              shortcutKeys: ((r = o.options.shortcutKeys) == null ? void 0 : r[0]) ?? ["mod", "Z"],
              isActive: () => e.can().undo(),
              icon: "Undo2",
              tooltip: t("editor.undo.tooltip")
            },
            redo: {
              action: () => {
                e.chain().focus().redo().run();
              },
              shortcutKeys: ((i = o.options.shortcutKeys) == null ? void 0 : i[1]) ?? ["shift", "mod", "Z"],
              isActive: () => e.can().redo(),
              icon: "Redo2",
              tooltip: t("editor.redo.tooltip")
            }
          }
        };
      }
    };
  }
});
export {
  le as History,
  rt as RichTextRedo,
  ot as RichTextUndo
};
