// reactjs-tiptap-editor@1.0.2/lib/ImageGif.js
import { m as B } from "./index-Dx-Dh7KS.js";
import { O as H, E as C, P as q, u as V, d as D, A as X, Q as Y, k as Z, J } from "./index-C52cuQ_Y.js";
import { N as Q, a as U } from "./clsx-_bq4MQQ7.js";
import { jsx as u, jsxs as z, Fragment as K } from "react/jsx-runtime";
import { useState as G, useMemo as L, useCallback as M, useEffect as $, useRef as ee } from "react";
import { I as j, j as W, k as te } from "./index-DtyAGMbq.js";
import "./theme.js";
import { P as ne, a as re, b as ae } from "./popover-BB-ZDnqk.js";
const O = {
  TOP_LEFT: "tl",
  TOP_RIGHT: "tr",
  BOTTOM_LEFT: "bl",
  BOTTOM_RIGHT: "br"
};
function ie(e) {
  var p, k;
  const [t, r] = G({
    width: j,
    height: j
  }), [c, a] = G({
    width: 0,
    height: 0
  }), [n] = G([
    O.TOP_LEFT,
    O.TOP_RIGHT,
    O.BOTTOM_LEFT,
    O.BOTTOM_RIGHT
  ]), [i, s] = G(!1), [l, v] = G({
    x: 0,
    y: 0,
    w: 0,
    h: 0,
    dir: ""
  }), { align: b } = (p = e == null ? void 0 : e.node) == null ? void 0 : p.attrs, g = L(() => {
    var w;
    const { src: o, alt: m, width: f, height: x } = (w = e == null ? void 0 : e.node) == null ? void 0 : w.attrs, A = H(f) ? `${f}px` : f, d = H(x) ? `${x}px` : x;
    return {
      src: o || void 0,
      alt: m || void 0,
      style: {
        width: A || void 0,
        height: d || void 0
      }
    };
  }, [(k = e == null ? void 0 : e.node) == null ? void 0 : k.attrs]), _ = L(() => {
    const {
      style: { width: o }
    } = g;
    return { width: o === "100%" ? o : void 0 };
  }, [g]);
  function h(o) {
    a({
      width: o.target.width,
      height: o.target.height
    });
  }
  function y() {
    const { editor: o, getPos: m } = e;
    o.commands.setNodeSelection(m());
  }
  const I = M(
    C(() => {
      const { editor: o } = e, { width: m } = getComputedStyle(o.view.dom);
      r((f) => ({
        ...f,
        width: Number.parseInt(m, 10)
      }));
    }, W),
    [e == null ? void 0 : e.editor]
  );
  function F(o, m) {
    o.preventDefault(), o.stopPropagation();
    const f = c.width, x = c.height, A = f / x;
    let d = Number(e.node.attrs.width), w = Number(e.node.attrs.height);
    const T = t.width;
    d && !w ? (d = d > T ? T : d, w = Math.round(d / A)) : w && !d ? (d = Math.round(w * A), d = d > T ? T : d) : !d && !w ? (d = f > T ? T : f, w = Math.round(d / A)) : d = d > T ? T : d, s(!0), v({
      x: o.clientX,
      y: o.clientY,
      w: d,
      h: w,
      dir: m
    });
  }
  const N = M(
    C((o) => {
      if (o.preventDefault(), o.stopPropagation(), !i)
        return;
      const { x: m, w: f, dir: x } = l, A = (o.clientX - m) * (/l/.test(x) ? -1 : 1), d = q(f + A, te, t.width);
      e.updateAttributes({
        width: d,
        height: null
      });
    }, W),
    [i, l, t, e.updateAttributes]
  ), E = M(
    (o) => {
      o.preventDefault(), o.stopPropagation(), i && (v({
        x: 0,
        y: 0,
        w: 0,
        h: 0,
        dir: ""
      }), s(!1), y());
    },
    [i, y]
  ), R = M(() => {
    document == null || document.addEventListener("mousemove", N, !0), document == null || document.addEventListener("mouseup", E, !0);
  }, [N, E]), P = M(() => {
    document == null || document.removeEventListener("mousemove", N, !0), document == null || document.removeEventListener("mouseup", E, !0);
  }, [N, E]);
  $(() => (i ? R() : P(), () => {
    P();
  }), [i, R, P]);
  const S = L(() => new ResizeObserver(() => I()), [I]);
  return $(() => (S.observe(e.editor.view.dom), () => {
    S.disconnect();
  }), [e.editor.view.dom, S]), /* @__PURE__ */ u(
    Q,
    {
      className: "image-view",
      style: { ..._, width: "100%", textAlign: b },
      children: /* @__PURE__ */ z(
        "div",
        {
          "data-drag-handle": !0,
          draggable: "true",
          style: _,
          className: `image-view__body ${e != null && e.selected ? "image-view__body--focused" : ""} ${i ? "image-view__body--resizing" : ""}`,
          children: [
            /* @__PURE__ */ u(
              "img",
              {
                alt: g.alt,
                className: "image-view__body__image block",
                height: "auto",
                onClick: y,
                onLoad: h,
                src: g.src,
                style: g.style
              }
            ),
            (e == null ? void 0 : e.editor.view.editable) && ((e == null ? void 0 : e.selected) || i) && /* @__PURE__ */ u("div", { className: "image-resizer", children: n == null ? void 0 : n.map((o) => /* @__PURE__ */ u(
              "span",
              {
                className: `image-resizer__handler image-resizer__handler--${o}`,
                onMouseDown: (m) => F(m, o)
              },
              `image-dir-${o}`
            )) })
          ]
        }
      )
    }
  );
}
async function se(e) {
  var c;
  const r = await (await fetch(`https://api.giphy.com/v1/gifs/trending?q=&limit=15&api_key=${e}`)).json();
  return (c = r == null ? void 0 : r.data) == null ? void 0 : c.map((a) => {
    var n, i, s;
    return {
      id: a == null ? void 0 : a.id,
      src: (n = a == null ? void 0 : a.images.original) == null ? void 0 : n.url,
      width: +((i = a == null ? void 0 : a.images.original) == null ? void 0 : i.width),
      height: +((s = a == null ? void 0 : a.images.original) == null ? void 0 : s.width)
    };
  });
}
async function oe(e, t) {
  var a;
  const c = await (await fetch(
    `https://api.giphy.com/v1/gifs/search?q=${e}&limit=15&api_key=${t}`
  )).json();
  return (a = c == null ? void 0 : c.data) == null ? void 0 : a.map((n) => {
    var i, s, l;
    return {
      id: n == null ? void 0 : n.id,
      src: (i = n == null ? void 0 : n.images.original) == null ? void 0 : i.url,
      width: +((s = n == null ? void 0 : n.images.original) == null ? void 0 : s.width),
      height: +((l = n == null ? void 0 : n.images.original) == null ? void 0 : l.width)
    };
  });
}
async function ce(e) {
  var n, i;
  const t = await fetch(
    `https://tenor.googleapis.com/v2/trending_terms?key=${e}&limit=10`
  ), r = await (t == null ? void 0 : t.json()), a = await (await fetch(
    `https://tenor.googleapis.com/v2/search?key=${e}&q=${(n = r == null ? void 0 : r.results) == null ? void 0 : n[0]}&limit=15`
  )).json();
  return (i = a == null ? void 0 : a.results) == null ? void 0 : i.map((s) => {
    var l, v, b, g, _, h, y, I;
    return {
      id: s == null ? void 0 : s.id,
      src: (v = (l = s == null ? void 0 : s.media_formats) == null ? void 0 : l.gif) == null ? void 0 : v.url,
      width: (_ = (g = (b = s == null ? void 0 : s.media_formats) == null ? void 0 : b.gif) == null ? void 0 : g.dims) == null ? void 0 : _[0],
      height: (I = (y = (h = s == null ? void 0 : s.media_formats) == null ? void 0 : h.gif) == null ? void 0 : y.dims) == null ? void 0 : I[1]
    };
  });
}
async function de(e, t) {
  var a;
  const c = await (await fetch(
    `https://tenor.googleapis.com/v2/search?key=${t}&q=${e}&limit=15`
  )).json();
  return (a = c == null ? void 0 : c.results) == null ? void 0 : a.map((n) => {
    var i, s, l, v, b, g, _, h;
    return {
      id: n == null ? void 0 : n.id,
      src: (s = (i = n == null ? void 0 : n.media_formats) == null ? void 0 : i.gif) == null ? void 0 : s.url,
      width: (b = (v = (l = n == null ? void 0 : n.media_formats) == null ? void 0 : l.gif) == null ? void 0 : v.dims) == null ? void 0 : b[0],
      height: (h = (_ = (g = n == null ? void 0 : n.media_formats) == null ? void 0 : g.gif) == null ? void 0 : _.dims) == null ? void 0 : h[1]
    };
  });
}
function le(e, t) {
  return {
    searchTrending: async () => t ? e === "giphy" ? se(t) : e === "tenor" ? ce(t) : [] : [],
    searchWord: async (a) => t ? e === "giphy" ? oe(a, t) : e === "tenor" ? de(a, t) : [] : []
  };
}
function he({ selectImage: e, apiKey: t, provider: r, children: c }) {
  const [a, n] = G(!1), [i, s] = G([]), { editorDisabled: l } = D(), v = ee(null), { searchTrending: b, searchWord: g } = le(r, t);
  $(() => {
    (async () => {
      const h = await b();
      s(h);
    })();
  }, []);
  const _ = M(
    Y(async (h) => {
      if (!h.target.value) {
        const I = await b();
        s(I);
        return;
      }
      const y = await g(h.target.value);
      s(y);
    }, 350),
    // Adjust the debounce delay as needed
    []
  );
  return /* @__PURE__ */ z(ne, { modal: !0, onOpenChange: n, open: a, children: [
    /* @__PURE__ */ u(re, { asChild: !0, disabled: l, children: c }),
    /* @__PURE__ */ u(
      ae,
      {
        align: "start",
        className: "richtext-size-full richtext-p-2",
        hideWhenDetached: !0,
        side: "bottom",
        children: t ? /* @__PURE__ */ z(K, { children: [
          /* @__PURE__ */ u("div", { className: "richtext-mb-[10px] richtext-w-full", children: /* @__PURE__ */ u(
            Z,
            {
              onChange: _,
              placeholder: "Search GIF",
              ref: v,
              type: "text"
            }
          ) }),
          /* @__PURE__ */ u("div", { className: "richtext-max-h-[280px] !richtext-max-w-[400px] richtext-overflow-y-auto", children: /* @__PURE__ */ u("div", { className: "richtext-grid richtext-grid-cols-2 richtext-gap-1", children: i != null && i.length ? i == null ? void 0 : i.map((h) => /* @__PURE__ */ u(
            "img",
            {
              alt: "",
              className: "richtext-cursor-pointer richtext-object-contain richtext-text-center",
              src: h.src,
              onClick: () => {
                e(h.src), n(!1);
              }
            },
            h.id
          )) : /* @__PURE__ */ u("p", { children: "No GIFs found" }) }) })
        ] }) : /* @__PURE__ */ u("div", { children: /* @__PURE__ */ u("p", { children: "Missing Giphy API Key" }) })
      }
    )
  ] });
}
function Te() {
  const e = V(ue.name), { action: t, icon: r, tooltip: c, apiKey: a, provider: n } = (e == null ? void 0 : e.componentProps) ?? {}, { editorDisabled: i } = D();
  return /* @__PURE__ */ u(he, { apiKey: a, provider: n, selectImage: (l) => {
    i || t && t(l);
  }, children: /* @__PURE__ */ u(X, { disabled: i, icon: r, tooltip: c }) });
}
const ue = /* @__PURE__ */ J.extend({
  name: "imageGif",
  //@ts-expect-error
  addOptions() {
    var e;
    return {
      ...(e = this.parent) == null ? void 0 : e.call(this),
      inline: !1,
      content: "",
      marks: "",
      group: "block",
      API_KEY: "",
      provider: "giphy",
      draggable: !1,
      selectable: !0,
      atom: !0,
      button: ({ editor: t, extension: r, t: c }) => {
        var i, s;
        const a = ((i = r == null ? void 0 : r.options) == null ? void 0 : i.provider) || "", n = ((s = r == null ? void 0 : r.options) == null ? void 0 : s.API_KEY) || "";
        return {
          componentProps: {
            action: (l) => {
              t.chain().focus().setImageGif({ src: l }).run();
            },
            isActive: () => !1,
            disabled: !1,
            icon: "GifIcon",
            tooltip: c("editor.imageGif.tooltip"),
            apiKey: n,
            provider: a
          }
        };
      }
    };
  },
  addAttributes() {
    var e;
    return {
      ...(e = this.parent) == null ? void 0 : e.call(this),
      width: {
        default: null,
        parseHTML: (t) => {
          const r = t.style.width || t.getAttribute("width") || "10";
          return r === void 0 ? null : Number.parseInt(`${r}`, 10);
        },
        renderHTML: (t) => ({
          width: t.width
        })
      },
      align: {
        default: "center",
        parseHTML: (t) => t.getAttribute("align"),
        renderHTML: (t) => ({
          align: t.align
        })
      }
    };
  },
  addNodeView() {
    return U(ie);
  },
  addCommands() {
    var e;
    return {
      ...(e = this.parent) == null ? void 0 : e.call(this),
      setImageGif: (t) => ({ commands: r }) => r.insertContent({
        type: this.name,
        attrs: t
      }),
      updateImageGif: (t) => ({ commands: r }) => r.updateAttributes(this.name, t),
      setAlignImageGif: (t) => ({ commands: r }) => r.updateAttributes(this.name, { align: t })
    };
  },
  renderHTML({ HTMLAttributes: e }) {
    const { align: t } = e;
    return [
      "div",
      // Parent element
      {
        style: t ? `text-align: ${t};` : "",
        class: "imageGIf"
      },
      [
        "img",
        B(
          // Always render the `height="auto"`
          {
            height: "auto"
          },
          this.options.HTMLAttributes,
          e
        )
      ]
    ];
  },
  parseHTML() {
    return [
      {
        tag: "div[class=imageGIf]",
        getAttrs: (e) => {
          const t = e.querySelector("img"), r = t == null ? void 0 : t.getAttribute("width");
          return {
            src: t == null ? void 0 : t.getAttribute("src"),
            alt: t == null ? void 0 : t.getAttribute("alt"),
            title: t == null ? void 0 : t.getAttribute("title"),
            width: r ? Number.parseInt(r, 10) : null,
            align: (t == null ? void 0 : t.getAttribute("align")) || e.style.textAlign || null
          };
        }
      }
    ];
  }
});
export {
  ue as ImageGif,
  Te as RichTextImageGif
};
