// reactjs-tiptap-editor@1.0.2/lib/ImportWord.js
import { E as M } from "./index-Dx-Dh7KS.js";
import { jsx as g, Fragment as w, jsxs as P } from "react/jsx-runtime";
import R from "mammoth";
import { useState as z, useRef as N } from "react";
import { h as k, u as D, d as K, w as U, x as q, A as G, y as J, z as Q } from "./index-C52cuQ_Y.js";
import "./theme.js";
import { u as V } from "./index-DtyAGMbq.js";
function X(e, t) {
  const s = atob(e.split(",")[1]), c = Array.from({ length: s.length });
  for (let i = 0; i < s.length; i++)
    c[i] = s.charCodeAt(i);
  const l = new Uint8Array(c);
  return new Blob([l], { type: t });
}
function Y(e, t) {
  return new File([e], t, { type: e.type });
}
function it() {
  const e = k(), t = D(I.name), {
    icon: s = void 0,
    tooltip: c = void 0,
    tooltipOptions: l = {},
    isActive: i = void 0,
    mammothOptions: x,
    limit: T,
    convert: h
  } = (t == null ? void 0 : t.componentProps) ?? {}, { editorDisabled: C } = K(i), u = U(I.name), { toast: E } = q(), { t: W } = V(), [B, y] = z(!1), b = N(null);
  function L() {
    var o;
    (o = b.current) == null || o.click();
  }
  function O(o) {
    const n = o.target.files[0];
    if (n) {
      if (n.size > T) {
        E({
          variant: "destructive",
          title: W("editor.importWord.limitSize")
        });
        return;
      }
      j(n);
    }
  }
  async function S(o) {
    const a = new DOMParser().parseFromString(o, "text/html"), p = a.querySelectorAll("img");
    if (p.length === 0)
      return a.body.innerHTML;
    if (J(e, Q.name)) {
      const d = u == null ? void 0 : u.options;
      if (d && typeof d.upload == "function") {
        const A = [];
        for (const f of p) {
          const m = f.getAttribute("src"), r = X(m, "image/jpeg"), H = Y(r, "image.jpeg");
          A.push(H);
        }
        const F = await d.upload(A);
        for (const [f, m] of p.entries()) {
          m.setAttribute("src", F[f].src);
          const r = m.parentElement;
          (r == null ? void 0 : r.tagName) === "P" && (r.insertAdjacentElement("beforebegin", m), !r.hasChildNodes() && r.textContent === "" && r.remove());
        }
        return a.body.innerHTML;
      } else
        return console.warn("Image Upload method found, skip image conversion"), a.body.innerHTML;
    } else
      return console.error("Image extension not found, unable to convert image"), a.body.innerHTML;
  }
  async function j(o) {
    y(!0);
    try {
      if (h) {
        const n = await h(o);
        v(n);
      } else {
        const n = await o.arrayBuffer(), { value: a } = await R.convertToHtml({ arrayBuffer: n }, x);
        v(a);
      }
    } finally {
      y(!1);
    }
  }
  async function v(o) {
    const n = await S(o);
    e.chain().setContent(n, !0).run();
  }
  return t ? /* @__PURE__ */ P(w, { children: [
    /* @__PURE__ */ g(
      G,
      {
        action: L,
        disabled: C,
        icon: s,
        loading: B,
        tooltip: c,
        tooltipOptions: l
      }
    ),
    /* @__PURE__ */ g(
      "input",
      {
        accept: ".docx",
        onChange: O,
        ref: b,
        type: "file",
        style: {
          display: "none"
        }
      }
    )
  ] }) : /* @__PURE__ */ g(w, {});
}
const I = /* @__PURE__ */ M.create({
  name: "importWord",
  //@ts-expect-error
  addOptions() {
    var e;
    return {
      ...(e = this.parent) == null ? void 0 : e.call(this),
      upload: void 0,
      convert: void 0,
      limit: 1024 * 1024 * 10,
      // 10 MB
      button: ({ extension: t, t: s }) => {
        const { convert: c, limit: l, mammothOptions: i } = t.options;
        return {
          componentProps: {
            convert: c,
            limit: l,
            mammothOptions: i,
            // action: () => editor.commands.setHorizontalRule(),
            // disabled: !editor.can().setHorizontalRule(),
            icon: "Word",
            shortcutKeys: t.options.shortcutKeys ?? ["alt", "mod", "S"],
            tooltip: s("editor.importWord.tooltip")
          }
        };
      }
    };
  }
});
export {
  I as ImportWord,
  it as RichTextImportWord
};
