// reactjs-tiptap-editor@1.0.2/lib/index-DtyAGMbq.js
import { useCallback as m } from "react";
import { b as a, S as c, y as h, v as n } from "./index-BG0kQamI.js";
const b = "en", D = 20, L = 1e5, w = 16, E = 10, B = 10, N = 2, R = [
  "#000000",
  "#262626",
  "#595959",
  "#8C8C8C",
  "#BFBFBF",
  "#D9D9D9",
  "#E9E9E9",
  "#F5F5F5",
  "#FAFAFA",
  "#FFFFFF",
  "#F5222D",
  "#FA541C",
  "#FA8C16",
  "#FADB14",
  "#52C41A",
  "#13C2C2",
  "#1890FF",
  "#2F54EB",
  "#722ED1",
  "#EB2F96",
  "#FFE8E6",
  "#FFECE0",
  "#FFEFD1",
  "#FCFCCA",
  "#E4F7D2",
  "#D3F5F0",
  "#D4EEFC",
  "#DEE8FC",
  "#EFE1FA",
  "#FAE1EB",
  "#FFA39E",
  "#FFBB96",
  "#FFD591",
  "#FFFB8F",
  "#B7EB8F",
  "#87E8DE",
  "#91D5FF",
  "#ADC6FF",
  "#D3ADF7",
  "#FFADD2",
  "#FF4D4F",
  "#FF7A45",
  "#FFA940",
  "#FFEC3D",
  "#73D13D",
  "#36CFC9",
  "#40A9FF",
  "#597EF7",
  "#9254DE",
  "#F759AB",
  "#CF1322",
  "#D4380D",
  "#D46B08",
  "#D4B106",
  "#389E0D",
  "#08979C",
  "#096DD9",
  "#1D39C4",
  "#531DAB",
  "#C41D7F",
  "#820014",
  "#871400",
  "#873800",
  "#614700",
  "#135200",
  "#00474F",
  "#003A8C",
  "#061178",
  "#22075E",
  "#780650"
], j = [
  "Default",
  "Inter",
  "Comic Sans MS, Comic Sans",
  "serif",
  "cursive",
  "Arial",
  "Arial Black",
  "Georgia",
  "Impact",
  "Tahoma",
  "Times New Roman",
  "Verdana",
  "Courier New",
  "Lucida Console",
  "Monaco",
  "monospace"
], W = [
  "Default",
  "10px",
  "11px",
  "12px",
  "14px",
  "16px",
  "18px",
  "20px",
  "22px",
  "24px",
  "26px",
  "28px",
  "36px",
  "48px",
  "72px"
];
var k = /* @__PURE__ */ ((t) => (t[t["size-small"] = 200] = "size-small", t[t["size-medium"] = 500] = "size-medium", t["size-large"] = "100%", t))(k || {}), f = /* @__PURE__ */ ((t) => (t[t["size-small"] = 480] = "size-small", t[t["size-medium"] = 640] = "size-medium", t["size-large"] = "100%", t))(f || {});
const P = ["Default", "1.5", "2", "2.5", "3", "3.5", "4"], U = ["Paragraph", 1, 2, 3, 4, 5, 6], y = {
  "editor.remove": "Remove",
  "editor.copy": "Copy",
  "editor.words": "WORDS",
  "editor.characters": "CHARACTERS",
  "editor.default": "Default",
  "editor.recent": "Recently Used",
  "editor.nofill": "No Fill",
  "editor.format": "Format Painter",
  "editor.delete": "Delete",
  "editor.edit": "Edit",
  "editor.settings": "Settings",
  "editor.table_of_content": "Table of Content",
  "editor.draghandle.tooltip": "Modify",
  "editor.draghandle.insertBlock": "Insert block",
  "editor.draghandle.grip": "Click for options, hold to drag",
  "editor.copyToClipboard": "CopyToClipboard",
  "editor.slash": "Press '/' for commands",
  "editor.slash.empty": "No Result",
  "editor.slash.format": "Format",
  "editor.slash.insert": "Insert",
  "editor.slash.embed": "Embed Services",
  "editor.content": "Please input content",
  "editor.fontFamily.tooltip": "Font Family",
  "editor.fontFamily.default.tooltip": "Default",
  "editor.moremark": "More Text Styles",
  "editor.size.small.tooltip": "Small",
  "editor.size.medium.tooltip": "Medium",
  "editor.size.large.tooltip": "Cover",
  "editor.bold.tooltip": "Bold",
  "editor.italic.tooltip": "Italic",
  "editor.underline.tooltip": "Underline",
  "editor.strike.tooltip": "Strike",
  "editor.color.tooltip": "Color",
  "editor.color.more": "More Colors",
  "editor.highlight.tooltip": "Highlight",
  "editor.lineheight.tooltip": "Line Height",
  "editor.heading.tooltip": "Headings",
  "editor.heading.h1.tooltip": "Heading 1",
  "editor.heading.h2.tooltip": "Heading 2",
  "editor.heading.h3.tooltip": "Heading 3",
  "editor.heading.h4.tooltip": "Heading 4",
  "editor.heading.h5.tooltip": "Heading 5",
  "editor.heading.h6.tooltip": "Heading 6",
  "editor.paragraph.tooltip": "Paragraph",
  "editor.textalign.tooltip": "Align",
  "editor.textalign.left.tooltip": "Left",
  "editor.textalign.center.tooltip": "Center",
  "editor.textalign.right.tooltip": "Right",
  "editor.textalign.justify.tooltip": "Justify",
  "editor.indent": "Indent",
  "editor.indent.indent": "Increase Indent",
  "editor.indent.outdent": "Decrease Indent",
  "editor.fontSize.tooltip": "Font size",
  "editor.fontSize.default.tooltip": "Default",
  "editor.superscript.tooltip": "Superscript",
  "editor.subscript.tooltip": "Subscript",
  "editor.bulletlist.tooltip": "Bullet List",
  "editor.orderedlist.tooltip": "Ordered List",
  "editor.tasklist.tooltip": "Task List",
  "editor.indent.tooltip": "Indent",
  "editor.outdent.tooltip": "Outdent",
  "editor.columns.tooltip": "Columns",
  "editor.link.tooltip": "Link",
  "editor.link.unlink.tooltip": "UnLink",
  "editor.link.open.tooltip": "Open Link",
  "editor.link.edit.tooltip": "Edit Link",
  "editor.link.dialog.title": "Insert Link",
  "editor.link.dialog.link": "Link",
  "editor.link.dialog.text": "Text",
  "editor.link.dialog.openInNewTab": "Open In New Tab",
  "editor.link.dialog.link.placeholder": "Link Address",
  "editor.link.dialog.text.placeholder": "Add Description",
  "editor.link.dialog.button.apply": "Apply",
  "editor.link.dialog.button.cancel": "Cancel",
  "editor.image.tooltip": "Image",
  "editor.image.dragger.tooltip": "Click or drag the image to the area to upload",
  "editor.image.float.left.tooltip": "Float left",
  "editor.image.float.none.tooltip": "Float none",
  "editor.image.float.right.tooltip": "Float right",
  "editor.image.dialog.title": "Add an image",
  "editor.image.dialog.tab.url": "Url",
  "editor.image.dialog.tab.upload": "Upload",
  "editor.image.dialog.tab.uploadCrop": "Upload & Crop",
  "editor.image.dialog.uploading": "Uploading",
  "editor.link.dialog.inline": "Inline",
  "editor.image.dialog.form.link": "Link",
  "editor.image.dialog.placeholder": "Link",
  "editor.image.dialog.form.alt": "Alt",
  "editor.image.dialog.form.aspectRatio": "Lock original aspect ratio",
  "editor.image.dialog.form.file": "File",
  "editor.image.dialog.button.apply": "Apply",
  "editor.video.tooltip": "Video",
  "editor.video.dialog.tab.upload": "Upload",
  "editor.video.dialog.uploading": "Uploading",
  "editor.video.dialog.title": "Embed or upload a video",
  "editor.video.dialog.link": "Link",
  "editor.video.dialog.placeholder": "Link",
  "editor.video.dialog.button.apply": "Apply",
  "editor.table.tooltip": "Table",
  "editor.table.menu.insert_table": "Insert Table",
  "editor.table.menu.insert_table.with_header_row": "With header row",
  "editor.table.menu.add_column_before": "Add Column Before",
  "editor.table.menu.add_column_after": "Add Column After",
  "editor.table.menu.delete_column": "Delete Column",
  "editor.table.menu.add_row_before": "Add Row Before",
  "editor.table.menu.add_row_after": "Add Row After",
  "editor.table.menu.delete_row": "Delete Row",
  "editor.table.menu.merge_or_split_cells": "Merge Or Split Cells",
  "editor.table.menu.delete_table": "Delete Table",
  "editor.blockquote.tooltip": "Blockquote",
  "editor.horizontalrule.tooltip": "Horizontal Rule",
  "editor.code.tooltip": "Code",
  "editor.codeblock.tooltip": "Code Block",
  "editor.clear.tooltip": "Clear Format",
  "editor.undo.tooltip": "Undo",
  "editor.redo.tooltip": "Redo",
  "editor.fullscreen.tooltip.fullscreen": "Fullscreen",
  "editor.fullscreen.tooltip.exit": "Fullscreen Exit",
  "editor.imageUpload.cancel": "Cancel",
  "editor.imageUpload.crop": "Crop",
  "editor.imageUpload.uploading": "Uploading...",
  "editor.imageUpload.alt": "Alt Text",
  "editor.imageUpload.fileTypeNotSupported": "File type not supported",
  "editor.imageUpload.fileSizeTooBig": "File size too big, Maximum size is",
  "editor.table.menu.insertColumnBefore": "Insert Column Before",
  "editor.table.menu.insertColumnAfter": "Insert Column After",
  "editor.table.menu.deleteColumn": "Delete Column",
  "editor.table.menu.insertRowAbove": "Insert Row Above",
  "editor.table.menu.insertRowBelow": "Insert Row Below",
  "editor.table.menu.deleteRow": "Delete Row",
  "editor.table.menu.mergeCells": "Merge Cells",
  "editor.table.menu.splitCells": "Split Cells",
  "editor.table.menu.deleteTable": "Delete Table",
  "editor.table.menu.setCellsBgColor": "Cell Background Color",
  "editor.emoji.tooltip": "Emoji",
  "editor.iframe.tooltip": "Iframe",
  "editor.searchAndReplace.tooltip": "Search and Replace",
  "editor.search.dialog.text": "Search",
  "editor.replace.dialog.text": "Replace",
  "editor.replaceAll.dialog.text": "Replace All",
  "editor.previous.dialog.text": "Previous",
  "editor.next.dialog.text": "Next",
  no_result_found: "No results found",
  "Smileys & People": "Smileys & People",
  "Animals & Nature": "Animals & Nature",
  "Food & Drink": "Food & Drink",
  Activity: "Activity",
  "Travel & Places": "Travel & Places",
  Object: "Objects",
  Symbol: "Symbols",
  Flags: "Flags",
  "Frequently used": "Frequently used",
  "editor.formula.dialog.text": "Formula",
  "editor.katex.tooltip": "Math Formula",
  "editor.exportPdf.tooltip": "Export PDF",
  "editor.exportWord.tooltip": "Export Word",
  "editor.importWord.tooltip": "Import Word",
  "editor.importWord.limitSize": "File size cannot exceed",
  "editor.importWord.converting": "Converting Word document, please wait...",
  "editor.importWord.convertError": "Error parsing Word document",
  "editor.importWord.importError": "Error importing Word document",
  "editor.textDirection.tooltip": "Text Direction",
  "editor.textDirection.auto.tooltip": "Auto",
  "editor.textDirection.ltr.tooltip": "Left to Right",
  "editor.textDirection.rtl.tooltip": "Right to Left",
  "editor.textDirection.unset.tooltip": "Unset",
  "editor.attachment.tooltip": "Attachment",
  "editor.attachment.uploading": "Uploading",
  "editor.attachment.please_upload": "Please upload the file",
  "editor.imageGif.tooltip": "Gif",
  "editor.replace.caseSensitive": "Case Sensitive",
  "editor.mermaid.tooltip": "Mermaid",
  "editor.twitter.tooltip": "Twitter",
  "editor.codeView.tooltip": "Code View",
  "editor.tooltip.flipX": "Flip Horizontal",
  "editor.tooltip.flipY": "Flip Vertical",
  "editor.upload.fileTypeNotSupported": "{fileName} file type not supported",
  "editor.upload.fileSizeTooBig": "{fileName} file size too big, maximum size is {size}MB",
  "editor.upload.error": "Error uploading file",
  "editor.callout.tooltip": "Callout",
  "editor.callout.dialog.title": "Insert Callout",
  "editor.callout.dialog.type": "Callout Type",
  "editor.callout.dialog.type.placeholder": "Select a callout type",
  "editor.callout.dialog.title.label": "Title",
  "editor.callout.dialog.title.placeholder": "Enter custom title (optional)",
  "editor.callout.dialog.body.label": "Body Content",
  "editor.callout.dialog.body.placeholder": "Enter body content (optional)",
  "editor.callout.dialog.button.apply": "Insert",
  "editor.callout.dialog.button.cancel": "Cancel",
  "editor.callout.edit.title": "Edit Callout",
  "editor.callout.type.note": "Note",
  "editor.callout.type.tip": "Tip",
  "editor.callout.type.important": "Important",
  "editor.callout.type.warning": "Warning",
  "editor.callout.type.caution": "Caution"
}, v = {
  "editor.remove": "Poista",
  "editor.copy": "Kopioi",
  "editor.words": "SANAT",
  "editor.characters": "MERKIT",
  "editor.default": "Oletus",
  "editor.recent": "Viimeksi käytetyt",
  "editor.nofill": "Ei täyttöä",
  "editor.format": "Muotoilun kopioija",
  "editor.delete": "Poista",
  "editor.edit": "Muokkaa",
  "editor.settings": "Asetukset",
  "editor.table_of_content": "Sisällysluettelo",
  "editor.draghandle.tooltip": "Muokkaa",
  "editor.draghandle.insertBlock": "Lisää lohko",
  "editor.draghandle.grip": "Napsauta valikkoon, pidä pohjassa siirtääksesi",
  "editor.copyToClipboard": "Kopioi leikepöydälle",
  "editor.slash": "Paina '/' nähdäksesi komennot",
  "editor.slash.empty": "Ei tuloksia",
  "editor.slash.format": "Muotoile",
  "editor.slash.insert": "Lisää",
  "editor.slash.embed": "Upota palvelut",
  "editor.content": "Syötä sisältö",
  "editor.fontFamily.tooltip": "Fontti",
  "editor.fontFamily.default.tooltip": "Oletus",
  "editor.moremark": "Lisää tekstityylejä",
  "editor.size.small.tooltip": "Pieni",
  "editor.size.medium.tooltip": "Keskikokoinen",
  "editor.size.large.tooltip": "Iso",
  "editor.bold.tooltip": "Lihavointi",
  "editor.italic.tooltip": "Kursivointi",
  "editor.underline.tooltip": "Alleviivaus",
  "editor.strike.tooltip": "Yliviivaus",
  "editor.color.tooltip": "Väri",
  "editor.color.more": "Lisää värejä",
  "editor.highlight.tooltip": "Korosta",
  "editor.lineheight.tooltip": "Riviväli",
  "editor.heading.tooltip": "Otsikot",
  "editor.heading.h1.tooltip": "Otsikko 1",
  "editor.heading.h2.tooltip": "Otsikko 2",
  "editor.heading.h3.tooltip": "Otsikko 3",
  "editor.heading.h4.tooltip": "Otsikko 4",
  "editor.heading.h5.tooltip": "Otsikko 5",
  "editor.heading.h6.tooltip": "Otsikko 6",
  "editor.paragraph.tooltip": "Kappale",
  "editor.textalign.tooltip": "Tasaus",
  "editor.textalign.left.tooltip": "Vasen",
  "editor.textalign.center.tooltip": "Keski",
  "editor.textalign.right.tooltip": "Oikea",
  "editor.textalign.justify.tooltip": "Tasaa",
  "editor.indent": "Sisennys",
  "editor.indent.indent": "Lisää sisennystä",
  "editor.indent.outdent": "Vähennä sisennystä",
  "editor.fontSize.tooltip": "Fontin koko",
  "editor.fontSize.default.tooltip": "Oletus",
  "editor.superscript.tooltip": "Yläindeksi",
  "editor.subscript.tooltip": "Alaindeksi",
  "editor.bulletlist.tooltip": "Lista",
  "editor.orderedlist.tooltip": "Numeroitu lista",
  "editor.tasklist.tooltip": "Tehtävälista",
  "editor.indent.tooltip": "Sisennä",
  "editor.outdent.tooltip": "Poista sisennys",
  "editor.columns.tooltip": "Sarakkeet",
  "editor.link.tooltip": "Linkki",
  "editor.link.unlink.tooltip": "Poista linkki",
  "editor.link.open.tooltip": "Avaa linkki",
  "editor.link.edit.tooltip": "Muokkaa linkkiä",
  "editor.link.dialog.title": "Lisää linkki",
  "editor.link.dialog.link": "Linkki",
  "editor.link.dialog.text": "Teksti",
  "editor.link.dialog.openInNewTab": "Avaa uudessa välilehdessä",
  "editor.link.dialog.link.placeholder": "Linkin osoite",
  "editor.link.dialog.text.placeholder": "Lisää kuvaus",
  "editor.link.dialog.button.apply": "Käytä",
  "editor.link.dialog.button.cancel": "Peruuta",
  "editor.image.tooltip": "Kuva",
  "editor.image.dragger.tooltip": "Klikkaa tai pudota kuva ladataksesi",
  "editor.image.float.left.tooltip": "Asemoi vasemmalle",
  "editor.image.float.none.tooltip": "Ei asemointia",
  "editor.image.float.right.tooltip": "Asemoi oikealle",
  "editor.image.dialog.title": "Lisää kuva",
  "editor.image.dialog.tab.url": "Url",
  "editor.image.dialog.tab.upload": "Lataa",
  "editor.image.dialog.tab.uploadCrop": "Lataa & Rajaa",
  "editor.image.dialog.uploading": "Ladataan",
  "editor.link.dialog.inline": "Rivissä",
  "editor.image.dialog.form.link": "Linkki",
  "editor.image.dialog.placeholder": "Linkki",
  "editor.image.dialog.form.alt": "Alt",
  "editor.image.dialog.form.aspectRatio": "Lukitse alkuperäinen kuvasuhde",
  "editor.image.dialog.form.file": "Tiedosto",
  "editor.image.dialog.button.apply": "Käytä",
  "editor.video.tooltip": "Video",
  "editor.video.dialog.tab.upload": "Lataa",
  "editor.video.dialog.uploading": "Ladataan",
  "editor.video.dialog.title": "Upota tai lataa video",
  "editor.video.dialog.link": "Linkki",
  "editor.video.dialog.placeholder": "Linkki",
  "editor.video.dialog.button.apply": "Käytä",
  "editor.table.tooltip": "Taulukko",
  "editor.table.menu.insert_table": "Lisää taulukko",
  "editor.table.menu.insert_table.with_header_row": "Otsikkorivillä",
  "editor.table.menu.add_column_before": "Lisää sarake ennen",
  "editor.table.menu.add_column_after": "Lisää sarake jälkeen",
  "editor.table.menu.delete_column": "Poista sarake",
  "editor.table.menu.add_row_before": "Lisää rivi ennen",
  "editor.table.menu.add_row_after": "Lisää rivi jälkeen",
  "editor.table.menu.delete_row": "Poista rivi",
  "editor.table.menu.merge_or_split_cells": "Yhdistä tai jaa solut",
  "editor.table.menu.delete_table": "Poista taulukko",
  "editor.blockquote.tooltip": "Lainaus",
  "editor.horizontalrule.tooltip": "Vaakaviiva",
  "editor.code.tooltip": "Koodi",
  "editor.codeblock.tooltip": "Koodilohko",
  "editor.clear.tooltip": "Tyhjennä muotoilu",
  "editor.undo.tooltip": "Kumoa",
  "editor.redo.tooltip": "Tee uudelleen",
  "editor.fullscreen.tooltip.fullscreen": "Koko näyttö",
  "editor.fullscreen.tooltip.exit": "Poistu koko näytöstä",
  "editor.imageUpload.cancel": "Peruuta",
  "editor.imageUpload.crop": "Rajaa",
  "editor.imageUpload.uploading": "Ladataan...",
  "editor.imageUpload.alt": "Alt-Teksti",
  "editor.imageUpload.fileTypeNotSupported": "Tiedostotyyppi ei tuettu",
  "editor.imageUpload.fileSizeTooBig": "Tiedosto liian suuri, suurin sallittu koko on",
  "editor.table.menu.insertColumnBefore": "Lisää sarake ennen",
  "editor.table.menu.insertColumnAfter": "Lisää sarake jälkeen",
  "editor.table.menu.deleteColumn": "Poista sarake",
  "editor.table.menu.insertRowAbove": "Lisää rivi yläpuolelle",
  "editor.table.menu.insertRowBelow": "Lisää rivi alapuolelle",
  "editor.table.menu.deleteRow": "Poista rivi",
  "editor.table.menu.mergeCells": "Yhdistä solut",
  "editor.table.menu.splitCells": "Jaa solut",
  "editor.table.menu.deleteTable": "Poista taulukko",
  "editor.table.menu.setCellsBgColor": "Solun taustaväri",
  "editor.emoji.tooltip": "Emoji",
  "editor.iframe.tooltip": "Iframe",
  "editor.searchAndReplace.tooltip": "Etsi ja korvaa",
  "editor.search.dialog.text": "Etsi",
  "editor.replace.dialog.text": "Korvaa",
  "editor.replaceAll.dialog.text": "Korvaa kaikki",
  "editor.previous.dialog.text": "Edellinen",
  "editor.next.dialog.text": "Seuraava",
  no_result_found: "Ei tuloksia",
  "Smileys & People": "Hymiöt & Ihmiset",
  "Animals & Nature": "Eläimet & Luonto",
  "Food & Drink": "Ruoka & Juoma",
  Activity: "Toiminta",
  "Travel & Places": "Matkat & Paikat",
  Object: "Esineet",
  Symbol: "Symbolit",
  Flags: "Liput",
  "Frequently used": "Usein käytetyt",
  "editor.formula.dialog.text": "Kaava",
  "editor.katex.tooltip": "Matemaattinen kaava",
  "editor.exportPdf.tooltip": "Vie PDF",
  "editor.exportWord.tooltip": "Vie Word",
  "editor.importWord.tooltip": "Tuo Word",
  "editor.importWord.limitSize": "Tiedoston koko ei saa ylittää",
  "editor.importWord.converting": "Muunnetaan Word-dokumenttia, odota...",
  "editor.importWord.convertError": "Virhe Word-dokumentin käsittelyssä",
  "editor.importWord.importError": "Virhe Word-dokumentin tuonnissa",
  "editor.textDirection.tooltip": "Tekstin suunta",
  "editor.textDirection.auto.tooltip": "Automaattinen",
  "editor.textDirection.ltr.tooltip": "Vasemmalta oikealle",
  "editor.textDirection.rtl.tooltip": "Oikealta vasemmalle",
  "editor.textDirection.unset.tooltip": "Poista asetus",
  "editor.attachment.tooltip": "Liite",
  "editor.attachment.uploading": "Ladataan",
  "editor.attachment.please_upload": "Lataa tiedosto",
  "editor.imageGif.tooltip": "Gif",
  "editor.replace.caseSensitive": "Huomioi isot ja pienet kirjaimet",
  "editor.mermaid.tooltip": "Mermaid",
  "editor.twitter.tooltip": "Twitter",
  "editor.codeView.tooltip": "Koodinäkymä",
  "editor.tooltip.flipX": "Käännä vaakasuunnassa",
  "editor.tooltip.flipY": "Käännä pystysuunnassa",
  "editor.upload.fileTypeNotSupported": "{fileName}-tiedostotyyppi ei tuettu",
  "editor.upload.fileSizeTooBig": "{fileName}-tiedosto liian suuri, suurin sallittu koko on {size}MB",
  "editor.upload.error": "Virhe tiedoston latauksessa",
  "editor.callout.tooltip": "Huomiolaatikko",
  "editor.callout.dialog.title": "Lisää huomiolaatikko",
  "editor.callout.dialog.type": "Huomiolaatikon tyyppi",
  "editor.callout.dialog.type.placeholder": "Valitse huomiolaatikon tyyppi",
  "editor.callout.dialog.title.label": "Otsikko",
  "editor.callout.dialog.title.placeholder": "Kirjoita mukautettu otsikko (valinnainen)",
  "editor.callout.dialog.body.label": "Sisältö",
  "editor.callout.dialog.body.placeholder": "Kirjoita sisältö (valinnainen)",
  "editor.callout.dialog.button.apply": "Lisää",
  "editor.callout.dialog.button.cancel": "Peruuta",
  "editor.callout.edit.title": "Muokkaa huomiolaatikkoa",
  "editor.callout.type.note": "Huomio",
  "editor.callout.type.tip": "Vinkki",
  "editor.callout.type.important": "Tärkeä",
  "editor.callout.type.warning": "Varoitus",
  "editor.callout.type.caution": "Varovaisuus"
}, x = {
  "editor.remove": "Eltávolítás",
  "editor.copy": "Másolás",
  "editor.words": "SZAVAK",
  "editor.characters": "KARAKTEREK",
  "editor.default": "Alapértelmezett",
  "editor.recent": "Gyakran használt",
  "editor.nofill": "Kitöltés törlése",
  "editor.format": "Formátummásoló",
  "editor.delete": "Törlés",
  "editor.edit": "Szerkesztés",
  "editor.settings": "Beállítások",
  "editor.table_of_content": "Tartalomjegyzék",
  "editor.draghandle.tooltip": "Módosítás",
  "editor.draghandle.insertBlock": "Blokk beszúrása",
  "editor.draghandle.grip": "Kattintson a menühöz, tartsa nyomva az áthúzáshoz",
  "editor.copyToClipboard": "Másolás vágólapra",
  "editor.slash": "Nyomja meg a '/' gombot a parancsokhoz",
  "editor.slash.empty": "Nincs találat",
  "editor.slash.format": "Formátum",
  "editor.slash.insert": "Beszúrás",
  "editor.slash.embed": "Szolgáltatások beágyazása",
  "editor.content": "Írjon be tartalmat",
  "editor.fontFamily.tooltip": "Betűtípus",
  "editor.fontFamily.default.tooltip": "Alapértelmezett",
  "editor.moremark": "Több betűstílus",
  "editor.size.small.tooltip": "Kicsi",
  "editor.size.medium.tooltip": "Közepes",
  "editor.size.large.tooltip": "Borítókép",
  "editor.bold.tooltip": "Félkövér",
  "editor.italic.tooltip": "Dőlt betű",
  "editor.underline.tooltip": "Aláhúzott",
  "editor.strike.tooltip": "Áthúzott",
  "editor.color.tooltip": "Szín",
  "editor.color.more": "Több szín",
  "editor.highlight.tooltip": "Kiemelés",
  "editor.lineheight.tooltip": "Sorköz",
  "editor.heading.tooltip": "Fejlécek",
  "editor.heading.h1.tooltip": "Fejléc 1",
  "editor.heading.h2.tooltip": "Fejléc 2",
  "editor.heading.h3.tooltip": "Fejléc 3",
  "editor.heading.h4.tooltip": "Fejléc 4",
  "editor.heading.h5.tooltip": "Fejléc 5",
  "editor.heading.h6.tooltip": "Fejléc 6",
  "editor.paragraph.tooltip": "Bekezdés",
  "editor.textalign.tooltip": "Igazítás",
  "editor.textalign.left.tooltip": "Bal",
  "editor.textalign.center.tooltip": "Közép",
  "editor.textalign.right.tooltip": "Jobb",
  "editor.textalign.justify.tooltip": "Sorkizárt",
  "editor.indent": "Behúzás",
  "editor.indent.indent": "Behúzás Növelése",
  "editor.indent.outdent": "Behúzás Csökkentése",
  "editor.fontSize.tooltip": "Betűméret",
  "editor.fontSize.default.tooltip": "Alapértelmezett",
  "editor.superscript.tooltip": "Felső index",
  "editor.subscript.tooltip": "Alsó index",
  "editor.bulletlist.tooltip": "Felsorolás",
  "editor.orderedlist.tooltip": "Rendezett lista",
  "editor.tasklist.tooltip": "Feladat lista",
  "editor.indent.tooltip": "Behúzás növelése",
  "editor.outdent.tooltip": "Behúzás csökkentése",
  "editor.columns.tooltip": "Oszlopok",
  "editor.link.tooltip": "Link",
  "editor.link.unlink.tooltip": "Link törlése",
  "editor.link.open.tooltip": "Link megnyitása",
  "editor.link.edit.tooltip": "Link szerkesztése",
  "editor.link.dialog.title": "Link beszúrása",
  "editor.link.dialog.link": "Link",
  "editor.link.dialog.text": "Szöveg",
  "editor.link.dialog.openInNewTab": "Megnyitás új lapon",
  "editor.link.dialog.link.placeholder": "Link célpont",
  "editor.link.dialog.text.placeholder": "Szöveg",
  "editor.link.dialog.button.apply": "Alkalmaz",
  "editor.link.dialog.button.cancel": "Mégsem",
  "editor.image.tooltip": "Kép",
  "editor.image.dragger.tooltip": "Kattintson vagy húzza a feltölteni kívánt képet a területre",
  "editor.image.float.left.tooltip": "Lebegtetés balra",
  "editor.image.float.none.tooltip": "Lebegtetés törlése",
  "editor.image.float.right.tooltip": "Lebegtetés jobbra",
  "editor.image.dialog.title": "Kép hozzáadása",
  "editor.image.dialog.tab.url": "URL",
  "editor.image.dialog.tab.upload": "Feltöltés",
  "editor.image.dialog.tab.uploadCrop": "Feltöltés és körbevágás",
  "editor.image.dialog.uploading": "Feltöltés alatt",
  "editor.link.dialog.inline": "Soron belüli",
  "editor.image.dialog.form.link": "Link",
  "editor.image.dialog.placeholder": "Link",
  "editor.image.dialog.form.alt": "Alternatív szöveg",
  "editor.image.dialog.form.aspectRatio": "Eredeti képarány megtartása",
  "editor.image.dialog.form.file": "Fájl",
  "editor.image.dialog.button.apply": "Alkalmaz",
  "editor.video.tooltip": "Videó",
  "editor.video.dialog.tab.upload": "Feltöltés",
  "editor.video.dialog.uploading": "Feltöltés alatt",
  "editor.video.dialog.title": "Videó beágyazása vagy feltöltése",
  "editor.video.dialog.link": "Link",
  "editor.video.dialog.placeholder": "Link",
  "editor.video.dialog.button.apply": "Alkalmaz",
  "editor.table.tooltip": "Táblázat",
  "editor.table.menu.insert_table": "Táblázat beszúrása",
  "editor.table.menu.insert_table.with_header_row": "Fejléc sorral",
  "editor.table.menu.add_column_before": "Oszlop beszúrása balra",
  "editor.table.menu.add_column_after": "Oszlop beszúrása jobbra",
  "editor.table.menu.delete_column": "Oszlop törlése",
  "editor.table.menu.add_row_before": "Sor beszúrása felülre",
  "editor.table.menu.add_row_after": "Sor beszúrása alulra",
  "editor.table.menu.delete_row": "Sor törlése",
  "editor.table.menu.merge_or_split_cells": "Cellák egyesítése vagy szétválasztása",
  "editor.table.menu.delete_table": "Táblázat törlése",
  "editor.blockquote.tooltip": "Blokkidézet",
  "editor.horizontalrule.tooltip": "Vízszintes vonal",
  "editor.code.tooltip": "Kód",
  "editor.codeblock.tooltip": "Kód Blokk",
  "editor.clear.tooltip": "Formatálás Törlése",
  "editor.undo.tooltip": "Visszavonás",
  "editor.redo.tooltip": "Visszaállítás",
  "editor.fullscreen.tooltip.fullscreen": "Teljes képernyős mód",
  "editor.fullscreen.tooltip.exit": "Kilépés teljes képernyős módból",
  "editor.imageUpload.cancel": "Mégsem",
  "editor.imageUpload.crop": "Körbevágás",
  "editor.imageUpload.uploading": "Feltöltés...",
  "editor.imageUpload.alt": "Alternatív szöveg",
  "editor.imageUpload.fileTypeNotSupported": "Fájltípus nem támogatott",
  "editor.imageUpload.fileSizeTooBig": "A fájlméret túl nagy, a maximum méret",
  "editor.table.menu.insertColumnBefore": "Oszlop beszúrása balra",
  "editor.table.menu.insertColumnAfter": "Oszlop beszúrása jobbra",
  "editor.table.menu.deleteColumn": "Oszlop törlése",
  "editor.table.menu.insertRowAbove": "Sor beszúrása felülre",
  "editor.table.menu.insertRowBelow": "Sor beszúrása alulra",
  "editor.table.menu.deleteRow": "Sor törlése",
  "editor.table.menu.mergeCells": "Cellák egyesítése",
  "editor.table.menu.splitCells": "Cellék szétválasztása",
  "editor.table.menu.deleteTable": "Táblázat törlése",
  "editor.table.menu.setCellsBgColor": "Cella háttérszín",
  "editor.emoji.tooltip": "Emoji",
  "editor.iframe.tooltip": "Beágyazás",
  "editor.searchAndReplace.tooltip": "Keresés és csere",
  "editor.search.dialog.text": "Keres",
  "editor.replace.dialog.text": "Csere",
  "editor.replaceAll.dialog.text": "Összes cseréje",
  "editor.previous.dialog.text": "Előző",
  "editor.next.dialog.text": "Következő",
  no_result_found: "Nincs találat",
  "Smileys & People": "Arcok és emberek",
  "Animals & Nature": "Állatok és természet",
  "Food & Drink": "Étel és ital",
  Activity: "Aktivitás",
  "Travel & Places": "Utazás és helyek",
  Object: "Tárgyak",
  Symbol: "Szimbólumok",
  Flags: "Zászlók",
  "Frequently used": "Gyakran használt",
  "editor.formula.dialog.text": "Formula",
  "editor.katex.tooltip": "Matematikai formula",
  "editor.exportPdf.tooltip": "PDF exportálása",
  "editor.exportWord.tooltip": "Word exportálása",
  "editor.importWord.tooltip": "Word importálása",
  "editor.importWord.limitSize": "A fájl mérete nem haladhatja meg",
  "editor.importWord.converting": "Word dokumentum konvertálása, kérjük, várjon...",
  "editor.importWord.convertError": "Hiba a Word-dokumentum elemzésekor",
  "editor.importWord.importError": "Hiba a Word dokumentum importálásakor",
  "editor.textDirection.tooltip": "Szövegirány",
  "editor.textDirection.auto.tooltip": "Automatikus",
  "editor.textDirection.ltr.tooltip": "Balról jobbra",
  "editor.textDirection.rtl.tooltip": "Jobbról balra",
  "editor.textDirection.unset.tooltip": "Beállítás visszavonása",
  "editor.attachment.tooltip": "Csatolmány",
  "editor.attachment.uploading": "Feltöltés alatt",
  "editor.attachment.please_upload": "Töltse fel a fájlt",
  "editor.imageGif.tooltip": "Gif",
  "editor.replace.caseSensitive": "Kisbetű-nagybetű érzékenység",
  "editor.mermaid.tooltip": "Mermaid",
  "editor.twitter.tooltip": "Twitter",
  "editor.codeView.tooltip": "Kódnézet",
  "editor.tooltip.flipX": "Vízszintes tükrözés",
  "editor.tooltip.flipY": "Függőleges tükrözés",
  "editor.upload.fileTypeNotSupported": "{fileName} fájltípus nem támogatott",
  "editor.upload.fileSizeTooBig": "{fileName} fájlméret túl nagy, maximális méret {size}MB",
  "editor.upload.error": "Hiba a fájl feltöltésekor",
  "editor.callout.tooltip": "Kiemelő doboz",
  "editor.callout.dialog.title": "Kiemelő doboz beszúrása",
  "editor.callout.dialog.type": "Kiemelő doboz típusa",
  "editor.callout.dialog.type.placeholder": "Válasszon egy kiemelő doboz típust",
  "editor.callout.dialog.title.label": "Cím",
  "editor.callout.dialog.title.placeholder": "Írjon be egyedi címet (opcionális)",
  "editor.callout.dialog.body.label": "Tartalom",
  "editor.callout.dialog.body.placeholder": "Írja be a tartalmat (opcionális)",
  "editor.callout.dialog.button.apply": "Beszúrás",
  "editor.callout.dialog.button.cancel": "Mégsem",
  "editor.callout.edit.title": "Kiemelő doboz szerkesztése",
  "editor.callout.type.note": "Megjegyzés",
  "editor.callout.type.tip": "Tipp",
  "editor.callout.type.important": "Fontos",
  "editor.callout.type.warning": "Figyelmeztetés",
  "editor.callout.type.caution": "Óvatosan"
}, C = {
  "editor.remove": "削除",
  "editor.copy": "コピー",
  "editor.words": "単語数",
  "editor.characters": "文字数",
  "editor.default": "デフォルト",
  "editor.recent": "最近使用した項目",
  "editor.nofill": "塗りなし",
  "editor.format": "書式のコピー",
  "editor.delete": "削除",
  "editor.edit": "編集",
  "editor.settings": "設定",
  "editor.table_of_content": "目次",
  "editor.draghandle.tooltip": "変更",
  "editor.draghandle.insertBlock": "ブロックを挿入",
  "editor.draghandle.grip": "クリックでオプション、長押しでドラッグ",
  "editor.copyToClipboard": "クリップボードにコピー",
  "editor.slash": "「/」でコマンドを表示",
  "editor.slash.empty": "該当なし",
  "editor.slash.format": "書式",
  "editor.slash.insert": "挿入",
  "editor.slash.embed": "埋め込みサービス",
  "editor.content": "内容を入力してください",
  "editor.fontFamily.tooltip": "フォント",
  "editor.fontFamily.default.tooltip": "デフォルト",
  "editor.moremark": "その他のテキストスタイル",
  "editor.size.small.tooltip": "小",
  "editor.size.medium.tooltip": "中",
  "editor.size.large.tooltip": "カバー",
  "editor.bold.tooltip": "太字",
  "editor.italic.tooltip": "斜体",
  "editor.underline.tooltip": "下線",
  "editor.strike.tooltip": "取り消し線",
  "editor.color.tooltip": "文字色",
  "editor.color.more": "その他の色",
  "editor.highlight.tooltip": "ハイライト",
  "editor.lineheight.tooltip": "行間",
  "editor.heading.tooltip": "見出し",
  "editor.heading.h1.tooltip": "見出し1",
  "editor.heading.h2.tooltip": "見出し2",
  "editor.heading.h3.tooltip": "見出し3",
  "editor.heading.h4.tooltip": "見出し4",
  "editor.heading.h5.tooltip": "見出し5",
  "editor.heading.h6.tooltip": "見出し6",
  "editor.paragraph.tooltip": "段落",
  "editor.textalign.tooltip": "配置",
  "editor.textalign.left.tooltip": "左揃え",
  "editor.textalign.center.tooltip": "中央揃え",
  "editor.textalign.right.tooltip": "右揃え",
  "editor.textalign.justify.tooltip": "両端揃え",
  "editor.indent": "インデント",
  "editor.indent.indent": "インデントを増やす",
  "editor.indent.outdent": "インデントを減らす",
  "editor.fontSize.tooltip": "フォントサイズ",
  "editor.fontSize.default.tooltip": "デフォルト",
  "editor.superscript.tooltip": "上付き",
  "editor.subscript.tooltip": "下付き",
  "editor.bulletlist.tooltip": "箇条書き",
  "editor.orderedlist.tooltip": "番号付きリスト",
  "editor.tasklist.tooltip": "タスクリスト",
  "editor.indent.tooltip": "インデント",
  "editor.outdent.tooltip": "アウトデント",
  "editor.columns.tooltip": "列",
  "editor.link.tooltip": "リンク",
  "editor.link.unlink.tooltip": "リンク解除",
  "editor.link.open.tooltip": "リンクを開く",
  "editor.link.edit.tooltip": "リンクを編集",
  "editor.link.dialog.title": "リンクを挿入",
  "editor.link.dialog.link": "リンク",
  "editor.link.dialog.text": "テキスト",
  "editor.link.dialog.openInNewTab": "新しいタブで開く",
  "editor.link.dialog.link.placeholder": "リンク先URL",
  "editor.link.dialog.text.placeholder": "説明を追加",
  "editor.link.dialog.button.apply": "適用",
  "editor.link.dialog.button.cancel": "キャンセル",
  "editor.image.tooltip": "画像",
  "editor.image.dragger.tooltip": "クリックするか、画像をドラッグしてアップロード",
  "editor.image.float.left.tooltip": "左寄せ",
  "editor.image.float.none.tooltip": "回り込みなし",
  "editor.image.float.right.tooltip": "右寄せ",
  "editor.image.dialog.title": "画像を追加",
  "editor.image.dialog.tab.url": "URL",
  "editor.image.dialog.tab.upload": "アップロード",
  "editor.image.dialog.tab.uploadCrop": "アップロードしてトリミング",
  "editor.image.dialog.uploading": "アップロード中",
  "editor.link.dialog.inline": "インライン",
  "editor.image.dialog.form.link": "リンク",
  "editor.image.dialog.placeholder": "リンク",
  "editor.image.dialog.form.alt": "代替テキスト",
  "editor.image.dialog.form.aspectRatio": "元の縦横比を維持",
  "editor.image.dialog.form.file": "ファイル",
  "editor.image.dialog.button.apply": "適用",
  "editor.video.tooltip": "動画",
  "editor.video.dialog.tab.upload": "アップロード",
  "editor.video.dialog.uploading": "アップロード中",
  "editor.video.dialog.title": "動画を埋め込むかアップロード",
  "editor.video.dialog.link": "リンク",
  "editor.video.dialog.placeholder": "リンク",
  "editor.video.dialog.button.apply": "適用",
  "editor.table.tooltip": "表",
  "editor.table.menu.insert_table": "表を挿入",
  "editor.table.menu.insert_table.with_header_row": "ヘッダー行あり",
  "editor.table.menu.add_column_before": "左に列を追加",
  "editor.table.menu.add_column_after": "右に列を追加",
  "editor.table.menu.delete_column": "列を削除",
  "editor.table.menu.add_row_before": "上に行を追加",
  "editor.table.menu.add_row_after": "下に行を追加",
  "editor.table.menu.delete_row": "行を削除",
  "editor.table.menu.merge_or_split_cells": "セルを結合／分割",
  "editor.table.menu.delete_table": "表を削除",
  "editor.blockquote.tooltip": "引用",
  "editor.horizontalrule.tooltip": "区切り線",
  "editor.code.tooltip": "コード",
  "editor.codeblock.tooltip": "コードブロック",
  "editor.clear.tooltip": "書式をクリア",
  "editor.undo.tooltip": "元に戻す",
  "editor.redo.tooltip": "やり直す",
  "editor.fullscreen.tooltip.fullscreen": "全画面表示",
  "editor.fullscreen.tooltip.exit": "全画面を終了",
  "editor.imageUpload.cancel": "キャンセル",
  "editor.imageUpload.crop": "トリミング",
  "editor.imageUpload.uploading": "アップロード中…",
  "editor.imageUpload.alt": "代替テキスト",
  "editor.imageUpload.fileTypeNotSupported": "このファイル形式はサポートされていません",
  "editor.imageUpload.fileSizeTooBig": "ファイルサイズが大きすぎます。上限は次のとおりです",
  "editor.table.menu.insertColumnBefore": "列を前に挿入",
  "editor.table.menu.insertColumnAfter": "列を後に挿入",
  "editor.table.menu.deleteColumn": "列を削除",
  "editor.table.menu.insertRowAbove": "行を上に挿入",
  "editor.table.menu.insertRowBelow": "行を下に挿入",
  "editor.table.menu.deleteRow": "行を削除",
  "editor.table.menu.mergeCells": "セルを結合",
  "editor.table.menu.splitCells": "セルを分割",
  "editor.table.menu.deleteTable": "表を削除",
  "editor.table.menu.setCellsBgColor": "セルの背景色",
  "editor.emoji.tooltip": "絵文字",
  "editor.iframe.tooltip": "iframe",
  "editor.searchAndReplace.tooltip": "検索と置換",
  "editor.search.dialog.text": "検索",
  "editor.replace.dialog.text": "置換",
  "editor.replaceAll.dialog.text": "すべて置換",
  "editor.previous.dialog.text": "前へ",
  "editor.next.dialog.text": "次へ",
  no_result_found: "結果が見つかりません",
  "Smileys & People": "顔文字と人",
  "Animals & Nature": "動物と自然",
  "Food & Drink": "食べ物と飲み物",
  Activity: "活動",
  "Travel & Places": "旅行と場所",
  Object: "物",
  Symbol: "記号",
  Flags: "旗",
  "Frequently used": "よく使う項目",
  "editor.formula.dialog.text": "数式",
  "editor.katex.tooltip": "数式（KaTeX）",
  "editor.exportPdf.tooltip": "PDFに書き出す",
  "editor.exportWord.tooltip": "Wordに書き出す",
  "editor.importWord.tooltip": "Wordを取り込む",
  "editor.importWord.limitSize": "ファイルサイズは次の上限を超えられません",
  "editor.importWord.converting": "Word文書を変換しています。しばらくお待ちください…",
  "editor.importWord.convertError": "Word文書の解析中にエラーが発生しました",
  "editor.importWord.importError": "Word文書の取り込み中にエラーが発生しました",
  "editor.textDirection.tooltip": "文字の向き",
  "editor.textDirection.auto.tooltip": "自動",
  "editor.textDirection.ltr.tooltip": "左から右",
  "editor.textDirection.rtl.tooltip": "右から左",
  "editor.textDirection.unset.tooltip": "未設定",
  "editor.attachment.tooltip": "添付ファイル",
  "editor.attachment.uploading": "アップロード中",
  "editor.attachment.please_upload": "ファイルをアップロードしてください",
  "editor.imageGif.tooltip": "GIF",
  "editor.replace.caseSensitive": "大文字と小文字を区別",
  "editor.mermaid.tooltip": "Mermaid",
  "editor.twitter.tooltip": "Twitter",
  "editor.codeView.tooltip": "コードビュー",
  "editor.tooltip.flipX": "左右反転",
  "editor.tooltip.flipY": "上下反転",
  "editor.upload.fileTypeNotSupported": "{fileName} のファイル形式はサポートされていません",
  "editor.upload.fileSizeTooBig": "{fileName} のサイズが大きすぎます。上限は {size}MB です",
  "editor.upload.error": "ファイルのアップロード中にエラーが発生しました",
  "editor.callout.tooltip": "コールアウト",
  "editor.callout.dialog.title": "コールアウトを挿入",
  "editor.callout.dialog.type": "コールアウトの種類",
  "editor.callout.dialog.type.placeholder": "種類を選択",
  "editor.callout.dialog.title.label": "タイトル",
  "editor.callout.dialog.title.placeholder": "タイトルを入力（任意）",
  "editor.callout.dialog.body.label": "本文",
  "editor.callout.dialog.body.placeholder": "本文を入力（任意）",
  "editor.callout.dialog.button.apply": "挿入",
  "editor.callout.dialog.button.cancel": "キャンセル",
  "editor.callout.edit.title": "コールアウトを編集",
  "editor.callout.type.note": "注記",
  "editor.callout.type.tip": "ヒント",
  "editor.callout.type.important": "重要",
  "editor.callout.type.warning": "警告",
  "editor.callout.type.caution": "注意"
}, z = {
  "editor.remove": "Remover",
  "editor.copy": "Copiar",
  "editor.words": "PALAVRAS",
  "editor.characters": "CARACTERES",
  "editor.default": "Padrão",
  "editor.recent": "Usado recentemente",
  "editor.nofill": "Sem preenchimento",
  "editor.format": "Pincel de formatação",
  "editor.delete": "Deletar",
  "editor.edit": "Editar",
  "editor.settings": "Configurações",
  "editor.table_of_content": "Tabela de conteúdos",
  "editor.draghandle.tooltip": "Modificar",
  "editor.draghandle.insertBlock": "Inserir bloco",
  "editor.draghandle.grip": "Clique para opções, segure para arrastar",
  "editor.copyToClipboard": "Copiar para a área de transferência",
  "editor.slash": "Pressione '/' para comandos",
  "editor.slash.empty": "Nenhum resultado",
  "editor.slash.format": "Formato",
  "editor.slash.insert": "Inserir",
  "editor.slash.embed": "Incorporar serviços",
  "editor.content": "Por favor, insira o conteúdo",
  "editor.fontFamily.tooltip": "Fonte",
  "editor.fontFamily.default.tooltip": "Padrão",
  "editor.moremark": "Mais estilos de texto",
  "editor.size.small.tooltip": "Pequeno",
  "editor.size.medium.tooltip": "Médio",
  "editor.size.large.tooltip": "Grande",
  "editor.bold.tooltip": "Negrito",
  "editor.italic.tooltip": "Itálico",
  "editor.underline.tooltip": "Sublinhado",
  "editor.strike.tooltip": "Riscado",
  "editor.color.tooltip": "Cor",
  "editor.color.more": "Mais cores",
  "editor.highlight.tooltip": "Destaque",
  "editor.lineheight.tooltip": "Altura da linha",
  "editor.heading.tooltip": "Cabeçalhos",
  "editor.heading.h1.tooltip": "Cabeçalho 1",
  "editor.heading.h2.tooltip": "Cabeçalho 2",
  "editor.heading.h3.tooltip": "Cabeçalho 3",
  "editor.heading.h4.tooltip": "Cabeçalho 4",
  "editor.heading.h5.tooltip": "Cabeçalho 5",
  "editor.heading.h6.tooltip": "Cabeçalho 6",
  "editor.paragraph.tooltip": "Parágrafo",
  "editor.textalign.tooltip": "Alinhar",
  "editor.textalign.left.tooltip": "Esquerda",
  "editor.textalign.center.tooltip": "Centro",
  "editor.textalign.right.tooltip": "Direita",
  "editor.textalign.justify.tooltip": "Justificar",
  "editor.indent": "Recuo",
  "editor.indent.indent": "Aumentar recuo",
  "editor.indent.outdent": "Diminuir recuo",
  "editor.fontSize.tooltip": "Tamanho da fonte",
  "editor.fontSize.default.tooltip": "Padrão",
  "editor.superscript.tooltip": "Sobrescrito",
  "editor.subscript.tooltip": "Subscrito",
  "editor.bulletlist.tooltip": "Lista de marcadores",
  "editor.orderedlist.tooltip": "Lista numerada",
  "editor.tasklist.tooltip": "Lista de tarefas",
  "editor.indent.tooltip": "Recuo",
  "editor.outdent.tooltip": "Diminuir recuo",
  "editor.columns.tooltip": "Colunas",
  "editor.link.tooltip": "Link",
  "editor.link.unlink.tooltip": "Desvincular",
  "editor.link.open.tooltip": "Abrir link",
  "editor.link.edit.tooltip": "Editar link",
  "editor.link.dialog.title": "Inserir link",
  "editor.link.dialog.link": "Link",
  "editor.link.dialog.text": "Texto",
  "editor.link.dialog.openInNewTab": "Abrir em nova guia",
  "editor.link.dialog.link.placeholder": "Endereço do link",
  "editor.link.dialog.text.placeholder": "Texto do link",
  "editor.link.dialog.button.apply": "Aplicar",
  "editor.link.dialog.button.cancel": "Cancelar",
  "editor.image.tooltip": "Imagem",
  "editor.image.dragger.tooltip": "Clique ou arraste a imagem para a área de upload",
  "editor.image.float.left.tooltip": "Flutuar à esquerda",
  "editor.image.float.none.tooltip": "Nenhum flutuante",
  "editor.image.float.right.tooltip": "Flutuar à direita",
  "editor.image.dialog.title": "Adicionar uma imagem",
  "editor.image.dialog.tab.url": "Url",
  "editor.image.dialog.tab.upload": "Enviar",
  "editor.link.dialog.inline": "Inline",
  "editor.image.dialog.tab.uploadCrop": "Enviar e cortar",
  "editor.image.dialog.uploading": "Enviando",
  "editor.image.dialog.form.link": "Link",
  "editor.image.dialog.placeholder": "Link",
  "editor.image.dialog.form.alt": "Alt",
  "editor.image.dialog.form.aspectRatio": "Bloquear proporção original",
  "editor.image.dialog.form.file": "Arquivo",
  "editor.image.dialog.button.apply": "Aplicar",
  "editor.video.tooltip": "Vídeo",
  "editor.video.dialog.tab.upload": "Enviar",
  "editor.video.dialog.uploading": "Enviando",
  "editor.video.dialog.title": "Incorporar ou enviar um vídeo",
  "editor.video.dialog.link": "Link",
  "editor.video.dialog.placeholder": "Link",
  "editor.video.dialog.button.apply": "Aplicar",
  "editor.table.tooltip": "Tabela",
  "editor.table.menu.insert_table": "Inserir tabela",
  "editor.table.menu.insert_table.with_header_row": "Com linha de cabeçalho",
  "editor.table.menu.add_column_before": "Adicionar coluna antes",
  "editor.table.menu.add_column_after": "Adicionar coluna depois",
  "editor.table.menu.delete_column": "Excluir coluna",
  "editor.table.menu.add_row_before": "Adicionar linha antes",
  "editor.table.menu.add_row_after": "Adicionar linha depois",
  "editor.table.menu.delete_row": "Deletar linha",
  "editor.table.menu.merge_or_split_cells": "Mesclar ou dividir células",
  "editor.table.menu.delete_table": "Deletar tabela",
  "editor.blockquote.tooltip": "Citação",
  "editor.horizontalrule.tooltip": "Regra horizontal",
  "editor.code.tooltip": "Código",
  "editor.codeblock.tooltip": "Bloco de código",
  "editor.clear.tooltip": "Limpar formato",
  "editor.undo.tooltip": "Desfazer",
  "editor.redo.tooltip": "Refazer",
  "editor.fullscreen.tooltip.fullscreen": "Tela cheia",
  "editor.fullscreen.tooltip.exit": "Sair da tela cheia",
  "editor.imageUpload.cancel": "Cancelar",
  "editor.imageUpload.crop": "Cortar",
  "editor.imageUpload.uploading": "Enviando...",
  "editor.imageUpload.alt": "Texto alternativo",
  "editor.imageUpload.fileTypeNotSupported": "Tipo de arquivo não suportado",
  "editor.imageUpload.fileSizeTooBig": "Tamanho do arquivo muito grande, tamanho máximo é",
  "editor.table.menu.insertColumnBefore": "Inserir coluna antes",
  "editor.table.menu.insertColumnAfter": "Inserir coluna depois",
  "editor.table.menu.deleteColumn": "Excluir coluna",
  "editor.table.menu.insertRowAbove": "Inserir linha acima",
  "editor.table.menu.insertRowBelow": "Inserir linha abaixo",
  "editor.table.menu.deleteRow": "Deletar linha",
  "editor.table.menu.mergeCells": "Mesclar células",
  "editor.table.menu.splitCells": "Dividir células",
  "editor.table.menu.deleteTable": "Deletar tabela",
  "editor.table.menu.setCellsBgColor": "Cor de fundo da célula",
  "editor.emoji.tooltip": "Emoji",
  "editor.iframe.tooltip": "Iframe",
  "editor.searchAndReplace.tooltip": "Pesquisar e substituir",
  "editor.search.dialog.text": "Pesquisar",
  "editor.replace.dialog.text": "Substituir",
  "editor.replaceAll.dialog.text": "Substituir tudo",
  "editor.previous.dialog.text": "Anterior",
  "editor.next.dialog.text": "Próximo",
  no_result_found: "Nenhum resultado encontrado",
  "Smileys & People": "Emoticons & Pessoas",
  "Animals & Nature": "Animais & Natureza",
  "Food & Drink": "Comida & Bebida",
  Activity: "Atividade",
  "Travel & Places": "Viagem & Lugares",
  Object: "Objetos",
  Symbol: "Símbolos",
  Flags: "Bandeiras",
  "Frequently used": "Frequentemente usado",
  "editor.formula.dialog.text": "Fórmula",
  "editor.katex.tooltip": "Fórmula matemática",
  "editor.exportPdf.tooltip": "Exportar PDF",
  "editor.exportWord.tooltip": "Exportar Word",
  "editor.importWord.tooltip": "Importar Word",
  "editor.importWord.limitSize": "O tamanho do arquivo não pode exceder",
  "editor.importWord.converting": "Convertendo documento do Word, aguarde...",
  "editor.importWord.convertError": "Erro ao analisar documento do Word",
  "editor.importWord.importError": "Erro ao importar documento do Word",
  "editor.textDirection.tooltip": "Direção do texto",
  "editor.textDirection.auto.tooltip": "Automático",
  "editor.textDirection.ltr.tooltip": "Esquerda para Direita",
  "editor.textDirection.rtl.tooltip": "Direita para Esquerda",
  "editor.textDirection.unset.tooltip": "Desconfigurar",
  "editor.attachment.tooltip": "Anexo",
  "editor.attachment.uploading": "Enviando",
  "editor.attachment.please_upload": "Por favor, envie o arquivo",
  "editor.imageGif.tooltip": "Gif",
  "editor.replace.caseSensitive": "Sensível a maiúsculas e minúsculas",
  "editor.mermaid.tooltip": "Mermaid",
  "editor.twitter.tooltip": "Twitter",
  "editor.codeView.tooltip": "Visualização de código",
  "editor.tooltip.flipX": "Inverter Horizontal",
  "editor.tooltip.flipY": "Inverter Vertical",
  "editor.upload.fileTypeNotSupported": "{fileName} tipo de arquivo não suportado",
  "editor.upload.fileSizeTooBig": "{fileName} tamanho do arquivo muito grande, o tamanho máximo é {size}MB",
  "editor.upload.error": "Erro ao enviar arquivo",
  "editor.callout.tooltip": "Caixa de destaque",
  "editor.callout.dialog.title": "Inserir caixa de destaque",
  "editor.callout.dialog.type": "Tipo de caixa de destaque",
  "editor.callout.dialog.type.placeholder": "Selecione um tipo de caixa de destaque",
  "editor.callout.dialog.title.label": "Título",
  "editor.callout.dialog.title.placeholder": "Digite um título personalizado (opcional)",
  "editor.callout.dialog.body.label": "Conteúdo",
  "editor.callout.dialog.body.placeholder": "Digite o conteúdo (opcional)",
  "editor.callout.dialog.button.apply": "Inserir",
  "editor.callout.dialog.button.cancel": "Cancelar",
  "editor.callout.edit.title": "Editar caixa de destaque",
  "editor.callout.type.note": "Nota",
  "editor.callout.type.tip": "Dica",
  "editor.callout.type.important": "Importante",
  "editor.callout.type.warning": "Aviso",
  "editor.callout.type.caution": "Cuidado"
}, T = {
  "editor.remove": "Xóa",
  "editor.copy": "Sao chép",
  "editor.words": "TỪ",
  "editor.characters": "KÝ TỰ",
  "editor.default": "Mặc định",
  "editor.recent": "Đã sử dụng gần đây",
  "editor.nofill": "Không tô",
  "editor.format": "Sao chép định dạng",
  "editor.delete": "Xóa",
  "editor.edit": "Chỉnh sửa",
  "editor.settings": "Cài đặt",
  "editor.table_of_content": "Mục lục",
  "editor.draghandle.tooltip": "Sửa đổi",
  "editor.draghandle.insertBlock": "Chèn khối",
  "editor.draghandle.grip": "Nhấn để mở menu, giữ để kéo",
  "editor.copyToClipboard": "CopyToClipboard",
  "editor.slash": "Nhấn '/' để biết lệnh",
  "editor.slash.empty": "Không có kết quả",
  "editor.slash.format": "Định dạng",
  "editor.slash.insert": "Chèn",
  "editor.slash.embed": "Dịch vụ nhúng",
  "editor.content": "Vui lòng nhập nội dung",
  "editor.moremark": "Nhiều kiểu văn bản hơn",
  "editor.size.small.tooltip": "Nhỏ",
  "editor.fontFamily.tooltip": "Phông chữ",
  "editor.fontFamily.default.tooltip": "Mặc định",
  "editor.size.medium.tooltip": "Trung bình",
  "editor.size.large.tooltip": "Lớn",
  "editor.bold.tooltip": "Đậm",
  "editor.italic.tooltip": "In nghiêng",
  "editor.underline.tooltip": "Gạch chân",
  "editor.strike.tooltip": "Gạch",
  "editor.color.tooltip": "Màu",
  "editor.color.more": "Nhiều màu hơn",
  "editor.highlight.tooltip": "Đánh dấu",
  "editor.lineheight.tooltip": "Chiều cao dòng",
  "editor.heading.tooltip": "Tiêu đề",
  "editor.heading.h1.tooltip": "Tiêu đề 1",
  "editor.heading.h2.tooltip": "Tiêu đề 2",
  "editor.heading.h3.tooltip": "Tiêu đề 3",
  "editor.heading.h4.tooltip": "Tiêu đề 4",
  "editor.heading.h5.tooltip": "Tiêu đề 5",
  "editor.heading.h6.tooltip": "Tiêu đề 6",
  "editor.paragraph.tooltip": "Đoạn văn",
  "editor.textalign.tooltip": "Căn chỉnh",
  "editor.textalign.left.tooltip": "Trái",
  "editor.textalign.center.tooltip": "Giữa",
  "editor.textalign.right.tooltip": "Phải",
  "editor.textalign.justify.tooltip": "Căn đều",
  "editor.indent": "Thụt lề",
  "editor.indent.indent": "Tăng thụt lề",
  "editor.indent.outdent": "Giảm thụt lề",
  "editor.fontSize.tooltip": "Kích thước phông chữ",
  "editor.fontSize.default.tooltip": "Mặc định",
  "editor.superscript.tooltip": "Chỉ số trên",
  "editor.subscript.tooltip": "Chỉ số dưới",
  "editor.bulletlist.tooltip": "Danh sách dấu đầu dòng",
  "editor.orderedlist.tooltip": "Danh sách có thứ tự",
  "editor.tasklist.tooltip": "Danh sách tác vụ",
  "editor.indent.tooltip": "Tăng thụt lề",
  "editor.outdent.tooltip": "Giảm thụt lề",
  "editor.columns.tooltip": "Cột",
  "editor.link.tooltip": "Liên kết",
  "editor.link.unlink.tooltip": "Hủy liên kết",
  "editor.link.open.tooltip": "Mở liên kết",
  "editor.link.edit.tooltip": "Chỉnh sửa liên kết",
  "editor.link.dialog.title": "Chèn liên kết",
  "editor.link.dialog.link": "Liên kết",
  "editor.link.dialog.text": "Văn bản",
  "editor.link.dialog.openInNewTab": "Mở trong tab mới",
  "editor.link.dialog.link.placeholder": "Địa chỉ liên kết",
  "editor.link.dialog.text.placeholder": "Thêm mô tả",
  "editor.link.dialog.button.apply": "Áp dụng",
  "editor.link.dialog.button.cancel": "Hủy",
  "editor.image.tooltip": "Hình ảnh",
  "editor.image.dragger.tooltip": "Nhấp hoặc kéo hình ảnh đến khu vực để tải lên",
  "editor.image.float.left.tooltip": "Trôi sang trái",
  "editor.image.float.none.tooltip": "Trôi không",
  "editor.image.float.right.tooltip": "Trôi sang phải",
  "editor.image.dialog.title": "Thêm hình ảnh",
  "editor.image.dialog.tab.url": "Url",
  "editor.image.dialog.tab.upload": "Tải lên",
  "editor.link.dialog.inline": "Inline",
  "editor.image.dialog.tab.uploadCrop": "Tải lên & Cắt",
  "editor.image.dialog.uploading": "Đang tải lên",
  "editor.image.dialog.form.link": "Liên kết",
  "editor.image.dialog.placeholder": "Liên kết",
  "editor.image.dialog.form.alt": "Alt",
  "editor.image.dialog.form.aspectRatio": "Khóa tỷ lệ khung hình gốc",
  "editor.image.dialog.form.file": "Tệp",
  "editor.image.dialog.button.apply": "Áp dụng",
  "editor.video.tooltip": "Video",
  "editor.video.dialog.tab.upload": "Tải lên",
  "editor.video.dialog.uploading": "Đang tải lên",
  "editor.video.dialog.title": "Nhúng hoặc tải lên video",
  "editor.video.dialog.link": "Liên kết",
  "editor.video.dialog.placeholder": "Liên kết",
  "editor.video.dialog.button.apply": "Áp dụng",
  "editor.table.tooltip": "Bảng",
  "editor.table.menu.insert_table": "Chèn Bảng",
  "editor.table.menu.insert_table.with_header_row": "Có hàng tiêu đề",
  "editor.table.menu.add_column_before": "Thêm Cột Trước",
  "editor.table.menu.add_column_after": "Thêm Cột Sau",
  "editor.table.menu.delete_column": "Xóa Cột",
  "editor.table.menu.add_row_before": "Thêm Hàng Trước",
  "editor.table.menu.add_row_after": "Thêm Hàng Sau",
  "editor.table.menu.delete_row": "Xóa Hàng",
  "editor.table.menu.merge_or_split_cells": "Gộp Hoặc Tách Ô",
  "editor.table.menu.delete_table": "Xóa Bảng",
  "editor.blockquote.tooltip": "Blockquote",
  "editor.horizontalrule.tooltip": "Quy tắc ngang",
  "editor.code.tooltip": "Mã",
  "editor.codeblock.tooltip": "Khối mã",
  "editor.clear.tooltip": "Xóa định dạng",
  "editor.undo.tooltip": "Hoàn tác",
  "editor.redo.tooltip": "Làm lại",
  "editor.fullscreen.tooltip.fullscreen": "Toàn màn hình",
  "editor.fullscreen.tooltip.exit": "Thoát toàn màn hình",
  "editor.imageUpload.cancel": "Hủy",
  "editor.imageUpload.crop": "Cắt",
  "editor.imageUpload.uploading": "Đang tải lên...",
  "editor.imageUpload.alt": "Văn bản thay thế",
  "editor.imageUpload.fileTypeNotSupported": "Loại tệp không được hỗ trợ",
  "editor.imageUpload.fileSizeTooBig": "Kích thước tệp quá lớn, Kích thước tối đa là",
  "editor.table.menu.insertColumnBefore": "Chèn cột trước",
  "editor.table.menu.insertColumnAfter": "Chèn cột sau",
  "editor.table.menu.deleteColumn": "Xóa cột",
  "editor.table.menu.insertRowAbove": "Chèn hàng ở trên",
  "editor.table.menu.insertRowBelow": "Chèn hàng ở dưới",
  "editor.table.menu.deleteRow": "Xóa hàng",
  "editor.table.menu.mergeCells": "Gộp các ô",
  "editor.table.menu.splitCells": "Chia ô",
  "editor.table.menu.deleteTable": "Xóa bảng",
  "editor.table.menu.setCellsBgColor": "Màu nền ô",
  "editor.emoji.tooltip": "Biểu tượng",
  "editor.iframe.tooltip": "Iframe",
  "editor.searchAndReplace.tooltip": "Tìm kiếm và thay thế",
  "editor.search.dialog.text": "Tìm kiếm",
  "editor.replace.dialog.text": "Thay thế",
  "editor.replaceAll.dialog.text": "Thay thế tất cả",
  "editor.previous.dialog.text": "Trước",
  "editor.next.dialog.text": "Tiếp theo",
  no_result_found: "Không tìm thấy kết quả",
  "Smileys & People": "Cười & Mọi người",
  "Animals & Nature": "Động vật & Thiên nhiên",
  "Food & Drink": "Thức ăn & Đồ uống",
  Activity: "Hoạt động",
  "Travel & Places": "Du lịch & Địa điểm",
  Object: "Vật",
  Symbol: "Ký hiệu",
  Flags: "Cờ",
  "Frequently used": "Thường xuyên sử dụng",
  "editor.formula.dialog.text": "Công thức",
  "editor.katex.tooltip": "Công thức toán học",
  "editor.exportPdf.tooltip": "Xuất PDF",
  "editor.exportWord.tooltip": "Xuất Word",
  "editor.importWord.tooltip": "Nhập Word",
  "editor.importWord.limitSize": "Kích thước tệp không được vượt quá",
  "editor.importWord.converting": "Đang chuyển đổi tài liệu Word, vui lòng đợi...",
  "editor.importWord.convertError": "Lỗi khi phân tích tài liệu Word",
  "editor.importWord.importError": "Lỗi khi nhập tài liệu Word",
  "editor.textDirection.tooltip": "Hướng văn bản",
  "editor.textDirection.auto.tooltip": "Tự động",
  "editor.textDirection.ltr.tooltip": "Trái sang Phải",
  "editor.textDirection.rtl.tooltip": "Phải sang Trái",
  "editor.textDirection.unset.tooltip": "Bỏ đặt",
  "editor.attachment.tooltip": "Đính kèm",
  "editor.attachment.uploading": "Đang tải lên",
  "editor.attachment.please_upload": "Vui lòng tải lên tệp",
  "editor.imageGif.tooltip": "Gif",
  "editor.replace.caseSensitive": "Phân biệt chữ hoa chữ thường",
  "editor.mermaid.tooltip": "Mermaid",
  "editor.twitter.tooltip": "Twitter",
  "editor.codeView.tooltip": "Xem mã",
  "editor.tooltip.flipX": "Lật Ngang",
  "editor.tooltip.flipY": "Lật Dọc",
  "editor.upload.fileTypeNotSupported": "{fileName} loại tệp không được hỗ trợ",
  "editor.upload.fileSizeTooBig": "{fileName} kích thước tệp quá lớn, kích thước tối đa là {size}MB",
  "editor.upload.error": "Lỗi khi tải lên tệp",
  "editor.callout.tooltip": "Hộp ghi chú",
  "editor.callout.dialog.title": "Chèn hộp ghi chú",
  "editor.callout.dialog.type": "Loại hộp ghi chú",
  "editor.callout.dialog.type.placeholder": "Chọn loại hộp ghi chú",
  "editor.callout.dialog.title.label": "Tiêu đề",
  "editor.callout.dialog.title.placeholder": "Nhập tiêu đề tùy chỉnh (tùy chọn)",
  "editor.callout.dialog.body.label": "Nội dung",
  "editor.callout.dialog.body.placeholder": "Nhập nội dung (tùy chọn)",
  "editor.callout.dialog.button.apply": "Chèn",
  "editor.callout.dialog.button.cancel": "Hủy",
  "editor.callout.edit.title": "Chỉnh sửa hộp ghi chú",
  "editor.callout.type.note": "Ghi chú",
  "editor.callout.type.tip": "Mẹo",
  "editor.callout.type.important": "Quan trọng",
  "editor.callout.type.warning": "Cảnh báo",
  "editor.callout.type.caution": "Thận trọng"
}, F = {
  "editor.remove": "移除",
  "editor.copy": "复制",
  "editor.words": "字数",
  "editor.characters": "字符数",
  "editor.default": "默认",
  "editor.recent": "最近使用",
  "editor.nofill": "无填充",
  "editor.format": "格式刷",
  "editor.delete": "删除",
  "editor.edit": "编辑",
  "editor.settings": "设置",
  "editor.table_of_content": "目录",
  "editor.draghandle.tooltip": "修改",
  "editor.draghandle.insertBlock": "插入块",
  "editor.draghandle.grip": "点击打开菜单，按住可拖拽",
  "editor.copyToClipboard": "复制到剪贴板",
  "editor.slash": "按 '/' 使用命令",
  "editor.slash.empty": "无结果",
  "editor.slash.format": "格式",
  "editor.slash.insert": "插入",
  "editor.slash.embed": "嵌入服务",
  "editor.content": "请输入内容",
  "editor.fontFamily.tooltip": "字体",
  "editor.fontFamily.default.tooltip": "默认",
  "editor.moremark": "更多文本样式",
  "editor.size.small.tooltip": "小",
  "editor.size.medium.tooltip": "中",
  "editor.size.large.tooltip": "封面",
  "editor.bold.tooltip": "粗体",
  "editor.italic.tooltip": "斜体",
  "editor.underline.tooltip": "下划线",
  "editor.strike.tooltip": "删除线",
  "editor.color.tooltip": "颜色",
  "editor.color.more": "更多颜色",
  "editor.highlight.tooltip": "高亮",
  "editor.lineheight.tooltip": "行高",
  "editor.heading.tooltip": "标题",
  "editor.heading.h1.tooltip": "标题 1",
  "editor.heading.h2.tooltip": "标题 2",
  "editor.heading.h3.tooltip": "标题 3",
  "editor.heading.h4.tooltip": "标题 4",
  "editor.heading.h5.tooltip": "标题 5",
  "editor.heading.h6.tooltip": "标题 6",
  "editor.paragraph.tooltip": "段落",
  "editor.textalign.tooltip": "对齐",
  "editor.textalign.left.tooltip": "左对齐",
  "editor.textalign.center.tooltip": "居中",
  "editor.textalign.right.tooltip": "右对齐",
  "editor.textalign.justify.tooltip": "两端对齐",
  "editor.indent": "缩进",
  "editor.indent.indent": "增加缩进",
  "editor.indent.outdent": "减少缩进",
  "editor.fontSize.tooltip": "字号",
  "editor.fontSize.default.tooltip": "默认",
  "editor.superscript.tooltip": "上标",
  "editor.subscript.tooltip": "下标",
  "editor.bulletlist.tooltip": "无序列表",
  "editor.orderedlist.tooltip": "有序列表",
  "editor.tasklist.tooltip": "任务列表",
  "editor.indent.tooltip": "缩进",
  "editor.outdent.tooltip": "减少缩进",
  "editor.columns.tooltip": "列",
  "editor.link.tooltip": "链接",
  "editor.link.unlink.tooltip": "取消链接",
  "editor.link.open.tooltip": "打开链接",
  "editor.link.edit.tooltip": "编辑链接",
  "editor.link.dialog.title": "插入链接",
  "editor.link.dialog.link": "链接",
  "editor.link.dialog.text": "文本",
  "editor.link.dialog.openInNewTab": "在新标签页中打开",
  "editor.link.dialog.link.placeholder": "链接地址",
  "editor.link.dialog.text.placeholder": "添加描述",
  "editor.link.dialog.button.apply": "应用",
  "editor.link.dialog.button.cancel": "取消",
  "editor.image.tooltip": "图片",
  "editor.image.dragger.tooltip": "点击或拖拽图片到此区域上传",
  "editor.image.float.left.tooltip": "左浮动",
  "editor.image.float.none.tooltip": "无浮动",
  "editor.image.float.right.tooltip": "右浮动",
  "editor.image.dialog.title": "添加图片",
  "editor.image.dialog.tab.url": "链接",
  "editor.image.dialog.tab.upload": "上传",
  "editor.link.dialog.inline": "行内嵌入图片",
  "editor.image.dialog.uploading": "上传中",
  "editor.image.dialog.form.link": "链接",
  "editor.image.dialog.placeholder": "链接",
  "editor.image.dialog.form.alt": "替代文本",
  "editor.image.dialog.form.aspectRatio": "锁定原始宽高比",
  "editor.image.dialog.form.file": "文件",
  "editor.image.dialog.button.apply": "应用",
  "editor.video.tooltip": "视频",
  "editor.video.dialog.tab.upload": "上传",
  "editor.image.dialog.tab.uploadCrop": "上传并裁剪",
  "editor.video.dialog.uploading": "上传中",
  "editor.video.dialog.title": "嵌入或上传视频",
  "editor.video.dialog.link": "链接",
  "editor.video.dialog.placeholder": "链接",
  "editor.video.dialog.button.apply": "应用",
  "editor.table.tooltip": "表格",
  "editor.table.menu.insert_table": "插入表格",
  "editor.table.menu.insert_table.with_header_row": "带标题行",
  "editor.table.menu.add_column_before": "在前面插入列",
  "editor.table.menu.add_column_after": "在后面插入列",
  "editor.table.menu.delete_column": "删除列",
  "editor.table.menu.add_row_before": "在上方插入行",
  "editor.table.menu.add_row_after": "在下方插入行",
  "editor.table.menu.delete_row": "删除行",
  "editor.table.menu.merge_or_split_cells": "合并或拆分单元格",
  "editor.table.menu.delete_table": "删除表格",
  "editor.blockquote.tooltip": "引用",
  "editor.horizontalrule.tooltip": "水平线",
  "editor.code.tooltip": "代码",
  "editor.codeblock.tooltip": "代码块",
  "editor.clear.tooltip": "清除格式",
  "editor.undo.tooltip": "撤销",
  "editor.redo.tooltip": "重做",
  "editor.fullscreen.tooltip.fullscreen": "全屏",
  "editor.fullscreen.tooltip.exit": "退出全屏",
  "editor.imageUpload.cancel": "取消",
  "editor.imageUpload.crop": "裁剪",
  "editor.imageUpload.uploading": "上传中...",
  "editor.imageUpload.alt": "替代文本",
  "editor.imageUpload.fileTypeNotSupported": "不支持的文件类型",
  "editor.imageUpload.fileSizeTooBig": "文件大小超出限制，最大大小为",
  "editor.table.menu.insertColumnBefore": "在前面插入列",
  "editor.table.menu.insertColumnAfter": "在后面插入列",
  "editor.table.menu.deleteColumn": "删除列",
  "editor.table.menu.insertRowAbove": "在上方插入行",
  "editor.table.menu.insertRowBelow": "在下方插入行",
  "editor.table.menu.deleteRow": "删除行",
  "editor.table.menu.mergeCells": "合并单元格",
  "editor.table.menu.splitCells": "拆分单元格",
  "editor.table.menu.deleteTable": "删除表格",
  "editor.table.menu.setCellsBgColor": "设置单元格背景色",
  "editor.emoji.tooltip": "表情",
  "editor.iframe.tooltip": "内嵌框架",
  "editor.searchAndReplace.tooltip": "搜索和替换",
  "editor.search.dialog.text": "搜索",
  "editor.replace.dialog.text": "替换",
  "editor.replaceAll.dialog.text": "全部替换",
  "editor.previous.dialog.text": "上一个",
  "editor.next.dialog.text": "下一个",
  no_result_found: "未找到结果",
  "Smileys & People": "笑脸和人",
  "Animals & Nature": "动物和自然",
  "Food & Drink": "食物和饮料",
  Activity: "活动",
  "Travel & Places": "旅行和地点",
  Object: "物体",
  Symbol: "符号",
  Flags: "旗帜",
  "Frequently used": "常用",
  "editor.formula.dialog.text": "公式",
  "editor.katex.tooltip": "数学公式",
  "editor.exportPdf.tooltip": "导出 PDF",
  "editor.exportWord.tooltip": "导出 Word",
  "editor.importWord.tooltip": "导入 Word",
  "editor.importWord.limitSize": "文件大小不能超过",
  "editor.importWord.converting": "正在转换 Word 文档，请稍后...",
  "editor.importWord.convertError": "解析 Word 文档出错",
  "editor.importWord.importError": "导入 Word 文档出错",
  "editor.textDirection.tooltip": "文本方向",
  "editor.textDirection.auto.tooltip": "自动",
  "editor.textDirection.ltr.tooltip": "从左到右",
  "editor.textDirection.rtl.tooltip": "从右到左",
  "editor.textDirection.unset.tooltip": "取消设置",
  "editor.attachment.tooltip": "附件",
  "editor.attachment.uploading": "上传中",
  "editor.attachment.please_upload": "请上传文件",
  "editor.imageGif.tooltip": "Gif",
  "editor.replace.caseSensitive": "区分大小写",
  "editor.mermaid.tooltip": "Mermaid",
  "editor.twitter.tooltip": "Twitter",
  "editor.codeView.tooltip": "代码视图",
  "editor.tooltip.flipX": "水平翻转",
  "editor.tooltip.flipY": "垂直翻转",
  "editor.upload.fileTypeNotSupported": "{fileName} 文件类型不受支持",
  "editor.upload.fileSizeTooBig": "{fileName} 文件大小超出限制，最大大小为 {size}MB",
  "editor.upload.error": "上传文件时出错",
  "editor.callout.tooltip": "标注框",
  "editor.callout.dialog.title": "插入标注框",
  "editor.callout.dialog.type": "标注框类型",
  "editor.callout.dialog.type.placeholder": "选择标注框类型",
  "editor.callout.dialog.title.label": "标题",
  "editor.callout.dialog.title.placeholder": "输入自定义标题（可选）",
  "editor.callout.dialog.body.label": "内容",
  "editor.callout.dialog.body.placeholder": "输入内容（可选）",
  "editor.callout.dialog.button.apply": "插入",
  "editor.callout.dialog.button.cancel": "取消",
  "editor.callout.edit.title": "编辑标注框",
  "editor.callout.type.note": "笔记",
  "editor.callout.type.tip": "提示",
  "editor.callout.type.important": "重要",
  "editor.callout.type.warning": "警告",
  "editor.callout.type.caution": "注意"
}, p = {
  currentLang: b,
  message: {
    en: y,
    hu_HU: x,
    vi: T,
    zh_CN: F,
    pt_BR: z,
    fi: v,
    ja: C
  }
}, i = h({
  currentLang: p.currentLang,
  message: p.message
});
function I() {
  const t = a(i).currentLang, o = a(i).message, e = c(i), s = m(
    (l, d) => {
      try {
        let r = (o[t] || {})[l] || l;
        return d && Object.entries(d).forEach(([u, g]) => {
          r = r.replace(new RegExp(`\\{${u}\\}`, "g"), String(g));
        }), r;
      } catch {
        return l;
      }
    },
    [o, t]
  );
  return {
    setLang: e,
    lang: t,
    t: s
  };
}
const M = {
  setLang: (t) => {
    n(i).setValue((o) => ({
      ...o,
      currentLang: t
    }));
  },
  setMessage: (t, o) => {
    n(i).setValue((e) => ({
      ...e,
      message: {
        ...e.message,
        [t]: {
          ...e.message[t],
          ...o
        }
      }
    }));
  }
};
export {
  R as C,
  j as D,
  U as H,
  L as I,
  B as T,
  f as V,
  v as a,
  x as b,
  C as c,
  M as d,
  z as e,
  T as f,
  F as g,
  W as h,
  P as i,
  w as j,
  D as k,
  y as l,
  E as m,
  N as n,
  k as o,
  I as u
};
