// reactjs-tiptap-editor@1.0.2/lib/index-BaR0Brts.js
import { N as b, w as g, m as v, r as z, E as _, a as Q, p as O, i as w, b as tt, c as et, d as nt, e as V } from "./index-Dx-Dh7KS.js";
import { Fragment as st } from "@tiptap/pm/model";
import { Plugin as rt } from "@tiptap/pm/state";
var it = Object.defineProperty, at = (t, e) => {
  for (var n in e)
    it(t, n, { get: e[n], enumerable: !0 });
}, ot = "listItem", P = "textStyle", H = /^\s*([-+*])\s$/, ct = b.create({
  name: "bulletList",
  addOptions() {
    return {
      itemTypeName: "listItem",
      HTMLAttributes: {},
      keepMarks: !1,
      keepAttributes: !1
    };
  },
  group: "block list",
  content() {
    return `${this.options.itemTypeName}+`;
  },
  parseHTML() {
    return [{ tag: "ul" }];
  },
  renderHTML({ HTMLAttributes: t }) {
    return ["ul", v(this.options.HTMLAttributes, t), 0];
  },
  markdownTokenName: "list",
  parseMarkdown: (t, e) => t.type !== "list" || t.ordered ? [] : {
    type: "bulletList",
    content: t.items ? e.parseChildren(t.items) : []
  },
  renderMarkdown: (t, e) => t.content ? e.renderChildren(t.content, `
`) : "",
  markdownOptions: {
    indentsContent: !0
  },
  addCommands() {
    return {
      toggleBulletList: () => ({ commands: t, chain: e }) => this.options.keepAttributes ? e().toggleList(this.name, this.options.itemTypeName, this.options.keepMarks).updateAttributes(ot, this.editor.getAttributes(P)).run() : t.toggleList(this.name, this.options.itemTypeName, this.options.keepMarks)
    };
  },
  addKeyboardShortcuts() {
    return {
      "Mod-Shift-8": () => this.editor.commands.toggleBulletList()
    };
  },
  addInputRules() {
    let t = g({
      find: H,
      type: this.type
    });
    return (this.options.keepMarks || this.options.keepAttributes) && (t = g({
      find: H,
      type: this.type,
      keepMarks: this.options.keepMarks,
      keepAttributes: this.options.keepAttributes,
      getAttributes: () => this.editor.getAttributes(P),
      editor: this.editor
    })), [t];
  }
}), ut = (t, e, n) => {
  const { selection: s } = t;
  if (!s.empty)
    return null;
  const { $from: r } = s;
  if (!r.parent.isTextblock || r.parentOffset !== r.parent.content.size)
    return null;
  let i = -1;
  for (let d = r.depth; d > 0; d -= 1)
    if (r.node(d).type.name === e) {
      i = d;
      break;
    }
  if (i < 0)
    return null;
  const o = r.node(i), a = r.index(i);
  if (a + 1 >= o.childCount)
    return null;
  const u = o.child(a + 1);
  if (!n.includes(u.type.name))
    return null;
  const h = t.schema.nodes[e];
  let p = !1;
  if (u.forEach((d) => {
    d.type === h && d.childCount > 1 && (p = !0);
  }), !p)
    return null;
  const c = t.doc.resolve(r.after()).nodeAfter;
  if (!c || !n.includes(c.type.name))
    return null;
  const l = [];
  return c.forEach((d) => {
    l.push(d);
  }), l.length === 0 ? null : {
    listItemDepth: i,
    nestedList: c,
    nestedListPos: r.after(),
    insertPos: r.after(i),
    items: l
  };
}, lt = (t, e, n, s) => {
  const r = ut(t, n, s);
  if (!r)
    return !1;
  const { selection: i } = t, { nestedList: o, nestedListPos: a, insertPos: u, items: h } = r, p = t.tr;
  p.delete(a, a + o.nodeSize);
  const c = p.mapping.map(u);
  return p.insert(c, st.from(h)), p.setSelection(i.map(p.doc, p.mapping)), e && e(p), !0;
}, dt = (t, e, n) => lt(t.state, t.view.dispatch, e, n), X = (t, e) => _.create({
  name: `${t}BranchingDeleteKeymap`,
  priority: 101,
  addKeyboardShortcuts() {
    const n = () => dt(this.editor, t, e);
    return {
      Delete: n,
      "Mod-Delete": n
    };
  }
}), F = [
  [1e3, "m"],
  [900, "cm"],
  [500, "d"],
  [400, "cd"],
  [100, "c"],
  [90, "xc"],
  [50, "l"],
  [40, "xl"],
  [10, "x"],
  [9, "ix"],
  [5, "v"],
  [4, "iv"],
  [1, "i"]
], T = "abcdefghijklmnopqrstuvwxyz", pt = "[a-zA-Z]{1,2}", S = String.raw`\d+|[ivxlcdmIVXLCDM]+|${pt}`;
function A(t) {
  let e = t, n = "";
  for (const [s, r] of F)
    for (; e >= s; )
      n += r, e -= s;
  return n;
}
function $(t) {
  return A(t).toUpperCase();
}
function U(t) {
  const e = t.toLowerCase();
  let n = 0, s = 0;
  for (; n < e.length; ) {
    let r = !1;
    for (const [i, o] of F)
      if (e.startsWith(o, n)) {
        s += i, n += o.length, r = !0;
        break;
      }
    if (!r)
      return 0;
  }
  return s;
}
function ht(t) {
  if (!/^[ivxlcdmIVXLCDM]+$/.test(t))
    return !1;
  const e = U(t);
  return e <= 0 ? !1 : (t === t.toLowerCase() ? A(e) : $(e)) === t;
}
function ft(t) {
  const e = t.toLowerCase();
  if (e.length === 1)
    return e.charCodeAt(0) - 97 + 1;
  if (e.length === 2) {
    const n = e.charCodeAt(0) - 97, s = e.charCodeAt(1) - 97;
    return (n + 1) * 26 + s + 1;
  }
  return 0;
}
function I(t) {
  if (t <= 26)
    return T[t - 1];
  const e = Math.floor((t - 1) / 26) - 1, n = (t - 1) % 26;
  return e < 0 ? T[n] : T[e] + T[n];
}
function x(t) {
  if (!(!t || /^\d+$/.test(t))) {
    if (ht(t))
      return t === t.toLowerCase() ? "i" : "I";
    if (/^[a-z]{1,2}$/.test(t))
      return "a";
    if (/^[A-Z]{1,2}$/.test(t))
      return "A";
  }
}
function R(t) {
  if (/^\d+$/.test(t))
    return parseInt(t, 10);
  const e = x(t);
  if (e === "i" || e === "I")
    return U(t);
  if (e === "a" || e === "A") {
    const s = ft(t);
    return s > 0 ? s : 1;
  }
  const n = parseInt(t, 10);
  return Number.isNaN(n) ? 1 : n;
}
function mt(t, e) {
  if (t === "numeric")
    return String(e);
  switch (t) {
    case "a":
      return I(e);
    case "A":
      return I(e).toUpperCase();
    case "i":
      return A(e);
    case "I":
      return $(e);
    default:
      return String(e);
  }
}
function Lt(t) {
  var e;
  if (t.length === 0)
    return !1;
  const n = (e = x(t[0])) != null ? e : "numeric", s = R(t[0]);
  if (s < 1)
    return !1;
  for (let r = 0; r < t.length; r++) {
    const i = mt(n, s + r);
    if (t[r] !== i)
      return !1;
  }
  return !0;
}
function kt(t) {
  return {
    type: x(t),
    start: R(t)
  };
}
function yt(t) {
  const { type: e, start: n } = kt(t), s = {};
  return e && (s.type = e), n !== 1 && (s.start = n), s;
}
function gt(t, e, n = ". ") {
  const s = e + 1;
  if (!t || t === "1")
    return `${s}${n}`;
  switch (t) {
    case "a":
      return `${I(s)}${n}`;
    case "A":
      return `${I(s).toUpperCase()}${n}`;
    case "i":
      return `${A(s)}${n}`;
    case "I":
      return `${$(s)}${n}`;
    default:
      return `${s}${n}`;
  }
}
function bt(t) {
  var e, n;
  const s = (e = t.tokens) == null ? void 0 : e[0];
  return !!(t.text && ((n = t.tokens) == null ? void 0 : n.length) === 1 && (s == null ? void 0 : s.type) === "list" && s.ordered && s.raw === t.text);
}
function vt(t, e) {
  return e.tokenizeInline ? e.parseInline(e.tokenizeInline(t)) : e.parseInline([
    {
      type: "text",
      raw: t,
      text: t
    }
  ]);
}
var Tt = b.create({
  name: "listItem",
  addOptions() {
    return {
      HTMLAttributes: {},
      bulletListTypeName: "bulletList",
      orderedListTypeName: "orderedList"
    };
  },
  content: "paragraph block*",
  defining: !0,
  parseHTML() {
    return [
      {
        tag: "li"
      }
    ];
  },
  renderHTML({ HTMLAttributes: t }) {
    return ["li", v(this.options.HTMLAttributes, t), 0];
  },
  markdownTokenName: "list_item",
  parseMarkdown: (t, e) => {
    var n;
    if (t.type !== "list_item")
      return [];
    const s = (n = e.parseBlockChildren) != null ? n : e.parseChildren;
    let r = [];
    if (t.tokens && t.tokens.length > 0) {
      if (bt(t))
        return {
          type: "listItem",
          content: [
            {
              type: "paragraph",
              content: vt(t.text || "", e)
            }
          ]
        };
      if (t.tokens.some((o) => o.type === "paragraph"))
        r = s(t.tokens);
      else {
        const o = t.tokens[0];
        if (o && o.type === "text" && o.tokens && o.tokens.length > 0) {
          if (r = [
            {
              type: "paragraph",
              content: e.parseInline(o.tokens)
            }
          ], t.tokens.length > 1) {
            const u = t.tokens.slice(1), h = s(u);
            r.push(...h);
          }
        } else
          r = s(t.tokens);
      }
    }
    return r.length === 0 && (r = [
      {
        type: "paragraph",
        content: []
      }
    ]), {
      type: "listItem",
      content: r
    };
  },
  renderMarkdown: (t, e, n) => z(
    t,
    e,
    (s) => {
      var r, i, o, a;
      if (s.parentType === "bulletList")
        return "- ";
      if (s.parentType === "orderedList") {
        const u = ((i = (r = s.meta) == null ? void 0 : r.parentAttrs) == null ? void 0 : i.start) || 1, h = (a = (o = s.meta) == null ? void 0 : o.parentAttrs) == null ? void 0 : a.type, p = u - 1 + (s.index || 0);
        return gt(h, p, ". ");
      }
      return "- ";
    },
    n
  ),
  addExtensions() {
    return [
      X(this.name, [
        this.options.bulletListTypeName,
        this.options.orderedListTypeName
      ])
    ];
  },
  addKeyboardShortcuts() {
    return {
      Enter: () => this.editor.commands.splitListItem(this.name),
      Tab: () => this.editor.commands.sinkListItem(this.name),
      "Shift-Tab": () => this.editor.commands.liftListItem(this.name)
    };
  }
}), It = {};
at(It, {
  findListItemPos: () => M,
  getNextListDepth: () => D,
  handleBackspace: () => C,
  handleDelete: () => E,
  hasListBefore: () => W,
  hasListItemAfter: () => At,
  hasListItemBefore: () => xt,
  listItemHasSubList: () => Mt,
  nextListIsDeeper: () => q,
  nextListIsHigher: () => G
});
var M = (t, e) => {
  const { $from: n } = e.selection, s = V(t, e.schema);
  let r = null, i = n.depth, o = n.pos, a = null;
  for (; i > 0 && a === null; )
    r = n.node(i), r.type === s ? a = i : (i -= 1, o -= 1);
  return a === null ? null : { $pos: e.doc.resolve(o), depth: a };
}, D = (t, e) => {
  const n = M(t, e);
  if (!n)
    return !1;
  const [, s] = nt(e, t, n.$pos.pos + 4);
  return s;
}, W = (t, e, n) => {
  const { $anchor: s } = t.selection, r = Math.max(0, s.pos - 2), i = t.doc.resolve(r).node();
  return !(!i || !n.includes(i.type.name));
}, C = (t, e, n) => {
  if (t.commands.undoInputRule())
    return !0;
  if (t.state.selection.from !== t.state.selection.to)
    return !1;
  if (!w(t.state, e) && W(t.state, e, n)) {
    const { $anchor: s } = t.state.selection, r = t.state.doc.resolve(s.before() - 1), i = [];
    r.node().descendants((u, h) => {
      u.type.name === e && i.push({ node: u, pos: h });
    });
    const o = i.at(-1);
    if (!o)
      return !1;
    const a = t.state.doc.resolve(r.start() + o.pos + 1);
    return t.chain().cut({ from: s.start() - 1, to: s.end() + 1 }, a.end()).joinForward().run();
  }
  return !w(t.state, e) || !tt(t.state) ? !1 : t.chain().liftListItem(e).run();
}, q = (t, e) => {
  const n = D(t, e), s = M(t, e);
  return !s || !n ? !1 : n > s.depth;
}, G = (t, e) => {
  const n = D(t, e), s = M(t, e);
  return !s || !n ? !1 : n < s.depth;
}, E = (t, e) => {
  if (!w(t.state, e) || !et(t.state, e))
    return !1;
  const { selection: n } = t.state, { $from: s, $to: r } = n;
  return !n.empty && s.sameParent(r) ? !1 : q(e, t.state) ? t.chain().focus(t.state.selection.from + 4).lift(e).joinBackward().run() : G(e, t.state) ? t.chain().joinForward().joinBackward().run() : t.commands.joinItemForward();
}, At = (t, e) => {
  var n;
  const { $anchor: s } = e.selection, r = e.doc.resolve(s.pos - s.parentOffset - 2);
  return !(r.index() === r.parent.childCount - 1 || ((n = r.nodeAfter) == null ? void 0 : n.type.name) !== t);
}, xt = (t, e) => {
  var n;
  const { $anchor: s } = e.selection, r = e.doc.resolve(s.pos - 2);
  return !(r.index() === 0 || ((n = r.nodeBefore) == null ? void 0 : n.type.name) !== t);
}, Mt = (t, e, n) => {
  if (!n)
    return !1;
  const s = V(t, e.schema);
  let r = !1;
  return n.descendants((i) => {
    i.type === s && (r = !0);
  }), r;
}, wt = _.create({
  name: "listKeymap",
  addOptions() {
    return {
      listTypes: [
        {
          itemName: "listItem",
          wrapperNames: ["bulletList", "orderedList"]
        },
        {
          itemName: "taskItem",
          wrapperNames: ["taskList"]
        }
      ]
    };
  },
  addKeyboardShortcuts() {
    return {
      Delete: ({ editor: t }) => {
        let e = !1;
        return this.options.listTypes.forEach(({ itemName: n }) => {
          t.state.schema.nodes[n] !== void 0 && E(t, n) && (e = !0);
        }), e;
      },
      "Mod-Delete": ({ editor: t }) => {
        let e = !1;
        return this.options.listTypes.forEach(({ itemName: n }) => {
          t.state.schema.nodes[n] !== void 0 && E(t, n) && (e = !0);
        }), e;
      },
      Backspace: ({ editor: t }) => {
        let e = !1;
        return this.options.listTypes.forEach(({ itemName: n, wrapperNames: s }) => {
          t.state.schema.nodes[n] !== void 0 && C(t, n, s) && (e = !0);
        }), e;
      },
      "Mod-Backspace": ({ editor: t }) => {
        let e = !1;
        return this.options.listTypes.forEach(({ itemName: n, wrapperNames: s }) => {
          t.state.schema.nodes[n] !== void 0 && C(t, n, s) && (e = !0);
        }), e;
      }
    };
  }
}), N = new RegExp(
  `^(\\s*)(${S})([.)])\\s+(.*)$`
), Ct = new RegExp(
  `^(\\s*)(${S})([.)])\\s+`
), Et = /^\s/;
function Nt(t) {
  return N.test(t.trimStart());
}
function _t(t) {
  const e = t.trimStart();
  return (
    // oxlint-disable-next-line prefer-string-starts-ends-with
    /^[-+*]\s+/.test(e) || Nt(e) || // oxlint-disable-next-line prefer-string-starts-ends-with
    /^>\s?/.test(e) || // oxlint-disable-next-line prefer-string-starts-ends-with
    /^```/.test(e) || // oxlint-disable-next-line prefer-string-starts-ends-with
    /^~~~/.test(e)
  );
}
function St(t) {
  const e = [], n = [];
  let s = !1;
  return t.forEach((r) => {
    if (s) {
      n.push(r);
      return;
    }
    if (r.trim() === "") {
      s = !0, n.push(r);
      return;
    }
    if (e.length > 0 && _t(r)) {
      s = !0, n.push(r);
      return;
    }
    e.push(r);
  }), {
    paragraphLines: e,
    blockLines: n
  };
}
function $t(t) {
  const e = [];
  let n = 0, s = 0;
  for (; n < t.length; ) {
    const r = t[n], i = r.match(N);
    if (!i)
      break;
    const [, o, a, u, h] = i, p = o.length, c = parseInt(a, 10), l = isNaN(c) ? x(a) : void 0, d = isNaN(c) ? R(a) : c, L = [h];
    let f = n + 1;
    const m = [r];
    let y = !1;
    for (; f < t.length; ) {
      const k = t[f];
      if (k.match(N))
        break;
      if (k.trim() === "")
        m.push(k), L.push(""), y = !0, f += 1;
      else if (k.match(Et)) {
        const J = k.length - k.trimStart().length, Y = p + a.length + 1;
        m.push(k), L.push(k.slice(Math.min(J, Y))), f += 1;
      } else {
        if (y)
          break;
        m.push(k), L.push(k), f += 1;
      }
    }
    e.push({
      indent: p,
      number: d,
      type: l,
      content: L.join(`
`).trim(),
      contentLines: L,
      raw: m.join(`
`)
    }), s = f, n = f;
  }
  return [e, s];
}
var Rt = new RegExp(
  `^(${S})([.)])\\s+(.+)$`
);
function Dt(t) {
  const e = t.split(`
`).filter((i) => i.trim().length > 0);
  if (e.length === 0)
    return null;
  const n = [];
  for (const i of e) {
    const o = i.trim().match(Rt);
    if (!o)
      return null;
    n.push({
      marker: o[1],
      content: o[3]
    });
  }
  const s = n.map((i) => i.marker);
  return Lt(s) ? {
    type: "orderedList",
    attrs: yt(n[0].marker),
    content: n.map((i) => ({
      type: "listItem",
      content: [
        {
          type: "paragraph",
          content: [{ type: "text", text: i.content }]
        }
      ]
    }))
  } : null;
}
function Z(t, e, n) {
  const s = [];
  let r = 0;
  for (; r < t.length; ) {
    const i = t[r];
    if (i.indent === e) {
      const { paragraphLines: o, blockLines: a } = St(i.contentLines), u = o.join(`
`).trim(), h = [];
      u && h.push({
        type: "paragraph",
        raw: u,
        tokens: n.inlineTokens(u)
      });
      const p = a.join(`
`).trim();
      if (p) {
        const d = n.blockTokens(p);
        h.push(...d);
      }
      let c = r + 1;
      const l = [];
      for (; c < t.length && t[c].indent > e; )
        l.push(t[c]), c += 1;
      if (l.length > 0) {
        const d = Math.min(...l.map((f) => f.indent)), L = Z(l, d, n);
        h.push({
          type: "list",
          ordered: !0,
          start: l[0].number,
          typeMarker: l[0].type,
          items: L,
          raw: l.map((f) => f.raw).join(`
`)
        });
      }
      s.push({
        type: "list_item",
        raw: i.raw,
        tokens: h
      }), r = c;
    } else
      r += 1;
  }
  return s;
}
function Ot(t, e) {
  return t.map((n) => {
    if (n.type !== "list_item")
      return e.parseChildren([n])[0];
    const s = [];
    return n.tokens && n.tokens.length > 0 && n.tokens.forEach((r) => {
      if (r.type === "paragraph" || r.type === "list" || r.type === "blockquote" || r.type === "code")
        s.push(...e.parseChildren([r]));
      else if (r.type === "text" && r.tokens) {
        const i = e.parseChildren([r]);
        s.push({
          type: "paragraph",
          content: i
        });
      } else {
        const i = e.parseChildren([r]);
        i.length > 0 && s.push(...i);
      }
    }), {
      type: "listItem",
      content: s
    };
  });
}
var Pt = "listItem", B = "textStyle", j = /^(\d+)\.\s$/;
function K(t) {
  const e = t.match(/list-style-type\s*:\s*([^;]+)/i);
  if (!e)
    return null;
  switch (e[1].trim().toLowerCase()) {
    case "upper-roman":
      return "I";
    case "lower-roman":
      return "i";
    case "upper-alpha":
    case "upper-latin":
      return "A";
    case "lower-alpha":
    case "lower-latin":
      return "a";
    default:
      return null;
  }
}
var Ht = b.create({
  name: "orderedList",
  addOptions() {
    return {
      itemTypeName: "listItem",
      HTMLAttributes: {},
      keepMarks: !1,
      keepAttributes: !1
    };
  },
  group: "block list",
  content() {
    return `${this.options.itemTypeName}+`;
  },
  addAttributes() {
    return {
      start: {
        default: 1,
        parseHTML: (t) => t.hasAttribute("start") ? parseInt(t.getAttribute("start") || "", 10) : 1
      },
      type: {
        default: null,
        parseHTML: (t) => {
          const e = t.getAttribute("type");
          if (e)
            return e;
          const n = t.getAttribute("style");
          if (n) {
            const r = K(n);
            if (r)
              return r;
          }
          const s = t.querySelector("li");
          if (s) {
            const r = s.getAttribute("style");
            if (r) {
              const i = K(r);
              if (i)
                return i;
            }
          }
          return null;
        }
      }
    };
  },
  parseHTML() {
    return [
      {
        tag: "ol"
      }
    ];
  },
  renderHTML({ HTMLAttributes: t }) {
    const { start: e, type: n, ...s } = t, r = v(this.options.HTMLAttributes, s);
    return e !== 1 && (r.start = e), n && n !== "1" && (r.type = n), ["ol", r, 0];
  },
  markdownTokenName: "list",
  parseMarkdown: (t, e) => {
    if (t.type !== "list" || !t.ordered)
      return [];
    const n = t.start || 1, s = t.typeMarker, r = t.items ? Ot(t.items, e) : [], i = {};
    return n !== 1 && (i.start = n), s && (i.type = s), Object.keys(i).length > 0 ? {
      type: "orderedList",
      attrs: i,
      content: r
    } : {
      type: "orderedList",
      content: r
    };
  },
  renderMarkdown: (t, e) => t.content ? e.renderChildren(t.content, `
`) : "",
  markdownTokenizer: {
    name: "orderedList",
    level: "block",
    start: (t) => {
      const e = t.match(Ct), n = e == null ? void 0 : e.index;
      return n !== void 0 ? n : -1;
    },
    tokenize: (t, e, n) => {
      var s, r;
      const i = t.split(`
`), [o, a] = $t(i);
      if (o.length === 0)
        return;
      const u = Z(o, 0, n);
      if (u.length === 0)
        return;
      const h = ((s = o[0]) == null ? void 0 : s.number) || 1, p = (r = o[0]) == null ? void 0 : r.type;
      return {
        type: "list",
        ordered: !0,
        start: h,
        typeMarker: p,
        items: u,
        raw: i.slice(0, a).join(`
`)
      };
    }
  },
  markdownOptions: {
    indentsContent: !0
  },
  addCommands() {
    return {
      toggleOrderedList: () => ({ commands: t, chain: e }) => this.options.keepAttributes ? e().toggleList(this.name, this.options.itemTypeName, this.options.keepMarks).updateAttributes(Pt, this.editor.getAttributes(B)).run() : t.toggleList(this.name, this.options.itemTypeName, this.options.keepMarks)
    };
  },
  addKeyboardShortcuts() {
    return {
      "Mod-Shift-7": () => this.editor.commands.toggleOrderedList()
    };
  },
  addProseMirrorPlugins() {
    return [
      new rt({
        props: {
          handlePaste: (t, e) => {
            var n, s;
            const r = (n = e.clipboardData) == null ? void 0 : n.getData("text/html");
            if (r != null && r.trim())
              return !1;
            const i = (s = e.clipboardData) == null ? void 0 : s.getData("text/plain");
            if (!i)
              return !1;
            const o = Dt(i);
            if (!o)
              return !1;
            try {
              const a = t.state.schema.nodeFromJSON(o), u = t.state.tr.replaceSelectionWith(a);
              return t.dispatch(u), !0;
            } catch {
              return !1;
            }
          }
        }
      })
    ];
  },
  addInputRules() {
    const t = (n, s) => (!s.attrs.type || s.attrs.type === "1") && s.childCount + s.attrs.start === +n[1];
    let e = g({
      find: j,
      type: this.type,
      getAttributes: (n) => ({ start: +n[1] }),
      joinPredicate: t
    });
    return (this.options.keepMarks || this.options.keepAttributes) && (e = g({
      find: j,
      type: this.type,
      keepMarks: this.options.keepMarks,
      keepAttributes: this.options.keepAttributes,
      getAttributes: (n) => ({ start: +n[1], ...this.editor.getAttributes(B) }),
      joinPredicate: t,
      editor: this.editor
    })), [e];
  }
}), Bt = /^\s*(\[([( |x])?\])\s$/, jt = b.create({
  name: "taskItem",
  addOptions() {
    return {
      nested: !1,
      HTMLAttributes: {},
      taskListTypeName: "taskList",
      a11y: void 0
    };
  },
  content() {
    return this.options.nested ? "paragraph block*" : "paragraph+";
  },
  defining: !0,
  addAttributes() {
    return {
      checked: {
        default: !1,
        keepOnSplit: !1,
        parseHTML: (t) => {
          const e = t.getAttribute("data-checked");
          return e === "" || e === "true";
        },
        renderHTML: (t) => ({
          "data-checked": t.checked
        })
      }
    };
  },
  parseHTML() {
    return [
      {
        tag: `li[data-type="${this.name}"]`,
        priority: 51
      }
    ];
  },
  renderHTML({ node: t, HTMLAttributes: e }) {
    return [
      "li",
      v(this.options.HTMLAttributes, e, {
        "data-type": this.name
      }),
      [
        "label",
        [
          "input",
          {
            type: "checkbox",
            checked: t.attrs.checked ? "checked" : null
          }
        ],
        ["span"]
      ],
      ["div", 0]
    ];
  },
  parseMarkdown: (t, e) => {
    const n = [];
    if (t.tokens && t.tokens.length > 0 ? n.push(e.createNode("paragraph", {}, e.parseInline(t.tokens))) : t.text ? n.push(e.createNode("paragraph", {}, [e.createNode("text", { text: t.text })])) : n.push(e.createNode("paragraph", {}, [])), t.nestedTokens && t.nestedTokens.length > 0) {
      const s = e.parseChildren(t.nestedTokens);
      n.push(...s);
    }
    return e.createNode("taskItem", { checked: t.checked || !1 }, n);
  },
  renderMarkdown: (t, e) => {
    var n;
    const r = `- [${(n = t.attrs) != null && n.checked ? "x" : " "}] `;
    return z(t, e, r);
  },
  addExtensions() {
    return this.options.nested ? [X(this.name, [this.options.taskListTypeName])] : [];
  },
  addKeyboardShortcuts() {
    const t = {
      Enter: () => this.editor.commands.splitListItem(this.name),
      "Shift-Tab": () => this.editor.commands.liftListItem(this.name)
    };
    return this.options.nested ? {
      ...t,
      Tab: () => this.editor.commands.sinkListItem(this.name)
    } : t;
  },
  addNodeView() {
    return ({ node: t, HTMLAttributes: e, getPos: n, editor: s }) => {
      const r = document.createElement("li"), i = document.createElement("label"), o = document.createElement("span"), a = document.createElement("input"), u = document.createElement("div"), h = (c) => {
        var l, d;
        a.ariaLabel = ((d = (l = this.options.a11y) == null ? void 0 : l.checkboxLabel) == null ? void 0 : d.call(l, c, a.checked)) || `Task item checkbox for ${c.textContent || "empty task item"}`;
      };
      h(t), i.contentEditable = "false", a.type = "checkbox", a.addEventListener("mousedown", (c) => c.preventDefault()), a.addEventListener("change", (c) => {
        if (!s.isEditable && !this.options.onReadOnlyChecked) {
          a.checked = !a.checked;
          return;
        }
        const { checked: l } = c.target;
        s.isEditable && typeof n == "function" && s.chain().focus(void 0, { scrollIntoView: !1 }).command(({ tr: d }) => {
          const L = n();
          if (typeof L != "number")
            return !1;
          const f = d.doc.nodeAt(L);
          return d.setNodeMarkup(L, void 0, {
            ...f == null ? void 0 : f.attrs,
            checked: l
          }), !0;
        }).run(), !s.isEditable && this.options.onReadOnlyChecked && (this.options.onReadOnlyChecked(t, l) || (a.checked = !a.checked));
      }), Object.entries(this.options.HTMLAttributes).forEach(([c, l]) => {
        r.setAttribute(c, l);
      }), r.dataset.checked = t.attrs.checked, a.checked = t.attrs.checked, i.append(a, o), r.append(i, u), Object.entries(e).forEach(([c, l]) => {
        r.setAttribute(c, l);
      });
      let p = new Set(Object.keys(e));
      return {
        dom: r,
        contentDOM: u,
        update: (c) => {
          if (c.type !== this.type)
            return !1;
          r.dataset.checked = c.attrs.checked, a.checked = c.attrs.checked, h(c);
          const l = s.extensionManager.attributes, d = Q(c, l), L = new Set(Object.keys(d)), f = this.options.HTMLAttributes;
          return p.forEach((m) => {
            L.has(m) || (m in f ? r.setAttribute(m, f[m]) : r.removeAttribute(m));
          }), Object.entries(d).forEach(([m, y]) => {
            y == null ? m in f ? r.setAttribute(m, f[m]) : r.removeAttribute(m) : r.setAttribute(m, y);
          }), p = L, !0;
        }
      };
    };
  },
  addInputRules() {
    return [
      g({
        find: Bt,
        type: this.type,
        getAttributes: (t) => ({
          checked: t[t.length - 1] === "x"
        })
      })
    ];
  }
}), Kt = b.create({
  name: "taskList",
  addOptions() {
    return {
      itemTypeName: "taskItem",
      HTMLAttributes: {}
    };
  },
  group: "block list",
  content() {
    return `${this.options.itemTypeName}+`;
  },
  parseHTML() {
    return [
      {
        tag: `ul[data-type="${this.name}"]`,
        priority: 51
      }
    ];
  },
  renderHTML({ HTMLAttributes: t }) {
    return [
      "ul",
      v(this.options.HTMLAttributes, t, { "data-type": this.name }),
      0
    ];
  },
  parseMarkdown: (t, e) => e.createNode("taskList", {}, e.parseChildren(t.items || [])),
  renderMarkdown: (t, e) => t.content ? e.renderChildren(t.content, `
`) : "",
  markdownTokenizer: {
    name: "taskList",
    level: "block",
    start(t) {
      var e;
      const n = (e = t.match(/^\s*[-+*]\s+\[([ xX])\]\s+/)) == null ? void 0 : e.index;
      return n !== void 0 ? n : -1;
    },
    tokenize(t, e, n) {
      const s = (i) => {
        const o = O(
          i,
          {
            itemPattern: /^(\s*)([-+*])\s+\[([ xX])\]\s+(.*)$/,
            extractItemData: (a) => ({
              indentLevel: a[1].length,
              mainContent: a[4],
              checked: a[3].toLowerCase() === "x"
            }),
            createToken: (a, u) => ({
              type: "taskItem",
              raw: "",
              mainContent: a.mainContent,
              indentLevel: a.indentLevel,
              checked: a.checked,
              text: a.mainContent,
              tokens: n.inlineTokens(a.mainContent),
              nestedTokens: u
            }),
            // Allow recursive nesting
            customNestedParser: s
          },
          n
        );
        return o ? [
          {
            type: "taskList",
            raw: o.raw,
            items: o.items
          }
        ] : n.blockTokens(i);
      }, r = O(
        t,
        {
          itemPattern: /^(\s*)([-+*])\s+\[([ xX])\]\s+(.*)$/,
          extractItemData: (i) => ({
            indentLevel: i[1].length,
            mainContent: i[4],
            checked: i[3].toLowerCase() === "x"
          }),
          createToken: (i, o) => ({
            type: "taskItem",
            raw: "",
            mainContent: i.mainContent,
            indentLevel: i.indentLevel,
            checked: i.checked,
            text: i.mainContent,
            tokens: n.inlineTokens(i.mainContent),
            nestedTokens: o
          }),
          // Use the recursive parser for nested content
          customNestedParser: s
        },
        n
      );
      if (r)
        return {
          type: "taskList",
          raw: r.raw,
          items: r.items
        };
    }
  },
  markdownOptions: {
    indentsContent: !0
  },
  addCommands() {
    return {
      toggleTaskList: () => ({ commands: t }) => t.toggleList(this.name, this.options.itemTypeName)
    };
  },
  addKeyboardShortcuts() {
    return {
      "Mod-Shift-9": () => this.editor.commands.toggleTaskList()
    };
  }
});
_.create({
  name: "listKit",
  addExtensions() {
    const t = [];
    return this.options.bulletList !== !1 && t.push(ct.configure(this.options.bulletList)), this.options.listItem !== !1 && t.push(Tt.configure(this.options.listItem)), this.options.listKeymap !== !1 && t.push(wt.configure(this.options.listKeymap)), this.options.orderedList !== !1 && t.push(Ht.configure(this.options.orderedList)), this.options.taskItem !== !1 && t.push(jt.configure(this.options.taskItem)), this.options.taskList !== !1 && t.push(Kt.configure(this.options.taskList)), t;
  }
});
export {
  ct as B,
  Ht as O,
  Kt as T,
  jt as a
};
