// reactjs-tiptap-editor@1.0.2/lib/index-C52cuQ_Y.js
import { jsxs as C, jsx as n, Fragment as K } from "react/jsx-runtime";
import * as O from "react";
import q, { useState as R, useMemo as F, useCallback as ue, useEffect as Se, useRef as ve } from "react";
import { Slot as ht } from "@radix-ui/react-slot";
import { y as Xt, b as Yt, S as qt } from "./index-BG0kQamI.js";
import "./theme.js";
import { c as ft, N as Zt, u as Kt, b as Jt, a as Qt } from "./clsx-_bq4MQQ7.js";
import * as ne from "@radix-ui/react-tabs";
import * as B from "@radix-ui/react-select";
import { NotebookPen as er, ExternalLink as tr, X as mt, Loader2 as rr, WrapText as ir, PencilRuler as nr, FlipHorizontal as or, FlipVertical as ar, CropIcon as sr, ChevronUp as gt, Paperclip as lr, Eye as cr, Settings as dr, ZoomOut as ur, ZoomIn as hr, BookMarked as fr, Sigma as mr, SmilePlusIcon as gr, Replace as Qe, SmilePlus as pr, Frame as xr, Columns2 as et, PanelRight as br, PanelLeft as vr, Clipboard as wr, Copy as yr, GripVertical as Cr, Plus as kr, Columns4 as Mr, Columns3 as Nr, Heading6 as Sr, Heading5 as zr, Heading4 as Tr, Heading3 as Ir, Heading2 as Lr, Heading1 as Ar, Pilcrow as Er, ChevronsUpDown as Rr, Trash as Vr, Trash2 as Dr, TableCellsSplit as Hr, TableCellsMerge as Or, BetweenVerticalEnd as jr, BetweenVerticalStart as Pr, BetweenHorizonalStart as _r, BetweenHorizonalEnd as Ur, Unlink as Br, Pencil as $r, Sparkles as Fr, Table as Wr, Minimize as Gr, Maximize as Xr, Video as Yr, ImageUp as qr, Link as Zr, ListTodo as Kr, ListOrdered as Jr, List as Qr, IndentDecrease as ei, IndentIncrease as ti, Type as ri, CodeXml as ii, Code as ni, Superscript as oi, Subscript as ai, ChevronDown as $e, Undo2 as si, Redo2 as li, PaintRoller as ci, Eraser as di, Minus as ui, Strikethrough as hi, Quote as fi, Underline as mi, Italic as gi, LoaderCircle as pi, Bold as xi, Check as pt } from "lucide-react";
import * as X from "@radix-ui/react-dialog";
import { N as xt, n as bi, m as Pe, aa as vi } from "./index-Dx-Dh7KS.js";
import { I as tt, j as rt, k as wi, u as ae, V as yi } from "./index-DtyAGMbq.js";
import * as bt from "@radix-ui/react-toggle";
import * as we from "@radix-ui/react-tooltip";
import { TextAlignRightIcon as Ci, TextAlignLeftIcon as ki, TextAlignJustifyIcon as Mi, TextAlignCenterIcon as Ni } from "@radix-ui/react-icons";
import * as vt from "@radix-ui/react-label";
import * as _e from "@radix-ui/react-checkbox";
import Si from "react-image-crop";
const Fe = "-", zi = (e) => {
  const t = Ii(e), {
    conflictingClassGroups: r,
    conflictingClassGroupModifiers: i
  } = e;
  return {
    getClassGroupId: (l) => {
      const s = l.split(Fe);
      return s[0] === "" && s.length !== 1 && s.shift(), wt(s, t) || Ti(l);
    },
    getConflictingClassGroupIds: (l, s) => {
      const c = r[l] || [];
      return s && i[l] ? [...c, ...i[l]] : c;
    }
  };
}, wt = (e, t) => {
  var l;
  if (e.length === 0)
    return t.classGroupId;
  const r = e[0], i = t.nextPart.get(r), o = i ? wt(e.slice(1), i) : void 0;
  if (o)
    return o;
  if (t.validators.length === 0)
    return;
  const a = e.join(Fe);
  return (l = t.validators.find(({
    validator: s
  }) => s(a))) == null ? void 0 : l.classGroupId;
}, it = /^\[(.+)\]$/, Ti = (e) => {
  if (it.test(e)) {
    const t = it.exec(e)[1], r = t == null ? void 0 : t.substring(0, t.indexOf(":"));
    if (r)
      return "arbitrary.." + r;
  }
}, Ii = (e) => {
  const {
    theme: t,
    prefix: r
  } = e, i = {
    nextPart: /* @__PURE__ */ new Map(),
    validators: []
  };
  return Ai(Object.entries(e.classGroups), r).forEach(([a, l]) => {
    Ue(l, i, a, t);
  }), i;
}, Ue = (e, t, r, i) => {
  e.forEach((o) => {
    if (typeof o == "string") {
      const a = o === "" ? t : nt(t, o);
      a.classGroupId = r;
      return;
    }
    if (typeof o == "function") {
      if (Li(o)) {
        Ue(o(i), t, r, i);
        return;
      }
      t.validators.push({
        validator: o,
        classGroupId: r
      });
      return;
    }
    Object.entries(o).forEach(([a, l]) => {
      Ue(l, nt(t, a), r, i);
    });
  });
}, nt = (e, t) => {
  let r = e;
  return t.split(Fe).forEach((i) => {
    r.nextPart.has(i) || r.nextPart.set(i, {
      nextPart: /* @__PURE__ */ new Map(),
      validators: []
    }), r = r.nextPart.get(i);
  }), r;
}, Li = (e) => e.isThemeGetter, Ai = (e, t) => t ? e.map(([r, i]) => {
  const o = i.map((a) => typeof a == "string" ? t + a : typeof a == "object" ? Object.fromEntries(Object.entries(a).map(([l, s]) => [t + l, s])) : a);
  return [r, o];
}) : e, Ei = (e) => {
  if (e < 1)
    return {
      get: () => {
      },
      set: () => {
      }
    };
  let t = 0, r = /* @__PURE__ */ new Map(), i = /* @__PURE__ */ new Map();
  const o = (a, l) => {
    r.set(a, l), t++, t > e && (t = 0, i = r, r = /* @__PURE__ */ new Map());
  };
  return {
    get(a) {
      let l = r.get(a);
      if (l !== void 0)
        return l;
      if ((l = i.get(a)) !== void 0)
        return o(a, l), l;
    },
    set(a, l) {
      r.has(a) ? r.set(a, l) : o(a, l);
    }
  };
}, yt = "!", Ri = (e) => {
  const {
    separator: t,
    experimentalParseClassName: r
  } = e, i = t.length === 1, o = t[0], a = t.length, l = (s) => {
    const c = [];
    let u = 0, f = 0, b;
    for (let g = 0; g < s.length; g++) {
      let y = s[g];
      if (u === 0) {
        if (y === o && (i || s.slice(g, g + a) === t)) {
          c.push(s.slice(f, g)), f = g + a;
          continue;
        }
        if (y === "/") {
          b = g;
          continue;
        }
      }
      y === "[" ? u++ : y === "]" && u--;
    }
    const d = c.length === 0 ? s : s.substring(f), w = d.startsWith(yt), m = w ? d.substring(1) : d, M = b && b > f ? b - f : void 0;
    return {
      modifiers: c,
      hasImportantModifier: w,
      baseClassName: m,
      maybePostfixModifierPosition: M
    };
  };
  return r ? (s) => r({
    className: s,
    parseClassName: l
  }) : l;
}, Vi = (e) => {
  if (e.length <= 1)
    return e;
  const t = [];
  let r = [];
  return e.forEach((i) => {
    i[0] === "[" ? (t.push(...r.sort(), i), r = []) : r.push(i);
  }), t.push(...r.sort()), t;
}, Di = (e) => ({
  cache: Ei(e.cacheSize),
  parseClassName: Ri(e),
  ...zi(e)
}), Hi = /\s+/, Oi = (e, t) => {
  const {
    parseClassName: r,
    getClassGroupId: i,
    getConflictingClassGroupIds: o
  } = t, a = [], l = e.trim().split(Hi);
  let s = "";
  for (let c = l.length - 1; c >= 0; c -= 1) {
    const u = l[c], {
      modifiers: f,
      hasImportantModifier: b,
      baseClassName: d,
      maybePostfixModifierPosition: w
    } = r(u);
    let m = !!w, M = i(m ? d.substring(0, w) : d);
    if (!M) {
      if (!m) {
        s = u + (s.length > 0 ? " " + s : s);
        continue;
      }
      if (M = i(d), !M) {
        s = u + (s.length > 0 ? " " + s : s);
        continue;
      }
      m = !1;
    }
    const g = Vi(f).join(":"), y = b ? g + yt : g, x = y + M;
    if (a.includes(x))
      continue;
    a.push(x);
    const L = o(M, m);
    for (let p = 0; p < L.length; ++p) {
      const I = L[p];
      a.push(y + I);
    }
    s = u + (s.length > 0 ? " " + s : s);
  }
  return s;
};
function ji() {
  let e = 0, t, r, i = "";
  for (; e < arguments.length; )
    (t = arguments[e++]) && (r = Ct(t)) && (i && (i += " "), i += r);
  return i;
}
const Ct = (e) => {
  if (typeof e == "string")
    return e;
  let t, r = "";
  for (let i = 0; i < e.length; i++)
    e[i] && (t = Ct(e[i])) && (r && (r += " "), r += t);
  return r;
};
function Pi(e, ...t) {
  let r, i, o, a = l;
  function l(c) {
    const u = t.reduce((f, b) => b(f), e());
    return r = Di(u), i = r.cache.get, o = r.cache.set, a = s, s(c);
  }
  function s(c) {
    const u = i(c);
    if (u)
      return u;
    const f = Oi(c, r);
    return o(c, f), f;
  }
  return function() {
    return a(ji.apply(null, arguments));
  };
}
const H = (e) => {
  const t = (r) => r[e] || [];
  return t.isThemeGetter = !0, t;
}, kt = /^\[(?:([a-z-]+):)?(.+)\]$/i, _i = /^\d+\/\d+$/, Ui = /* @__PURE__ */ new Set(["px", "full", "screen"]), Bi = /^(\d+(\.\d+)?)?(xs|sm|md|lg|xl)$/, $i = /\d+(%|px|r?em|[sdl]?v([hwib]|min|max)|pt|pc|in|cm|mm|cap|ch|ex|r?lh|cq(w|h|i|b|min|max))|\b(calc|min|max|clamp)\(.+\)|^0$/, Fi = /^(rgba?|hsla?|hwb|(ok)?(lab|lch)|color-mix)\(.+\)$/, Wi = /^(inset_)?-?((\d+)?\.?(\d+)[a-z]+|0)_-?((\d+)?\.?(\d+)[a-z]+|0)/, Gi = /^(url|image|image-set|cross-fade|element|(repeating-)?(linear|radial|conic)-gradient)\(.+\)$/, Z = (e) => oe(e) || Ui.has(e) || _i.test(e), ee = (e) => se(e, "length", en), oe = (e) => !!e && !Number.isNaN(Number(e)), Ee = (e) => se(e, "number", oe), he = (e) => !!e && Number.isInteger(Number(e)), Xi = (e) => e.endsWith("%") && oe(e.slice(0, -1)), N = (e) => kt.test(e), te = (e) => Bi.test(e), Yi = /* @__PURE__ */ new Set(["length", "size", "percentage"]), qi = (e) => se(e, Yi, Mt), Zi = (e) => se(e, "position", Mt), Ki = /* @__PURE__ */ new Set(["image", "url"]), Ji = (e) => se(e, Ki, rn), Qi = (e) => se(e, "", tn), fe = () => !0, se = (e, t, r) => {
  const i = kt.exec(e);
  return i ? i[1] ? typeof t == "string" ? i[1] === t : t.has(i[1]) : r(i[2]) : !1;
}, en = (e) => (
  // `colorFunctionRegex` check is necessary because color functions can have percentages in them which which would be incorrectly classified as lengths.
  // For example, `hsl(0 0% 0%)` would be classified as a length without this check.
  // I could also use lookbehind assertion in `lengthUnitRegex` but that isn't supported widely enough.
  $i.test(e) && !Fi.test(e)
), Mt = () => !1, tn = (e) => Wi.test(e), rn = (e) => Gi.test(e), nn = () => {
  const e = H("colors"), t = H("spacing"), r = H("blur"), i = H("brightness"), o = H("borderColor"), a = H("borderRadius"), l = H("borderSpacing"), s = H("borderWidth"), c = H("contrast"), u = H("grayscale"), f = H("hueRotate"), b = H("invert"), d = H("gap"), w = H("gradientColorStops"), m = H("gradientColorStopPositions"), M = H("inset"), g = H("margin"), y = H("opacity"), x = H("padding"), L = H("saturate"), p = H("scale"), I = H("sepia"), v = H("skew"), U = H("space"), z = H("translate"), j = () => ["auto", "contain", "none"], A = () => ["auto", "hidden", "clip", "visible", "scroll"], $ = () => ["auto", N, t], h = () => [N, t], D = () => ["", Z, ee], E = () => ["auto", oe, N], k = () => ["bottom", "center", "left", "left-bottom", "left-top", "right", "right-bottom", "right-top", "top"], T = () => ["solid", "dashed", "dotted", "double", "none"], S = () => ["normal", "multiply", "screen", "overlay", "darken", "lighten", "color-dodge", "color-burn", "hard-light", "soft-light", "difference", "exclusion", "hue", "saturation", "color", "luminosity"], _ = () => ["start", "end", "center", "between", "around", "evenly", "stretch"], P = () => ["", "0", N], W = () => ["auto", "avoid", "all", "avoid-page", "page", "left", "right", "column"], Y = () => [oe, N];
  return {
    cacheSize: 500,
    separator: ":",
    theme: {
      colors: [fe],
      spacing: [Z, ee],
      blur: ["none", "", te, N],
      brightness: Y(),
      borderColor: [e],
      borderRadius: ["none", "", "full", te, N],
      borderSpacing: h(),
      borderWidth: D(),
      contrast: Y(),
      grayscale: P(),
      hueRotate: Y(),
      invert: P(),
      gap: h(),
      gradientColorStops: [e],
      gradientColorStopPositions: [Xi, ee],
      inset: $(),
      margin: $(),
      opacity: Y(),
      padding: h(),
      saturate: Y(),
      scale: Y(),
      sepia: P(),
      skew: Y(),
      space: h(),
      translate: h()
    },
    classGroups: {
      // Layout
      /**
       * Aspect Ratio
       * @see https://tailwindcss.com/docs/aspect-ratio
       */
      aspect: [{
        aspect: ["auto", "square", "video", N]
      }],
      /**
       * Container
       * @see https://tailwindcss.com/docs/container
       */
      container: ["container"],
      /**
       * Columns
       * @see https://tailwindcss.com/docs/columns
       */
      columns: [{
        columns: [te]
      }],
      /**
       * Break After
       * @see https://tailwindcss.com/docs/break-after
       */
      "break-after": [{
        "break-after": W()
      }],
      /**
       * Break Before
       * @see https://tailwindcss.com/docs/break-before
       */
      "break-before": [{
        "break-before": W()
      }],
      /**
       * Break Inside
       * @see https://tailwindcss.com/docs/break-inside
       */
      "break-inside": [{
        "break-inside": ["auto", "avoid", "avoid-page", "avoid-column"]
      }],
      /**
       * Box Decoration Break
       * @see https://tailwindcss.com/docs/box-decoration-break
       */
      "box-decoration": [{
        "box-decoration": ["slice", "clone"]
      }],
      /**
       * Box Sizing
       * @see https://tailwindcss.com/docs/box-sizing
       */
      box: [{
        box: ["border", "content"]
      }],
      /**
       * Display
       * @see https://tailwindcss.com/docs/display
       */
      display: ["block", "inline-block", "inline", "flex", "inline-flex", "table", "inline-table", "table-caption", "table-cell", "table-column", "table-column-group", "table-footer-group", "table-header-group", "table-row-group", "table-row", "flow-root", "grid", "inline-grid", "contents", "list-item", "hidden"],
      /**
       * Floats
       * @see https://tailwindcss.com/docs/float
       */
      float: [{
        float: ["right", "left", "none", "start", "end"]
      }],
      /**
       * Clear
       * @see https://tailwindcss.com/docs/clear
       */
      clear: [{
        clear: ["left", "right", "both", "none", "start", "end"]
      }],
      /**
       * Isolation
       * @see https://tailwindcss.com/docs/isolation
       */
      isolation: ["isolate", "isolation-auto"],
      /**
       * Object Fit
       * @see https://tailwindcss.com/docs/object-fit
       */
      "object-fit": [{
        object: ["contain", "cover", "fill", "none", "scale-down"]
      }],
      /**
       * Object Position
       * @see https://tailwindcss.com/docs/object-position
       */
      "object-position": [{
        object: [...k(), N]
      }],
      /**
       * Overflow
       * @see https://tailwindcss.com/docs/overflow
       */
      overflow: [{
        overflow: A()
      }],
      /**
       * Overflow X
       * @see https://tailwindcss.com/docs/overflow
       */
      "overflow-x": [{
        "overflow-x": A()
      }],
      /**
       * Overflow Y
       * @see https://tailwindcss.com/docs/overflow
       */
      "overflow-y": [{
        "overflow-y": A()
      }],
      /**
       * Overscroll Behavior
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      overscroll: [{
        overscroll: j()
      }],
      /**
       * Overscroll Behavior X
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      "overscroll-x": [{
        "overscroll-x": j()
      }],
      /**
       * Overscroll Behavior Y
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      "overscroll-y": [{
        "overscroll-y": j()
      }],
      /**
       * Position
       * @see https://tailwindcss.com/docs/position
       */
      position: ["static", "fixed", "absolute", "relative", "sticky"],
      /**
       * Top / Right / Bottom / Left
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      inset: [{
        inset: [M]
      }],
      /**
       * Right / Left
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      "inset-x": [{
        "inset-x": [M]
      }],
      /**
       * Top / Bottom
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      "inset-y": [{
        "inset-y": [M]
      }],
      /**
       * Start
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      start: [{
        start: [M]
      }],
      /**
       * End
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      end: [{
        end: [M]
      }],
      /**
       * Top
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      top: [{
        top: [M]
      }],
      /**
       * Right
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      right: [{
        right: [M]
      }],
      /**
       * Bottom
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      bottom: [{
        bottom: [M]
      }],
      /**
       * Left
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      left: [{
        left: [M]
      }],
      /**
       * Visibility
       * @see https://tailwindcss.com/docs/visibility
       */
      visibility: ["visible", "invisible", "collapse"],
      /**
       * Z-Index
       * @see https://tailwindcss.com/docs/z-index
       */
      z: [{
        z: ["auto", he, N]
      }],
      // Flexbox and Grid
      /**
       * Flex Basis
       * @see https://tailwindcss.com/docs/flex-basis
       */
      basis: [{
        basis: $()
      }],
      /**
       * Flex Direction
       * @see https://tailwindcss.com/docs/flex-direction
       */
      "flex-direction": [{
        flex: ["row", "row-reverse", "col", "col-reverse"]
      }],
      /**
       * Flex Wrap
       * @see https://tailwindcss.com/docs/flex-wrap
       */
      "flex-wrap": [{
        flex: ["wrap", "wrap-reverse", "nowrap"]
      }],
      /**
       * Flex
       * @see https://tailwindcss.com/docs/flex
       */
      flex: [{
        flex: ["1", "auto", "initial", "none", N]
      }],
      /**
       * Flex Grow
       * @see https://tailwindcss.com/docs/flex-grow
       */
      grow: [{
        grow: P()
      }],
      /**
       * Flex Shrink
       * @see https://tailwindcss.com/docs/flex-shrink
       */
      shrink: [{
        shrink: P()
      }],
      /**
       * Order
       * @see https://tailwindcss.com/docs/order
       */
      order: [{
        order: ["first", "last", "none", he, N]
      }],
      /**
       * Grid Template Columns
       * @see https://tailwindcss.com/docs/grid-template-columns
       */
      "grid-cols": [{
        "grid-cols": [fe]
      }],
      /**
       * Grid Column Start / End
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-start-end": [{
        col: ["auto", {
          span: ["full", he, N]
        }, N]
      }],
      /**
       * Grid Column Start
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-start": [{
        "col-start": E()
      }],
      /**
       * Grid Column End
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-end": [{
        "col-end": E()
      }],
      /**
       * Grid Template Rows
       * @see https://tailwindcss.com/docs/grid-template-rows
       */
      "grid-rows": [{
        "grid-rows": [fe]
      }],
      /**
       * Grid Row Start / End
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-start-end": [{
        row: ["auto", {
          span: [he, N]
        }, N]
      }],
      /**
       * Grid Row Start
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-start": [{
        "row-start": E()
      }],
      /**
       * Grid Row End
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-end": [{
        "row-end": E()
      }],
      /**
       * Grid Auto Flow
       * @see https://tailwindcss.com/docs/grid-auto-flow
       */
      "grid-flow": [{
        "grid-flow": ["row", "col", "dense", "row-dense", "col-dense"]
      }],
      /**
       * Grid Auto Columns
       * @see https://tailwindcss.com/docs/grid-auto-columns
       */
      "auto-cols": [{
        "auto-cols": ["auto", "min", "max", "fr", N]
      }],
      /**
       * Grid Auto Rows
       * @see https://tailwindcss.com/docs/grid-auto-rows
       */
      "auto-rows": [{
        "auto-rows": ["auto", "min", "max", "fr", N]
      }],
      /**
       * Gap
       * @see https://tailwindcss.com/docs/gap
       */
      gap: [{
        gap: [d]
      }],
      /**
       * Gap X
       * @see https://tailwindcss.com/docs/gap
       */
      "gap-x": [{
        "gap-x": [d]
      }],
      /**
       * Gap Y
       * @see https://tailwindcss.com/docs/gap
       */
      "gap-y": [{
        "gap-y": [d]
      }],
      /**
       * Justify Content
       * @see https://tailwindcss.com/docs/justify-content
       */
      "justify-content": [{
        justify: ["normal", ..._()]
      }],
      /**
       * Justify Items
       * @see https://tailwindcss.com/docs/justify-items
       */
      "justify-items": [{
        "justify-items": ["start", "end", "center", "stretch"]
      }],
      /**
       * Justify Self
       * @see https://tailwindcss.com/docs/justify-self
       */
      "justify-self": [{
        "justify-self": ["auto", "start", "end", "center", "stretch"]
      }],
      /**
       * Align Content
       * @see https://tailwindcss.com/docs/align-content
       */
      "align-content": [{
        content: ["normal", ..._(), "baseline"]
      }],
      /**
       * Align Items
       * @see https://tailwindcss.com/docs/align-items
       */
      "align-items": [{
        items: ["start", "end", "center", "baseline", "stretch"]
      }],
      /**
       * Align Self
       * @see https://tailwindcss.com/docs/align-self
       */
      "align-self": [{
        self: ["auto", "start", "end", "center", "stretch", "baseline"]
      }],
      /**
       * Place Content
       * @see https://tailwindcss.com/docs/place-content
       */
      "place-content": [{
        "place-content": [..._(), "baseline"]
      }],
      /**
       * Place Items
       * @see https://tailwindcss.com/docs/place-items
       */
      "place-items": [{
        "place-items": ["start", "end", "center", "baseline", "stretch"]
      }],
      /**
       * Place Self
       * @see https://tailwindcss.com/docs/place-self
       */
      "place-self": [{
        "place-self": ["auto", "start", "end", "center", "stretch"]
      }],
      // Spacing
      /**
       * Padding
       * @see https://tailwindcss.com/docs/padding
       */
      p: [{
        p: [x]
      }],
      /**
       * Padding X
       * @see https://tailwindcss.com/docs/padding
       */
      px: [{
        px: [x]
      }],
      /**
       * Padding Y
       * @see https://tailwindcss.com/docs/padding
       */
      py: [{
        py: [x]
      }],
      /**
       * Padding Start
       * @see https://tailwindcss.com/docs/padding
       */
      ps: [{
        ps: [x]
      }],
      /**
       * Padding End
       * @see https://tailwindcss.com/docs/padding
       */
      pe: [{
        pe: [x]
      }],
      /**
       * Padding Top
       * @see https://tailwindcss.com/docs/padding
       */
      pt: [{
        pt: [x]
      }],
      /**
       * Padding Right
       * @see https://tailwindcss.com/docs/padding
       */
      pr: [{
        pr: [x]
      }],
      /**
       * Padding Bottom
       * @see https://tailwindcss.com/docs/padding
       */
      pb: [{
        pb: [x]
      }],
      /**
       * Padding Left
       * @see https://tailwindcss.com/docs/padding
       */
      pl: [{
        pl: [x]
      }],
      /**
       * Margin
       * @see https://tailwindcss.com/docs/margin
       */
      m: [{
        m: [g]
      }],
      /**
       * Margin X
       * @see https://tailwindcss.com/docs/margin
       */
      mx: [{
        mx: [g]
      }],
      /**
       * Margin Y
       * @see https://tailwindcss.com/docs/margin
       */
      my: [{
        my: [g]
      }],
      /**
       * Margin Start
       * @see https://tailwindcss.com/docs/margin
       */
      ms: [{
        ms: [g]
      }],
      /**
       * Margin End
       * @see https://tailwindcss.com/docs/margin
       */
      me: [{
        me: [g]
      }],
      /**
       * Margin Top
       * @see https://tailwindcss.com/docs/margin
       */
      mt: [{
        mt: [g]
      }],
      /**
       * Margin Right
       * @see https://tailwindcss.com/docs/margin
       */
      mr: [{
        mr: [g]
      }],
      /**
       * Margin Bottom
       * @see https://tailwindcss.com/docs/margin
       */
      mb: [{
        mb: [g]
      }],
      /**
       * Margin Left
       * @see https://tailwindcss.com/docs/margin
       */
      ml: [{
        ml: [g]
      }],
      /**
       * Space Between X
       * @see https://tailwindcss.com/docs/space
       */
      "space-x": [{
        "space-x": [U]
      }],
      /**
       * Space Between X Reverse
       * @see https://tailwindcss.com/docs/space
       */
      "space-x-reverse": ["space-x-reverse"],
      /**
       * Space Between Y
       * @see https://tailwindcss.com/docs/space
       */
      "space-y": [{
        "space-y": [U]
      }],
      /**
       * Space Between Y Reverse
       * @see https://tailwindcss.com/docs/space
       */
      "space-y-reverse": ["space-y-reverse"],
      // Sizing
      /**
       * Width
       * @see https://tailwindcss.com/docs/width
       */
      w: [{
        w: ["auto", "min", "max", "fit", "svw", "lvw", "dvw", N, t]
      }],
      /**
       * Min-Width
       * @see https://tailwindcss.com/docs/min-width
       */
      "min-w": [{
        "min-w": [N, t, "min", "max", "fit"]
      }],
      /**
       * Max-Width
       * @see https://tailwindcss.com/docs/max-width
       */
      "max-w": [{
        "max-w": [N, t, "none", "full", "min", "max", "fit", "prose", {
          screen: [te]
        }, te]
      }],
      /**
       * Height
       * @see https://tailwindcss.com/docs/height
       */
      h: [{
        h: [N, t, "auto", "min", "max", "fit", "svh", "lvh", "dvh"]
      }],
      /**
       * Min-Height
       * @see https://tailwindcss.com/docs/min-height
       */
      "min-h": [{
        "min-h": [N, t, "min", "max", "fit", "svh", "lvh", "dvh"]
      }],
      /**
       * Max-Height
       * @see https://tailwindcss.com/docs/max-height
       */
      "max-h": [{
        "max-h": [N, t, "min", "max", "fit", "svh", "lvh", "dvh"]
      }],
      /**
       * Size
       * @see https://tailwindcss.com/docs/size
       */
      size: [{
        size: [N, t, "auto", "min", "max", "fit"]
      }],
      // Typography
      /**
       * Font Size
       * @see https://tailwindcss.com/docs/font-size
       */
      "font-size": [{
        text: ["base", te, ee]
      }],
      /**
       * Font Smoothing
       * @see https://tailwindcss.com/docs/font-smoothing
       */
      "font-smoothing": ["antialiased", "subpixel-antialiased"],
      /**
       * Font Style
       * @see https://tailwindcss.com/docs/font-style
       */
      "font-style": ["italic", "not-italic"],
      /**
       * Font Weight
       * @see https://tailwindcss.com/docs/font-weight
       */
      "font-weight": [{
        font: ["thin", "extralight", "light", "normal", "medium", "semibold", "bold", "extrabold", "black", Ee]
      }],
      /**
       * Font Family
       * @see https://tailwindcss.com/docs/font-family
       */
      "font-family": [{
        font: [fe]
      }],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-normal": ["normal-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-ordinal": ["ordinal"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-slashed-zero": ["slashed-zero"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-figure": ["lining-nums", "oldstyle-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-spacing": ["proportional-nums", "tabular-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-fraction": ["diagonal-fractions", "stacked-fractions"],
      /**
       * Letter Spacing
       * @see https://tailwindcss.com/docs/letter-spacing
       */
      tracking: [{
        tracking: ["tighter", "tight", "normal", "wide", "wider", "widest", N]
      }],
      /**
       * Line Clamp
       * @see https://tailwindcss.com/docs/line-clamp
       */
      "line-clamp": [{
        "line-clamp": ["none", oe, Ee]
      }],
      /**
       * Line Height
       * @see https://tailwindcss.com/docs/line-height
       */
      leading: [{
        leading: ["none", "tight", "snug", "normal", "relaxed", "loose", Z, N]
      }],
      /**
       * List Style Image
       * @see https://tailwindcss.com/docs/list-style-image
       */
      "list-image": [{
        "list-image": ["none", N]
      }],
      /**
       * List Style Type
       * @see https://tailwindcss.com/docs/list-style-type
       */
      "list-style-type": [{
        list: ["none", "disc", "decimal", N]
      }],
      /**
       * List Style Position
       * @see https://tailwindcss.com/docs/list-style-position
       */
      "list-style-position": [{
        list: ["inside", "outside"]
      }],
      /**
       * Placeholder Color
       * @deprecated since Tailwind CSS v3.0.0
       * @see https://tailwindcss.com/docs/placeholder-color
       */
      "placeholder-color": [{
        placeholder: [e]
      }],
      /**
       * Placeholder Opacity
       * @see https://tailwindcss.com/docs/placeholder-opacity
       */
      "placeholder-opacity": [{
        "placeholder-opacity": [y]
      }],
      /**
       * Text Alignment
       * @see https://tailwindcss.com/docs/text-align
       */
      "text-alignment": [{
        text: ["left", "center", "right", "justify", "start", "end"]
      }],
      /**
       * Text Color
       * @see https://tailwindcss.com/docs/text-color
       */
      "text-color": [{
        text: [e]
      }],
      /**
       * Text Opacity
       * @see https://tailwindcss.com/docs/text-opacity
       */
      "text-opacity": [{
        "text-opacity": [y]
      }],
      /**
       * Text Decoration
       * @see https://tailwindcss.com/docs/text-decoration
       */
      "text-decoration": ["underline", "overline", "line-through", "no-underline"],
      /**
       * Text Decoration Style
       * @see https://tailwindcss.com/docs/text-decoration-style
       */
      "text-decoration-style": [{
        decoration: [...T(), "wavy"]
      }],
      /**
       * Text Decoration Thickness
       * @see https://tailwindcss.com/docs/text-decoration-thickness
       */
      "text-decoration-thickness": [{
        decoration: ["auto", "from-font", Z, ee]
      }],
      /**
       * Text Underline Offset
       * @see https://tailwindcss.com/docs/text-underline-offset
       */
      "underline-offset": [{
        "underline-offset": ["auto", Z, N]
      }],
      /**
       * Text Decoration Color
       * @see https://tailwindcss.com/docs/text-decoration-color
       */
      "text-decoration-color": [{
        decoration: [e]
      }],
      /**
       * Text Transform
       * @see https://tailwindcss.com/docs/text-transform
       */
      "text-transform": ["uppercase", "lowercase", "capitalize", "normal-case"],
      /**
       * Text Overflow
       * @see https://tailwindcss.com/docs/text-overflow
       */
      "text-overflow": ["truncate", "text-ellipsis", "text-clip"],
      /**
       * Text Wrap
       * @see https://tailwindcss.com/docs/text-wrap
       */
      "text-wrap": [{
        text: ["wrap", "nowrap", "balance", "pretty"]
      }],
      /**
       * Text Indent
       * @see https://tailwindcss.com/docs/text-indent
       */
      indent: [{
        indent: h()
      }],
      /**
       * Vertical Alignment
       * @see https://tailwindcss.com/docs/vertical-align
       */
      "vertical-align": [{
        align: ["baseline", "top", "middle", "bottom", "text-top", "text-bottom", "sub", "super", N]
      }],
      /**
       * Whitespace
       * @see https://tailwindcss.com/docs/whitespace
       */
      whitespace: [{
        whitespace: ["normal", "nowrap", "pre", "pre-line", "pre-wrap", "break-spaces"]
      }],
      /**
       * Word Break
       * @see https://tailwindcss.com/docs/word-break
       */
      break: [{
        break: ["normal", "words", "all", "keep"]
      }],
      /**
       * Hyphens
       * @see https://tailwindcss.com/docs/hyphens
       */
      hyphens: [{
        hyphens: ["none", "manual", "auto"]
      }],
      /**
       * Content
       * @see https://tailwindcss.com/docs/content
       */
      content: [{
        content: ["none", N]
      }],
      // Backgrounds
      /**
       * Background Attachment
       * @see https://tailwindcss.com/docs/background-attachment
       */
      "bg-attachment": [{
        bg: ["fixed", "local", "scroll"]
      }],
      /**
       * Background Clip
       * @see https://tailwindcss.com/docs/background-clip
       */
      "bg-clip": [{
        "bg-clip": ["border", "padding", "content", "text"]
      }],
      /**
       * Background Opacity
       * @deprecated since Tailwind CSS v3.0.0
       * @see https://tailwindcss.com/docs/background-opacity
       */
      "bg-opacity": [{
        "bg-opacity": [y]
      }],
      /**
       * Background Origin
       * @see https://tailwindcss.com/docs/background-origin
       */
      "bg-origin": [{
        "bg-origin": ["border", "padding", "content"]
      }],
      /**
       * Background Position
       * @see https://tailwindcss.com/docs/background-position
       */
      "bg-position": [{
        bg: [...k(), Zi]
      }],
      /**
       * Background Repeat
       * @see https://tailwindcss.com/docs/background-repeat
       */
      "bg-repeat": [{
        bg: ["no-repeat", {
          repeat: ["", "x", "y", "round", "space"]
        }]
      }],
      /**
       * Background Size
       * @see https://tailwindcss.com/docs/background-size
       */
      "bg-size": [{
        bg: ["auto", "cover", "contain", qi]
      }],
      /**
       * Background Image
       * @see https://tailwindcss.com/docs/background-image
       */
      "bg-image": [{
        bg: ["none", {
          "gradient-to": ["t", "tr", "r", "br", "b", "bl", "l", "tl"]
        }, Ji]
      }],
      /**
       * Background Color
       * @see https://tailwindcss.com/docs/background-color
       */
      "bg-color": [{
        bg: [e]
      }],
      /**
       * Gradient Color Stops From Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-from-pos": [{
        from: [m]
      }],
      /**
       * Gradient Color Stops Via Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-via-pos": [{
        via: [m]
      }],
      /**
       * Gradient Color Stops To Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-to-pos": [{
        to: [m]
      }],
      /**
       * Gradient Color Stops From
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-from": [{
        from: [w]
      }],
      /**
       * Gradient Color Stops Via
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-via": [{
        via: [w]
      }],
      /**
       * Gradient Color Stops To
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-to": [{
        to: [w]
      }],
      // Borders
      /**
       * Border Radius
       * @see https://tailwindcss.com/docs/border-radius
       */
      rounded: [{
        rounded: [a]
      }],
      /**
       * Border Radius Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-s": [{
        "rounded-s": [a]
      }],
      /**
       * Border Radius End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-e": [{
        "rounded-e": [a]
      }],
      /**
       * Border Radius Top
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-t": [{
        "rounded-t": [a]
      }],
      /**
       * Border Radius Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-r": [{
        "rounded-r": [a]
      }],
      /**
       * Border Radius Bottom
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-b": [{
        "rounded-b": [a]
      }],
      /**
       * Border Radius Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-l": [{
        "rounded-l": [a]
      }],
      /**
       * Border Radius Start Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-ss": [{
        "rounded-ss": [a]
      }],
      /**
       * Border Radius Start End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-se": [{
        "rounded-se": [a]
      }],
      /**
       * Border Radius End End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-ee": [{
        "rounded-ee": [a]
      }],
      /**
       * Border Radius End Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-es": [{
        "rounded-es": [a]
      }],
      /**
       * Border Radius Top Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-tl": [{
        "rounded-tl": [a]
      }],
      /**
       * Border Radius Top Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-tr": [{
        "rounded-tr": [a]
      }],
      /**
       * Border Radius Bottom Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-br": [{
        "rounded-br": [a]
      }],
      /**
       * Border Radius Bottom Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-bl": [{
        "rounded-bl": [a]
      }],
      /**
       * Border Width
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w": [{
        border: [s]
      }],
      /**
       * Border Width X
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-x": [{
        "border-x": [s]
      }],
      /**
       * Border Width Y
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-y": [{
        "border-y": [s]
      }],
      /**
       * Border Width Start
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-s": [{
        "border-s": [s]
      }],
      /**
       * Border Width End
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-e": [{
        "border-e": [s]
      }],
      /**
       * Border Width Top
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-t": [{
        "border-t": [s]
      }],
      /**
       * Border Width Right
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-r": [{
        "border-r": [s]
      }],
      /**
       * Border Width Bottom
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-b": [{
        "border-b": [s]
      }],
      /**
       * Border Width Left
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-l": [{
        "border-l": [s]
      }],
      /**
       * Border Opacity
       * @see https://tailwindcss.com/docs/border-opacity
       */
      "border-opacity": [{
        "border-opacity": [y]
      }],
      /**
       * Border Style
       * @see https://tailwindcss.com/docs/border-style
       */
      "border-style": [{
        border: [...T(), "hidden"]
      }],
      /**
       * Divide Width X
       * @see https://tailwindcss.com/docs/divide-width
       */
      "divide-x": [{
        "divide-x": [s]
      }],
      /**
       * Divide Width X Reverse
       * @see https://tailwindcss.com/docs/divide-width
       */
      "divide-x-reverse": ["divide-x-reverse"],
      /**
       * Divide Width Y
       * @see https://tailwindcss.com/docs/divide-width
       */
      "divide-y": [{
        "divide-y": [s]
      }],
      /**
       * Divide Width Y Reverse
       * @see https://tailwindcss.com/docs/divide-width
       */
      "divide-y-reverse": ["divide-y-reverse"],
      /**
       * Divide Opacity
       * @see https://tailwindcss.com/docs/divide-opacity
       */
      "divide-opacity": [{
        "divide-opacity": [y]
      }],
      /**
       * Divide Style
       * @see https://tailwindcss.com/docs/divide-style
       */
      "divide-style": [{
        divide: T()
      }],
      /**
       * Border Color
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color": [{
        border: [o]
      }],
      /**
       * Border Color X
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-x": [{
        "border-x": [o]
      }],
      /**
       * Border Color Y
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-y": [{
        "border-y": [o]
      }],
      /**
       * Border Color S
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-s": [{
        "border-s": [o]
      }],
      /**
       * Border Color E
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-e": [{
        "border-e": [o]
      }],
      /**
       * Border Color Top
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-t": [{
        "border-t": [o]
      }],
      /**
       * Border Color Right
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-r": [{
        "border-r": [o]
      }],
      /**
       * Border Color Bottom
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-b": [{
        "border-b": [o]
      }],
      /**
       * Border Color Left
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-l": [{
        "border-l": [o]
      }],
      /**
       * Divide Color
       * @see https://tailwindcss.com/docs/divide-color
       */
      "divide-color": [{
        divide: [o]
      }],
      /**
       * Outline Style
       * @see https://tailwindcss.com/docs/outline-style
       */
      "outline-style": [{
        outline: ["", ...T()]
      }],
      /**
       * Outline Offset
       * @see https://tailwindcss.com/docs/outline-offset
       */
      "outline-offset": [{
        "outline-offset": [Z, N]
      }],
      /**
       * Outline Width
       * @see https://tailwindcss.com/docs/outline-width
       */
      "outline-w": [{
        outline: [Z, ee]
      }],
      /**
       * Outline Color
       * @see https://tailwindcss.com/docs/outline-color
       */
      "outline-color": [{
        outline: [e]
      }],
      /**
       * Ring Width
       * @see https://tailwindcss.com/docs/ring-width
       */
      "ring-w": [{
        ring: D()
      }],
      /**
       * Ring Width Inset
       * @see https://tailwindcss.com/docs/ring-width
       */
      "ring-w-inset": ["ring-inset"],
      /**
       * Ring Color
       * @see https://tailwindcss.com/docs/ring-color
       */
      "ring-color": [{
        ring: [e]
      }],
      /**
       * Ring Opacity
       * @see https://tailwindcss.com/docs/ring-opacity
       */
      "ring-opacity": [{
        "ring-opacity": [y]
      }],
      /**
       * Ring Offset Width
       * @see https://tailwindcss.com/docs/ring-offset-width
       */
      "ring-offset-w": [{
        "ring-offset": [Z, ee]
      }],
      /**
       * Ring Offset Color
       * @see https://tailwindcss.com/docs/ring-offset-color
       */
      "ring-offset-color": [{
        "ring-offset": [e]
      }],
      // Effects
      /**
       * Box Shadow
       * @see https://tailwindcss.com/docs/box-shadow
       */
      shadow: [{
        shadow: ["", "inner", "none", te, Qi]
      }],
      /**
       * Box Shadow Color
       * @see https://tailwindcss.com/docs/box-shadow-color
       */
      "shadow-color": [{
        shadow: [fe]
      }],
      /**
       * Opacity
       * @see https://tailwindcss.com/docs/opacity
       */
      opacity: [{
        opacity: [y]
      }],
      /**
       * Mix Blend Mode
       * @see https://tailwindcss.com/docs/mix-blend-mode
       */
      "mix-blend": [{
        "mix-blend": [...S(), "plus-lighter", "plus-darker"]
      }],
      /**
       * Background Blend Mode
       * @see https://tailwindcss.com/docs/background-blend-mode
       */
      "bg-blend": [{
        "bg-blend": S()
      }],
      // Filters
      /**
       * Filter
       * @deprecated since Tailwind CSS v3.0.0
       * @see https://tailwindcss.com/docs/filter
       */
      filter: [{
        filter: ["", "none"]
      }],
      /**
       * Blur
       * @see https://tailwindcss.com/docs/blur
       */
      blur: [{
        blur: [r]
      }],
      /**
       * Brightness
       * @see https://tailwindcss.com/docs/brightness
       */
      brightness: [{
        brightness: [i]
      }],
      /**
       * Contrast
       * @see https://tailwindcss.com/docs/contrast
       */
      contrast: [{
        contrast: [c]
      }],
      /**
       * Drop Shadow
       * @see https://tailwindcss.com/docs/drop-shadow
       */
      "drop-shadow": [{
        "drop-shadow": ["", "none", te, N]
      }],
      /**
       * Grayscale
       * @see https://tailwindcss.com/docs/grayscale
       */
      grayscale: [{
        grayscale: [u]
      }],
      /**
       * Hue Rotate
       * @see https://tailwindcss.com/docs/hue-rotate
       */
      "hue-rotate": [{
        "hue-rotate": [f]
      }],
      /**
       * Invert
       * @see https://tailwindcss.com/docs/invert
       */
      invert: [{
        invert: [b]
      }],
      /**
       * Saturate
       * @see https://tailwindcss.com/docs/saturate
       */
      saturate: [{
        saturate: [L]
      }],
      /**
       * Sepia
       * @see https://tailwindcss.com/docs/sepia
       */
      sepia: [{
        sepia: [I]
      }],
      /**
       * Backdrop Filter
       * @deprecated since Tailwind CSS v3.0.0
       * @see https://tailwindcss.com/docs/backdrop-filter
       */
      "backdrop-filter": [{
        "backdrop-filter": ["", "none"]
      }],
      /**
       * Backdrop Blur
       * @see https://tailwindcss.com/docs/backdrop-blur
       */
      "backdrop-blur": [{
        "backdrop-blur": [r]
      }],
      /**
       * Backdrop Brightness
       * @see https://tailwindcss.com/docs/backdrop-brightness
       */
      "backdrop-brightness": [{
        "backdrop-brightness": [i]
      }],
      /**
       * Backdrop Contrast
       * @see https://tailwindcss.com/docs/backdrop-contrast
       */
      "backdrop-contrast": [{
        "backdrop-contrast": [c]
      }],
      /**
       * Backdrop Grayscale
       * @see https://tailwindcss.com/docs/backdrop-grayscale
       */
      "backdrop-grayscale": [{
        "backdrop-grayscale": [u]
      }],
      /**
       * Backdrop Hue Rotate
       * @see https://tailwindcss.com/docs/backdrop-hue-rotate
       */
      "backdrop-hue-rotate": [{
        "backdrop-hue-rotate": [f]
      }],
      /**
       * Backdrop Invert
       * @see https://tailwindcss.com/docs/backdrop-invert
       */
      "backdrop-invert": [{
        "backdrop-invert": [b]
      }],
      /**
       * Backdrop Opacity
       * @see https://tailwindcss.com/docs/backdrop-opacity
       */
      "backdrop-opacity": [{
        "backdrop-opacity": [y]
      }],
      /**
       * Backdrop Saturate
       * @see https://tailwindcss.com/docs/backdrop-saturate
       */
      "backdrop-saturate": [{
        "backdrop-saturate": [L]
      }],
      /**
       * Backdrop Sepia
       * @see https://tailwindcss.com/docs/backdrop-sepia
       */
      "backdrop-sepia": [{
        "backdrop-sepia": [I]
      }],
      // Tables
      /**
       * Border Collapse
       * @see https://tailwindcss.com/docs/border-collapse
       */
      "border-collapse": [{
        border: ["collapse", "separate"]
      }],
      /**
       * Border Spacing
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing": [{
        "border-spacing": [l]
      }],
      /**
       * Border Spacing X
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing-x": [{
        "border-spacing-x": [l]
      }],
      /**
       * Border Spacing Y
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing-y": [{
        "border-spacing-y": [l]
      }],
      /**
       * Table Layout
       * @see https://tailwindcss.com/docs/table-layout
       */
      "table-layout": [{
        table: ["auto", "fixed"]
      }],
      /**
       * Caption Side
       * @see https://tailwindcss.com/docs/caption-side
       */
      caption: [{
        caption: ["top", "bottom"]
      }],
      // Transitions and Animation
      /**
       * Tranisition Property
       * @see https://tailwindcss.com/docs/transition-property
       */
      transition: [{
        transition: ["none", "all", "", "colors", "opacity", "shadow", "transform", N]
      }],
      /**
       * Transition Duration
       * @see https://tailwindcss.com/docs/transition-duration
       */
      duration: [{
        duration: Y()
      }],
      /**
       * Transition Timing Function
       * @see https://tailwindcss.com/docs/transition-timing-function
       */
      ease: [{
        ease: ["linear", "in", "out", "in-out", N]
      }],
      /**
       * Transition Delay
       * @see https://tailwindcss.com/docs/transition-delay
       */
      delay: [{
        delay: Y()
      }],
      /**
       * Animation
       * @see https://tailwindcss.com/docs/animation
       */
      animate: [{
        animate: ["none", "spin", "ping", "pulse", "bounce", N]
      }],
      // Transforms
      /**
       * Transform
       * @see https://tailwindcss.com/docs/transform
       */
      transform: [{
        transform: ["", "gpu", "none"]
      }],
      /**
       * Scale
       * @see https://tailwindcss.com/docs/scale
       */
      scale: [{
        scale: [p]
      }],
      /**
       * Scale X
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-x": [{
        "scale-x": [p]
      }],
      /**
       * Scale Y
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-y": [{
        "scale-y": [p]
      }],
      /**
       * Rotate
       * @see https://tailwindcss.com/docs/rotate
       */
      rotate: [{
        rotate: [he, N]
      }],
      /**
       * Translate X
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-x": [{
        "translate-x": [z]
      }],
      /**
       * Translate Y
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-y": [{
        "translate-y": [z]
      }],
      /**
       * Skew X
       * @see https://tailwindcss.com/docs/skew
       */
      "skew-x": [{
        "skew-x": [v]
      }],
      /**
       * Skew Y
       * @see https://tailwindcss.com/docs/skew
       */
      "skew-y": [{
        "skew-y": [v]
      }],
      /**
       * Transform Origin
       * @see https://tailwindcss.com/docs/transform-origin
       */
      "transform-origin": [{
        origin: ["center", "top", "top-right", "right", "bottom-right", "bottom", "bottom-left", "left", "top-left", N]
      }],
      // Interactivity
      /**
       * Accent Color
       * @see https://tailwindcss.com/docs/accent-color
       */
      accent: [{
        accent: ["auto", e]
      }],
      /**
       * Appearance
       * @see https://tailwindcss.com/docs/appearance
       */
      appearance: [{
        appearance: ["none", "auto"]
      }],
      /**
       * Cursor
       * @see https://tailwindcss.com/docs/cursor
       */
      cursor: [{
        cursor: ["auto", "default", "pointer", "wait", "text", "move", "help", "not-allowed", "none", "context-menu", "progress", "cell", "crosshair", "vertical-text", "alias", "copy", "no-drop", "grab", "grabbing", "all-scroll", "col-resize", "row-resize", "n-resize", "e-resize", "s-resize", "w-resize", "ne-resize", "nw-resize", "se-resize", "sw-resize", "ew-resize", "ns-resize", "nesw-resize", "nwse-resize", "zoom-in", "zoom-out", N]
      }],
      /**
       * Caret Color
       * @see https://tailwindcss.com/docs/just-in-time-mode#caret-color-utilities
       */
      "caret-color": [{
        caret: [e]
      }],
      /**
       * Pointer Events
       * @see https://tailwindcss.com/docs/pointer-events
       */
      "pointer-events": [{
        "pointer-events": ["none", "auto"]
      }],
      /**
       * Resize
       * @see https://tailwindcss.com/docs/resize
       */
      resize: [{
        resize: ["none", "y", "x", ""]
      }],
      /**
       * Scroll Behavior
       * @see https://tailwindcss.com/docs/scroll-behavior
       */
      "scroll-behavior": [{
        scroll: ["auto", "smooth"]
      }],
      /**
       * Scroll Margin
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-m": [{
        "scroll-m": h()
      }],
      /**
       * Scroll Margin X
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mx": [{
        "scroll-mx": h()
      }],
      /**
       * Scroll Margin Y
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-my": [{
        "scroll-my": h()
      }],
      /**
       * Scroll Margin Start
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-ms": [{
        "scroll-ms": h()
      }],
      /**
       * Scroll Margin End
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-me": [{
        "scroll-me": h()
      }],
      /**
       * Scroll Margin Top
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mt": [{
        "scroll-mt": h()
      }],
      /**
       * Scroll Margin Right
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mr": [{
        "scroll-mr": h()
      }],
      /**
       * Scroll Margin Bottom
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mb": [{
        "scroll-mb": h()
      }],
      /**
       * Scroll Margin Left
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-ml": [{
        "scroll-ml": h()
      }],
      /**
       * Scroll Padding
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-p": [{
        "scroll-p": h()
      }],
      /**
       * Scroll Padding X
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-px": [{
        "scroll-px": h()
      }],
      /**
       * Scroll Padding Y
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-py": [{
        "scroll-py": h()
      }],
      /**
       * Scroll Padding Start
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-ps": [{
        "scroll-ps": h()
      }],
      /**
       * Scroll Padding End
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pe": [{
        "scroll-pe": h()
      }],
      /**
       * Scroll Padding Top
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pt": [{
        "scroll-pt": h()
      }],
      /**
       * Scroll Padding Right
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pr": [{
        "scroll-pr": h()
      }],
      /**
       * Scroll Padding Bottom
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pb": [{
        "scroll-pb": h()
      }],
      /**
       * Scroll Padding Left
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pl": [{
        "scroll-pl": h()
      }],
      /**
       * Scroll Snap Align
       * @see https://tailwindcss.com/docs/scroll-snap-align
       */
      "snap-align": [{
        snap: ["start", "end", "center", "align-none"]
      }],
      /**
       * Scroll Snap Stop
       * @see https://tailwindcss.com/docs/scroll-snap-stop
       */
      "snap-stop": [{
        snap: ["normal", "always"]
      }],
      /**
       * Scroll Snap Type
       * @see https://tailwindcss.com/docs/scroll-snap-type
       */
      "snap-type": [{
        snap: ["none", "x", "y", "both"]
      }],
      /**
       * Scroll Snap Type Strictness
       * @see https://tailwindcss.com/docs/scroll-snap-type
       */
      "snap-strictness": [{
        snap: ["mandatory", "proximity"]
      }],
      /**
       * Touch Action
       * @see https://tailwindcss.com/docs/touch-action
       */
      touch: [{
        touch: ["auto", "none", "manipulation"]
      }],
      /**
       * Touch Action X
       * @see https://tailwindcss.com/docs/touch-action
       */
      "touch-x": [{
        "touch-pan": ["x", "left", "right"]
      }],
      /**
       * Touch Action Y
       * @see https://tailwindcss.com/docs/touch-action
       */
      "touch-y": [{
        "touch-pan": ["y", "up", "down"]
      }],
      /**
       * Touch Action Pinch Zoom
       * @see https://tailwindcss.com/docs/touch-action
       */
      "touch-pz": ["touch-pinch-zoom"],
      /**
       * User Select
       * @see https://tailwindcss.com/docs/user-select
       */
      select: [{
        select: ["none", "text", "all", "auto"]
      }],
      /**
       * Will Change
       * @see https://tailwindcss.com/docs/will-change
       */
      "will-change": [{
        "will-change": ["auto", "scroll", "contents", "transform", N]
      }],
      // SVG
      /**
       * Fill
       * @see https://tailwindcss.com/docs/fill
       */
      fill: [{
        fill: [e, "none"]
      }],
      /**
       * Stroke Width
       * @see https://tailwindcss.com/docs/stroke-width
       */
      "stroke-w": [{
        stroke: [Z, ee, Ee]
      }],
      /**
       * Stroke
       * @see https://tailwindcss.com/docs/stroke
       */
      stroke: [{
        stroke: [e, "none"]
      }],
      // Accessibility
      /**
       * Screen Readers
       * @see https://tailwindcss.com/docs/screen-readers
       */
      sr: ["sr-only", "not-sr-only"],
      /**
       * Forced Color Adjust
       * @see https://tailwindcss.com/docs/forced-color-adjust
       */
      "forced-color-adjust": [{
        "forced-color-adjust": ["auto", "none"]
      }]
    },
    conflictingClassGroups: {
      overflow: ["overflow-x", "overflow-y"],
      overscroll: ["overscroll-x", "overscroll-y"],
      inset: ["inset-x", "inset-y", "start", "end", "top", "right", "bottom", "left"],
      "inset-x": ["right", "left"],
      "inset-y": ["top", "bottom"],
      flex: ["basis", "grow", "shrink"],
      gap: ["gap-x", "gap-y"],
      p: ["px", "py", "ps", "pe", "pt", "pr", "pb", "pl"],
      px: ["pr", "pl"],
      py: ["pt", "pb"],
      m: ["mx", "my", "ms", "me", "mt", "mr", "mb", "ml"],
      mx: ["mr", "ml"],
      my: ["mt", "mb"],
      size: ["w", "h"],
      "font-size": ["leading"],
      "fvn-normal": ["fvn-ordinal", "fvn-slashed-zero", "fvn-figure", "fvn-spacing", "fvn-fraction"],
      "fvn-ordinal": ["fvn-normal"],
      "fvn-slashed-zero": ["fvn-normal"],
      "fvn-figure": ["fvn-normal"],
      "fvn-spacing": ["fvn-normal"],
      "fvn-fraction": ["fvn-normal"],
      "line-clamp": ["display", "overflow"],
      rounded: ["rounded-s", "rounded-e", "rounded-t", "rounded-r", "rounded-b", "rounded-l", "rounded-ss", "rounded-se", "rounded-ee", "rounded-es", "rounded-tl", "rounded-tr", "rounded-br", "rounded-bl"],
      "rounded-s": ["rounded-ss", "rounded-es"],
      "rounded-e": ["rounded-se", "rounded-ee"],
      "rounded-t": ["rounded-tl", "rounded-tr"],
      "rounded-r": ["rounded-tr", "rounded-br"],
      "rounded-b": ["rounded-br", "rounded-bl"],
      "rounded-l": ["rounded-tl", "rounded-bl"],
      "border-spacing": ["border-spacing-x", "border-spacing-y"],
      "border-w": ["border-w-s", "border-w-e", "border-w-t", "border-w-r", "border-w-b", "border-w-l"],
      "border-w-x": ["border-w-r", "border-w-l"],
      "border-w-y": ["border-w-t", "border-w-b"],
      "border-color": ["border-color-s", "border-color-e", "border-color-t", "border-color-r", "border-color-b", "border-color-l"],
      "border-color-x": ["border-color-r", "border-color-l"],
      "border-color-y": ["border-color-t", "border-color-b"],
      "scroll-m": ["scroll-mx", "scroll-my", "scroll-ms", "scroll-me", "scroll-mt", "scroll-mr", "scroll-mb", "scroll-ml"],
      "scroll-mx": ["scroll-mr", "scroll-ml"],
      "scroll-my": ["scroll-mt", "scroll-mb"],
      "scroll-p": ["scroll-px", "scroll-py", "scroll-ps", "scroll-pe", "scroll-pt", "scroll-pr", "scroll-pb", "scroll-pl"],
      "scroll-px": ["scroll-pr", "scroll-pl"],
      "scroll-py": ["scroll-pt", "scroll-pb"],
      touch: ["touch-x", "touch-y", "touch-pz"],
      "touch-x": ["touch"],
      "touch-y": ["touch"],
      "touch-pz": ["touch"]
    },
    conflictingClassGroupModifiers: {
      "font-size": ["leading"]
    }
  };
}, on = /* @__PURE__ */ Pi(nn);
function V(...e) {
  return on(ft(e));
}
let Re;
function Ve() {
  return Re === void 0 && (Re = navigator.platform.includes("Mac")), Re;
}
function an(e) {
  return `${e}`.toLowerCase() === "mod" ? Ve() ? "⌘" : "Ctrl" : `${e}`.toLowerCase() === "alt" ? Ve() ? "⌥" : "Alt" : `${e}`.toLowerCase() === "shift" ? Ve() ? "⇧" : "Shift" : e;
}
function sn(e) {
  return e.map(an).join(" ");
}
const Nt = q.forwardRef(
  (e, t) => {
    const {
      icon: r = void 0,
      // title = undefined,
      tooltip: i = void 0,
      disabled: o = !1,
      customClass: a = "",
      // color = undefined,
      loading: l = void 0,
      shortcutKeys: s = void 0,
      tooltipOptions: c = {},
      action: u = void 0,
      isActive: f = void 0,
      children: b,
      asChild: d = !1,
      upload: w = !1,
      initialDisplayedColor: m = void 0,
      dataState: M = !1,
      ...g
    } = e, y = St[r], x = d ? ht : zt, L = (p) => {
      if (o) {
        p.preventDefault();
        return;
      }
      u == null || u(p);
    };
    return /* @__PURE__ */ C(Hn, { children: [
      /* @__PURE__ */ n(On, { asChild: !0, children: /* @__PURE__ */ C(
        x,
        {
          className: V("richtext-h-[32px] richtext-w-[32px]", a),
          "data-state": M ? "on" : "off",
          disabled: o,
          onClick: L,
          ref: t,
          size: "sm",
          ...g,
          children: [
            y && /* @__PURE__ */ n(y, { className: "richtext-size-4" }),
            b
          ]
        }
      ) }),
      i && /* @__PURE__ */ n(Tt, { ...c, className: "richtext-tooltip", children: /* @__PURE__ */ C("div", { className: "richtext-flex richtext-max-w-24 richtext-flex-col richtext-items-center richtext-text-center", children: [
        /* @__PURE__ */ n("div", { children: i }),
        !!(s != null && s.length) && /* @__PURE__ */ n("span", { children: sn(s) })
      ] }) })
    ] });
  }
);
function ln(e) {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", ...e, children: /* @__PURE__ */ n(
    "path",
    {
      fill: "currentColor",
      d: "M19 12h-2v3h-3v2h5zM7 9h3V7H5v5h2zm14-6H3a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h18a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2m0 16H3V5h18z"
    }
  ) });
}
function cn(e) {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", ...e, children: /* @__PURE__ */ n(
    "path",
    {
      fill: "none",
      stroke: "currentColor",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: "2",
      d: "M6 15h15m0 4H6m9-8h6m0-4h-6M9 9h1a1 1 0 1 1-1 1V7.5a2 2 0 0 1 2-2M3 9h1a1 1 0 1 1-1 1V7.5a2 2 0 0 1 2-2"
    }
  ) });
}
function dn() {
  return /* @__PURE__ */ n(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      xmlnsXlink: "http://www.w3.org/1999/xlink",
      "aria-hidden": "true",
      role: "img",
      className: "richtext-h-4 richtext-w-4",
      width: "1em",
      height: "1em",
      viewBox: "0 0 24 24",
      children: /* @__PURE__ */ n(
        "path",
        {
          fill: "none",
          stroke: "currentColor",
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: "1.5",
          d: "M5.5 3c1.404 0 2.107 0 2.611.38c.219.164.406.375.552.62C9 4.568 9 5.358 9 6.938v10.125c0 1.58 0 2.37-.337 2.937a2.1 2.1 0 0 1-.552.621c-.504.38-1.207.38-2.611.38s-2.107 0-2.611-.38a2.1 2.1 0 0 1-.552-.62C2 19.432 2 18.642 2 17.062V6.938c0-1.58 0-2.37.337-2.938a2.1 2.1 0 0 1 .552-.62C3.393 3 4.096 3 5.5 3M20 11.938v5.124c0 1.58 0 2.37-.337 2.938a2.1 2.1 0 0 1-.552.62c-.504.38-1.207.38-2.611.38s-2.107 0-2.611-.38a2.1 2.1 0 0 1-.552-.62C13 19.433 13 18.642 13 17.063V6.938c0-1.58 0-2.37.337-2.938M22 9l-6-6m6 0l-6 6",
          color: "currentColor"
        }
      )
    }
  );
}
function un() {
  return /* @__PURE__ */ n(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      xmlnsXlink: "http://www.w3.org/1999/xlink",
      "aria-hidden": "true",
      role: "img",
      className: "richtext-h-4 richtext-w-4",
      width: "1em",
      height: "1em",
      viewBox: "0 0 24 24",
      children: /* @__PURE__ */ n(
        "path",
        {
          fill: "none",
          stroke: "currentColor",
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: "1.5",
          d: "M21 18.5c0 1.404 0 2.107-.38 2.611a2.1 2.1 0 0 1-.62.552c-.567.337-1.358.337-2.937.337H6.938c-1.58 0-2.37 0-2.938-.337a2.1 2.1 0 0 1-.62-.552C3 20.607 3 19.904 3 18.5s0-2.107.38-2.611c.163-.218.374-.406.62-.552C4.567 15 5.357 15 6.938 15h10.125c1.58 0 2.37 0 2.937.337c.246.146.457.334.62.552c.38.504.38 1.207.38 2.611M12.063 4H6.937C5.358 4 4.568 4 4 4.337a2.1 2.1 0 0 0-.62.552C3 5.393 3 6.096 3 7.5s0 2.107.38 2.611c.163.218.374.406.62.552C4.567 11 5.357 11 6.938 11h10.125c1.58 0 2.37 0 2.937-.337M21 8l-6-6m6 0l-6 6",
          color: "currentColor"
        }
      )
    }
  );
}
function hn(e) {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", ...e, children: /* @__PURE__ */ n(
    "path",
    {
      fill: "currentColor",
      d: "M14 2H6a2 2 0 0 0-2 2v16c0 1.11.89 2 2 2h12c1.11 0 2-.89 2-2V8zm4 18H6V4h7v5h5zm-.65-10l-2.1 9h-1.4l-1.8-6.79l-1.8 6.79h-1.4l-2.2-9h1.5l1.4 6.81l1.8-6.81h1.3l1.8 6.81l1.4-6.81z"
    }
  ) });
}
function ge(e) {
  const t = St[e.name];
  return t ? /* @__PURE__ */ n(
    t,
    {
      onClick: e == null ? void 0 : e.onClick,
      className: `richtext-h-4 richtext-w-4 ${(e == null ? void 0 : e.className) || ""}`
    }
  ) : null;
}
function fn(e) {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", ...e, children: /* @__PURE__ */ n(
    "path",
    {
      fill: "currentColor",
      d: "M21 22H3v-2h18zm0-18H3V2h18zm-11 9.7h4l-2-5.4zM11.2 6h1.7l4.7 12h-2l-.9-2.6H9.4L8.5 18h-2z"
    }
  ) });
}
function mn(e) {
  return /* @__PURE__ */ n(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      xmlnsXlink: "http://www.w3.org/1999/xlink",
      "aria-hidden": "true",
      role: "img",
      width: "1em",
      height: "1em",
      viewBox: "0 0 48 48",
      ...e,
      children: /* @__PURE__ */ n(
        "path",
        {
          fill: "currentColor",
          stroke: "currentColor",
          strokeLinejoin: "round",
          strokeWidth: 4,
          d: "M36 19L24 31L12 19z"
        }
      )
    }
  );
}
function gn(e) {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", ...e, children: /* @__PURE__ */ n("path", { fill: "currentColor", d: "M9 7v10h6v-2h-4V7z" }) });
}
function pn(e) {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", ...e, children: /* @__PURE__ */ n(
    "path",
    {
      fill: "currentColor",
      d: "M9 7c-1.1 0-2 .9-2 2v8h2V9h2v7h2V9h2v8h2V9a2 2 0 0 0-2-2z"
    }
  ) });
}
function xn(e) {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", ...e, children: /* @__PURE__ */ n(
    "path",
    {
      fill: "currentColor",
      d: "M11 7c-1.1 0-2 .9-2 2v2a2 2 0 0 0 2 2h2v2H9v2h4c1.11 0 2-.89 2-2v-2a2 2 0 0 0-2-2h-2V9h4V7z"
    }
  ) });
}
function bn() {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", children: /* @__PURE__ */ n(
    "path",
    {
      fill: "currentColor",
      d: "M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6l1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6l-1.4-1.4z"
    }
  ) });
}
function vn() {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", children: /* @__PURE__ */ n(
    "path",
    {
      fill: "none",
      stroke: "currentColor",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: "2",
      d: "M14 4h4a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1h-4a1 1 0 0 1-1-1V5a1 1 0 0 1 1-1m-9 8h4m-2-2v4"
    }
  ) });
}
function wn() {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", children: /* @__PURE__ */ n(
    "path",
    {
      fill: "none",
      stroke: "currentColor",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: "2",
      d: "M6 4h4a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1H6a1 1 0 0 1-1-1V5a1 1 0 0 1 1-1m9 8h4m-2-2v4"
    }
  ) });
}
function yn() {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", children: /* @__PURE__ */ n(
    "path",
    {
      fill: "none",
      stroke: "currentColor",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: "2",
      d: "M12 21v-4m0-4V9m0-4V3m-2 18h4M8 5v4h11l2-2l-2-2zm6 8v4H6l-2-2l2-2z"
    }
  ) });
}
function Cn() {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", children: /* @__PURE__ */ n(
    "path",
    {
      fill: "currentColor",
      d: "M23.943 19.806a.2.2 0 0 0-.168-.034c-1.26-1.855-2.873-3.61-4.419-5.315l-.252-.284c-.001-.073-.067-.12-.134-.15l-.084-.084c-.05-.1-.169-.167-.286-.1c-.47.234-.907.585-1.327.919c-.554.434-1.109.87-1.63 1.354a5 5 0 0 0-.588.618c-.084.117-.017.217.084.267c-.37.368-.74.736-1.109 1.12a.2.2 0 0 0-.05.134c0 .05.033.1.067.117l.655.502v.016c.924.92 2.554 2.173 4.285 3.527c.251.201.52.402.773.602c.117.134.234.285.335.418c.05.066.169.084.236.033c.033.034.084.067.118.1a.24.24 0 0 0 .1.034a.15.15 0 0 0 .135-.066a.24.24 0 0 0 .033-.1c.017 0 .017.016.034.016a.2.2 0 0 0 .134-.05l3.058-3.327c.12-.116.014-.267 0-.267m-7.628-.134l-1.546-1.17l-.15-.1c-.035-.017-.068-.05-.102-.067l-.117-.1c.66-.66 1.33-1.308 2-1.956c-.488.484-1.463 1.906-1.261 2.373c.002 0 .018.042.067.084zm4.1 3.126l-1.277-.97a27 27 0 0 0-1.58-1.504c.69.518 1.277.97 1.361 1.053c.673.585.638.485 1.093.87l.554.4c-.074.103-.151.148-.151.151m.336.25l-.034-.016a1 1 0 0 0 .152-.117zM.588 3.476c.033.217.084.435.117.636c.201 1.103.403 2.106.772 2.858l.152.568c.05.217.134.485.219.552a67 67 0 0 0 3.578 2.942a.18.18 0 0 0 .219 0s0 .016.016.016a.15.15 0 0 0 .118.05a.2.2 0 0 0 .134-.05c1.798-1.989 3.142-3.627 4.1-4.998c.068-.066.084-.167.084-.25c.067-.067.118-.151.185-.201c.067-.067.067-.184 0-.235l-.017-.016c0-.033-.017-.084-.05-.1c-.42-.401-.722-.685-1.042-.986a94 94 0 0 1-2.352-2.273c-.017-.017-.034-.034-.067-.034c-.336-.117-1.025-.234-1.882-.385c-1.277-.216-3.008-.517-4.57-.986c0 0-.101 0-.118.017l-.05.05C.05.714.022.707 0 .718c.017.1.017.167.05.284c0 .033.068.301.068.334zm7.19 4.78l-.033.034a.036.036 0 0 1 .033-.034M6.553 2.238c.101.1.521.502.622.585c-.437-.2-1.529-.702-2.034-.869c.505.1 1.194.201 1.412.284M.79 1.403c.252.434.454 1.939.655 3.41c-.118-.469-.201-.936-.302-1.372C.992 2.673.84 1.988.638 1.386c.124 0 .152.021.152.017m-.286-.369c0-.016 0-.033-.017-.033c.085 0 .135.017.202.05c0 .006-.145-.017-.185-.017m23.17-.217c.017-.066-.336-.367-.219-.384c.253-.017.253-.401 0-.401c-.335.017-.688.1-1.008.15c-.587.117-1.192.234-1.78.367a80 80 0 0 0-3.949.937c-.403.117-.857.2-1.243.401c-.135.067-.118.2-.05.284c-.034.017-.051.017-.085.034c-.117.017-.218.034-.335.05c-.102.017-.152.1-.135.2c0 .017.017.05.017.067c-.706.936-1.496 1.923-2.353 2.976c-.84.969-1.73 1.989-2.62 3.042c-2.84 3.31-6.05 7.07-9.594 10.38a.16.16 0 0 0 0 .234c.016.016.033.033.05.033c-.05.05-.101.085-.152.134q-.05.05-.05.1a.4.4 0 0 0-.067.084c-.067.067-.067.184.017.234c.067.066.185.066.235-.017c.017-.017.017-.033.033-.033a.265.265 0 0 1 .37 0c.202.217.404.435.588.618l-.42-.35c-.067-.067-.184-.05-.234.016c-.068.066-.051.184.016.234l4.469 3.727c.034.034.067.034.118.034a.15.15 0 0 0 .117-.05l.101-.1c.017.016.05.016.067.016c.05 0 .084-.016.118-.05c6.049-6.05 10.922-10.614 16.5-14.693c.05-.033.067-.1.067-.15c.067 0 .118-.05.15-.117c1.026-3.125 1.228-5.9 1.295-7.27c0-.059.016-.038.016-.068c.017-.033.017-.05.017-.05a.98.98 0 0 0-.067-.619m-10.82 4.915c.268-.301.537-.619.806-.903c-1.73 2.273-4.603 5.767-8.67 9.929c2.773-3.059 5.562-6.218 7.864-9.026M5.14 23.466c-.016-.017-.016-.017 0-.017zm2.504-2.156c.135-.15.27-.284.42-.434c0 0 0 .016.017.016c-.224.198-.433.418-.437.418m.69-.668c.099-.1.14-.173.284-.318c.992-1.02 2.017-2.04 3.059-3.076l.016-.016c.252-.2.555-.418.824-.619a228 228 0 0 0-4.184 4.029M14.852 3.91c-.554.719-1.176 1.671-1.697 2.423c-1.646 2.374-6.94 8.174-7.057 8.274a1190 1190 0 0 1-4.839 4.597l-.1.1c-.085-.1-.085-.25.016-.334C8.652 11.966 13.19 6.133 15.021 3.576c-.05.116-.084.216-.168.334zm2.906 3.427c-.671-.386-.99-.987-.806-1.572l.05-.2a.8.8 0 0 1 .085-.167a1.9 1.9 0 0 1 .756-.703c.016 0 .033 0 .05-.016c-.017-.034-.017-.084-.017-.134c.017-.1.085-.167.202-.167c.202 0 .824.184 1.059.384c.067.05.134.117.202.184c.084.1.218.268.285.401c.034.017.067.184.118.268c.033.134.067.284.05.418c-.017.016 0 .116-.017.116a1.6 1.6 0 0 1-.218.619c-.03.03.006.012-.05.067a1.2 1.2 0 0 1-.32.334a1.49 1.49 0 0 1-1.26.234a2 2 0 0 0-.169-.066m4.37 1.403c0 .017-.017.05 0 .067c-.034 0-.05.017-.085.034a110 110 0 0 0-3.915 3.025c1.11-.986 2.218-1.989 3.378-2.975c.336-.301.571-.686.638-1.12l.168-1.003v-.033c.085-.201.404-.118.353.1c-.004-.001-.173.795-.537 1.905"
    }
  ) });
}
function kn() {
  return /* @__PURE__ */ C("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 32 32", children: [
    /* @__PURE__ */ n(
      "path",
      {
        fill: "currentColor",
        d: "M30 18v-2h-6v10h2v-4h3v-2h-3v-2zm-11 8h-4V16h4a3.003 3.003 0 0 1 3 3v4a3.003 3.003 0 0 1-3 3m-2-2h2a1 1 0 0 0 1-1v-4a1 1 0 0 0-1-1h-2zm-6-8H6v10h2v-3h3a2.003 2.003 0 0 0 2-2v-3a2 2 0 0 0-2-2m-3 5v-3h3l.001 3z"
      }
    ),
    /* @__PURE__ */ n(
      "path",
      {
        fill: "currentColor",
        d: "M22 14v-4a.91.91 0 0 0-.3-.7l-7-7A.9.9 0 0 0 14 2H4a2.006 2.006 0 0 0-2 2v24a2 2 0 0 0 2 2h16v-2H4V4h8v6a2.006 2.006 0 0 0 2 2h6v2Zm-8-4V4.4l5.6 5.6Z"
      }
    )
  ] });
}
function Mn() {
  return /* @__PURE__ */ C(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      width: "1em",
      height: "1em",
      className: "icon",
      viewBox: "0 0 1024 1024",
      children: [
        /* @__PURE__ */ n(
          "path",
          {
            fill: "currentColor",
            d: "M679.253 402.364 618.77 561.015l-60.348-158.651a30.04 30.04 0 0 0-30.447-18.637 29.76 29.76 0 0 0-30.447 18.637l-60.416 158.651-60.416-158.651a30.515 30.515 0 0 0-38.843-17.272 28.945 28.945 0 0 0-17.954 37.547l88.815 233.267c4.369 11.469 15.7 19.115 28.398 19.115a30.31 30.31 0 0 0 28.468-19.115l62.395-163.908 62.396 163.84c4.437 11.605 15.701 19.183 28.4 19.183a30.31 30.31 0 0 0 28.466-19.115l88.747-233.267a28.945 28.945 0 0 0-17.886-37.547 30.447 30.447 0 0 0-38.912 17.272zm219.478 395.605-51.883-29.218c-28.672-16.18-52.224-3.072-52.224 29.082v.273H643.209a29.833 29.833 0 0 0-30.31 29.354c0 16.18 13.584 29.218 30.31 29.218h151.825c1.092 30.516 24.03 43.077 52.224 27.648l51.063-27.989c29.013-15.906 29.15-42.189.41-58.368"
          }
        ),
        /* @__PURE__ */ n(
          "path",
          {
            fill: "currentColor",
            d: "m810.667 913.135-.478.068H201.796c-19.865 0-36.727-11.673-36.727-25.6v-618.36h154.965c51.268 0 92.911-39.39 92.911-87.858v-87.86H810.19c19.797 0 36.522 11.742 36.522 25.669V739.26h61.987V119.262c0-46.421-44.169-84.241-98.51-84.241H328.364l-225.28 194.56v658.09c0 46.285 44.236 84.105 98.713 84.105H810.19c43.759 0 80.554-24.713 93.32-58.573h-92.842zM350.89 94.89v86.562c0 16.11-13.858 29.286-30.925 29.286H216.815L350.959 94.891z"
          }
        )
      ]
    }
  );
}
function Nn() {
  return /* @__PURE__ */ n("svg", { height: "1em", viewBox: "0 0 15 15", width: "1em", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ n(
    "path",
    {
      d: "M2.5 10.5H2v.5h.5zm2 0v.5H5v-.5zm9-7h.5v-.207l-.146-.147zm-3-3l.354-.354L10.707 0H10.5zM2 6v4.5h1V6zm.5 5h2v-1h-2zm2.5-.5v-2H4v2zM3 7h2V6H3zM2 5V1.5H1V5zm11-1.5V5h1V3.5zM2.5 1h8V0h-8zm7.646-.146l3 3l.708-.708l-3-3zM2 1.5a.5.5 0 0 1 .5-.5V0A1.5 1.5 0 0 0 1 1.5zM1 12v1.5h1V12zm1.5 3h10v-1h-10zM14 13.5V12h-1v1.5zM12.5 15a1.5 1.5 0 0 0 1.5-1.5h-1a.5.5 0 0 1-.5.5zM1 13.5A1.5 1.5 0 0 0 2.5 15v-1a.5.5 0 0 1-.5-.5zM6 7h3V6H6zm0 4h3v-1H6zm1-4.5v4h1v-4zm3.5.5H13V6h-2.5zM10 6v5h1V6zm.5 3H12V8h-1.5z",
      fill: "currentColor"
    }
  ) });
}
function Sn() {
  return /* @__PURE__ */ n("svg", { height: "1em", viewBox: "0 0 24 24", width: "1em", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ C(
    "g",
    {
      fill: "none",
      stroke: "currentColor",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: "2",
      children: [
        /* @__PURE__ */ n("path", { d: "M14 3v4a1 1 0 0 0 1 1h4" }),
        /* @__PURE__ */ n("path", { d: "M5 12V5a2 2 0 0 1 2-2h7l5 5v4M2 21v-6m3 0v6m-3-3h3m15-3v6h2m-9 0v-6l2 3l2-3v6m-9.5-6h3M9 15v6" })
      ]
    }
  ) });
}
function zn() {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 20 20", children: /* @__PURE__ */ n(
    "path",
    {
      fill: "currentColor",
      d: "M6.5 3a.75.75 0 0 1 .697.471l3 7.5a.75.75 0 0 1-1.393.557L7.992 9.5H5.008l-.811 2.028a.75.75 0 0 1-1.393-.556l3-7.5A.75.75 0 0 1 6.5 3m0 2.77L5.608 8h1.784zm8.28-1.55a.75.75 0 1 0-1.06 1.06l.72.72h-3.69a.75.75 0 0 0 0 1.5h3.69l-.72.72a.75.75 0 0 0 1.06 1.06l2-2a.75.75 0 0 0 0-1.06zm0 7.5a.75.75 0 1 0-1.06 1.06l.72.72H3.75a.75.75 0 0 0 0 1.5h10.69l-.72.72a.75.75 0 1 0 1.06 1.06l2-2a.75.75 0 0 0 0-1.06z"
    }
  ) });
}
function Tn() {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 48 48", children: /* @__PURE__ */ C("g", { fill: "none", stroke: "currentColor", strokeWidth: "4", children: [
    /* @__PURE__ */ n("circle", { cx: "10", cy: "24", r: "4" }),
    /* @__PURE__ */ n("circle", { cx: "38", cy: "10", r: "4" }),
    /* @__PURE__ */ n("circle", { cx: "38", cy: "24", r: "4" }),
    /* @__PURE__ */ n("circle", { cx: "38", cy: "38", r: "4" }),
    /* @__PURE__ */ n("path", { strokeLinecap: "round", strokeLinejoin: "round", d: "M34 38H22V10h12M14 24h20" })
  ] }) });
}
function In() {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 20 20", children: /* @__PURE__ */ n(
    "path",
    {
      fill: "currentColor",
      d: "M13.5 3a.75.75 0 0 0-.697.471l-3 7.5a.75.75 0 0 0 1.393.557l.812-2.028h2.984l.811 2.028a.75.75 0 0 0 1.393-.556l-3-7.5A.75.75 0 0 0 13.5 3m0 2.77L14.392 8h-1.784zM5.22 4.22a.75.75 0 0 1 1.06 1.06L5.56 6h3.69a.75.75 0 0 1 0 1.5H5.56l.72.72a.75.75 0 0 1-1.06 1.06l-2-2a.75.75 0 0 1 0-1.06zm0 7.5a.75.75 0 0 1 1.06 1.06l-.72.72h10.69a.75.75 0 0 1 0 1.5H5.56l.72.72a.75.75 0 1 1-1.06 1.06l-2-2a.75.75 0 0 1 0-1.06z"
    }
  ) });
}
function Ln() {
  return /* @__PURE__ */ n("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", children: /* @__PURE__ */ n(
    "path",
    {
      fill: "none",
      stroke: "currentColor",
      strokeLinecap: "round",
      strokeLinejoin: "round",
      strokeWidth: "1.5",
      d: "m3 21l7.548-7.548M21 3l-7.548 7.548m0 0L8 3H3l7.548 10.452m2.904-2.904L21 21h-5l-5.452-7.548",
      color: "currentColor"
    }
  ) });
}
function An() {
  return /* @__PURE__ */ n(
    "svg",
    {
      xmlns: "http://www.w3.org/2000/svg",
      xmlnsXlink: "http://www.w3.org/1999/xlink",
      "aria-hidden": "true",
      role: "img",
      width: "1em",
      height: "1em",
      viewBox: "0 0 24 24",
      children: /* @__PURE__ */ n(
        "path",
        {
          fill: "currentColor",
          d: "M14 2H6a2 2 0 0 0-2 2v16c0 1.11.89 2 2 2h12c1.11 0 2-.89 2-2V8zm4 18H6V4h7v5h5zm-.65-10l-2.1 9h-1.4l-1.8-6.79l-1.8 6.79h-1.4l-2.2-9h1.5l1.4 6.81l1.8-6.81h1.3l1.8 6.81l1.4-6.81z"
        }
      )
    }
  );
}
const St = {
  Bold: xi,
  LoaderCircle: pi,
  Italic: gi,
  Underline: mi,
  Quote: fi,
  TextQuote: cn,
  Strikethrough: hi,
  Minus: ui,
  Eraser: di,
  PaintRoller: ci,
  Redo2: li,
  Undo2: si,
  AlignCenter: Ni,
  AlignJustify: Mi,
  AlignLeft: ki,
  AlignRight: Ci,
  ChevronDown: $e,
  Subscript: ai,
  Superscript: oi,
  Code: ni,
  Code2: ii,
  Type: ri,
  IndentIncrease: ti,
  IndentDecrease: ei,
  List: Qr,
  ListOrdered: Jr,
  ListTodo: Kr,
  Link: Zr,
  ImageUp: qr,
  Video: Yr,
  Maximize: Xr,
  Minimize: Gr,
  Table: Wr,
  Sparkles: Fr,
  Pencil: $r,
  Unlink: Br,
  BetweenHorizonalEnd: Ur,
  BetweenHorizonalStart: _r,
  BetweenVerticalStart: Pr,
  BetweenVerticalEnd: jr,
  TableCellsMerge: Or,
  TableCellsSplit: Hr,
  Trash2: Dr,
  Trash: Vr,
  Replace: Qe,
  ChevronsUpDown: Rr,
  LineHeight: fn,
  Word: hn,
  HeadingParagraph: Er,
  Heading1: Ar,
  Heading2: Lr,
  Heading3: Ir,
  Heading4: Tr,
  Heading5: zr,
  Heading6: Sr,
  Columns2: et,
  Columns3: Nr,
  Columns4: Mr,
  Plus: kr,
  Grip: Cr,
  Copy: yr,
  Clipboard: wr,
  PanelLeft: vr,
  PanelRight: br,
  Columns: et,
  Iframe: xr,
  MenuDown: mn,
  SizeS: xn,
  SizeM: pn,
  SizeL: gn,
  AspectRatio: ln,
  Emoji: pr,
  DeleteColumn: dn,
  DeleteRow: un,
  SearchAndReplace: Qe,
  EmojiIcon: gr,
  KatexIcon: mr,
  ExportPdf: kn,
  ExportWord: Mn,
  ImportWord: An,
  ColumnAddLeft: vn,
  ColumnAddRight: wn,
  BookMarked: fr,
  Excalidraw: Cn,
  ZoomIn: hr,
  ZoomOut: ur,
  Settings: dr,
  Eye: cr,
  TextDirection: yn,
  LeftToRight: zn,
  RightToLeft: In,
  Attachment: lr,
  GifIcon: Nn,
  ChevronUp: gt,
  Crop: sr,
  Mermaid: Tn,
  Twitter: Ln,
  CodeView: bn,
  FlipX: ar,
  FlipY: or,
  PencilRuler: nr,
  WrapText: ir,
  Loader: rr,
  X: mt,
  Html: Sn,
  ExternalLink: tr,
  Callout: er
}, ot = (e) => typeof e == "boolean" ? `${e}` : e === 0 ? "0" : e, at = ft, We = (e, t) => (r) => {
  var i;
  if ((t == null ? void 0 : t.variants) == null) return at(e, r == null ? void 0 : r.class, r == null ? void 0 : r.className);
  const { variants: o, defaultVariants: a } = t, l = Object.keys(o).map((u) => {
    const f = r == null ? void 0 : r[u], b = a == null ? void 0 : a[u];
    if (f === null) return null;
    const d = ot(f) || ot(b);
    return o[u][d];
  }), s = r && Object.entries(r).reduce((u, f) => {
    let [b, d] = f;
    return d === void 0 || (u[b] = d), u;
  }, {}), c = t == null || (i = t.compoundVariants) === null || i === void 0 ? void 0 : i.reduce((u, f) => {
    let { class: b, className: d, ...w } = f;
    return Object.entries(w).every((m) => {
      let [M, g] = m;
      return Array.isArray(g) ? g.includes({
        ...a,
        ...s
      }[M]) : {
        ...a,
        ...s
      }[M] === g;
    }) ? [
      ...u,
      b,
      d
    ] : u;
  }, []);
  return at(e, l, c, r == null ? void 0 : r.class, r == null ? void 0 : r.className);
}, En = We(
  "richtext-inline-flex richtext-items-center richtext-justify-center richtext-whitespace-nowrap richtext-rounded-md richtext-text-sm richtext-font-medium richtext-ring-offset-background richtext-transition-colors focus-visible:richtext-outline-none focus-visible:richtext-ring-2 focus-visible:richtext-ring-ring focus-visible:richtext-ring-offset-1 disabled:richtext-pointer-events-none disabled:richtext-opacity-50",
  {
    variants: {
      variant: {
        default: "!richtext-bg-primary !richtext-text-primary-foreground hover:!richtext-bg-primary/90",
        destructive: "richtext-bg-destructive richtext-text-destructive-foreground hover:richtext-bg-destructive/90",
        outline: "richtext-border richtext-border-input richtext-bg-background richtext-text-foreground hover:richtext-bg-accent hover:richtext-text-accent-foreground",
        secondary: "richtext-bg-secondary richtext-text-secondary-foreground hover:richtext-bg-secondary/80",
        ghost: "hover:richtext-bg-accent hover:richtext-text-accent-foreground",
        link: "richtext-text-primary richtext-underline-offset-4 hover:richtext-underline"
      },
      size: {
        default: "richtext-h-10 richtext-px-4 richtext-py-2",
        sm: "richtext-h-9 richtext-rounded-md richtext-px-3",
        lg: "richtext-h-11 richtext-rounded-md richtext-px-8",
        icon: "richtext-size-10"
      }
    },
    defaultVariants: {
      variant: "default",
      size: "default"
    }
  }
), G = O.forwardRef(
  ({ className: e, variant: t, size: r, asChild: i = !1, ...o }, a) => /* @__PURE__ */ n(i ? ht : "button", { className: V(En({ variant: t, size: r, className: e })), ref: a, ...o })
);
G.displayName = "Button";
const ie = O.forwardRef(
  ({ className: e, type: t, ...r }, i) => /* @__PURE__ */ n(
    "input",
    {
      ref: i,
      type: t,
      className: V(
        "richtext-flex richtext-h-10 richtext-w-full richtext-rounded-md !richtext-border richtext-border-input richtext-bg-transparent richtext-px-3 richtext-py-2 richtext-text-sm richtext-text-foreground richtext-ring-offset-background file:richtext-border-0 file:richtext-bg-transparent file:richtext-text-sm file:richtext-font-medium placeholder:richtext-text-muted-foreground focus-visible:richtext-outline-none focus-visible:richtext-ring-1 focus-visible:richtext-ring-ring focus-visible:richtext-ring-offset-1 disabled:richtext-cursor-not-allowed disabled:richtext-opacity-50",
        e
      ),
      ...r
    }
  )
);
ie.displayName = "Input";
const Rn = We(
  "richtext-text-sm richtext-font-medium richtext-leading-none richtext-text-foreground peer-disabled:richtext-cursor-not-allowed peer-disabled:richtext-opacity-70"
), be = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(vt.Root, { className: V(Rn(), e), ref: r, ...t }));
be.displayName = vt.Root.displayName;
const Le = ne.Root, ye = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(
  ne.List,
  {
    ref: r,
    className: V(
      "richtext-inline-flex richtext-h-10 richtext-items-center richtext-justify-center richtext-rounded-md richtext-bg-muted richtext-p-1 richtext-text-muted-foreground",
      e
    ),
    ...t
  }
));
ye.displayName = ne.List.displayName;
const J = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(
  ne.Trigger,
  {
    ref: r,
    className: V(
      "richtext-inline-flex richtext-items-center richtext-justify-center richtext-whitespace-nowrap richtext-rounded-sm richtext-px-3 richtext-py-1.5 richtext-text-sm richtext-font-medium richtext-ring-offset-background richtext-transition-all focus-visible:richtext-outline-none focus-visible:richtext-ring-2 focus-visible:richtext-ring-ring focus-visible:richtext-ring-offset-1 disabled:richtext-pointer-events-none disabled:richtext-opacity-50 data-[state=active]:richtext-bg-background data-[state=active]:richtext-text-foreground data-[state=active]:richtext-shadow-sm",
      e
    ),
    ...t
  }
));
J.displayName = ne.Trigger.displayName;
const Q = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(
  ne.Content,
  {
    ref: r,
    className: V(
      "richtext-mt-2 richtext-ring-offset-background focus-visible:richtext-outline-none focus-visible:richtext-ring-2 focus-visible:richtext-ring-ring focus-visible:richtext-ring-offset-1",
      e
    ),
    ...t
  }
));
Q.displayName = ne.Content.displayName;
const Vn = O.forwardRef(
  ({ className: e, ...t }, r) => /* @__PURE__ */ n(
    "textarea",
    {
      ref: r,
      className: V(
        "richtext-flex richtext-min-h-[80px] richtext-w-full richtext-rounded-md !richtext-border richtext-border-input richtext-bg-transparent richtext-px-3 richtext-py-2 richtext-text-sm richtext-ring-offset-background placeholder:richtext-text-muted-foreground focus-visible:richtext-outline-none focus-visible:richtext-ring-2 focus-visible:richtext-ring-ring focus-visible:richtext-ring-offset-1 disabled:richtext-cursor-not-allowed disabled:richtext-opacity-50",
        e
      ),
      ...t
    }
  )
);
Vn.displayName = "Textarea";
const Dn = We(
  "richtext-inline-flex richtext-items-center richtext-justify-center richtext-rounded-md richtext-text-sm richtext-font-medium richtext-ring-offset-background richtext-transition-colors hover:richtext-bg-muted hover:richtext-text-muted-foreground focus-visible:richtext-outline-none focus-visible:richtext-ring-2 focus-visible:richtext-ring-ring focus-visible:richtext-ring-offset-1 disabled:richtext-pointer-events-none disabled:richtext-opacity-50 data-[state=on]:richtext-bg-accent data-[state=on]:richtext-text-accent-foreground",
  {
    variants: {
      variant: {
        default: "richtext-bg-transparent",
        outline: "richtext-border richtext-border-input richtext-bg-transparent hover:richtext-bg-accent hover:richtext-text-accent-foreground"
      },
      size: {
        default: "richtext-h-10 richtext-px-3",
        sm: "richtext-h-9 richtext-px-2",
        lg: "richtext-h-11 richtext-px-5"
      }
    },
    defaultVariants: {
      variant: "default",
      size: "default"
    }
  }
), zt = O.forwardRef(({ className: e, variant: t, size: r, ...i }, o) => /* @__PURE__ */ n(
  bt.Root,
  {
    ref: o,
    className: V(Dn({ variant: t, size: r, className: e })),
    ...i
  }
));
zt.displayName = bt.Root.displayName;
const ra = we.Provider, Hn = we.Root, On = we.Trigger, Tt = O.forwardRef(({ className: e, sideOffset: t = 4, ...r }, i) => /* @__PURE__ */ n(
  we.Content,
  {
    "data-richtext-portal": !0,
    ref: i,
    sideOffset: t,
    className: V(
      "richtext-z-50 richtext-overflow-hidden richtext-rounded-md !richtext-border-none richtext-bg-primary richtext-px-3 richtext-py-1.5 richtext-text-sm richtext-text-primary-foreground richtext-shadow-md richtext-animate-in richtext-fade-in-0 richtext-zoom-in-95 data-[side=bottom]:richtext-slide-in-from-top-2 data-[side=left]:richtext-slide-in-from-right-2 data-[side=right]:richtext-slide-in-from-left-2 data-[side=top]:richtext-slide-in-from-bottom-2",
      e
    ),
    ...r
  }
));
Tt.displayName = we.Content.displayName;
const ia = B.Root, na = B.Value, jn = O.forwardRef(({ className: e, children: t, ...r }, i) => /* @__PURE__ */ C(
  B.Trigger,
  {
    ref: i,
    className: V(
      "richtext-flex richtext-h-10 richtext-w-full richtext-items-center richtext-justify-between richtext-rounded-md richtext-border richtext-border-input richtext-bg-background richtext-px-3 richtext-py-2 richtext-text-sm richtext-text-foreground richtext-ring-offset-background placeholder:richtext-text-muted-foreground focus:richtext-outline-none focus:richtext-ring-2 focus:richtext-ring-ring focus:richtext-ring-offset-1 disabled:richtext-cursor-not-allowed disabled:richtext-opacity-50 [&>span]:richtext-line-clamp-1",
      e
    ),
    ...r,
    children: [
      t,
      /* @__PURE__ */ n(B.Icon, { asChild: !0, children: /* @__PURE__ */ n($e, { className: "richtext-size-4 richtext-opacity-50" }) })
    ]
  }
));
jn.displayName = B.Trigger.displayName;
const It = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(
  B.ScrollUpButton,
  {
    ref: r,
    className: V(
      "richtext-flex richtext-cursor-default richtext-items-center richtext-justify-center richtext-py-1",
      e
    ),
    ...t,
    children: /* @__PURE__ */ n(gt, { className: "richtext-size-4" })
  }
));
It.displayName = B.ScrollUpButton.displayName;
const Lt = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(
  B.ScrollDownButton,
  {
    ref: r,
    className: V(
      "richtext-flex richtext-cursor-default richtext-items-center richtext-justify-center richtext-py-1",
      e
    ),
    ...t,
    children: /* @__PURE__ */ n($e, { className: "richtext-size-4" })
  }
));
Lt.displayName = B.ScrollDownButton.displayName;
const Pn = O.forwardRef(({ className: e, children: t, position: r = "popper", ...i }, o) => /* @__PURE__ */ n(B.Portal, { children: /* @__PURE__ */ C(
  B.Content,
  {
    "data-richtext-portal": !0,
    position: r,
    ref: o,
    className: V(
      "richtext-relative richtext-z-50 richtext-max-h-60 richtext-min-w-[8rem] richtext-overflow-hidden richtext-overflow-y-auto richtext-rounded-md richtext-border !richtext-border-border richtext-bg-popover richtext-text-popover-foreground richtext-shadow-md data-[state=open]:richtext-animate-in data-[state=closed]:richtext-animate-out data-[state=closed]:richtext-fade-out-0 data-[state=open]:richtext-fade-in-0 data-[state=closed]:richtext-zoom-out-95 data-[state=open]:richtext-zoom-in-95 data-[side=bottom]:richtext-slide-in-from-top-2 data-[side=left]:richtext-slide-in-from-right-2 data-[side=right]:richtext-slide-in-from-left-2 data-[side=top]:richtext-slide-in-from-bottom-2",
      r === "popper" && "data-[side=bottom]:richtext-translate-y-1 data-[side=left]:richtext--translate-x-1 data-[side=right]:richtext-translate-x-1 data-[side=top]:richtext--translate-y-1",
      e
    ),
    ...i,
    children: [
      /* @__PURE__ */ n(It, {}),
      /* @__PURE__ */ n(
        B.Viewport,
        {
          className: V(
            "richtext-p-1",
            r === "popper" && "richtext-h-[var(--radix-select-trigger-height)] richtext-w-full richtext-min-w-[var(--radix-select-trigger-width)]"
          ),
          children: t
        }
      ),
      /* @__PURE__ */ n(Lt, {})
    ]
  }
) }));
Pn.displayName = B.Content.displayName;
const _n = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(
  B.Label,
  {
    className: V(
      "richtext-py-1.5 richtext-pl-8 richtext-pr-2 richtext-text-sm richtext-font-semibold",
      e
    ),
    ref: r,
    ...t
  }
));
_n.displayName = B.Label.displayName;
const Un = O.forwardRef(({ className: e, children: t, ...r }, i) => /* @__PURE__ */ C(
  B.Item,
  {
    ref: i,
    className: V(
      "richtext-relative richtext-flex richtext-w-full richtext-cursor-default richtext-select-none richtext-items-center richtext-rounded-sm richtext-py-1.5 richtext-pl-8 richtext-pr-2 richtext-text-sm richtext-outline-none focus:richtext-bg-accent focus:richtext-text-accent-foreground data-[disabled]:richtext-pointer-events-none data-[disabled]:richtext-opacity-50",
      e
    ),
    ...r,
    children: [
      /* @__PURE__ */ n("span", { className: "richtext-absolute richtext-left-2 richtext-flex richtext-size-3.5 richtext-items-center richtext-justify-center", children: /* @__PURE__ */ n(B.ItemIndicator, { children: /* @__PURE__ */ n(pt, { className: "richtext-size-4" }) }) }),
      /* @__PURE__ */ n(B.ItemText, { children: t })
    ]
  }
));
Un.displayName = B.Item.displayName;
const Bn = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(
  B.Separator,
  {
    className: V("richtext--mx-1 richtext-my-1 richtext-h-px richtext-bg-muted", e),
    ref: r,
    ...t
  }
));
Bn.displayName = B.Separator.displayName;
const $n = 1, Fn = 1e6;
let De = 0;
function Wn() {
  return De = (De + 1) % Number.MAX_SAFE_INTEGER, De.toString();
}
const He = /* @__PURE__ */ new Map();
function st(e) {
  if (He.has(e))
    return;
  const t = setTimeout(() => {
    He.delete(e), pe({
      type: "REMOVE_TOAST",
      toastId: e
    });
  }, Fn);
  He.set(e, t);
}
function Gn(e, t) {
  switch (t.type) {
    case "ADD_TOAST":
      return {
        ...e,
        toasts: [t.toast, ...e.toasts].slice(0, $n)
      };
    case "UPDATE_TOAST":
      return {
        ...e,
        toasts: e.toasts.map((r) => r.id === t.toast.id ? { ...r, ...t.toast } : r)
      };
    case "DISMISS_TOAST": {
      const { toastId: r } = t;
      return r ? st(r) : e.toasts.forEach((i) => {
        st(i.id);
      }), {
        ...e,
        toasts: e.toasts.map(
          (i) => i.id === r || r === void 0 ? {
            ...i,
            open: !1
          } : i
        )
      };
    }
    case "REMOVE_TOAST":
      return t.toastId === void 0 ? {
        ...e,
        toasts: []
      } : {
        ...e,
        toasts: e.toasts.filter((r) => r.id !== t.toastId)
      };
  }
}
const Me = [];
let Ne = { toasts: [] };
function pe(e) {
  Ne = Gn(Ne, e), Me.forEach((t) => {
    t(Ne);
  });
}
function Xn({ ...e }) {
  const t = Wn(), r = (o) => pe({
    type: "UPDATE_TOAST",
    toast: { ...o, id: t }
  }), i = () => pe({ type: "DISMISS_TOAST", toastId: t });
  return pe({
    type: "ADD_TOAST",
    toast: {
      ...e,
      id: t,
      open: !0,
      onOpenChange: (o) => {
        o || i();
      }
    }
  }), {
    id: t,
    dismiss: i,
    update: r
  };
}
function Ge() {
  const [e, t] = O.useState(Ne);
  return O.useEffect(() => (Me.push(t), () => {
    const r = Me.indexOf(t);
    r > -1 && Me.splice(r, 1);
  }), [e]), {
    ...e,
    toast: Xn,
    dismiss: (r) => pe({ type: "DISMISS_TOAST", toastId: r })
  };
}
const Xe = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(
  _e.Root,
  {
    ref: r,
    className: V(
      "!richtext-peer !richtext-h-4 !richtext-w-4 !richtext-shrink-0 !richtext-rounded-sm !richtext-border !richtext-border-primary !richtext-p-0 !richtext-ring-offset-background focus-visible:!richtext-outline-none focus-visible:!richtext-ring-2 focus-visible:!richtext-ring-ring focus-visible:!richtext-ring-offset-1 disabled:!richtext-cursor-not-allowed disabled:!richtext-opacity-50 data-[state=checked]:!richtext-bg-primary data-[state=checked]:!richtext-text-primary-foreground",
      e
    ),
    ...t,
    children: /* @__PURE__ */ n(
      _e.Indicator,
      {
        className: V(
          "!richtext-flex !richtext-items-center !richtext-justify-center !richtext-p-0 !richtext-text-current"
        ),
        children: /* @__PURE__ */ n(pt, { className: "!richtext-h-4 !richtext-w-4" })
      }
    )
  }
));
Xe.displayName = _e.Root.displayName;
function Yn(e) {
  return e = e || /* @__PURE__ */ new Map(), {
    /**
     * A Map of event names to registered handler functions.
     */
    all: e,
    /**
     * Register an event handler for the given type.
     * @param {string|symbol} type Type of event to listen for, or `'*'` for all events
     * @param {Function} handler Function to call in response to given event
     * @memberOf mitt
     */
    on(t, r) {
      const i = e.get(t);
      i ? i.push(r) : e.set(t, [r]);
    },
    /**
     * Remove an event handler for the given type.
     * If `handler` is omitted, all handlers of the given type are removed.
     * @param {string|symbol} type Type of event to unregister `handler` from (`'*'` to remove a wildcard handler)
     * @param {Function} [handler] Handler function to remove
     * @memberOf mitt
     */
    off(t, r) {
      const i = e.get(t);
      i && (r ? i.splice(i.indexOf(r) >>> 0, 1) : e.set(t, []));
    },
    /**
     * Invoke all handlers for the given type.
     * If present, `'*'` handlers are invoked after type-matched handlers.
     *
     * Note: Manually firing '*' handlers is not supported.
     *
     * @param {string|symbol} type The event type to invoke
     * @param {Any} [evt] Any value (object is recommended and powerful), passed to each handler
     * @memberOf mitt
     */
    emit(t, r) {
      let i = e.get(t);
      i && i.slice().map((o) => {
        o(r);
      }), i = e.get("*"), i && i.slice().map((o) => {
        o(t, r);
      });
    }
  };
}
const Ye = Yn(), At = q.createContext(Ye), qn = () => q.useContext(At);
function Et(e, t) {
  const r = qn();
  q.useEffect(() => (t.map((i) => r.on(i, e)), () => {
    t.map((i) => r.off(i, e));
  }), [r, t, e]);
}
const oa = Ye.emit;
function aa({ children: e }) {
  return /* @__PURE__ */ n(At.Provider, { value: Ye, children: e });
}
const Ce = X.Root, qe = X.Trigger, Zn = X.Portal, Rt = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(
  X.Overlay,
  {
    ref: r,
    className: V(
      "richtext-fixed richtext-inset-0 richtext-z-50 richtext-bg-black/80 data-[state=open]:richtext-animate-in data-[state=closed]:richtext-animate-out data-[state=closed]:richtext-fade-out-0 data-[state=open]:richtext-fade-in-0",
      e
    ),
    ...t
  }
));
Rt.displayName = X.Overlay.displayName;
const le = O.forwardRef(({ className: e, children: t, ...r }, i) => /* @__PURE__ */ C(Zn, { "data-richtext-portal": !0, children: [
  /* @__PURE__ */ n(Rt, {}),
  /* @__PURE__ */ C(
    X.Content,
    {
      "data-richtext-portal": !0,
      ref: i,
      className: V(
        "richtext-fixed richtext-left-[50%] richtext-top-[50%] richtext-z-50 richtext-grid richtext-w-full richtext-max-w-lg richtext-translate-x-[-50%] richtext-translate-y-[-50%] richtext-gap-4 richtext-border !richtext-border-border richtext-bg-background richtext-p-6 richtext-shadow-lg richtext-duration-200 data-[state=open]:richtext-animate-in data-[state=closed]:richtext-animate-out data-[state=closed]:richtext-fade-out-0 data-[state=open]:richtext-fade-in-0 data-[state=closed]:richtext-zoom-out-95 data-[state=open]:richtext-zoom-in-95 data-[state=closed]:richtext-slide-out-to-left-1/2 data-[state=closed]:richtext-slide-out-to-top-[48%] data-[state=open]:richtext-slide-in-from-left-1/2 data-[state=open]:richtext-slide-in-from-top-[48%] sm:richtext-rounded-lg",
        e
      ),
      ...r,
      children: [
        t,
        /* @__PURE__ */ C(X.Close, { className: "richtext-absolute richtext-right-4 richtext-top-4 richtext-rounded-sm richtext-opacity-70 richtext-ring-offset-background richtext-transition-opacity hover:richtext-opacity-100 focus:richtext-outline-none focus:richtext-ring-2 focus:richtext-ring-ring focus:richtext-ring-offset-1 disabled:richtext-pointer-events-none data-[state=open]:richtext-bg-accent data-[state=open]:richtext-text-muted-foreground", children: [
          /* @__PURE__ */ n(mt, { className: "richtext-size-4 richtext-text-accent-foreground" }),
          /* @__PURE__ */ n("span", { className: "richtext-sr-only richtext-text-accent-foreground", children: "Close" })
        ] })
      ]
    }
  )
] }));
le.displayName = X.Content.displayName;
function Kn({ className: e, ...t }) {
  return /* @__PURE__ */ n(
    "div",
    {
      className: V(
        "richtext-flex richtext-flex-col richtext-space-y-1.5 richtext-text-center sm:richtext-text-left",
        e
      ),
      ...t
    }
  );
}
Kn.displayName = "DialogHeader";
function Vt({ className: e, ...t }) {
  return /* @__PURE__ */ n(
    "div",
    {
      className: V(
        "richtext-flex richtext-flex-col-reverse sm:richtext-flex-row sm:richtext-justify-end sm:richtext-space-x-2",
        e
      ),
      ...t
    }
  );
}
Vt.displayName = "DialogFooter";
const ce = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(
  X.Title,
  {
    ref: r,
    className: V(
      "richtext-text-lg richtext-font-semibold richtext-leading-none richtext-tracking-tight richtext-text-foreground",
      e
    ),
    ...t
  }
));
ce.displayName = X.Title.displayName;
const Jn = O.forwardRef(({ className: e, ...t }, r) => /* @__PURE__ */ n(
  X.Description,
  {
    className: V("richtext-text-sm richtext-text-muted-foreground", e),
    ref: r,
    ...t
  }
));
Jn.displayName = X.Description.displayName;
var Qn = /(?:^|\s)(!\[(.+|:?)]\((\S+)(?:(?:\s+)["'](\S+)["'])?\))$/, eo = xt.create({
  name: "image",
  addOptions() {
    return {
      inline: !1,
      allowBase64: !1,
      HTMLAttributes: {},
      resize: !1
    };
  },
  inline() {
    return this.options.inline;
  },
  group() {
    return this.options.inline ? "inline" : "block";
  },
  draggable: !0,
  addAttributes() {
    return {
      src: {
        default: null
      },
      alt: {
        default: null
      },
      title: {
        default: null
      },
      width: {
        default: null
      },
      height: {
        default: null
      }
    };
  },
  parseHTML() {
    return [
      {
        tag: this.options.allowBase64 ? "img[src]" : 'img[src]:not([src^="data:"])'
      }
    ];
  },
  renderHTML({ HTMLAttributes: e }) {
    return ["img", Pe(this.options.HTMLAttributes, e)];
  },
  parseMarkdown: (e, t) => t.createNode("image", {
    src: e.href,
    title: e.title,
    alt: e.text
  }),
  renderMarkdown: (e) => {
    var t, r, i, o, a, l;
    const s = (r = (t = e.attrs) == null ? void 0 : t.src) != null ? r : "", c = (o = (i = e.attrs) == null ? void 0 : i.alt) != null ? o : "", u = (l = (a = e.attrs) == null ? void 0 : a.title) != null ? l : "";
    return u ? `![${c}](${s} "${u}")` : `![${c}](${s})`;
  },
  addNodeView() {
    if (!this.options.resize || !this.options.resize.enabled || typeof document > "u")
      return null;
    const { directions: e, minWidth: t, minHeight: r, alwaysPreserveAspectRatio: i } = this.options.resize;
    return ({ node: o, getPos: a, HTMLAttributes: l, editor: s }) => {
      const c = document.createElement("img");
      c.draggable = !1;
      const u = Pe(this.options.HTMLAttributes, l);
      Object.entries(u).forEach(([d, w]) => {
        if (w != null)
          switch (d) {
            case "width":
            case "height":
              break;
            default:
              c.setAttribute(d, w);
              break;
          }
      }), u.src !== null && (c.src = u.src);
      const f = new vi({
        element: c,
        editor: s,
        node: o,
        getPos: a,
        onResize: (d, w) => {
          c.style.width = `${d}px`, c.style.height = `${w}px`;
        },
        onCommit: (d, w) => {
          const m = a();
          m !== void 0 && this.editor.chain().setNodeSelection(m).updateAttributes(this.name, {
            width: d,
            height: w
          }).run();
        },
        onUpdate: (d, w, m) => d.type === o.type,
        options: {
          directions: e,
          min: {
            width: t,
            height: r
          },
          preserveAspectRatio: i === !0
        }
      }), b = f.dom;
      return b.style.visibility = "hidden", b.style.pointerEvents = "none", c.onload = () => {
        b.style.visibility = "", b.style.pointerEvents = "";
      }, f;
    };
  },
  addCommands() {
    return {
      setImage: (e) => ({ commands: t }) => t.insertContent({
        type: this.name,
        attrs: e
      })
    };
  },
  addInputRules() {
    return [
      bi({
        find: Qn,
        type: this.type,
        getAttributes: (e) => {
          const [, , t, r, i] = e;
          return { src: r, alt: t, title: i };
        }
      })
    ];
  }
}), to = eo, ro = typeof global == "object" && global && global.Object === Object && global, io = typeof self == "object" && self && self.Object === Object && self, Dt = ro || io || Function("return this")(), ze = Dt.Symbol, Ht = Object.prototype, no = Ht.hasOwnProperty, oo = Ht.toString, me = ze ? ze.toStringTag : void 0;
function ao(e) {
  var t = no.call(e, me), r = e[me];
  try {
    e[me] = void 0;
    var i = !0;
  } catch {
  }
  var o = oo.call(e);
  return i && (t ? e[me] = r : delete e[me]), o;
}
var so = Object.prototype, lo = so.toString;
function co(e) {
  return lo.call(e);
}
var uo = "[object Null]", ho = "[object Undefined]", lt = ze ? ze.toStringTag : void 0;
function Ot(e) {
  return e == null ? e === void 0 ? ho : uo : lt && lt in Object(e) ? ao(e) : co(e);
}
function jt(e) {
  return e != null && typeof e == "object";
}
var fo = "[object Symbol]";
function mo(e) {
  return typeof e == "symbol" || jt(e) && Ot(e) == fo;
}
var go = /\s/;
function po(e) {
  for (var t = e.length; t-- && go.test(e.charAt(t)); )
    ;
  return t;
}
var xo = /^\s+/;
function bo(e) {
  return e && e.slice(0, po(e) + 1).replace(xo, "");
}
function Te(e) {
  var t = typeof e;
  return e != null && (t == "object" || t == "function");
}
var ct = NaN, vo = /^[-+]0x[0-9a-f]+$/i, wo = /^0b[01]+$/i, yo = /^0o[0-7]+$/i, Co = parseInt;
function xe(e) {
  if (typeof e == "number")
    return e;
  if (mo(e))
    return ct;
  if (Te(e)) {
    var t = typeof e.valueOf == "function" ? e.valueOf() : e;
    e = Te(t) ? t + "" : t;
  }
  if (typeof e != "string")
    return e === 0 ? e : +e;
  e = bo(e);
  var r = wo.test(e);
  return r || yo.test(e) ? Co(e.slice(2), r ? 2 : 8) : vo.test(e) ? ct : +e;
}
function ko(e, t, r) {
  return e === e && (r !== void 0 && (e = e <= r ? e : r), t !== void 0 && (e = e >= t ? e : t)), e;
}
function Mo(e, t, r) {
  return r === void 0 && (r = t, t = void 0), r !== void 0 && (r = xe(r), r = r === r ? r : 0), t !== void 0 && (t = xe(t), t = t === t ? t : 0), ko(xe(e), t, r);
}
var Oe = function() {
  return Dt.Date.now();
}, No = "Expected a function", So = Math.max, zo = Math.min;
function To(e, t, r) {
  var i, o, a, l, s, c, u = 0, f = !1, b = !1, d = !0;
  if (typeof e != "function")
    throw new TypeError(No);
  t = xe(t) || 0, Te(r) && (f = !!r.leading, b = "maxWait" in r, a = b ? So(xe(r.maxWait) || 0, t) : a, d = "trailing" in r ? !!r.trailing : d);
  function w(v) {
    var U = i, z = o;
    return i = o = void 0, u = v, l = e.apply(z, U), l;
  }
  function m(v) {
    return u = v, s = setTimeout(y, t), f ? w(v) : l;
  }
  function M(v) {
    var U = v - c, z = v - u, j = t - U;
    return b ? zo(j, a - z) : j;
  }
  function g(v) {
    var U = v - c, z = v - u;
    return c === void 0 || U >= t || U < 0 || b && z >= a;
  }
  function y() {
    var v = Oe();
    if (g(v))
      return x(v);
    s = setTimeout(y, M(v));
  }
  function x(v) {
    return s = void 0, d && i ? w(v) : (i = o = void 0, l);
  }
  function L() {
    s !== void 0 && clearTimeout(s), u = 0, i = c = o = s = void 0;
  }
  function p() {
    return s === void 0 ? l : x(Oe());
  }
  function I() {
    var v = Oe(), U = g(v);
    if (i = arguments, o = this, c = v, U) {
      if (s === void 0)
        return m(c);
      if (b)
        return clearTimeout(s), s = setTimeout(y, t), w(c);
    }
    return s === void 0 && (s = setTimeout(y, t)), l;
  }
  return I.cancel = L, I.flush = p, I;
}
var Io = "[object Number]";
function dt(e) {
  return typeof e == "number" || jt(e) && Ot(e) == Io;
}
var Lo = "Expected a function";
function ut(e, t, r) {
  var i = !0, o = !0;
  if (typeof e != "function")
    throw new TypeError(Lo);
  return Te(r) && (i = "leading" in r ? !!r.leading : i, o = "trailing" in r ? !!r.trailing : o), To(e, t, {
    leading: i,
    maxWait: t,
    trailing: o
  });
}
const ke = {
  TOP_LEFT: "tl",
  TOP_RIGHT: "tr",
  BOTTOM_LEFT: "bl",
  BOTTOM_RIGHT: "br"
};
function Ao(e) {
  var j, A, $;
  const [t, r] = R({
    width: tt,
    height: tt
  }), [i, o] = R({
    width: 0,
    height: 0
  }), [a] = R([
    ke.TOP_LEFT,
    ke.TOP_RIGHT,
    ke.BOTTOM_LEFT,
    ke.BOTTOM_RIGHT
  ]), [l, s] = R(!1), [c, u] = R({
    x: 0,
    y: 0,
    w: 0,
    h: 0,
    dir: ""
  }), { align: f, inline: b } = (j = e == null ? void 0 : e.node) == null ? void 0 : j.attrs, d = b === !0 || b === "true", w = d && (f === "left" || f === "right"), m = F(() => {
    var Je;
    const { src: h, alt: D, width: E, height: k, flipX: T, flipY: S } = (Je = e == null ? void 0 : e.node) == null ? void 0 : Je.attrs, _ = dt(E) ? `${E}px` : E, P = dt(k) ? `${k}px` : k, W = [];
    T && W.push("rotateX(180deg)"), S && W.push("rotateY(180deg)");
    const Y = W.join(" ");
    return {
      src: h || void 0,
      alt: D || void 0,
      style: {
        width: _ || void 0,
        height: P || void 0,
        transform: Y || "none",
        ...w ? { float: f } : {}
      }
    };
  }, [(A = e == null ? void 0 : e.node) == null ? void 0 : A.attrs, w, f]), M = F(() => {
    const {
      style: { width: h }
    } = m;
    return { width: h === "100%" ? h : void 0 };
  }, [m]);
  function g(h) {
    o({
      width: h.target.width,
      height: h.target.height
    });
  }
  function y() {
    const { editor: h, getPos: D } = e;
    h.commands.setNodeSelection(D());
  }
  const x = ue(
    ut(() => {
      const { editor: h } = e, { width: D } = getComputedStyle(h.view.dom);
      r((E) => ({
        ...E,
        width: Number.parseInt(D, 10)
      }));
    }, rt),
    [e == null ? void 0 : e.editor]
  );
  function L(h, D) {
    h.preventDefault(), h.stopPropagation();
    const E = i.width, k = i.height, T = E / k;
    let S = Number(e.node.attrs.width), _ = Number(e.node.attrs.height);
    const P = t.width;
    S && !_ ? (S = S > P ? P : S, _ = Math.round(S / T)) : _ && !S ? (S = Math.round(_ * T), S = S > P ? P : S) : !S && !_ ? (S = E > P ? P : E, _ = Math.round(S / T)) : S = S > P ? P : S, s(!0), u({
      x: h.clientX,
      y: h.clientY,
      w: S,
      h: _,
      dir: D
    });
  }
  const p = ue(
    ut((h) => {
      if (h.preventDefault(), h.stopPropagation(), !l)
        return;
      const { x: D, w: E, dir: k } = c, T = (h.clientX - D) * (/l/.test(k) ? -1 : 1), S = Mo(E + T, wi, t.width);
      e.updateAttributes({
        width: S,
        height: null
      });
    }, rt),
    [l, c, t, e.updateAttributes]
  ), I = ue(
    (h) => {
      h.preventDefault(), h.stopPropagation(), l && (u({
        x: 0,
        y: 0,
        w: 0,
        h: 0,
        dir: ""
      }), s(!1), y());
    },
    [l, y]
  ), v = ue(() => {
    document == null || document.addEventListener("mousemove", p, !0), document == null || document.addEventListener("mouseup", I, !0);
  }, [p, I]), U = ue(() => {
    document == null || document.removeEventListener("mousemove", p, !0), document == null || document.removeEventListener("mouseup", I, !0);
  }, [p, I]);
  Se(() => (l ? v() : U(), () => {
    U();
  }), [l, v, U]);
  const z = F(() => new ResizeObserver(() => x()), [x]);
  return Se(() => (z.observe(e.editor.view.dom), () => {
    z.disconnect();
  }), [e.editor.view.dom, z]), /* @__PURE__ */ n(
    Zt,
    {
      as: "span",
      className: "image-view",
      style: {
        float: w ? f : void 0,
        margin: w ? f === "left" ? "1em 1em 1em 0" : "1em 0 1em 1em" : void 0,
        display: d ? "inline" : "block",
        textAlign: w ? void 0 : f,
        width: (($ = m.style) == null ? void 0 : $.width) ?? "auto",
        ...w ? {} : M
      },
      children: /* @__PURE__ */ C(
        "span",
        {
          "data-drag-handle": !0,
          draggable: "true",
          style: M,
          className: `image-view__body ${e != null && e.selected ? "image-view__body--focused" : ""} ${l ? "image-view__body--resizing" : ""}`,
          children: [
            /* @__PURE__ */ n(
              "img",
              {
                alt: m.alt,
                className: "image-view__body__image block",
                height: "auto",
                onClick: y,
                onLoad: g,
                src: m.src,
                style: m.style
              }
            ),
            (e == null ? void 0 : e.editor.view.editable) && ((e == null ? void 0 : e.selected) || l) && /* @__PURE__ */ n("span", { className: "image-resizer", children: a == null ? void 0 : a.map((h) => /* @__PURE__ */ n(
              "span",
              {
                className: `image-resizer__handler image-resizer__handler--${h}`,
                onMouseDown: (D) => L(D, h)
              },
              `image-dir-${h}`
            )) })
          ]
        }
      )
    }
  );
}
function Eo(e) {
  const { editor: t } = Kt(), r = O.useMemo(
    () => t,
    [e, t]
  );
  return Jt({
    editor: r,
    selector(o) {
      return o.editor ? {
        editor: o.editor,
        editorState: o.editor.state,
        canCommand: o.editor.can
      } : {
        editor: null,
        editorState: void 0,
        canCommand: void 0
      };
    }
  }) || { editor: null };
}
function re() {
  return Eo().editor;
}
const Pt = Xt(!1);
function _t() {
  return Yt(Pt);
}
function sa() {
  return qt(Pt);
}
const Ut = () => !1;
function la(e = Ut) {
  const t = _t(), [r, i] = R(() => {
    const s = e();
    return typeof s == "boolean" ? !s : s;
  }), o = re();
  Se(() => {
    if (!o || !e) return;
    const s = () => {
      const c = e();
      i(typeof c == "boolean" ? !c : c);
    };
    return s(), o.on("selectionUpdate", s), o.on("transaction", s), () => {
      o.off("selectionUpdate", s), o.off("transaction", s);
    };
  }, [o, e]);
  const a = F(() => !t || !o ? !0 : typeof r == "boolean" ? r : !1, [t, o, r]), l = F(() => !t || !o, [t, o]);
  return {
    disabled: a,
    // can not action, opacity < 1
    dataState: r,
    // true => show background, false => no background
    editorDisabled: l
  };
}
function Ae(e = Ut) {
  const t = _t(), [r, i] = R({}), [o, a] = R(e()), l = re();
  Se(() => {
    if (!l || !e) return;
    const u = () => {
      a(e());
    };
    return u(), l.on("selectionUpdate", u), () => {
      l.off("selectionUpdate", u);
    };
  }, [r, l, e]);
  const s = F(() => !t || !l, [t, l]), c = F(() => !t || !l, [t, l]);
  return {
    disabled: s,
    // can not action, opacity < 1
    dataState: o,
    // true => show background, false => no background
    editorDisabled: c,
    update: () => i({})
    // force update
  };
}
function de(e) {
  const t = re();
  return F(() => t ? t.extensionManager.extensions.find(
    (i) => i.name === e
  ) : null, [t, e]);
}
function ca(e, t, r) {
  return e < t ? t : e > r ? r : e;
}
const Ro = (e) => typeof e == "number", Vo = (e) => typeof e == "string", Do = (e) => typeof e == "function";
function Ho(e, t = "px") {
  if (!e) return e;
  const r = Ro(e) ? String(e) : e, i = Number.parseFloat(r), o = r.match(/[%a-z]+$/i), a = o ? o[0] : t;
  return Number.isNaN(i) ? e : i + a;
}
function da(e, t) {
  if (!e)
    return !1;
  const { extensions: r = [] } = (e == null ? void 0 : e.extensionManager) ?? {};
  return !!r.find((o) => o.name === t);
}
function ua(e) {
  return e.map((t) => Vo(t) ? { value: t, name: t } : t);
}
function Bt(e) {
  const t = re(), r = de(e), { t: i } = ae();
  return F(() => {
    if (!t || !r || !i)
      return null;
    const { button: o } = r.options;
    return !o || !Do(o) ? null : o({
      editor: t,
      extension: r,
      t: i
    });
  }, [t, r, i]);
}
function Oo(e, t) {
  const r = e.type, i = e.name.toLowerCase(), o = i.split(".").pop();
  if (!o) return !1;
  const l = r || {
    heif: "image/heif",
    heic: "image/heic",
    dng: "image/x-adobe-dng",
    cr2: "image/x-canon-cr2",
    nef: "image/x-nikon-nef",
    arw: "image/x-sony-arw",
    raf: "image/x-fuji-raf",
    orf: "image/x-olympus-orf"
  }[o];
  return t.some((s) => {
    if (s.startsWith("."))
      return i.endsWith(s);
    if (s.endsWith("/*")) {
      const c = s.split("/")[0];
      return l == null ? void 0 : l.startsWith(c + "/");
    }
    return l === s;
  });
}
function Ze(e, t) {
  const { acceptMimes: r, maxSize: i, t: o, toast: a } = t, l = [];
  return (Array.isArray(e) ? e : Object.values(e)).forEach((c) => {
    if (!Oo(c, r)) {
      t.onError ? t.onError({
        type: "type",
        message: o("editor.upload.fileTypeNotSupported", {
          fileName: c.name
        }),
        file: c
      }) : a({
        variant: "default",
        title: o("editor.upload.fileTypeNotSupported", {
          fileName: c.name
        })
      });
      return;
    }
    if (c.size > i) {
      const u = (i / 1024 / 1024).toFixed(2);
      t.onError ? t.onError({
        type: "size",
        message: o("editor.upload.fileSizeTooBig", {
          fileName: c.name,
          size: u
        }),
        file: c
      }) : a({
        variant: "default",
        title: o("editor.upload.fileSizeTooBig", {
          fileName: c.name,
          size: u
        })
      });
      return;
    }
    l.push(c);
  }), l;
}
function ha() {
  const { t: e } = ae(), { toast: t } = Ge(), r = re(), i = Bt(Ie.name), { icon: o, tooltip: a } = (i == null ? void 0 : i.componentProps) ?? {}, { editorDisabled: l } = Ae(), [s, c] = R(!1), [u, f] = R(!1), [b, d] = R(!1), w = de(Ie.name), [m, M] = R(""), [g, y] = R(""), x = ve(null), L = (w == null ? void 0 : w.options.defaultInline) || !1, [p, I] = R(L), v = F(() => (w == null ? void 0 : w.options) || Ke, [w]);
  async function U(A) {
    var D;
    const $ = (D = A == null ? void 0 : A.target) == null ? void 0 : D.files;
    if (!r || r.isDestroyed || $.length === 0 || b) {
      A.target.value = "";
      return;
    }
    const h = Ze($, {
      acceptMimes: v == null ? void 0 : v.acceptMimes,
      maxSize: v == null ? void 0 : v.maxSize,
      t: e,
      toast: t,
      onError: v.onError
    });
    if (h.length <= 0) {
      A.target.value = "";
      return;
    }
    d(!0);
    try {
      if (v != null && v.multiple) {
        const E = h.map(async (T) => {
          let S = "";
          return v.upload ? S = await v.upload(T) : S = URL.createObjectURL(T), S;
        });
        (await Promise.all(E)).forEach((T) => {
          r.chain().focus().setImageInline({ src: T, inline: p, alt: g }).run();
        });
      } else {
        const E = h[0];
        let k = "";
        v.upload ? k = await v.upload(E) : k = URL.createObjectURL(E), r.chain().focus().setImageInline({ src: k, inline: p, alt: g }).run();
      }
      c(!1), y(""), I(L);
    } catch (E) {
      console.error("Error uploading image", E), v.onError ? v.onError({
        type: "upload",
        message: e("editor.upload.error")
      }) : t({
        variant: "destructive",
        title: e("editor.upload.error")
      });
    } finally {
      d(!1), A.target.value = "";
    }
  }
  function z(A) {
    A.preventDefault(), A.stopPropagation(), r.chain().focus().setImageInline({ src: m, inline: p, alt: g }).run(), c(!1), I(L), M(""), y("");
  }
  function j(A) {
    var $;
    A.preventDefault(), ($ = x.current) == null || $.click();
  }
  return i ? /* @__PURE__ */ C(
    Ce,
    {
      onOpenChange: (A) => {
        c(A), A || f(!1);
      },
      open: s,
      children: [
        /* @__PURE__ */ n(qe, { asChild: !0, children: /* @__PURE__ */ n(
          Nt,
          {
            disabled: l,
            icon: o,
            tooltip: a,
            action: () => {
              l || c(!0);
            }
          }
        ) }),
        /* @__PURE__ */ C(le, { className: u ? "richtext-hidden" : void 0, children: [
          /* @__PURE__ */ n(ce, { children: e("editor.image.dialog.title") }),
          /* @__PURE__ */ C(
            Le,
            {
              activationMode: "manual",
              defaultValue: v.resourceImage === "both" || v.resourceImage === "upload" ? "upload" : "link",
              children: [
                v.resourceImage === "both" && /* @__PURE__ */ C(ye, { className: "richtext-grid richtext-w-full richtext-grid-cols-2", children: [
                  /* @__PURE__ */ n(J, { value: "upload", children: e("editor.image.dialog.tab.upload") }),
                  /* @__PURE__ */ n(J, { value: "link", children: e("editor.image.dialog.tab.url") })
                ] }),
                /* @__PURE__ */ C("div", { className: "richtext-my-[10px] richtext-flex richtext-items-center richtext-gap-[4px]", children: [
                  /* @__PURE__ */ n(
                    Xe,
                    {
                      checked: p,
                      onCheckedChange: (A) => {
                        I(A);
                      }
                    }
                  ),
                  /* @__PURE__ */ n(be, { children: e("editor.link.dialog.inline") })
                ] }),
                v.enableAlt && /* @__PURE__ */ C("div", { className: "richtext-my-[10px]", children: [
                  /* @__PURE__ */ n(be, { className: "mb-[6px]", children: e("editor.imageUpload.alt") }),
                  /* @__PURE__ */ n(ie, { onChange: (A) => y(A.target.value), required: !0, type: "text", value: g })
                ] }),
                /* @__PURE__ */ C(Q, { value: "upload", children: [
                  /* @__PURE__ */ C("div", { className: "richtext-flex richtext-items-center richtext-gap-[10px]", children: [
                    /* @__PURE__ */ n(
                      G,
                      {
                        className: "richtext-mt-1 richtext-w-full",
                        disabled: b,
                        onClick: j,
                        size: "sm",
                        children: b ? /* @__PURE__ */ C(K, { children: [
                          e("editor.imageUpload.uploading"),
                          /* @__PURE__ */ n(ge, { className: "richtext-ml-1 richtext-animate-spin", name: "Loader" })
                        ] }) : e("editor.image.dialog.tab.upload")
                      }
                    ),
                    /* @__PURE__ */ n(
                      Ft,
                      {
                        alt: g,
                        disabled: b,
                        editor: r,
                        imageInline: p,
                        onClose: () => {
                          c(!1), f(!1), y(""), I(L);
                        },
                        onOpenChange: f
                      }
                    )
                  ] }),
                  /* @__PURE__ */ n(
                    "input",
                    {
                      accept: v.acceptMimes.join(",") || "image/*",
                      multiple: v.multiple,
                      onChange: U,
                      ref: x,
                      style: { display: "none" },
                      type: "file"
                    }
                  )
                ] }),
                /* @__PURE__ */ n(Q, { value: "link", children: /* @__PURE__ */ n("form", { onSubmit: z, children: /* @__PURE__ */ C("div", { className: "richtext-flex richtext-items-center richtext-gap-2", children: [
                  /* @__PURE__ */ n(
                    ie,
                    {
                      autoFocus: !0,
                      onChange: (A) => M(A.target.value),
                      placeholder: e("editor.image.dialog.placeholder"),
                      required: !0,
                      type: "url",
                      value: m
                    }
                  ),
                  /* @__PURE__ */ n(G, { type: "submit", children: e("editor.image.dialog.button.apply") })
                ] }) }) })
              ]
            }
          )
        ] })
      ]
    }
  ) : /* @__PURE__ */ n(K, {});
}
const Ke = {
  acceptMimes: ["image/jpeg", "image/gif", "image/png", "image/jpg"],
  maxSize: 1024 * 1024 * 5,
  // 5MB
  multiple: !0,
  resourceImage: "both",
  defaultInline: !1,
  enableAlt: !0
};
function $t(e) {
  return e == null ? null : e === !0 || e === "" || e === "true";
}
function je(e) {
  return $t(e) ?? !1;
}
const Ie = /* @__PURE__ */ to.extend({
  group: "inline",
  inline: !0,
  defining: !0,
  draggable: !0,
  selectable: !0,
  addOptions() {
    var e;
    return {
      ...Ke,
      ...(e = this.parent) == null ? void 0 : e.call(this),
      upload: () => Promise.reject("Image Upload Function"),
      button: ({
        editor: t,
        extension: r,
        t: i
      }) => {
        var o, a;
        return {
          componentProps: {
            action: () => !0,
            upload: r.options.upload,
            /* If setImage is not available(when Image Component is not imported), the button is disabled */
            disabled: !((a = (o = t.can()).setImage) != null && a.call(o, {})),
            icon: "ImageUp",
            tooltip: i("editor.image.tooltip")
          }
        };
      }
    };
  },
  addAttributes() {
    var e;
    return {
      ...(e = this.parent) == null ? void 0 : e.call(this),
      flipX: {
        default: !1
      },
      flipY: {
        default: !1
      },
      width: {
        default: null,
        parseHTML: (t) => {
          const r = t.style.width || t.getAttribute("width") || null;
          return r ? Number.parseInt(r, 10) : null;
        },
        renderHTML: (t) => ({
          width: t.width
        })
      },
      align: {
        default: "center",
        parseHTML: (t) => t.getAttribute("align"),
        renderHTML: (t) => ({
          align: t.align
        })
      },
      inline: {
        default: !1,
        parseHTML: (t) => $t(t.getAttribute("inline")),
        renderHTML: (t) => ({
          inline: t.inline ? "true" : null
        })
      },
      alt: {
        default: "",
        parseHTML: (t) => t.getAttribute("alt"),
        renderHTML: (t) => ({
          alt: t.alt
        })
      }
    };
  },
  addNodeView() {
    return Qt(Ao);
  },
  addCommands() {
    var e;
    return {
      ...(e = this.parent) == null ? void 0 : e.call(this),
      setImageInline: (t) => ({ commands: r }) => r.insertContent({
        type: this.name,
        attrs: {
          ...t,
          inline: t.inline ?? this.options.defaultInline
        }
      }),
      updateImage: (t) => ({ commands: r }) => r.updateAttributes(this.name, t),
      setAlignImage: (t) => ({ commands: r }) => r.updateAttributes(this.name, { align: t })
    };
  },
  renderHTML({ HTMLAttributes: e }) {
    const { flipX: t, flipY: r, align: i, inline: o } = e, a = je(o), l = a && (i === "left" || i === "right"), s = t || r ? `transform: rotateX(${t ? "180" : "0"}deg) rotateY(${r ? "180" : "0"}deg);` : "", c = !a && i === "center" ? "margin: 0 auto;" : !a && i === "right" ? "margin-left: auto;" : a ? "" : "margin-right: auto;", u = a ? "" : "display: block;", f = l ? `float: ${i};` : "", d = `${u}${c}${f}${l ? i === "left" ? "margin: 1em 1em 1em 0;" : "margin: 1em 0 1em 1em;" : ""}${s}`, w = {
      ...e,
      inline: a ? "true" : null
    };
    return [
      "span",
      {
        class: "image"
      },
      [
        "img",
        Pe(
          {
            height: "auto",
            style: d || null
          },
          this.options.HTMLAttributes,
          w
        )
      ]
    ];
  },
  parseHTML() {
    return [
      {
        tag: "span.image",
        getAttrs: (e) => {
          const t = e.querySelector("img"), r = t == null ? void 0 : t.getAttribute("width"), i = (t == null ? void 0 : t.getAttribute("flipx")) || !1, o = (t == null ? void 0 : t.getAttribute("flipy")) || !1;
          return {
            src: t == null ? void 0 : t.getAttribute("src"),
            alt: t == null ? void 0 : t.getAttribute("alt"),
            caption: t == null ? void 0 : t.getAttribute("caption"),
            width: r ? Number.parseInt(r, 10) : null,
            align: (t == null ? void 0 : t.getAttribute("align")) || e.style.textAlign || null,
            inline: je(t == null ? void 0 : t.getAttribute("inline")),
            flipX: i === "true",
            flipY: o === "true"
          };
        }
      },
      {
        tag: "div[class=image]",
        getAttrs: (e) => {
          const t = e.querySelector("img"), r = t == null ? void 0 : t.getAttribute("width"), i = (t == null ? void 0 : t.getAttribute("flipx")) || !1, o = (t == null ? void 0 : t.getAttribute("flipy")) || !1;
          return {
            src: t == null ? void 0 : t.getAttribute("src"),
            alt: t == null ? void 0 : t.getAttribute("alt"),
            caption: t == null ? void 0 : t.getAttribute("caption"),
            width: r ? Number.parseInt(r, 10) : null,
            align: (t == null ? void 0 : t.getAttribute("align")) || e.style.textAlign || null,
            inline: je(t == null ? void 0 : t.getAttribute("inline")),
            flipX: i === "true",
            flipY: o === "true"
          };
        }
      },
      {
        tag: 'img[src]:not([src^="data:"])'
      }
    ];
  }
  // addProseMirrorPlugins() {
  //   const validateFile = (file: File): boolean => {
  //     // @ts-expect-error
  //     if (!this.options.acceptMimes.includes(file.type)) {
  //       // toast({ description: t.value('editor.imageUpload.fileTypeNotSupported'), duration: 2000 });
  //       return false;
  //     }
  //     // @ts-expect-error
  //     if (file.size > this.options.maxSize) {
  //       // toast({
  //       //   description: `${t.value('editor.imageUpload.fileSizeTooBig')} ${formatFileSize(
  //       //     this.options.maxSize,
  //       //   )}.`,
  //       //   duration: 2000,
  //       // });
  //       return false;
  //     }
  //     return true;
  //   };
  //   const uploadFn = createImageUpload({
  //     validateFn: validateFile,
  //     onUpload: this.options.upload as any,
  //     // postUpload: this.options.postUpload,
  //     defaultInline: this.options.defaultInline,
  //   });
  //   return [
  //     UploadImagesPlugin(),
  //     new Plugin({
  //       key: new PluginKey(`richtextCustomPlugin${this.name}`),
  //       props: {
  //         handlePaste: (view, event) => {
  //           const hasFiles =
  //               event.clipboardData &&
  //               event.clipboardData.files &&
  //               event.clipboardData.files?.length;
  //           if (!hasFiles) {
  //             return;
  //           }
  //           const items = [...(event.clipboardData.files || [])];
  //           if (items.some(x => x.type === 'text/html')) {
  //             return false;
  //           }
  //           return handleImagePaste(view, event, uploadFn);
  //         },
  //         handleDrop: (view, event, _, moved) => {
  //           if (!(event instanceof DragEvent) || !event.dataTransfer) {
  //             return false;
  //           }
  //           handleImageDrop(view, event, moved, uploadFn);
  //           return false;
  //         },
  //       },
  //     }),
  //   ];
  // },
});
function fa(e) {
  return e.replace(/^.*\/|\..+$/g, "");
}
function ma(e) {
  return e.split(".").pop();
}
function ga(e) {
  return e < 1024 ? `${e} Byte` : e < 1024 * 1024 ? `${(e / 1024).toFixed(2)} KB` : `${(e / 1024 / 1024).toFixed(2)} MB`;
}
function pa(e) {
  return e ? e === "application/pdf" ? "pdf" : e.startsWith("application/") && [".document", "word"].some((t) => e.includes(t)) ? "word" : e.startsWith("application/") && ["presentation"].some((t) => e.includes(t)) ? "excel" : e.startsWith("application/") && ["sheet"].some((t) => e.includes(t)) ? "ppt" : e.startsWith("image") ? "image" : e.startsWith("audio") ? "audio" : e.startsWith("video") ? "video" : "file" : "file";
}
function jo(e) {
  return new Promise((t) => {
    const r = new FileReader();
    r.addEventListener(
      "load",
      () => {
        t({
          alt: e.name,
          src: r.result
        });
      },
      !1
    ), r.readAsDataURL(e);
  });
}
function Po(e, t) {
  const r = e.split(","), i = r[0].match(/:(.*?);/)[1], o = atob(r[r.length - 1]);
  let a = o.length;
  const l = new Uint8Array(a);
  for (; a--; )
    l[a] = o.charCodeAt(a);
  return new File([l], t, { type: i });
}
function Ft({ editor: e, imageInline: t, onClose: r, onOpenChange: i, disabled: o, alt: a }) {
  var D, E;
  const { t: l } = ae(), { toast: s } = Ge(), [c, u] = R(!1), [f, b] = R(!1), d = q.useRef(null), [w, m] = q.useState(), [M, g] = q.useState(""), y = ve(null), [x, L] = R({
    src: "",
    file: null
  }), p = de(Ie.name), I = F(() => (p == null ? void 0 : p.options) ?? {}, [p]);
  function v(k) {
    if (d.current && k.width && k.height) {
      const T = U(d.current, k);
      g(T);
    }
  }
  function U(k, T) {
    const S = document.createElement("canvas"), _ = k.naturalWidth / k.width, P = k.naturalHeight / k.height;
    S.width = T.width * _, S.height = T.height * P;
    const W = S.getContext("2d");
    return W && (W.imageSmoothingEnabled = !1, W.drawImage(
      k,
      T.x * _,
      T.y * P,
      T.width * _,
      T.height * P,
      0,
      0,
      T.width * _,
      T.height * P
    )), S.toDataURL("image/png", 1);
  }
  const z = q.useCallback(() => {
    y.current && (y.current.value = "");
  }, []), j = q.useCallback(
    (k) => {
      u(k), i == null || i(k), k || (m(void 0), g(""), L({ src: "", file: null }), z());
    },
    [i, z]
  ), A = q.useCallback(async () => {
    var k;
    if (!f) {
      b(!0);
      try {
        const T = Po(M, ((k = x == null ? void 0 : x.file) == null ? void 0 : k.name) || "image.png");
        let S = "";
        I.upload ? S = await I.upload(T) : S = URL.createObjectURL(T), e.chain().focus().setImageInline({ src: S, inline: t, alt: a }).run(), j(!1), r();
      } catch (T) {
        console.error("Error cropping image", T);
      } finally {
        b(!1);
      }
    }
  }, [
    a,
    M,
    e,
    j,
    t,
    f,
    r,
    (D = x == null ? void 0 : x.file) == null ? void 0 : D.name,
    I
  ]);
  function $(k) {
    var T;
    k.preventDefault(), (T = y.current) == null || T.click();
  }
  const h = async (k) => {
    var W;
    const T = (W = k == null ? void 0 : k.target) == null ? void 0 : W.files;
    if (!e || e.isDestroyed || T.length === 0) {
      k.target.value = "";
      return;
    }
    const S = Ze(T, {
      acceptMimes: I == null ? void 0 : I.acceptMimes,
      maxSize: I == null ? void 0 : I.maxSize,
      t: l,
      toast: s,
      onError: I.onError
    });
    if (S.length <= 0) {
      k.target.value = "";
      return;
    }
    const _ = S[0], P = await jo(_);
    j(!0), L({
      src: P.src,
      file: _
    });
  };
  return /* @__PURE__ */ C(K, { children: [
    /* @__PURE__ */ n(
      G,
      {
        className: "richtext-mt-1 richtext-w-full",
        disabled: o,
        onClick: $,
        size: "sm",
        children: l("editor.image.dialog.tab.uploadCrop")
      }
    ),
    /* @__PURE__ */ C(Ce, { open: c, onOpenChange: j, children: [
      /* @__PURE__ */ n(qe, {}),
      /* @__PURE__ */ C(le, { children: [
        /* @__PURE__ */ n(ce, { children: l("editor.image.dialog.tab.uploadCrop") }),
        /* @__PURE__ */ n("div", { children: x.src && /* @__PURE__ */ n(
          Si,
          {
            className: "richtext-w-full",
            crop: w,
            onChange: (k) => m(k),
            onComplete: (k) => v(k),
            children: /* @__PURE__ */ n("img", { alt: "Crop me", ref: d, src: x.src })
          }
        ) }),
        /* @__PURE__ */ C(Vt, { children: [
          /* @__PURE__ */ C(
            G,
            {
              disabled: f,
              onClick: () => {
                j(!1);
              },
              children: [
                l("editor.imageUpload.cancel"),
                /* @__PURE__ */ n(ge, { className: "richtext-ml-1", name: "Trash2" })
              ]
            }
          ),
          /* @__PURE__ */ n(G, { className: "richtext-w-fit", disabled: f || !w, onClick: A, children: f ? /* @__PURE__ */ C(K, { children: [
            l("editor.imageUpload.uploading"),
            /* @__PURE__ */ n(ge, { className: "richtext-ml-1 richtext-animate-spin", name: "Loader" })
          ] }) : /* @__PURE__ */ C(K, { children: [
            l("editor.imageUpload.crop"),
            /* @__PURE__ */ n(ge, { className: "richtext-ml-1", name: "Crop" })
          ] }) })
        ] })
      ] })
    ] }),
    /* @__PURE__ */ n(
      "input",
      {
        accept: ((E = I == null ? void 0 : I.acceptMimes) == null ? void 0 : E.join(",")) || "image/*",
        multiple: !1,
        onChange: h,
        ref: y,
        style: { display: "none" },
        type: "file"
      }
    )
  ] });
}
const Wt = {
  UPLOAD_IMAGE: (e) => `UPLOAD_IMAGE-${e}`,
  UPLOAD_VIDEO: (e) => `UPLOAD_VIDEO-${e}`,
  EXCALIDRAW: (e) => `EXCALIDRAW-${e}`
};
function _o() {
  const { t: e } = ae(), { toast: t } = Ge(), r = re(), { editorDisabled: i } = Ae(), [o, a] = R(!1), [l, s] = R(!1), c = Wt.UPLOAD_IMAGE(r.id);
  Et(a, [c]);
  const [u, f] = R(!1), b = de(Ie.name), [d, w] = R(""), [m, M] = R(""), g = ve(null), y = (b == null ? void 0 : b.options.defaultInline) || !1, [x, L] = R(y), p = F(() => (b == null ? void 0 : b.options) || Ke, [b]);
  async function I(z) {
    var $;
    const j = ($ = z == null ? void 0 : z.target) == null ? void 0 : $.files;
    if (!r || r.isDestroyed || j.length === 0 || u) {
      z.target.value = "";
      return;
    }
    const A = Ze(j, {
      acceptMimes: p == null ? void 0 : p.acceptMimes,
      maxSize: p == null ? void 0 : p.maxSize,
      t: e,
      toast: t,
      onError: p.onError
    });
    if (A.length <= 0) {
      z.target.value = "";
      return;
    }
    f(!0);
    try {
      if (p != null && p.multiple) {
        const h = A.map(async (E) => {
          let k = "";
          return p.upload ? k = await p.upload(E) : k = URL.createObjectURL(E), k;
        });
        (await Promise.all(h)).forEach((E) => {
          r.chain().focus().setImageInline({ src: E, inline: x, alt: m }).run();
        });
      } else {
        const h = A[0];
        let D = "";
        p.upload ? D = await p.upload(h) : D = URL.createObjectURL(h), r.chain().focus().setImageInline({ src: D, inline: x, alt: m }).run();
      }
      a(!1), M(""), L(y);
    } catch (h) {
      console.error("Error uploading image", h), p.onError ? p.onError({
        type: "upload",
        message: e("editor.upload.error")
      }) : t({
        variant: "destructive",
        title: e("editor.upload.error")
      });
    } finally {
      f(!1), z.target.value = "";
    }
  }
  function v(z) {
    z.preventDefault(), z.stopPropagation(), r.chain().focus().setImageInline({ src: d, inline: x, alt: m }).run(), a(!1), L(y), w(""), M("");
  }
  function U(z) {
    var j;
    z.preventDefault(), (j = g.current) == null || j.click();
  }
  return i ? /* @__PURE__ */ n(K, {}) : /* @__PURE__ */ n(
    Ce,
    {
      onOpenChange: (z) => {
        a(z), z || s(!1);
      },
      open: o,
      children: /* @__PURE__ */ C(le, { className: l ? "richtext-hidden" : void 0, children: [
        /* @__PURE__ */ n(ce, { children: e("editor.image.dialog.title") }),
        /* @__PURE__ */ C(
          Le,
          {
            activationMode: "manual",
            defaultValue: p.resourceImage === "both" || p.resourceImage === "upload" ? "upload" : "link",
            children: [
              p.resourceImage === "both" && /* @__PURE__ */ C(ye, { className: "richtext-grid richtext-w-full richtext-grid-cols-2", children: [
                /* @__PURE__ */ n(J, { value: "upload", children: e("editor.image.dialog.tab.upload") }),
                /* @__PURE__ */ n(J, { value: "link", children: e("editor.image.dialog.tab.url") })
              ] }),
              /* @__PURE__ */ C("div", { className: "richtext-my-[10px] richtext-flex richtext-items-center richtext-gap-[4px]", children: [
                /* @__PURE__ */ n(
                  Xe,
                  {
                    checked: x,
                    onCheckedChange: (z) => {
                      L(z);
                    }
                  }
                ),
                /* @__PURE__ */ n(be, { children: e("editor.link.dialog.inline") })
              ] }),
              p.enableAlt && /* @__PURE__ */ C("div", { className: "richtext-my-[10px]", children: [
                /* @__PURE__ */ n(be, { className: "mb-[6px]", children: e("editor.imageUpload.alt") }),
                /* @__PURE__ */ n(ie, { onChange: (z) => M(z.target.value), required: !0, type: "text", value: m })
              ] }),
              /* @__PURE__ */ C(Q, { value: "upload", children: [
                /* @__PURE__ */ C("div", { className: "richtext-flex richtext-items-center richtext-gap-[10px]", children: [
                  /* @__PURE__ */ n(
                    G,
                    {
                      className: "richtext-mt-1 richtext-w-full",
                      disabled: u,
                      onClick: U,
                      size: "sm",
                      children: u ? /* @__PURE__ */ C(K, { children: [
                        e("editor.imageUpload.uploading"),
                        /* @__PURE__ */ n(ge, { className: "richtext-ml-1 richtext-animate-spin", name: "Loader" })
                      ] }) : e("editor.image.dialog.tab.upload")
                    }
                  ),
                  /* @__PURE__ */ n(
                    Ft,
                    {
                      alt: m,
                      disabled: u,
                      editor: r,
                      imageInline: x,
                      onClose: () => {
                        a(!1), s(!1), M(""), L(y);
                      },
                      onOpenChange: s
                    }
                  )
                ] }),
                /* @__PURE__ */ n(
                  "input",
                  {
                    accept: p.acceptMimes.join(",") || "image/*",
                    multiple: p.multiple,
                    onChange: I,
                    ref: g,
                    style: { display: "none" },
                    type: "file"
                  }
                )
              ] }),
              /* @__PURE__ */ n(Q, { value: "link", children: /* @__PURE__ */ n("form", { onSubmit: v, children: /* @__PURE__ */ C("div", { className: "richtext-flex richtext-items-center richtext-gap-2", children: [
                /* @__PURE__ */ n(
                  ie,
                  {
                    autoFocus: !0,
                    onChange: (z) => w(z.target.value),
                    placeholder: e("editor.image.dialog.placeholder"),
                    required: !0,
                    type: "url",
                    value: d
                  }
                ),
                /* @__PURE__ */ n(G, { type: "submit", children: e("editor.image.dialog.button.apply") })
              ] }) }) })
            ]
          }
        )
      ] })
    }
  );
}
function Gt(e, t) {
  let r;
  try {
    r = new URL(e);
  } catch {
    return !1;
  }
  return !(t != null && t.length) || t.length === 1 && t[0] === "." ? !0 : t.some((i) => {
    if (i.includes("*")) {
      const o = i.replace(/\./g, String.raw`\.`).replace(/\*/g, ".*");
      return new RegExp(`^${o}$`).test(r.hostname);
    }
    return r.hostname.includes(i);
  });
}
function xa() {
  const { t: e } = ae(), t = re(), r = Bt(Be.name), { icon: i, tooltip: o } = (r == null ? void 0 : r.componentProps) ?? {}, { editorDisabled: a } = Ae(), [l, s] = R(""), c = ve(null), [u, f] = R(""), [b, d] = R(!1), w = de(Be.name), m = F(() => (w == null ? void 0 : w.options) ?? {}, [w]);
  async function M(x) {
    var v;
    const L = (v = x == null ? void 0 : x.target) == null ? void 0 : v.files;
    if (!t || t.isDestroyed || L.length === 0)
      return;
    const p = L[0];
    let I = "";
    m.upload ? I = await m.upload(p) : I = URL.createObjectURL(p), t.chain().focus().setVideo({
      src: I,
      width: "100%"
    }).run(), d(!1);
  }
  function g(x) {
    x.preventDefault(), x.stopPropagation(), l && (t.chain().focus().setVideo({
      src: l,
      width: "100%"
    }).run(), d(!1), s(""));
  }
  function y(x) {
    var L;
    x.preventDefault(), (L = c.current) == null || L.click();
  }
  return /* @__PURE__ */ C(Ce, { onOpenChange: d, open: b, children: [
    /* @__PURE__ */ n(qe, { asChild: !0, children: /* @__PURE__ */ n(
      Nt,
      {
        disabled: a,
        icon: i,
        tooltip: o,
        action: () => {
          a || d(!0);
        }
      }
    ) }),
    /* @__PURE__ */ C(le, { children: [
      /* @__PURE__ */ n(ce, { children: e("editor.video.dialog.title") }),
      /* @__PURE__ */ C(
        Le,
        {
          activationMode: "manual",
          defaultValue: (m == null ? void 0 : m.resourceVideo) === "both" || (m == null ? void 0 : m.resourceVideo) === "upload" ? "upload" : "link",
          children: [
            /* @__PURE__ */ C(ye, { className: "richtext-grid richtext-w-full richtext-grid-cols-2", children: [
              ((m == null ? void 0 : m.resourceVideo) === "both" || (m == null ? void 0 : m.resourceVideo) === "upload") && /* @__PURE__ */ n(J, { value: "upload", children: e("editor.video.dialog.tab.upload") }),
              ((m == null ? void 0 : m.resourceVideo) === "both" || (m == null ? void 0 : m.resourceVideo) === "link") && /* @__PURE__ */ n(J, { value: "link", children: e("editor.video.dialog.link") })
            ] }),
            /* @__PURE__ */ C(Q, { value: "upload", children: [
              /* @__PURE__ */ n("div", { className: "richtext-flex richtext-items-center richtext-gap-[10px]", children: /* @__PURE__ */ n(G, { className: "richtext-mt-1 richtext-w-full", onClick: y, size: "sm", children: e("editor.video.dialog.tab.upload") }) }),
              /* @__PURE__ */ n(
                "input",
                {
                  accept: "video/*",
                  multiple: !0,
                  onChange: M,
                  ref: c,
                  type: "file",
                  style: {
                    display: "none"
                  }
                }
              )
            ] }),
            /* @__PURE__ */ C(Q, { value: "link", children: [
              /* @__PURE__ */ n("div", { children: /* @__PURE__ */ C("div", { className: "richtext-flex richtext-items-center richtext-gap-2", children: [
                /* @__PURE__ */ n(
                  ie,
                  {
                    autoFocus: !0,
                    placeholder: e("editor.video.dialog.placeholder"),
                    required: !0,
                    type: "url",
                    value: l,
                    onBlur: (x) => {
                      const L = x.target.value, p = m.videoProviders || ["."];
                      L && !Gt(L, p) ? f("Invalid video URL") : f("");
                    },
                    onChange: (x) => {
                      s(x.target.value);
                    }
                  }
                ),
                /* @__PURE__ */ n(G, { onClick: g, type: "button", children: e("editor.video.dialog.button.apply") })
              ] }) }),
              u && /* @__PURE__ */ n("div", { className: "richtext-my-[5px] richtext-text-red-500", children: u })
            ] })
          ]
        }
      )
    ] })
  ] });
}
function Uo(e) {
  e = e.replace("https://youtu.be/", "https://www.youtube.com/watch?v=").replace("watch?v=", "embed/");
  const t = e.match(/^https:\/\/www\.youtube\.com\/shorts\/([a-zA-Z0-9_-]+)/);
  t && (e = `https://www.youtube.com/embed/${t[1]}`);
  const r = e.match(/^https:\/\/vimeo\.com\/(\d+)(?:\/([a-zA-Z0-9]+))?/);
  if (r) {
    const o = r[1], a = r[2];
    a ? e = `https://player.vimeo.com/video/${o}?h=${a}` : e = `https://player.vimeo.com/video/${o}`;
  }
  return /^https?:\/\/www.bilibili.com\/video\/.*/i.test(e) && (e = e.replace(/\?.*$/, "").replace("https://www.bilibili.com/video/", "https://player.bilibili.com/player.html?bvid=")), e.includes("drive.google.com") && (e = e.replace("/view", "/preview")), e;
}
const Be = /* @__PURE__ */ xt.create({
  name: "video",
  group: "block",
  atom: !0,
  draggable: !0,
  //@ts-expect-error
  addOptions() {
    return {
      divider: !1,
      spacer: !1,
      allowFullscreen: !0,
      upload: void 0,
      frameborder: !1,
      resourceVideo: "both",
      width: yi["size-medium"],
      HTMLAttributes: {
        class: "iframe-wrapper"
        // style: 'display: flex;justify-content: center;',
      },
      button: ({ editor: e, t }) => {
        var r, i;
        return {
          componentProps: {
            action: () => {
            },
            isActive: () => e.isActive("video") || !1,
            /* If setVideo is not available(when Video Component is not imported), the button is disabled */
            disabled: !((i = (r = e.can()).setVideo) != null && i.call(r, {})),
            icon: "Video",
            tooltip: t("editor.video.tooltip"),
            videoProviders: ["."],
            editor: e
          }
        };
      }
    };
  },
  addAttributes() {
    return {
      src: {
        default: null,
        renderHTML: ({ src: e }) => ({
          src: e ? Uo(e) : null
        })
      },
      width: {
        default: this.options.width,
        renderHTML: ({ width: e }) => ({
          width: Ho(e)
        })
      },
      frameborder: {
        default: this.options.frameborder ? 1 : 0,
        parseHTML: () => this.options.frameborder ? 1 : 0
      },
      allowfullscreen: {
        default: this.options.allowFullscreen,
        parseHTML: () => this.options.allowFullscreen
      },
      align: {
        default: "center",
        // Default alignment
        renderHTML: ({ align: e }) => ({
          align: e
        })
      }
    };
  },
  parseHTML() {
    return [
      {
        tag: "div[data-video] iframe"
      }
    ];
  },
  renderHTML({ HTMLAttributes: e }) {
    const { width: t = "100%", align: r = "center" } = e ?? {}, i = {
      ...e,
      width: "100%",
      height: "100%"
    }, o = `position: relative;overflow: hidden;display: flex;flex: 1;max-width: ${t};`, a = `flex: 1;padding-bottom: ${9 / 16 * 100}%;`, f = ["div", { style: `display: flex; justify-content: ${r};` }, ["div", { style: o }, ["div", { style: a }], ["iframe", i]]];
    return ["div", {
      ...this.options.HTMLAttributes,
      class: "iframe-wrapper",
      "data-video": ""
    }, f];
  },
  addCommands() {
    return {
      setVideo: (e) => ({ commands: t }) => t.insertContent({
        type: this.name,
        attrs: e
      }),
      updateVideo: (e) => ({ commands: t }) => t.updateAttributes(this.name, e)
    };
  }
});
function Bo() {
  const { t: e } = ae(), t = re(), { editorDisabled: r } = Ae(), [i, o] = R(""), a = ve(null), [l, s] = R(""), [c, u] = R(!1), f = de(Be.name), b = Wt.UPLOAD_VIDEO(t.id);
  Et(u, [b]);
  const d = F(() => (f == null ? void 0 : f.options) ?? {}, [f]);
  async function w(g) {
    var p;
    const y = (p = g == null ? void 0 : g.target) == null ? void 0 : p.files;
    if (!t || t.isDestroyed || y.length === 0)
      return;
    const x = y[0];
    let L = "";
    d.upload ? L = await d.upload(x) : L = URL.createObjectURL(x), t.chain().focus().setVideo({
      src: L,
      width: "100%"
    }).run(), u(!1);
  }
  function m(g) {
    g.preventDefault(), g.stopPropagation(), i && (t.chain().focus().setVideo({
      src: i,
      width: "100%"
    }).run(), u(!1), o(""));
  }
  function M(g) {
    var y;
    g.preventDefault(), (y = a.current) == null || y.click();
  }
  return r ? /* @__PURE__ */ n(K, {}) : /* @__PURE__ */ n(Ce, { onOpenChange: u, open: c, children: /* @__PURE__ */ C(le, { children: [
    /* @__PURE__ */ n(ce, { children: e("editor.video.dialog.title") }),
    /* @__PURE__ */ C(
      Le,
      {
        activationMode: "manual",
        defaultValue: (d == null ? void 0 : d.resourceVideo) === "both" || (d == null ? void 0 : d.resourceVideo) === "upload" ? "upload" : "link",
        children: [
          /* @__PURE__ */ C(ye, { className: "richtext-grid richtext-w-full richtext-grid-cols-2", children: [
            ((d == null ? void 0 : d.resourceVideo) === "both" || (d == null ? void 0 : d.resourceVideo) === "upload") && /* @__PURE__ */ n(J, { value: "upload", children: e("editor.video.dialog.tab.upload") }),
            ((d == null ? void 0 : d.resourceVideo) === "both" || (d == null ? void 0 : d.resourceVideo) === "link") && /* @__PURE__ */ n(J, { value: "link", children: e("editor.video.dialog.link") })
          ] }),
          /* @__PURE__ */ C(Q, { value: "upload", children: [
            /* @__PURE__ */ n("div", { className: "richtext-flex richtext-items-center richtext-gap-[10px]", children: /* @__PURE__ */ n(G, { className: "richtext-mt-1 richtext-w-full", onClick: M, size: "sm", children: e("editor.video.dialog.tab.upload") }) }),
            /* @__PURE__ */ n(
              "input",
              {
                accept: "video/*",
                multiple: !0,
                onChange: w,
                ref: a,
                type: "file",
                style: {
                  display: "none"
                }
              }
            )
          ] }),
          /* @__PURE__ */ C(Q, { value: "link", children: [
            /* @__PURE__ */ n("div", { children: /* @__PURE__ */ C("div", { className: "richtext-flex richtext-items-center richtext-gap-2", children: [
              /* @__PURE__ */ n(
                ie,
                {
                  autoFocus: !0,
                  placeholder: e("editor.video.dialog.placeholder"),
                  required: !0,
                  type: "url",
                  value: i,
                  onBlur: (g) => {
                    const y = g.target.value, x = d.videoProviders || ["."];
                    y && !Gt(y, x) ? s("Invalid video URL") : s("");
                  },
                  onChange: (g) => {
                    o(g.target.value);
                  }
                }
              ),
              /* @__PURE__ */ n(G, { onClick: m, type: "button", children: e("editor.video.dialog.button.apply") })
            ] }) }),
            l && /* @__PURE__ */ n("div", { className: "richtext-my-[5px] richtext-text-red-500", children: l })
          ] })
        ]
      }
    )
  ] }) });
}
function $o() {
  return /* @__PURE__ */ C(K, { children: [
    /* @__PURE__ */ n(_o, {}),
    /* @__PURE__ */ n(Bo, {})
  ] });
}
process.env.NODE_ENV !== "production" && ($o.displayName = "SlashDialogTrigger");
export {
  ma as $,
  Nt as A,
  G as B,
  Xe as C,
  Ce as D,
  ut as E,
  ca as F,
  Vn as G,
  Po as H,
  ge as I,
  to as J,
  Wt as K,
  be as L,
  oa as M,
  Et as N,
  dt as O,
  Mo as P,
  To as Q,
  mn as R,
  ia as S,
  Hn as T,
  sa as U,
  aa as V,
  ra as W,
  $o as X,
  pa as Y,
  Mn as Z,
  kn as _,
  On as a,
  fa as a0,
  ga as a1,
  Kn as a2,
  Be as a3,
  Ke as a4,
  ha as a5,
  xa as a6,
  Tt as b,
  V as c,
  Ae as d,
  la as e,
  ua as f,
  sn as g,
  re as h,
  St as i,
  _t as j,
  ie as k,
  zt as l,
  an as m,
  qe as n,
  le as o,
  ce as p,
  jn as q,
  na as r,
  Pn as s,
  Un as t,
  Bt as u,
  Vt as v,
  de as w,
  Ge as x,
  da as y,
  Ie as z
};
