// reactjs-tiptap-editor@1.0.2/lib/index.js
import { jsx as s, Fragment as f, jsxs as j } from "react/jsx-runtime";
import { E as N } from "./clsx-_bq4MQQ7.js";
import { useEffect as l, useId as M } from "react";
import { U as P, V as A, W as $, X as O } from "./index-C52cuQ_Y.js";
import { useTheme as D, THEME as m } from "./theme.js";
const H = `
.reactjs-tiptap-editor {
  button,
  input:where([type=button]),
  input:where([type=reset]),
  input:where([type=submit]) {
    -webkit-appearance: button;
    background-color: transparent;
    background-image: none;
  }

  input,
  optgroup,
  select {
    font-family: inherit;
    font-feature-settings: inherit;
    font-variation-settings: inherit;
    font-size: 100%;
    font-weight: inherit;
    line-height: inherit;
    letter-spacing: inherit;
    color: inherit;
  }

  button {
    font-family: inherit;
    font-feature-settings: inherit;
    font-variation-settings: inherit;
    font-size: 100%;
    font-weight: inherit;
    line-height: inherit;
    letter-spacing: inherit;
    color: inherit;
  }

  *,
  ::before,
  ::after {
    box-sizing: border-box;
    border-width: 0;
    border-style: solid;
  }


  hr {
    height: 0;
    color: inherit;
    border-top-width: 1px;
  }

  a {
    color: inherit;
    text-decoration: inherit;
  }

  b,
  strong {
    font-weight: bolder;
  }

  code,
  kbd,
  samp,
  pre {
    font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, Liberation Mono, Courier New, monospace;
    font-feature-settings: normal;
    font-variation-settings: normal;
    font-size: 1em;
  }

  table {
    text-indent: 0;
    border-color: inherit;
    border-collapse: collapse;
  }

  input {
    border-width: 1px;
  }

  input::placeholder {
    opacity: 1;
  }

  button, input {
    cursor: pointer;
    color: inherit;
  }

  img,
  svg,
  video,
  canvas,
  audio,
  iframe,
  embed,
  object {
    display: block;
    vertical-align: middle;
  }

  img,
  video {
    max-width: 100%;
    height: auto;
  }
}
`;
function L({ editor: e }) {
  const n = P();
  l(() => {
    n(e == null ? void 0 : e.isEditable);
  }, [e == null ? void 0 : e.isEditable]);
  const t = () => {
    n(e == null ? void 0 : e.isEditable);
  };
  return l(() => (e && e.on("update", t), () => {
    e && e.off("update", t);
  }), [e]), /* @__PURE__ */ s(f, {});
}
const g = "data-rc-order", E = "data-rc-priority", z = "rc-util-key", d = /* @__PURE__ */ new Map();
function Q(e, n) {
  if (!e)
    return !1;
  if (e.contains)
    return e.contains(n);
  let t = n;
  for (; t; ) {
    if (t === e)
      return !0;
    t = t.parentNode;
  }
  return !1;
}
function v({ mark: e } = {}) {
  return e ? e.startsWith("data-") ? e : `data-${e}` : z;
}
function h(e) {
  return e.attachTo ? e.attachTo : document.querySelector("head") || document.body;
}
function _(e) {
  return e === "queue" ? "prependQueue" : e ? "prepend" : "append";
}
function x(e) {
  return [...(d.get(e) || e).children].filter(
    (n) => n.tagName === "STYLE"
  );
}
function y(e, n = {}) {
  const { csp: t, prepend: o, priority: r = 0 } = n, a = _(o), c = a === "prependQueue", i = document.createElement("style");
  i.setAttribute(g, a), c && r && i.setAttribute(E, `${r}`), t != null && t.nonce && (i.nonce = t == null ? void 0 : t.nonce), i.innerHTML = e;
  const u = h(n), { firstChild: R } = u;
  if (o) {
    if (c) {
      const p = x(u).filter((b) => {
        if (!["prepend", "prependQueue"].includes(b.getAttribute(g)))
          return !1;
        const T = Number(b.getAttribute(E) || 0);
        return r >= T;
      });
      if (p.length > 0)
        return u.insertBefore(i, p.at(-1).nextSibling), i;
    }
    R.before(i);
  } else
    u.append(i);
  return i;
}
function S(e, n = {}) {
  const t = h(n);
  return x(t).find((o) => o.getAttribute(v(n)) === e);
}
function w(e, n = {}) {
  const t = S(e, n);
  t && t.remove();
}
function I(e, n) {
  const t = d.get(e);
  if (!t || !Q(document, t)) {
    const o = y("", n), { parentNode: r } = o;
    d.set(e, r), o.remove();
  }
}
function C(e, n, t = {}) {
  var c, i, u;
  const o = h(t);
  I(o, t);
  const r = S(n, t);
  if (r)
    return (c = t.csp) != null && c.nonce && r.nonce !== ((i = t.csp) == null ? void 0 : i.nonce) && (r.nonce = (u = t.csp) == null ? void 0 : u.nonce), r.innerHTML !== e && (r.innerHTML = e), r;
  const a = y(e, t);
  return a.setAttribute(v(t), n), a;
}
function V() {
  const { theme: e, color: n, borderRadius: t } = D();
  return l(() => {
    const o = e || "light", r = n || "default";
    let a = m[o][r];
    if (!a) {
      a = m.light.default;
      return;
    }
    return C(
      `
      .reactjs-tiptap-editor, .reactjs-tiptap-editor *,
      .reactjs-tiptap-editor-theme, .reactjs-tiptap-editor-theme *,
      div[data-richtext-portal], div[data-richtext-portal] * {
        ${Object.entries(a).map(([c, i]) => typeof t == "string" && c === "radius" ? `--${c}: ${t};` : `--${c}: ${i};`).join(`
`)}
      }
      `,
      "richtext-theme",
      {
        priority: 50
      }
    ), () => {
      w("richtext-theme");
    };
  }, [e, n, t]), /* @__PURE__ */ s(f, {});
}
function W({ editor: e, children: n }) {
  const t = M();
  return l(() => (C(H, "react-tiptap-reset"), () => {
    w("react-tiptap-reset");
  }), []), l(() => {
    e && (e.id = t);
  }, [t, e]), e ? /* @__PURE__ */ s("div", { className: "reactjs-tiptap-editor", children: /* @__PURE__ */ s(A, { children: /* @__PURE__ */ j(N.Provider, { value: { editor: e }, children: [
    /* @__PURE__ */ s($, { delayDuration: 0, disableHoverableContent: !0, children: n }),
    /* @__PURE__ */ s(L, { editor: e }),
    /* @__PURE__ */ s(O, {}),
    /* @__PURE__ */ s(V, {})
  ] }) }) }) : /* @__PURE__ */ s(f, {});
}
export {
  W as RichTextProvider
};
