// reactjs-tiptap-editor@1.0.2/lib/Italic.js
import { M as g, h as o, j as i, m as y } from "./index-Dx-Dh7KS.js";
import { jsx as d, Fragment as h } from "react/jsx-runtime";
import { u as f, d as I, A as v } from "./index-C52cuQ_Y.js";
import "react";
import "./theme.js";
var M = /(?:^|\s)(\*(?!\s+\*)((?:[^*]+))\*(?!\s+\*))$/, k = /(?:^|\s)(\*(?!\s+\*)((?:[^*]+))\*(?!\s+\*))/g, A = /(?:^|\s)(_(?!\s+_)((?:[^_]+))_(?!\s+_))$/, _ = /(?:^|\s)(_(?!\s+_)((?:[^_]+))_(?!\s+_))/g, R = g.create({
  name: "italic",
  addOptions() {
    return {
      HTMLAttributes: {}
    };
  },
  parseHTML() {
    return [
      {
        tag: "em"
      },
      {
        tag: "i",
        getAttrs: (t) => t.style.fontStyle !== "normal" && null
      },
      {
        style: "font-style=normal",
        clearMark: (t) => t.type.name === this.name
      },
      {
        style: "font-style=italic"
      }
    ];
  },
  renderHTML({ HTMLAttributes: t }) {
    return ["em", y(this.options.HTMLAttributes, t), 0];
  },
  addCommands() {
    return {
      setItalic: () => ({ commands: t }) => t.setMark(this.name),
      toggleItalic: () => ({ commands: t }) => t.toggleMark(this.name),
      unsetItalic: () => ({ commands: t }) => t.unsetMark(this.name)
    };
  },
  markdownTokenName: "em",
  parseMarkdown: (t, e) => e.applyMark("italic", e.parseInline(t.tokens || [])),
  markdownOptions: {
    htmlReopen: {
      open: "<em>",
      close: "</em>"
    }
  },
  renderMarkdown: (t, e) => `*${e.renderChildren(t)}*`,
  addKeyboardShortcuts() {
    return {
      "Mod-i": () => this.editor.commands.toggleItalic(),
      "Mod-I": () => this.editor.commands.toggleItalic()
    };
  },
  addInputRules() {
    return [
      i({
        find: M,
        type: this.type
      }),
      i({
        find: A,
        type: this.type
      })
    ];
  },
  addPasteRules() {
    return [
      o({
        find: k,
        type: this.type
      }),
      o({
        find: _,
        type: this.type
      })
    ];
  }
}), x = R;
function L() {
  const t = f(T.name), {
    icon: e = void 0,
    tooltip: r = void 0,
    shortcutKeys: s = void 0,
    tooltipOptions: l = {},
    action: n = void 0,
    isActive: c = void 0
  } = (t == null ? void 0 : t.componentProps) ?? {}, { dataState: u, disabled: a, update: m } = I(c), p = () => {
    a || n && (n(), m());
  };
  return t ? /* @__PURE__ */ d(
    v,
    {
      action: p,
      dataState: u,
      disabled: a,
      icon: e,
      shortcutKeys: s,
      tooltip: r,
      tooltipOptions: l
    }
  ) : /* @__PURE__ */ d(h, {});
}
const T = /* @__PURE__ */ x.extend({
  //@ts-expect-error
  addOptions() {
    var t;
    return {
      ...(t = this.parent) == null ? void 0 : t.call(this),
      button({ editor: e, t: r, extension: s }) {
        return {
          componentProps: {
            action: () => e.commands.toggleItalic(),
            isActive: () => e.isActive("italic") || !1,
            disabled: !1,
            shortcutKeys: s.options.shortcutKeys ?? ["mod", "I"],
            icon: "Italic",
            tooltip: r("editor.italic.tooltip")
          }
        };
      }
    };
  }
});
export {
  T as Italic,
  L as RichTextItalic
};
