// reactjs-tiptap-editor@1.0.2/lib/Katex-BPbXjcZQ.js
import { N as Io, n as Po, m as Ro } from "./index-Dx-Dh7KS.js";
import { N as $o, a as Eo } from "./clsx-_bq4MQQ7.js";
import { jsx as er, jsxs as _r } from "react/jsx-runtime";
import Xa from "katex";
import { useState as Ur, useRef as xo, useEffect as Bo, useMemo as Hr, useCallback as To } from "react";
import { u as Mo, d as jo, h as Co, D as Do, n as Fo, A as No, o as _o, p as Wo, L as un, G as fn, v as ko, B as Lo } from "./index-C52cuQ_Y.js";
import "./theme.js";
import { a as Uo, c as Gr, g as Ho } from "./_commonjsHelpers-B4e78b8K.js";
import { u as Go } from "./index-DtyAGMbq.js";
var ae, sn;
function Qa() {
  if (sn) return ae;
  sn = 1;
  var r = Object.prototype.toString;
  return ae = function(n) {
    var o = r.call(n), t = o === "[object Arguments]";
    return t || (t = o !== "[object Array]" && n !== null && typeof n == "object" && typeof n.length == "number" && n.length >= 0 && r.call(n.callee) === "[object Function]"), t;
  }, ae;
}
var oe, ln;
function zo() {
  if (ln) return oe;
  ln = 1;
  var r;
  if (!Object.keys) {
    var e = Object.prototype.hasOwnProperty, n = Object.prototype.toString, o = Qa(), t = Object.prototype.propertyIsEnumerable, f = !t.call({ toString: null }, "toString"), i = t.call(function() {
    }, "prototype"), a = [
      "toString",
      "toLocaleString",
      "valueOf",
      "hasOwnProperty",
      "isPrototypeOf",
      "propertyIsEnumerable",
      "constructor"
    ], u = function(y) {
      var h = y.constructor;
      return h && h.prototype === y;
    }, c = {
      $applicationCache: !0,
      $console: !0,
      $external: !0,
      $frame: !0,
      $frameElement: !0,
      $frames: !0,
      $innerHeight: !0,
      $innerWidth: !0,
      $onmozfullscreenchange: !0,
      $onmozfullscreenerror: !0,
      $outerHeight: !0,
      $outerWidth: !0,
      $pageXOffset: !0,
      $pageYOffset: !0,
      $parent: !0,
      $scrollLeft: !0,
      $scrollTop: !0,
      $scrollX: !0,
      $scrollY: !0,
      $self: !0,
      $webkitIndexedDB: !0,
      $webkitStorageInfo: !0,
      $window: !0
    }, v = (function() {
      if (typeof window > "u")
        return !1;
      for (var y in window)
        try {
          if (!c["$" + y] && e.call(window, y) && window[y] !== null && typeof window[y] == "object")
            try {
              u(window[y]);
            } catch {
              return !0;
            }
        } catch {
          return !0;
        }
      return !1;
    })(), p = function(y) {
      if (typeof window > "u" || !v)
        return u(y);
      try {
        return u(y);
      } catch {
        return !1;
      }
    };
    r = function(h) {
      var S = h !== null && typeof h == "object", R = n.call(h) === "[object Function]", w = o(h), q = S && n.call(h) === "[object String]", A = [];
      if (!S && !R && !w)
        throw new TypeError("Object.keys called on a non-object");
      var m = i && R;
      if (q && h.length > 0 && !e.call(h, 0))
        for (var x = 0; x < h.length; ++x)
          A.push(String(x));
      if (w && h.length > 0)
        for (var X = 0; X < h.length; ++X)
          A.push(String(X));
      else
        for (var H in h)
          !(m && H === "prototype") && e.call(h, H) && A.push(String(H));
      if (f)
        for (var F = p(h), D = 0; D < a.length; ++D)
          !(F && a[D] === "constructor") && e.call(h, a[D]) && A.push(a[D]);
      return A;
    };
  }
  return oe = r, oe;
}
var ie, cn;
function Wt() {
  if (cn) return ie;
  cn = 1;
  var r = Array.prototype.slice, e = Qa(), n = Object.keys, o = n ? function(i) {
    return n(i);
  } : zo(), t = Object.keys;
  return o.shim = function() {
    if (Object.keys) {
      var i = (function() {
        var a = Object.keys(arguments);
        return a && a.length === arguments.length;
      })(1, 2);
      i || (Object.keys = function(u) {
        return e(u) ? t(r.call(u)) : t(u);
      });
    } else
      Object.keys = o;
    return Object.keys || o;
  }, ie = o, ie;
}
var ue, pn;
function zr() {
  if (pn) return ue;
  pn = 1;
  var r = Object.defineProperty || !1;
  if (r)
    try {
      r({}, "a", { value: 1 });
    } catch {
      r = !1;
    }
  return ue = r, ue;
}
var fe, yn;
function kt() {
  return yn || (yn = 1, fe = SyntaxError), fe;
}
var se, vn;
function nr() {
  return vn || (vn = 1, se = TypeError), se;
}
var le, gn;
function Vo() {
  return gn || (gn = 1, le = Object.getOwnPropertyDescriptor), le;
}
var ce, hn;
function Or() {
  if (hn) return ce;
  hn = 1;
  var r = /* @__PURE__ */ Vo();
  if (r)
    try {
      r([], "length");
    } catch {
      r = null;
    }
  return ce = r, ce;
}
var pe, dn;
function Lt() {
  if (dn) return pe;
  dn = 1;
  var r = /* @__PURE__ */ zr(), e = /* @__PURE__ */ kt(), n = /* @__PURE__ */ nr(), o = /* @__PURE__ */ Or();
  return pe = function(f, i, a) {
    if (!f || typeof f != "object" && typeof f != "function")
      throw new n("`obj` must be an object or a function`");
    if (typeof i != "string" && typeof i != "symbol")
      throw new n("`property` must be a string or a symbol`");
    if (arguments.length > 3 && typeof arguments[3] != "boolean" && arguments[3] !== null)
      throw new n("`nonEnumerable`, if provided, must be a boolean or null");
    if (arguments.length > 4 && typeof arguments[4] != "boolean" && arguments[4] !== null)
      throw new n("`nonWritable`, if provided, must be a boolean or null");
    if (arguments.length > 5 && typeof arguments[5] != "boolean" && arguments[5] !== null)
      throw new n("`nonConfigurable`, if provided, must be a boolean or null");
    if (arguments.length > 6 && typeof arguments[6] != "boolean")
      throw new n("`loose`, if provided, must be a boolean");
    var u = arguments.length > 3 ? arguments[3] : null, c = arguments.length > 4 ? arguments[4] : null, v = arguments.length > 5 ? arguments[5] : null, p = arguments.length > 6 ? arguments[6] : !1, y = !!o && o(f, i);
    if (r)
      r(f, i, {
        configurable: v === null && y ? y.configurable : !v,
        enumerable: u === null && y ? y.enumerable : !u,
        value: a,
        writable: c === null && y ? y.writable : !c
      });
    else if (p || !u && !c && !v)
      f[i] = a;
    else
      throw new e("This environment does not support defining a property as non-configurable, non-writable, or non-enumerable.");
  }, pe;
}
var ye, mn;
function Ut() {
  if (mn) return ye;
  mn = 1;
  var r = /* @__PURE__ */ zr(), e = function() {
    return !!r;
  };
  return e.hasArrayLengthDefineBug = function() {
    if (!r)
      return null;
    try {
      return r([], "length", { value: 1 }).length !== 1;
    } catch {
      return !0;
    }
  }, ye = e, ye;
}
var ve, Sn;
function wr() {
  if (Sn) return ve;
  Sn = 1;
  var r = Wt(), e = typeof Symbol == "function" && typeof Symbol("foo") == "symbol", n = Object.prototype.toString, o = Array.prototype.concat, t = /* @__PURE__ */ Lt(), f = function(c) {
    return typeof c == "function" && n.call(c) === "[object Function]";
  }, i = /* @__PURE__ */ Ut()(), a = function(c, v, p, y) {
    if (v in c) {
      if (y === !0) {
        if (c[v] === p)
          return;
      } else if (!f(y) || !y())
        return;
    }
    i ? t(c, v, p, !0) : t(c, v, p);
  }, u = function(c, v) {
    var p = arguments.length > 2 ? arguments[2] : {}, y = r(v);
    e && (y = o.call(y, Object.getOwnPropertySymbols(v)));
    for (var h = 0; h < y.length; h += 1)
      a(c, y[h], v[y[h]], p[y[h]]);
  };
  return u.supportsDescriptors = !!i, ve = u, ve;
}
var ge = { exports: {} }, he, bn;
function Ht() {
  return bn || (bn = 1, he = Object), he;
}
var de, An;
function Za() {
  return An || (An = 1, de = Error), de;
}
var me, On;
function Ko() {
  return On || (On = 1, me = EvalError), me;
}
var Se, wn;
function Jo() {
  return wn || (wn = 1, Se = RangeError), Se;
}
var be, qn;
function Yo() {
  return qn || (qn = 1, be = ReferenceError), be;
}
var Ae, In;
function Xo() {
  return In || (In = 1, Ae = URIError), Ae;
}
var Oe, Pn;
function Qo() {
  return Pn || (Pn = 1, Oe = Math.abs), Oe;
}
var we, Rn;
function Zo() {
  return Rn || (Rn = 1, we = Math.floor), we;
}
var qe, $n;
function ri() {
  return $n || ($n = 1, qe = Math.max), qe;
}
var Ie, En;
function ei() {
  return En || (En = 1, Ie = Math.min), Ie;
}
var Pe, xn;
function ti() {
  return xn || (xn = 1, Pe = Math.pow), Pe;
}
var Re, Bn;
function ni() {
  return Bn || (Bn = 1, Re = Math.round), Re;
}
var $e, Tn;
function ai() {
  return Tn || (Tn = 1, $e = Number.isNaN || function(e) {
    return e !== e;
  }), $e;
}
var Ee, Mn;
function oi() {
  if (Mn) return Ee;
  Mn = 1;
  var r = /* @__PURE__ */ ai();
  return Ee = function(n) {
    return r(n) || n === 0 ? n : n < 0 ? -1 : 1;
  }, Ee;
}
var xe, jn;
function Vr() {
  return jn || (jn = 1, xe = function() {
    if (typeof Symbol != "function" || typeof Object.getOwnPropertySymbols != "function")
      return !1;
    if (typeof Symbol.iterator == "symbol")
      return !0;
    var e = {}, n = Symbol("test"), o = Object(n);
    if (typeof n == "string" || Object.prototype.toString.call(n) !== "[object Symbol]" || Object.prototype.toString.call(o) !== "[object Symbol]")
      return !1;
    var t = 42;
    e[n] = t;
    for (var f in e)
      return !1;
    if (typeof Object.keys == "function" && Object.keys(e).length !== 0 || typeof Object.getOwnPropertyNames == "function" && Object.getOwnPropertyNames(e).length !== 0)
      return !1;
    var i = Object.getOwnPropertySymbols(e);
    if (i.length !== 1 || i[0] !== n || !Object.prototype.propertyIsEnumerable.call(e, n))
      return !1;
    if (typeof Object.getOwnPropertyDescriptor == "function") {
      var a = (
        /** @type {PropertyDescriptor} */
        Object.getOwnPropertyDescriptor(e, n)
      );
      if (a.value !== t || a.enumerable !== !0)
        return !1;
    }
    return !0;
  }), xe;
}
var Be, Cn;
function Gt() {
  if (Cn) return Be;
  Cn = 1;
  var r = typeof Symbol < "u" && Symbol, e = Vr();
  return Be = function() {
    return typeof r != "function" || typeof Symbol != "function" || typeof r("foo") != "symbol" || typeof Symbol("bar") != "symbol" ? !1 : e();
  }, Be;
}
var Te, Dn;
function ro() {
  return Dn || (Dn = 1, Te = typeof Reflect < "u" && Reflect.getPrototypeOf || null), Te;
}
var Me, Fn;
function eo() {
  if (Fn) return Me;
  Fn = 1;
  var r = /* @__PURE__ */ Ht();
  return Me = r.getPrototypeOf || null, Me;
}
var je, Nn;
function ii() {
  if (Nn) return je;
  Nn = 1;
  var r = "Function.prototype.bind called on incompatible ", e = Object.prototype.toString, n = Math.max, o = "[object Function]", t = function(u, c) {
    for (var v = [], p = 0; p < u.length; p += 1)
      v[p] = u[p];
    for (var y = 0; y < c.length; y += 1)
      v[y + u.length] = c[y];
    return v;
  }, f = function(u, c) {
    for (var v = [], p = c, y = 0; p < u.length; p += 1, y += 1)
      v[y] = u[p];
    return v;
  }, i = function(a, u) {
    for (var c = "", v = 0; v < a.length; v += 1)
      c += a[v], v + 1 < a.length && (c += u);
    return c;
  };
  return je = function(u) {
    var c = this;
    if (typeof c != "function" || e.apply(c) !== o)
      throw new TypeError(r + c);
    for (var v = f(arguments, 1), p, y = function() {
      if (this instanceof p) {
        var q = c.apply(
          this,
          t(v, arguments)
        );
        return Object(q) === q ? q : this;
      }
      return c.apply(
        u,
        t(v, arguments)
      );
    }, h = n(0, c.length - v.length), S = [], R = 0; R < h; R++)
      S[R] = "$" + R;
    if (p = Function("binder", "return function (" + i(S, ",") + "){ return binder.apply(this,arguments); }")(y), c.prototype) {
      var w = function() {
      };
      w.prototype = c.prototype, p.prototype = new w(), w.prototype = null;
    }
    return p;
  }, je;
}
var Ce, _n;
function jr() {
  if (_n) return Ce;
  _n = 1;
  var r = ii();
  return Ce = Function.prototype.bind || r, Ce;
}
var De, Wn;
function zt() {
  return Wn || (Wn = 1, De = Function.prototype.call), De;
}
var Fe, kn;
function Vt() {
  return kn || (kn = 1, Fe = Function.prototype.apply), Fe;
}
var Ne, Ln;
function ui() {
  return Ln || (Ln = 1, Ne = typeof Reflect < "u" && Reflect && Reflect.apply), Ne;
}
var _e, Un;
function to() {
  if (Un) return _e;
  Un = 1;
  var r = jr(), e = Vt(), n = zt(), o = ui();
  return _e = o || r.call(n, e), _e;
}
var We, Hn;
function Kt() {
  if (Hn) return We;
  Hn = 1;
  var r = jr(), e = /* @__PURE__ */ nr(), n = zt(), o = to();
  return We = function(f) {
    if (f.length < 1 || typeof f[0] != "function")
      throw new e("a function is required");
    return o(r, n, f);
  }, We;
}
var ke, Gn;
function fi() {
  if (Gn) return ke;
  Gn = 1;
  var r = Kt(), e = /* @__PURE__ */ Or(), n;
  try {
    n = /** @type {{ __proto__?: typeof Array.prototype }} */
    [].__proto__ === Array.prototype;
  } catch (i) {
    if (!i || typeof i != "object" || !("code" in i) || i.code !== "ERR_PROTO_ACCESS")
      throw i;
  }
  var o = !!n && e && e(
    Object.prototype,
    /** @type {keyof typeof Object.prototype} */
    "__proto__"
  ), t = Object, f = t.getPrototypeOf;
  return ke = o && typeof o.get == "function" ? r([o.get]) : typeof f == "function" ? (
    /** @type {import('./get')} */
    function(a) {
      return f(a == null ? a : t(a));
    }
  ) : !1, ke;
}
var Le, zn;
function Jt() {
  if (zn) return Le;
  zn = 1;
  var r = ro(), e = eo(), n = /* @__PURE__ */ fi();
  return Le = r ? function(t) {
    return r(t);
  } : e ? function(t) {
    if (!t || typeof t != "object" && typeof t != "function")
      throw new TypeError("getProto: not an object");
    return e(t);
  } : n ? function(t) {
    return n(t);
  } : null, Le;
}
var Ue, Vn;
function Yt() {
  if (Vn) return Ue;
  Vn = 1;
  var r = Function.prototype.call, e = Object.prototype.hasOwnProperty, n = jr();
  return Ue = n.call(r, e), Ue;
}
var He, Kn;
function pr() {
  if (Kn) return He;
  Kn = 1;
  var r, e = /* @__PURE__ */ Ht(), n = /* @__PURE__ */ Za(), o = /* @__PURE__ */ Ko(), t = /* @__PURE__ */ Jo(), f = /* @__PURE__ */ Yo(), i = /* @__PURE__ */ kt(), a = /* @__PURE__ */ nr(), u = /* @__PURE__ */ Xo(), c = /* @__PURE__ */ Qo(), v = /* @__PURE__ */ Zo(), p = /* @__PURE__ */ ri(), y = /* @__PURE__ */ ei(), h = /* @__PURE__ */ ti(), S = /* @__PURE__ */ ni(), R = /* @__PURE__ */ oi(), w = Function, q = function(j) {
    try {
      return w('"use strict"; return (' + j + ").constructor;")();
    } catch {
    }
  }, A = /* @__PURE__ */ Or(), m = /* @__PURE__ */ zr(), x = function() {
    throw new a();
  }, X = A ? (function() {
    try {
      return arguments.callee, x;
    } catch {
      try {
        return A(arguments, "callee").get;
      } catch {
        return x;
      }
    }
  })() : x, H = Gt()(), F = Jt(), D = eo(), I = ro(), N = Vt(), L = zt(), _ = {}, U = typeof Uint8Array > "u" || !F ? r : F(Uint8Array), V = {
    __proto__: null,
    "%AggregateError%": typeof AggregateError > "u" ? r : AggregateError,
    "%Array%": Array,
    "%ArrayBuffer%": typeof ArrayBuffer > "u" ? r : ArrayBuffer,
    "%ArrayIteratorPrototype%": H && F ? F([][Symbol.iterator]()) : r,
    "%AsyncFromSyncIteratorPrototype%": r,
    "%AsyncFunction%": _,
    "%AsyncGenerator%": _,
    "%AsyncGeneratorFunction%": _,
    "%AsyncIteratorPrototype%": _,
    "%Atomics%": typeof Atomics > "u" ? r : Atomics,
    "%BigInt%": typeof BigInt > "u" ? r : BigInt,
    "%BigInt64Array%": typeof BigInt64Array > "u" ? r : BigInt64Array,
    "%BigUint64Array%": typeof BigUint64Array > "u" ? r : BigUint64Array,
    "%Boolean%": Boolean,
    "%DataView%": typeof DataView > "u" ? r : DataView,
    "%Date%": Date,
    "%decodeURI%": decodeURI,
    "%decodeURIComponent%": decodeURIComponent,
    "%encodeURI%": encodeURI,
    "%encodeURIComponent%": encodeURIComponent,
    "%Error%": n,
    "%eval%": eval,
    // eslint-disable-line no-eval
    "%EvalError%": o,
    "%Float16Array%": typeof Float16Array > "u" ? r : Float16Array,
    "%Float32Array%": typeof Float32Array > "u" ? r : Float32Array,
    "%Float64Array%": typeof Float64Array > "u" ? r : Float64Array,
    "%FinalizationRegistry%": typeof FinalizationRegistry > "u" ? r : FinalizationRegistry,
    "%Function%": w,
    "%GeneratorFunction%": _,
    "%Int8Array%": typeof Int8Array > "u" ? r : Int8Array,
    "%Int16Array%": typeof Int16Array > "u" ? r : Int16Array,
    "%Int32Array%": typeof Int32Array > "u" ? r : Int32Array,
    "%isFinite%": isFinite,
    "%isNaN%": isNaN,
    "%IteratorPrototype%": H && F ? F(F([][Symbol.iterator]())) : r,
    "%JSON%": typeof JSON == "object" ? JSON : r,
    "%Map%": typeof Map > "u" ? r : Map,
    "%MapIteratorPrototype%": typeof Map > "u" || !H || !F ? r : F((/* @__PURE__ */ new Map())[Symbol.iterator]()),
    "%Math%": Math,
    "%Number%": Number,
    "%Object%": e,
    "%Object.getOwnPropertyDescriptor%": A,
    "%parseFloat%": parseFloat,
    "%parseInt%": parseInt,
    "%Promise%": typeof Promise > "u" ? r : Promise,
    "%Proxy%": typeof Proxy > "u" ? r : Proxy,
    "%RangeError%": t,
    "%ReferenceError%": f,
    "%Reflect%": typeof Reflect > "u" ? r : Reflect,
    "%RegExp%": RegExp,
    "%Set%": typeof Set > "u" ? r : Set,
    "%SetIteratorPrototype%": typeof Set > "u" || !H || !F ? r : F((/* @__PURE__ */ new Set())[Symbol.iterator]()),
    "%SharedArrayBuffer%": typeof SharedArrayBuffer > "u" ? r : SharedArrayBuffer,
    "%String%": String,
    "%StringIteratorPrototype%": H && F ? F(""[Symbol.iterator]()) : r,
    "%Symbol%": H ? Symbol : r,
    "%SyntaxError%": i,
    "%ThrowTypeError%": X,
    "%TypedArray%": U,
    "%TypeError%": a,
    "%Uint8Array%": typeof Uint8Array > "u" ? r : Uint8Array,
    "%Uint8ClampedArray%": typeof Uint8ClampedArray > "u" ? r : Uint8ClampedArray,
    "%Uint16Array%": typeof Uint16Array > "u" ? r : Uint16Array,
    "%Uint32Array%": typeof Uint32Array > "u" ? r : Uint32Array,
    "%URIError%": u,
    "%WeakMap%": typeof WeakMap > "u" ? r : WeakMap,
    "%WeakRef%": typeof WeakRef > "u" ? r : WeakRef,
    "%WeakSet%": typeof WeakSet > "u" ? r : WeakSet,
    "%Function.prototype.call%": L,
    "%Function.prototype.apply%": N,
    "%Object.defineProperty%": m,
    "%Object.getPrototypeOf%": D,
    "%Math.abs%": c,
    "%Math.floor%": v,
    "%Math.max%": p,
    "%Math.min%": y,
    "%Math.pow%": h,
    "%Math.round%": S,
    "%Math.sign%": R,
    "%Reflect.getPrototypeOf%": I
  };
  if (F)
    try {
      null.error;
    } catch (j) {
      var ur = F(F(j));
      V["%Error.prototype%"] = ur;
    }
  var Ir = function j(O) {
    var P;
    if (O === "%AsyncFunction%")
      P = q("async function () {}");
    else if (O === "%GeneratorFunction%")
      P = q("function* () {}");
    else if (O === "%AsyncGeneratorFunction%")
      P = q("async function* () {}");
    else if (O === "%AsyncGenerator%") {
      var b = j("%AsyncGeneratorFunction%");
      b && (P = b.prototype);
    } else if (O === "%AsyncIteratorPrototype%") {
      var B = j("%AsyncGenerator%");
      B && F && (P = F(B.prototype));
    }
    return V[O] = P, P;
  }, mr = {
    __proto__: null,
    "%ArrayBufferPrototype%": ["ArrayBuffer", "prototype"],
    "%ArrayPrototype%": ["Array", "prototype"],
    "%ArrayProto_entries%": ["Array", "prototype", "entries"],
    "%ArrayProto_forEach%": ["Array", "prototype", "forEach"],
    "%ArrayProto_keys%": ["Array", "prototype", "keys"],
    "%ArrayProto_values%": ["Array", "prototype", "values"],
    "%AsyncFunctionPrototype%": ["AsyncFunction", "prototype"],
    "%AsyncGenerator%": ["AsyncGeneratorFunction", "prototype"],
    "%AsyncGeneratorPrototype%": ["AsyncGeneratorFunction", "prototype", "prototype"],
    "%BooleanPrototype%": ["Boolean", "prototype"],
    "%DataViewPrototype%": ["DataView", "prototype"],
    "%DatePrototype%": ["Date", "prototype"],
    "%ErrorPrototype%": ["Error", "prototype"],
    "%EvalErrorPrototype%": ["EvalError", "prototype"],
    "%Float32ArrayPrototype%": ["Float32Array", "prototype"],
    "%Float64ArrayPrototype%": ["Float64Array", "prototype"],
    "%FunctionPrototype%": ["Function", "prototype"],
    "%Generator%": ["GeneratorFunction", "prototype"],
    "%GeneratorPrototype%": ["GeneratorFunction", "prototype", "prototype"],
    "%Int8ArrayPrototype%": ["Int8Array", "prototype"],
    "%Int16ArrayPrototype%": ["Int16Array", "prototype"],
    "%Int32ArrayPrototype%": ["Int32Array", "prototype"],
    "%JSONParse%": ["JSON", "parse"],
    "%JSONStringify%": ["JSON", "stringify"],
    "%MapPrototype%": ["Map", "prototype"],
    "%NumberPrototype%": ["Number", "prototype"],
    "%ObjectPrototype%": ["Object", "prototype"],
    "%ObjProto_toString%": ["Object", "prototype", "toString"],
    "%ObjProto_valueOf%": ["Object", "prototype", "valueOf"],
    "%PromisePrototype%": ["Promise", "prototype"],
    "%PromiseProto_then%": ["Promise", "prototype", "then"],
    "%Promise_all%": ["Promise", "all"],
    "%Promise_reject%": ["Promise", "reject"],
    "%Promise_resolve%": ["Promise", "resolve"],
    "%RangeErrorPrototype%": ["RangeError", "prototype"],
    "%ReferenceErrorPrototype%": ["ReferenceError", "prototype"],
    "%RegExpPrototype%": ["RegExp", "prototype"],
    "%SetPrototype%": ["Set", "prototype"],
    "%SharedArrayBufferPrototype%": ["SharedArrayBuffer", "prototype"],
    "%StringPrototype%": ["String", "prototype"],
    "%SymbolPrototype%": ["Symbol", "prototype"],
    "%SyntaxErrorPrototype%": ["SyntaxError", "prototype"],
    "%TypedArrayPrototype%": ["TypedArray", "prototype"],
    "%TypeErrorPrototype%": ["TypeError", "prototype"],
    "%Uint8ArrayPrototype%": ["Uint8Array", "prototype"],
    "%Uint8ClampedArrayPrototype%": ["Uint8ClampedArray", "prototype"],
    "%Uint16ArrayPrototype%": ["Uint16Array", "prototype"],
    "%Uint32ArrayPrototype%": ["Uint32Array", "prototype"],
    "%URIErrorPrototype%": ["URIError", "prototype"],
    "%WeakMapPrototype%": ["WeakMap", "prototype"],
    "%WeakSetPrototype%": ["WeakSet", "prototype"]
  }, ar = jr(), G = /* @__PURE__ */ Yt(), Sr = ar.call(L, Array.prototype.concat), Pr = ar.call(N, Array.prototype.splice), br = ar.call(L, String.prototype.replace), yr = ar.call(L, String.prototype.slice), g = ar.call(L, RegExp.prototype.exec), d = /[^%.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|%$))/g, E = /\\(\\)?/g, $ = function(O) {
    var P = yr(O, 0, 1), b = yr(O, -1);
    if (P === "%" && b !== "%")
      throw new i("invalid intrinsic syntax, expected closing `%`");
    if (b === "%" && P !== "%")
      throw new i("invalid intrinsic syntax, expected opening `%`");
    var B = [];
    return br(O, d, function(z, K, J, Y) {
      B[B.length] = J ? br(Y, E, "$1") : K || z;
    }), B;
  }, M = function(O, P) {
    var b = O, B;
    if (G(mr, b) && (B = mr[b], b = "%" + B[0] + "%"), G(V, b)) {
      var z = V[b];
      if (z === _ && (z = Ir(b)), typeof z > "u" && !P)
        throw new a("intrinsic " + O + " exists, but is not available. Please file an issue!");
      return {
        alias: B,
        name: b,
        value: z
      };
    }
    throw new i("intrinsic " + O + " does not exist!");
  };
  return He = function(O, P) {
    if (typeof O != "string" || O.length === 0)
      throw new a("intrinsic name must be a non-empty string");
    if (arguments.length > 1 && typeof P != "boolean")
      throw new a('"allowMissing" argument must be a boolean');
    if (g(/^%?[^%]*%?$/, O) === null)
      throw new i("`%` may not be present anywhere but at the beginning and end of the intrinsic name");
    var b = $(O), B = b.length > 0 ? b[0] : "", z = M("%" + B + "%", P), K = z.name, J = z.value, Y = !1, Z = z.alias;
    Z && (B = Z[0], Pr(b, Sr([0, 1], Z)));
    for (var fr = 1, or = !0; fr < b.length; fr += 1) {
      var rr = b[fr], sr = yr(rr, 0, 1), lr = yr(rr, -1);
      if ((sr === '"' || sr === "'" || sr === "`" || lr === '"' || lr === "'" || lr === "`") && sr !== lr)
        throw new i("property names with quotes must have matching quotes");
      if ((rr === "constructor" || !or) && (Y = !0), B += "." + rr, K = "%" + B + "%", G(V, K))
        J = V[K];
      else if (J != null) {
        if (!(rr in J)) {
          if (!P)
            throw new a("base intrinsic for " + O + " exists, but the property is not available.");
          return;
        }
        if (A && fr + 1 >= b.length) {
          var cr = A(J, rr);
          or = !!cr, or && "get" in cr && !("originalValue" in cr.get) ? J = cr.get : J = J[rr];
        } else
          or = G(J, rr), J = J[rr];
        or && !Y && (V[K] = J);
      }
    }
    return J;
  }, He;
}
var Ge, Jn;
function si() {
  if (Jn) return Ge;
  Jn = 1;
  var r = /* @__PURE__ */ pr(), e = /* @__PURE__ */ Lt(), n = /* @__PURE__ */ Ut()(), o = /* @__PURE__ */ Or(), t = /* @__PURE__ */ nr(), f = r("%Math.floor%");
  return Ge = function(a, u) {
    if (typeof a != "function")
      throw new t("`fn` is not a function");
    if (typeof u != "number" || u < 0 || u > 4294967295 || f(u) !== u)
      throw new t("`length` must be a positive 32-bit integer");
    var c = arguments.length > 2 && !!arguments[2], v = !0, p = !0;
    if ("length" in a && o) {
      var y = o(a, "length");
      y && !y.configurable && (v = !1), y && !y.writable && (p = !1);
    }
    return (v || p || !c) && (n ? e(
      /** @type {Parameters<define>[0]} */
      a,
      "length",
      u,
      !0,
      !0
    ) : e(
      /** @type {Parameters<define>[0]} */
      a,
      "length",
      u
    )), a;
  }, Ge;
}
var ze, Yn;
function li() {
  if (Yn) return ze;
  Yn = 1;
  var r = jr(), e = Vt(), n = to();
  return ze = function() {
    return n(r, e, arguments);
  }, ze;
}
var Xn;
function xr() {
  return Xn || (Xn = 1, (function(r) {
    var e = /* @__PURE__ */ si(), n = /* @__PURE__ */ zr(), o = Kt(), t = li();
    r.exports = function(i) {
      var a = o(arguments), u = i.length - (arguments.length - 1);
      return e(
        a,
        1 + (u > 0 ? u : 0),
        !0
      );
    }, n ? n(r.exports, "apply", { value: t }) : r.exports.apply = t;
  })(ge)), ge.exports;
}
var Ve, Qn;
function Q() {
  if (Qn) return Ve;
  Qn = 1;
  var r = /* @__PURE__ */ pr(), e = Kt(), n = e([r("%String.prototype.indexOf%")]);
  return Ve = function(t, f) {
    var i = (
      /** @type {(this: unknown, ...args: unknown[]) => unknown} */
      r(t, !!f)
    );
    return typeof i == "function" && n(t, ".prototype.") > -1 ? e(
      /** @type {const} */
      [i]
    ) : i;
  }, Ve;
}
var Ke, Zn;
function no() {
  if (Zn) return Ke;
  Zn = 1;
  var r = Wt(), e = Vr()(), n = /* @__PURE__ */ Q(), o = /* @__PURE__ */ Ht(), t = n("Array.prototype.push"), f = n("Object.prototype.propertyIsEnumerable"), i = e ? o.getOwnPropertySymbols : null;
  return Ke = function(u, c) {
    if (u == null)
      throw new TypeError("target must be an object");
    var v = o(u);
    if (arguments.length === 1)
      return v;
    for (var p = 1; p < arguments.length; ++p) {
      var y = o(arguments[p]), h = r(y), S = e && (o.getOwnPropertySymbols || i);
      if (S)
        for (var R = S(y), w = 0; w < R.length; ++w) {
          var q = R[w];
          f(y, q) && t(h, q);
        }
      for (var A = 0; A < h.length; ++A) {
        var m = h[A];
        if (f(y, m)) {
          var x = y[m];
          v[m] = x;
        }
      }
    }
    return v;
  }, Ke;
}
var Je, ra;
function ao() {
  if (ra) return Je;
  ra = 1;
  var r = no(), e = function() {
    if (!Object.assign)
      return !1;
    for (var o = "abcdefghijklmnopqrst", t = o.split(""), f = {}, i = 0; i < t.length; ++i)
      f[t[i]] = t[i];
    var a = Object.assign({}, f), u = "";
    for (var c in a)
      u += c;
    return o !== u;
  }, n = function() {
    if (!Object.assign || !Object.preventExtensions)
      return !1;
    var o = Object.preventExtensions({ 1: 2 });
    try {
      Object.assign(o, "xy");
    } catch {
      return o[1] === "y";
    }
    return !1;
  };
  return Je = function() {
    return !Object.assign || e() || n() ? r : Object.assign;
  }, Je;
}
var Ye, ea;
function ci() {
  if (ea) return Ye;
  ea = 1;
  var r = wr(), e = ao();
  return Ye = function() {
    var o = e();
    return r(
      Object,
      { assign: o },
      { assign: function() {
        return Object.assign !== o;
      } }
    ), o;
  }, Ye;
}
var Xe, ta;
function pi() {
  if (ta) return Xe;
  ta = 1;
  var r = wr(), e = xr(), n = no(), o = ao(), t = ci(), f = e.apply(o()), i = function(u, c) {
    return f(Object, arguments);
  };
  return r(i, {
    getPolyfill: o,
    implementation: n,
    shim: t
  }), Xe = i, Xe;
}
var Qe, na;
function oo() {
  if (na) return Qe;
  na = 1;
  var r = /* @__PURE__ */ pr(), e = xr(), n = e(r("String.prototype.indexOf"));
  return Qe = function(t, f) {
    var i = r(t, !!f);
    return typeof i == "function" && n(t, ".prototype.") > -1 ? e(i) : i;
  }, Qe;
}
var Ze, aa;
function yi() {
  if (aa) return Ze;
  aa = 1;
  var r = function() {
    return typeof (function() {
    }).name == "string";
  }, e = Object.getOwnPropertyDescriptor;
  if (e)
    try {
      e([], "length");
    } catch {
      e = null;
    }
  r.functionsHaveConfigurableNames = function() {
    if (!r() || !e)
      return !1;
    var t = e(function() {
    }, "name");
    return !!t && !!t.configurable;
  };
  var n = Function.prototype.bind;
  return r.boundFunctionsHaveNames = function() {
    return r() && typeof n == "function" && (function() {
    }).bind().name !== "";
  }, Ze = r, Ze;
}
var rt, oa;
function vi() {
  if (oa) return rt;
  oa = 1;
  var r = /* @__PURE__ */ Lt(), e = /* @__PURE__ */ Ut()(), n = yi().functionsHaveConfigurableNames(), o = /* @__PURE__ */ nr();
  return rt = function(f, i) {
    if (typeof f != "function")
      throw new o("`fn` is not a function");
    var a = arguments.length > 2 && !!arguments[2];
    return (!a || n) && (e ? r(
      /** @type {Parameters<define>[0]} */
      f,
      "name",
      i,
      !0,
      !0
    ) : r(
      /** @type {Parameters<define>[0]} */
      f,
      "name",
      i
    )), f;
  }, rt;
}
var et, ia;
function io() {
  if (ia) return et;
  ia = 1;
  var r = vi(), e = /* @__PURE__ */ nr(), n = Object;
  return et = r(function() {
    if (this == null || this !== n(this))
      throw new e("RegExp.prototype.flags getter called on non-object");
    var t = "";
    return this.hasIndices && (t += "d"), this.global && (t += "g"), this.ignoreCase && (t += "i"), this.multiline && (t += "m"), this.dotAll && (t += "s"), this.unicode && (t += "u"), this.unicodeSets && (t += "v"), this.sticky && (t += "y"), t;
  }, "get flags", !0), et;
}
var tt, ua;
function uo() {
  if (ua) return tt;
  ua = 1;
  var r = io(), e = wr().supportsDescriptors, n = Object.getOwnPropertyDescriptor;
  return tt = function() {
    if (e && /a/mig.flags === "gim") {
      var t = n(RegExp.prototype, "flags");
      if (t && typeof t.get == "function" && "dotAll" in RegExp.prototype && "hasIndices" in RegExp.prototype) {
        var f = "", i = {};
        if (Object.defineProperty(i, "hasIndices", {
          get: function() {
            f += "d";
          }
        }), Object.defineProperty(i, "sticky", {
          get: function() {
            f += "y";
          }
        }), t.get.call(i), f === "dy")
          return t.get;
      }
    }
    return r;
  }, tt;
}
var nt, fa;
function gi() {
  if (fa) return nt;
  fa = 1;
  var r = wr().supportsDescriptors, e = uo(), n = /* @__PURE__ */ Or(), o = Object.defineProperty, t = /* @__PURE__ */ Za(), f = Jt(), i = /a/;
  return nt = function() {
    if (!r || !f)
      throw new t("RegExp.prototype.flags requires a true ES5 environment that supports property descriptors");
    var u = e(), c = f(i), v = n(c, "flags");
    return (!v || v.get !== u) && o(c, "flags", {
      configurable: !0,
      enumerable: !1,
      get: u
    }), u;
  }, nt;
}
var at, sa;
function hi() {
  if (sa) return at;
  sa = 1;
  var r = wr(), e = xr(), n = io(), o = uo(), t = gi(), f = e(o());
  return r(f, {
    getPolyfill: o,
    implementation: n,
    shim: t
  }), at = f, at;
}
var Mr = { exports: {} }, ot, la;
function qr() {
  if (la) return ot;
  la = 1;
  var r = Vr();
  return ot = function() {
    return r() && !!Symbol.toStringTag;
  }, ot;
}
var it, ca;
function fo() {
  if (ca) return it;
  ca = 1;
  var r = qr()(), e = /* @__PURE__ */ Q(), n = e("Object.prototype.toString"), o = function(a) {
    return r && a && typeof a == "object" && Symbol.toStringTag in a ? !1 : n(a) === "[object Arguments]";
  }, t = function(a) {
    return o(a) ? !0 : a !== null && typeof a == "object" && "length" in a && typeof a.length == "number" && a.length >= 0 && n(a) !== "[object Array]" && "callee" in a && n(a.callee) === "[object Function]";
  }, f = (function() {
    return o(arguments);
  })();
  return o.isLegacyArguments = t, it = f ? o : t, it;
}
const di = {}, mi = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: di
}, Symbol.toStringTag, { value: "Module" })), Si = /* @__PURE__ */ Uo(mi);
var ut, pa;
function Kr() {
  if (pa) return ut;
  pa = 1;
  var r = typeof Map == "function" && Map.prototype, e = Object.getOwnPropertyDescriptor && r ? Object.getOwnPropertyDescriptor(Map.prototype, "size") : null, n = r && e && typeof e.get == "function" ? e.get : null, o = r && Map.prototype.forEach, t = typeof Set == "function" && Set.prototype, f = Object.getOwnPropertyDescriptor && t ? Object.getOwnPropertyDescriptor(Set.prototype, "size") : null, i = t && f && typeof f.get == "function" ? f.get : null, a = t && Set.prototype.forEach, u = typeof WeakMap == "function" && WeakMap.prototype, c = u ? WeakMap.prototype.has : null, v = typeof WeakSet == "function" && WeakSet.prototype, p = v ? WeakSet.prototype.has : null, y = typeof WeakRef == "function" && WeakRef.prototype, h = y ? WeakRef.prototype.deref : null, S = Boolean.prototype.valueOf, R = Object.prototype.toString, w = Function.prototype.toString, q = String.prototype.match, A = String.prototype.slice, m = String.prototype.replace, x = String.prototype.toUpperCase, X = String.prototype.toLowerCase, H = RegExp.prototype.test, F = Array.prototype.concat, D = Array.prototype.join, I = Array.prototype.slice, N = Math.floor, L = typeof BigInt == "function" ? BigInt.prototype.valueOf : null, _ = Object.getOwnPropertySymbols, U = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? Symbol.prototype.toString : null, V = typeof Symbol == "function" && typeof Symbol.iterator == "object", ur = typeof Symbol == "function" && Symbol.toStringTag && (typeof Symbol.toStringTag === V || !0) ? Symbol.toStringTag : null, Ir = Object.prototype.propertyIsEnumerable, mr = (typeof Reflect == "function" ? Reflect.getPrototypeOf : Object.getPrototypeOf) || ([].__proto__ === Array.prototype ? function(s) {
    return s.__proto__;
  } : null);
  function ar(s, l) {
    if (s === 1 / 0 || s === -1 / 0 || s !== s || s && s > -1e3 && s < 1e3 || H.call(/e/, l))
      return l;
    var C = /[0-9](?=(?:[0-9]{3})+(?![0-9]))/g;
    if (typeof s == "number") {
      var W = s < 0 ? -N(-s) : N(s);
      if (W !== s) {
        var k = String(W), T = A.call(l, k.length + 1);
        return m.call(k, C, "$&_") + "." + m.call(m.call(T, /([0-9]{3})/g, "$&_"), /_$/, "");
      }
    }
    return m.call(l, C, "$&_");
  }
  var G = Si, Sr = G.custom, Pr = z(Sr) ? Sr : null, br = {
    __proto__: null,
    double: '"',
    single: "'"
  }, yr = {
    __proto__: null,
    double: /(["\\])/g,
    single: /(['\\])/g
  };
  ut = function s(l, C, W, k) {
    var T = C || {};
    if (Y(T, "quoteStyle") && !Y(br, T.quoteStyle))
      throw new TypeError('option "quoteStyle" must be "single" or "double"');
    if (Y(T, "maxStringLength") && (typeof T.maxStringLength == "number" ? T.maxStringLength < 0 && T.maxStringLength !== 1 / 0 : T.maxStringLength !== null))
      throw new TypeError('option "maxStringLength", if provided, must be a positive integer, Infinity, or `null`');
    var hr = Y(T, "customInspect") ? T.customInspect : !0;
    if (typeof hr != "boolean" && hr !== "symbol")
      throw new TypeError("option \"customInspect\", if provided, must be `true`, `false`, or `'symbol'`");
    if (Y(T, "indent") && T.indent !== null && T.indent !== "	" && !(parseInt(T.indent, 10) === T.indent && T.indent > 0))
      throw new TypeError('option "indent" must be "\\t", an integer > 0, or `null`');
    if (Y(T, "numericSeparator") && typeof T.numericSeparator != "boolean")
      throw new TypeError('option "numericSeparator", if provided, must be `true` or `false`');
    var Ar = T.numericSeparator;
    if (typeof l > "u")
      return "undefined";
    if (l === null)
      return "null";
    if (typeof l == "boolean")
      return l ? "true" : "false";
    if (typeof l == "string")
      return Rr(l, T);
    if (typeof l == "number") {
      if (l === 0)
        return 1 / 0 / l > 0 ? "0" : "-0";
      var tr = String(l);
      return Ar ? ar(l, tr) : tr;
    }
    if (typeof l == "bigint") {
      var dr = String(l) + "n";
      return Ar ? ar(l, dr) : dr;
    }
    var Xr = typeof T.depth > "u" ? 5 : T.depth;
    if (typeof W > "u" && (W = 0), W >= Xr && Xr > 0 && typeof l == "object")
      return $(l) ? "[Array]" : "[Object]";
    var $r = Oo(T, W);
    if (typeof k > "u")
      k = [];
    else if (or(k, l) >= 0)
      return "[Circular]";
    function ir(Er, Nr, qo) {
      if (Nr && (k = I.call(k), k.push(Nr)), qo) {
        var on = {
          depth: T.depth
        };
        return Y(T, "quoteStyle") && (on.quoteStyle = T.quoteStyle), s(Er, on, W + 1, k);
      }
      return s(Er, T, W + 1, k);
    }
    if (typeof l == "function" && !j(l)) {
      var Qt = fr(l), Zt = Dr(l, ir);
      return "[Function" + (Qt ? ": " + Qt : " (anonymous)") + "]" + (Zt.length > 0 ? " { " + D.call(Zt, ", ") + " }" : "");
    }
    if (z(l)) {
      var rn = V ? m.call(String(l), /^(Symbol\(.*\))_[^)]*$/, "$1") : U.call(l);
      return typeof l == "object" && !V ? gr(rn) : rn;
    }
    if (vr(l)) {
      for (var Tr = "<" + X.call(String(l.nodeName)), Qr = l.attributes || [], Fr = 0; Fr < Qr.length; Fr++)
        Tr += " " + Qr[Fr].name + "=" + g(d(Qr[Fr].value), "double", T);
      return Tr += ">", l.childNodes && l.childNodes.length && (Tr += "..."), Tr += "</" + X.call(String(l.nodeName)) + ">", Tr;
    }
    if ($(l)) {
      if (l.length === 0)
        return "[]";
      var Zr = Dr(l, ir);
      return $r && !Ao(Zr) ? "[" + Yr(Zr, $r) + "]" : "[ " + D.call(Zr, ", ") + " ]";
    }
    if (O(l)) {
      var re = Dr(l, ir);
      return !("cause" in Error.prototype) && "cause" in l && !Ir.call(l, "cause") ? "{ [" + String(l) + "] " + D.call(F.call("[cause]: " + ir(l.cause), re), ", ") + " }" : re.length === 0 ? "[" + String(l) + "]" : "{ [" + String(l) + "] " + D.call(re, ", ") + " }";
    }
    if (typeof l == "object" && hr) {
      if (Pr && typeof l[Pr] == "function" && G)
        return G(l, { depth: Xr - W });
      if (hr !== "symbol" && typeof l.inspect == "function")
        return l.inspect();
    }
    if (rr(l)) {
      var en = [];
      return o && o.call(l, function(Er, Nr) {
        en.push(ir(Nr, l, !0) + " => " + ir(Er, l));
      }), Xt("Map", n.call(l), en, $r);
    }
    if (cr(l)) {
      var tn = [];
      return a && a.call(l, function(Er) {
        tn.push(ir(Er, l));
      }), Xt("Set", i.call(l), tn, $r);
    }
    if (sr(l))
      return Jr("WeakMap");
    if (Cr(l))
      return Jr("WeakSet");
    if (lr(l))
      return Jr("WeakRef");
    if (b(l))
      return gr(ir(Number(l)));
    if (K(l))
      return gr(ir(L.call(l)));
    if (B(l))
      return gr(S.call(l));
    if (P(l))
      return gr(ir(String(l)));
    if (typeof window < "u" && l === window)
      return "{ [object Window] }";
    if (typeof globalThis < "u" && l === globalThis || typeof Gr < "u" && l === Gr)
      return "{ [object globalThis] }";
    if (!M(l) && !j(l)) {
      var ee = Dr(l, ir), nn = mr ? mr(l) === Object.prototype : l instanceof Object || l.constructor === Object, te = l instanceof Object ? "" : "null prototype", an = !nn && ur && Object(l) === l && ur in l ? A.call(Z(l), 8, -1) : te ? "Object" : "", wo = nn || typeof l.constructor != "function" ? "" : l.constructor.name ? l.constructor.name + " " : "", ne = wo + (an || te ? "[" + D.call(F.call([], an || [], te || []), ": ") + "] " : "");
      return ee.length === 0 ? ne + "{}" : $r ? ne + "{" + Yr(ee, $r) + "}" : ne + "{ " + D.call(ee, ", ") + " }";
    }
    return String(l);
  };
  function g(s, l, C) {
    var W = C.quoteStyle || l, k = br[W];
    return k + s + k;
  }
  function d(s) {
    return m.call(String(s), /"/g, "&quot;");
  }
  function E(s) {
    return !ur || !(typeof s == "object" && (ur in s || typeof s[ur] < "u"));
  }
  function $(s) {
    return Z(s) === "[object Array]" && E(s);
  }
  function M(s) {
    return Z(s) === "[object Date]" && E(s);
  }
  function j(s) {
    return Z(s) === "[object RegExp]" && E(s);
  }
  function O(s) {
    return Z(s) === "[object Error]" && E(s);
  }
  function P(s) {
    return Z(s) === "[object String]" && E(s);
  }
  function b(s) {
    return Z(s) === "[object Number]" && E(s);
  }
  function B(s) {
    return Z(s) === "[object Boolean]" && E(s);
  }
  function z(s) {
    if (V)
      return s && typeof s == "object" && s instanceof Symbol;
    if (typeof s == "symbol")
      return !0;
    if (!s || typeof s != "object" || !U)
      return !1;
    try {
      return U.call(s), !0;
    } catch {
    }
    return !1;
  }
  function K(s) {
    if (!s || typeof s != "object" || !L)
      return !1;
    try {
      return L.call(s), !0;
    } catch {
    }
    return !1;
  }
  var J = Object.prototype.hasOwnProperty || function(s) {
    return s in this;
  };
  function Y(s, l) {
    return J.call(s, l);
  }
  function Z(s) {
    return R.call(s);
  }
  function fr(s) {
    if (s.name)
      return s.name;
    var l = q.call(w.call(s), /^function\s*([\w$]+)/);
    return l ? l[1] : null;
  }
  function or(s, l) {
    if (s.indexOf)
      return s.indexOf(l);
    for (var C = 0, W = s.length; C < W; C++)
      if (s[C] === l)
        return C;
    return -1;
  }
  function rr(s) {
    if (!n || !s || typeof s != "object")
      return !1;
    try {
      n.call(s);
      try {
        i.call(s);
      } catch {
        return !0;
      }
      return s instanceof Map;
    } catch {
    }
    return !1;
  }
  function sr(s) {
    if (!c || !s || typeof s != "object")
      return !1;
    try {
      c.call(s, c);
      try {
        p.call(s, p);
      } catch {
        return !0;
      }
      return s instanceof WeakMap;
    } catch {
    }
    return !1;
  }
  function lr(s) {
    if (!h || !s || typeof s != "object")
      return !1;
    try {
      return h.call(s), !0;
    } catch {
    }
    return !1;
  }
  function cr(s) {
    if (!i || !s || typeof s != "object")
      return !1;
    try {
      i.call(s);
      try {
        n.call(s);
      } catch {
        return !0;
      }
      return s instanceof Set;
    } catch {
    }
    return !1;
  }
  function Cr(s) {
    if (!p || !s || typeof s != "object")
      return !1;
    try {
      p.call(s, p);
      try {
        c.call(s, c);
      } catch {
        return !0;
      }
      return s instanceof WeakSet;
    } catch {
    }
    return !1;
  }
  function vr(s) {
    return !s || typeof s != "object" ? !1 : typeof HTMLElement < "u" && s instanceof HTMLElement ? !0 : typeof s.nodeName == "string" && typeof s.getAttribute == "function";
  }
  function Rr(s, l) {
    if (s.length > l.maxStringLength) {
      var C = s.length - l.maxStringLength, W = "... " + C + " more character" + (C > 1 ? "s" : "");
      return Rr(A.call(s, 0, l.maxStringLength), l) + W;
    }
    var k = yr[l.quoteStyle || "single"];
    k.lastIndex = 0;
    var T = m.call(m.call(s, k, "\\$1"), /[\x00-\x1f]/g, Br);
    return g(T, "single", l);
  }
  function Br(s) {
    var l = s.charCodeAt(0), C = {
      8: "b",
      9: "t",
      10: "n",
      12: "f",
      13: "r"
    }[l];
    return C ? "\\" + C : "\\x" + (l < 16 ? "0" : "") + x.call(l.toString(16));
  }
  function gr(s) {
    return "Object(" + s + ")";
  }
  function Jr(s) {
    return s + " { ? }";
  }
  function Xt(s, l, C, W) {
    var k = W ? Yr(C, W) : D.call(C, ", ");
    return s + " (" + l + ") {" + k + "}";
  }
  function Ao(s) {
    for (var l = 0; l < s.length; l++)
      if (or(s[l], `
`) >= 0)
        return !1;
    return !0;
  }
  function Oo(s, l) {
    var C;
    if (s.indent === "	")
      C = "	";
    else if (typeof s.indent == "number" && s.indent > 0)
      C = D.call(Array(s.indent + 1), " ");
    else
      return null;
    return {
      base: C,
      prev: D.call(Array(l + 1), C)
    };
  }
  function Yr(s, l) {
    if (s.length === 0)
      return "";
    var C = `
` + l.prev + l.base;
    return C + D.call(s, "," + C) + `
` + l.prev;
  }
  function Dr(s, l) {
    var C = $(s), W = [];
    if (C) {
      W.length = s.length;
      for (var k = 0; k < s.length; k++)
        W[k] = Y(s, k) ? l(s[k], s) : "";
    }
    var T = typeof _ == "function" ? _(s) : [], hr;
    if (V) {
      hr = {};
      for (var Ar = 0; Ar < T.length; Ar++)
        hr["$" + T[Ar]] = T[Ar];
    }
    for (var tr in s)
      Y(s, tr) && (C && String(Number(tr)) === tr && tr < s.length || V && hr["$" + tr] instanceof Symbol || (H.call(/[^\w$]/, tr) ? W.push(l(tr, s) + ": " + l(s[tr], s)) : W.push(tr + ": " + l(s[tr], s))));
    if (typeof _ == "function")
      for (var dr = 0; dr < T.length; dr++)
        Ir.call(s, T[dr]) && W.push("[" + l(T[dr]) + "]: " + l(s[T[dr]], s));
    return W;
  }
  return ut;
}
var ft, ya;
function bi() {
  if (ya) return ft;
  ya = 1;
  var r = /* @__PURE__ */ Kr(), e = /* @__PURE__ */ nr(), n = function(a, u, c) {
    for (var v = a, p; (p = v.next) != null; v = p)
      if (p.key === u)
        return v.next = p.next, c || (p.next = /** @type {NonNullable<typeof list.next>} */
        a.next, a.next = p), p;
  }, o = function(a, u) {
    if (a) {
      var c = n(a, u);
      return c && c.value;
    }
  }, t = function(a, u, c) {
    var v = n(a, u);
    v ? v.value = c : a.next = /** @type {import('./list.d.ts').ListNode<typeof value, typeof key>} */
    {
      // eslint-disable-line no-param-reassign, no-extra-parens
      key: u,
      next: a.next,
      value: c
    };
  }, f = function(a, u) {
    return a ? !!n(a, u) : !1;
  }, i = function(a, u) {
    if (a)
      return n(a, u, !0);
  };
  return ft = function() {
    var u, c = {
      assert: function(v) {
        if (!c.has(v))
          throw new e("Side channel does not contain " + r(v));
      },
      delete: function(v) {
        var p = u && u.next, y = i(u, v);
        return y && p && p === y && (u = void 0), !!y;
      },
      get: function(v) {
        return o(u, v);
      },
      has: function(v) {
        return f(u, v);
      },
      set: function(v, p) {
        u || (u = {
          next: void 0
        }), t(
          /** @type {NonNullable<typeof $o>} */
          u,
          v,
          p
        );
      }
    };
    return c;
  }, ft;
}
var st, va;
function so() {
  if (va) return st;
  va = 1;
  var r = /* @__PURE__ */ pr(), e = /* @__PURE__ */ Q(), n = /* @__PURE__ */ Kr(), o = /* @__PURE__ */ nr(), t = r("%Map%", !0), f = e("Map.prototype.get", !0), i = e("Map.prototype.set", !0), a = e("Map.prototype.has", !0), u = e("Map.prototype.delete", !0), c = e("Map.prototype.size", !0);
  return st = !!t && /** @type {Exclude<import('.'), false>} */
  function() {
    var p, y = {
      assert: function(h) {
        if (!y.has(h))
          throw new o("Side channel does not contain " + n(h));
      },
      delete: function(h) {
        if (p) {
          var S = u(p, h);
          return c(p) === 0 && (p = void 0), S;
        }
        return !1;
      },
      get: function(h) {
        if (p)
          return f(p, h);
      },
      has: function(h) {
        return p ? a(p, h) : !1;
      },
      set: function(h, S) {
        p || (p = new t()), i(p, h, S);
      }
    };
    return y;
  }, st;
}
var lt, ga;
function Ai() {
  if (ga) return lt;
  ga = 1;
  var r = /* @__PURE__ */ pr(), e = /* @__PURE__ */ Q(), n = /* @__PURE__ */ Kr(), o = so(), t = /* @__PURE__ */ nr(), f = r("%WeakMap%", !0), i = e("WeakMap.prototype.get", !0), a = e("WeakMap.prototype.set", !0), u = e("WeakMap.prototype.has", !0), c = e("WeakMap.prototype.delete", !0);
  return lt = f ? (
    /** @type {Exclude<import('.'), false>} */
    function() {
      var p, y, h = {
        assert: function(S) {
          if (!h.has(S))
            throw new t("Side channel does not contain " + n(S));
        },
        delete: function(S) {
          if (f && S && (typeof S == "object" || typeof S == "function")) {
            if (p)
              return c(p, S);
          } else if (o && y)
            return y.delete(S);
          return !1;
        },
        get: function(S) {
          return f && S && (typeof S == "object" || typeof S == "function") && p ? i(p, S) : y && y.get(S);
        },
        has: function(S) {
          return f && S && (typeof S == "object" || typeof S == "function") && p ? u(p, S) : !!y && y.has(S);
        },
        set: function(S, R) {
          f && S && (typeof S == "object" || typeof S == "function") ? (p || (p = new f()), a(p, S, R)) : o && (y || (y = o()), y.set(S, R));
        }
      };
      return h;
    }
  ) : o, lt;
}
var ct, ha;
function lo() {
  if (ha) return ct;
  ha = 1;
  var r = /* @__PURE__ */ nr(), e = /* @__PURE__ */ Kr(), n = bi(), o = so(), t = Ai(), f = t || o || n;
  return ct = function() {
    var a, u = {
      assert: function(c) {
        if (!u.has(c))
          throw new r("Side channel does not contain " + e(c));
      },
      delete: function(c) {
        return !!a && a.delete(c);
      },
      get: function(c) {
        return a && a.get(c);
      },
      has: function(c) {
        return !!a && a.has(c);
      },
      set: function(c, v) {
        a || (a = f()), a.set(c, v);
      }
    };
    return u;
  }, ct;
}
var pt, da;
function Oi() {
  if (da) return pt;
  da = 1;
  var r = /* @__PURE__ */ Yt(), e = lo()(), n = /* @__PURE__ */ nr(), o = {
    assert: function(t, f) {
      if (!t || typeof t != "object" && typeof t != "function")
        throw new n("`O` is not an object");
      if (typeof f != "string")
        throw new n("`slot` must be a string");
      if (e.assert(t), !o.has(t, f))
        throw new n("`" + f + "` is not present on `O`");
    },
    get: function(t, f) {
      if (!t || typeof t != "object" && typeof t != "function")
        throw new n("`O` is not an object");
      if (typeof f != "string")
        throw new n("`slot` must be a string");
      var i = e.get(t);
      return i && i[
        /** @type {SaltedInternalSlot} */
        "$" + f
      ];
    },
    has: function(t, f) {
      if (!t || typeof t != "object" && typeof t != "function")
        throw new n("`O` is not an object");
      if (typeof f != "string")
        throw new n("`slot` must be a string");
      var i = e.get(t);
      return !!i && r(
        i,
        /** @type {SaltedInternalSlot} */
        "$" + f
      );
    },
    set: function(t, f, i) {
      if (!t || typeof t != "object" && typeof t != "function")
        throw new n("`O` is not an object");
      if (typeof f != "string")
        throw new n("`slot` must be a string");
      var a = e.get(t);
      a || (a = {}, e.set(t, a)), a[
        /** @type {SaltedInternalSlot} */
        "$" + f
      ] = i;
    }
  };
  return Object.freeze && Object.freeze(o), pt = o, pt;
}
var yt, ma;
function wi() {
  if (ma) return yt;
  ma = 1;
  var r = Oi(), e = /* @__PURE__ */ kt(), n = typeof StopIteration == "object" ? StopIteration : null;
  return yt = function(t) {
    if (!n)
      throw new e("this environment lacks StopIteration");
    r.set(t, "[[Done]]", !1);
    var f = {
      next: (
        /** @type {() => IteratorResult<T>} */
        function() {
          var a = (
            /** @type {typeof origIterator} */
            r.get(this, "[[Iterator]]")
          ), u = !!r.get(a, "[[Done]]");
          try {
            return {
              done: u,
              // eslint-disable-next-line no-extra-parens
              value: u ? void 0 : (
                /** @type {T} */
                a.next()
              )
            };
          } catch (c) {
            if (r.set(a, "[[Done]]", !0), c !== n)
              throw c;
            return {
              done: !0,
              value: void 0
            };
          }
        }
      )
    };
    return r.set(f, "[[Iterator]]", t), f;
  }, yt;
}
var vt, Sa;
function co() {
  if (Sa) return vt;
  Sa = 1;
  var r = {}.toString;
  return vt = Array.isArray || function(e) {
    return r.call(e) == "[object Array]";
  }, vt;
}
var gt, ba;
function po() {
  if (ba) return gt;
  ba = 1;
  var r = /* @__PURE__ */ Q(), e = r("String.prototype.valueOf"), n = function(a) {
    try {
      return e(a), !0;
    } catch {
      return !1;
    }
  }, o = r("Object.prototype.toString"), t = "[object String]", f = qr()();
  return gt = function(a) {
    return typeof a == "string" ? !0 : !a || typeof a != "object" ? !1 : f ? n(a) : o(a) === t;
  }, gt;
}
var ht, Aa;
function yo() {
  if (Aa) return ht;
  Aa = 1;
  var r = typeof Map == "function" && Map.prototype ? Map : null, e = typeof Set == "function" && Set.prototype ? Set : null, n;
  r || (n = function(i) {
    return !1;
  });
  var o = r ? Map.prototype.has : null, t = e ? Set.prototype.has : null;
  return !n && !o && (n = function(i) {
    return !1;
  }), ht = n || function(i) {
    if (!i || typeof i != "object")
      return !1;
    try {
      if (o.call(i), t)
        try {
          t.call(i);
        } catch {
          return !0;
        }
      return i instanceof r;
    } catch {
    }
    return !1;
  }, ht;
}
var dt, Oa;
function vo() {
  if (Oa) return dt;
  Oa = 1;
  var r = typeof Map == "function" && Map.prototype ? Map : null, e = typeof Set == "function" && Set.prototype ? Set : null, n;
  e || (n = function(i) {
    return !1;
  });
  var o = r ? Map.prototype.has : null, t = e ? Set.prototype.has : null;
  return !n && !t && (n = function(i) {
    return !1;
  }), dt = n || function(i) {
    if (!i || typeof i != "object")
      return !1;
    try {
      if (t.call(i), o)
        try {
          o.call(i);
        } catch {
          return !0;
        }
      return i instanceof e;
    } catch {
    }
    return !1;
  }, dt;
}
var wa;
function qi() {
  if (wa) return Mr.exports;
  wa = 1;
  var r = /* @__PURE__ */ fo(), e = /* @__PURE__ */ wi();
  if (Gt()() || Vr()()) {
    var n = Symbol.iterator;
    Mr.exports = function(I) {
      if (I != null && typeof I[n] < "u")
        return I[n]();
      if (r(I))
        return Array.prototype[n].call(I);
    };
  } else {
    var o = co(), t = po(), f = /* @__PURE__ */ pr(), i = f("%Map%", !0), a = f("%Set%", !0), u = oo(), c = u("Array.prototype.push"), v = u("String.prototype.charCodeAt"), p = u("String.prototype.slice"), y = function(I, N) {
      var L = I.length;
      if (N + 1 >= L)
        return N + 1;
      var _ = v(I, N);
      if (_ < 55296 || _ > 56319)
        return N + 1;
      var U = v(I, N + 1);
      return U < 56320 || U > 57343 ? N + 1 : N + 2;
    }, h = function(I) {
      var N = 0;
      return {
        next: function() {
          var _ = N >= I.length, U;
          return _ || (U = I[N], N += 1), {
            done: _,
            value: U
          };
        }
      };
    }, S = function(I, N) {
      if (o(I) || r(I))
        return h(I);
      if (t(I)) {
        var L = 0;
        return {
          next: function() {
            var U = y(I, L), V = p(I, L, U);
            return L = U, {
              done: U > I.length,
              value: V
            };
          }
        };
      }
      if (N && typeof I["_es6-shim iterator_"] < "u")
        return I["_es6-shim iterator_"]();
    };
    if (!i && !a)
      Mr.exports = function(I) {
        if (I != null)
          return S(I, !0);
      };
    else {
      var R = /* @__PURE__ */ yo(), w = /* @__PURE__ */ vo(), q = u("Map.prototype.forEach", !0), A = u("Set.prototype.forEach", !0);
      if (typeof process > "u" || !process.versions || !process.versions.node)
        var m = u("Map.prototype.iterator", !0), x = u("Set.prototype.iterator", !0);
      var X = u("Map.prototype.@@iterator", !0) || u("Map.prototype._es6-shim iterator_", !0), H = u("Set.prototype.@@iterator", !0) || u("Set.prototype._es6-shim iterator_", !0), F = function(I) {
        if (R(I)) {
          if (m)
            return e(m(I));
          if (X)
            return X(I);
          if (q) {
            var N = [];
            return q(I, function(_, U) {
              c(N, [U, _]);
            }), h(N);
          }
        }
        if (w(I)) {
          if (x)
            return e(x(I));
          if (H)
            return H(I);
          if (A) {
            var L = [];
            return A(I, function(_) {
              c(L, _);
            }), h(L);
          }
        }
      };
      Mr.exports = function(I) {
        return F(I) || S(I);
      };
    }
  }
  return Mr.exports;
}
var mt, qa;
function go() {
  if (qa) return mt;
  qa = 1;
  var r = function(e) {
    return e !== e;
  };
  return mt = function(n, o) {
    return n === 0 && o === 0 ? 1 / n === 1 / o : !!(n === o || r(n) && r(o));
  }, mt;
}
var St, Ia;
function ho() {
  if (Ia) return St;
  Ia = 1;
  var r = go();
  return St = function() {
    return typeof Object.is == "function" ? Object.is : r;
  }, St;
}
var bt, Pa;
function Ii() {
  if (Pa) return bt;
  Pa = 1;
  var r = ho(), e = wr();
  return bt = function() {
    var o = r();
    return e(Object, { is: o }, {
      is: function() {
        return Object.is !== o;
      }
    }), o;
  }, bt;
}
var At, Ra;
function Pi() {
  if (Ra) return At;
  Ra = 1;
  var r = wr(), e = xr(), n = go(), o = ho(), t = Ii(), f = e(o(), Object);
  return r(f, {
    getPolyfill: o,
    implementation: n,
    shim: t
  }), At = f, At;
}
var Ot, $a;
function mo() {
  if ($a) return Ot;
  $a = 1;
  var r = xr(), e = /* @__PURE__ */ Q(), n = /* @__PURE__ */ pr(), o = n("%ArrayBuffer%", !0), t = e("ArrayBuffer.prototype.byteLength", !0), f = e("Object.prototype.toString"), i = !!o && !t && new o(0).slice, a = !!i && r(i);
  return Ot = t || a ? function(c) {
    if (!c || typeof c != "object")
      return !1;
    try {
      return t ? t(c) : a(c, 0), !0;
    } catch {
      return !1;
    }
  } : o ? function(c) {
    return f(c) === "[object ArrayBuffer]";
  } : function(c) {
    return !1;
  }, Ot;
}
var wt, Ea;
function Ri() {
  if (Ea) return wt;
  Ea = 1;
  var r = /* @__PURE__ */ Q(), e = r("Date.prototype.getDay"), n = function(a) {
    try {
      return e(a), !0;
    } catch {
      return !1;
    }
  }, o = r("Object.prototype.toString"), t = "[object Date]", f = qr()();
  return wt = function(a) {
    return typeof a != "object" || a === null ? !1 : f ? n(a) : o(a) === t;
  }, wt;
}
var qt, xa;
function So() {
  if (xa) return qt;
  xa = 1;
  var r = /* @__PURE__ */ Q(), e = qr()(), n = /* @__PURE__ */ Yt(), o = /* @__PURE__ */ Or(), t;
  if (e) {
    var f = r("RegExp.prototype.exec"), i = {}, a = function() {
      throw i;
    }, u = {
      toString: a,
      valueOf: a
    };
    typeof Symbol.toPrimitive == "symbol" && (u[Symbol.toPrimitive] = a), t = function(y) {
      if (!y || typeof y != "object")
        return !1;
      var h = (
        /** @type {NonNullable<typeof gOPD>} */
        o(
          /** @type {{ lastIndex?: unknown }} */
          y,
          "lastIndex"
        )
      ), S = h && n(h, "value");
      if (!S)
        return !1;
      try {
        f(
          y,
          /** @type {string} */
          /** @type {unknown} */
          u
        );
      } catch (R) {
        return R === i;
      }
    };
  } else {
    var c = r("Object.prototype.toString"), v = "[object RegExp]";
    t = function(y) {
      return !y || typeof y != "object" && typeof y != "function" ? !1 : c(y) === v;
    };
  }
  return qt = t, qt;
}
var It, Ba;
function $i() {
  if (Ba) return It;
  Ba = 1;
  var r = /* @__PURE__ */ Q(), e = r("SharedArrayBuffer.prototype.byteLength", !0);
  return It = e ? function(o) {
    if (!o || typeof o != "object")
      return !1;
    try {
      return e(o), !0;
    } catch {
      return !1;
    }
  } : function(o) {
    return !1;
  }, It;
}
var Pt, Ta;
function Ei() {
  if (Ta) return Pt;
  Ta = 1;
  var r = /* @__PURE__ */ Q(), e = r("Number.prototype.toString"), n = function(a) {
    try {
      return e(a), !0;
    } catch {
      return !1;
    }
  }, o = r("Object.prototype.toString"), t = "[object Number]", f = qr()();
  return Pt = function(a) {
    return typeof a == "number" ? !0 : !a || typeof a != "object" ? !1 : f ? n(a) : o(a) === t;
  }, Pt;
}
var Rt, Ma;
function xi() {
  if (Ma) return Rt;
  Ma = 1;
  var r = /* @__PURE__ */ Q(), e = r("Boolean.prototype.toString"), n = r("Object.prototype.toString"), o = function(a) {
    try {
      return e(a), !0;
    } catch {
      return !1;
    }
  }, t = "[object Boolean]", f = qr()();
  return Rt = function(a) {
    return typeof a == "boolean" ? !0 : a === null || typeof a != "object" ? !1 : f ? o(a) : n(a) === t;
  }, Rt;
}
var Wr = { exports: {} }, $t, ja;
function Bi() {
  if (ja) return $t;
  ja = 1;
  var r = /* @__PURE__ */ Q(), e = So(), n = r("RegExp.prototype.exec"), o = /* @__PURE__ */ nr();
  return $t = function(f) {
    if (!e(f))
      throw new o("`regex` must be a RegExp");
    return function(a) {
      return n(f, a) !== null;
    };
  }, $t;
}
var Ca;
function Ti() {
  if (Ca) return Wr.exports;
  Ca = 1;
  var r = /* @__PURE__ */ Q(), e = r("Object.prototype.toString"), n = Gt()(), o = /* @__PURE__ */ Bi();
  if (n) {
    var t = r("Symbol.prototype.toString"), f = o(/^Symbol\(.*\)$/), i = function(u) {
      return typeof u.valueOf() != "symbol" ? !1 : f(t(u));
    };
    Wr.exports = function(u) {
      if (typeof u == "symbol")
        return !0;
      if (!u || typeof u != "object" || e(u) !== "[object Symbol]")
        return !1;
      try {
        return i(u);
      } catch {
        return !1;
      }
    };
  } else
    Wr.exports = function(u) {
      return !1;
    };
  return Wr.exports;
}
var kr = { exports: {} }, Et, Da;
function Mi() {
  if (Da) return Et;
  Da = 1;
  var r = typeof BigInt < "u" && BigInt;
  return Et = function() {
    return typeof r == "function" && typeof BigInt == "function" && typeof r(42) == "bigint" && typeof BigInt(42) == "bigint";
  }, Et;
}
var Fa;
function ji() {
  if (Fa) return kr.exports;
  Fa = 1;
  var r = Mi()();
  if (r) {
    var e = BigInt.prototype.valueOf, n = function(t) {
      try {
        return e.call(t), !0;
      } catch {
      }
      return !1;
    };
    kr.exports = function(t) {
      return t === null || typeof t > "u" || typeof t == "boolean" || typeof t == "string" || typeof t == "number" || typeof t == "symbol" || typeof t == "function" ? !1 : typeof t == "bigint" ? !0 : n(t);
    };
  } else
    kr.exports = function(t) {
      return !1;
    };
  return kr.exports;
}
var xt, Na;
function Ci() {
  if (Na) return xt;
  Na = 1;
  var r = po(), e = Ei(), n = xi(), o = Ti(), t = ji();
  return xt = function(i) {
    if (i == null || typeof i != "object" && typeof i != "function")
      return null;
    if (r(i))
      return "String";
    if (e(i))
      return "Number";
    if (n(i))
      return "Boolean";
    if (o(i))
      return "Symbol";
    if (t(i))
      return "BigInt";
  }, xt;
}
var Bt, _a;
function Di() {
  if (_a) return Bt;
  _a = 1;
  var r = typeof WeakMap == "function" && WeakMap.prototype ? WeakMap : null, e = typeof WeakSet == "function" && WeakSet.prototype ? WeakSet : null, n;
  r || (n = function(i) {
    return !1;
  });
  var o = r ? r.prototype.has : null, t = e ? e.prototype.has : null;
  return !n && !o && (n = function(i) {
    return !1;
  }), Bt = n || function(i) {
    if (!i || typeof i != "object")
      return !1;
    try {
      if (o.call(i, o), t)
        try {
          t.call(i, t);
        } catch {
          return !0;
        }
      return i instanceof r;
    } catch {
    }
    return !1;
  }, Bt;
}
var Lr = { exports: {} }, Wa;
function Fi() {
  if (Wa) return Lr.exports;
  Wa = 1;
  var r = /* @__PURE__ */ pr(), e = /* @__PURE__ */ Q(), n = r("%WeakSet%", !0), o = e("WeakSet.prototype.has", !0);
  if (o) {
    var t = e("WeakMap.prototype.has", !0);
    Lr.exports = function(i) {
      if (!i || typeof i != "object")
        return !1;
      try {
        if (o(i, o), t)
          try {
            t(i, t);
          } catch {
            return !0;
          }
        return i instanceof n;
      } catch {
      }
      return !1;
    };
  } else
    Lr.exports = function(i) {
      return !1;
    };
  return Lr.exports;
}
var Tt, ka;
function Ni() {
  if (ka) return Tt;
  ka = 1;
  var r = /* @__PURE__ */ yo(), e = /* @__PURE__ */ vo(), n = Di(), o = /* @__PURE__ */ Fi();
  return Tt = function(f) {
    if (f && typeof f == "object") {
      if (r(f))
        return "Map";
      if (e(f))
        return "Set";
      if (n(f))
        return "WeakMap";
      if (o(f))
        return "WeakSet";
    }
    return !1;
  }, Tt;
}
var Mt, La;
function _i() {
  if (La) return Mt;
  La = 1;
  var r = Function.prototype.toString, e = typeof Reflect == "object" && Reflect !== null && Reflect.apply, n, o;
  if (typeof e == "function" && typeof Object.defineProperty == "function")
    try {
      n = Object.defineProperty({}, "length", {
        get: function() {
          throw o;
        }
      }), o = {}, e(function() {
        throw 42;
      }, null, n);
    } catch (A) {
      A !== o && (e = null);
    }
  else
    e = null;
  var t = /^\s*class\b/, f = function(m) {
    try {
      var x = r.call(m);
      return t.test(x);
    } catch {
      return !1;
    }
  }, i = function(m) {
    try {
      return f(m) ? !1 : (r.call(m), !0);
    } catch {
      return !1;
    }
  }, a = Object.prototype.toString, u = "[object Object]", c = "[object Function]", v = "[object GeneratorFunction]", p = "[object HTMLAllCollection]", y = "[object HTML document.all class]", h = "[object HTMLCollection]", S = typeof Symbol == "function" && !!Symbol.toStringTag, R = !(0 in [,]), w = function() {
    return !1;
  };
  if (typeof document == "object") {
    var q = document.all;
    a.call(q) === a.call(document.all) && (w = function(m) {
      if ((R || !m) && (typeof m > "u" || typeof m == "object"))
        try {
          var x = a.call(m);
          return (x === p || x === y || x === h || x === u) && m("") == null;
        } catch {
        }
      return !1;
    });
  }
  return Mt = e ? function(m) {
    if (w(m))
      return !0;
    if (!m || typeof m != "function" && typeof m != "object")
      return !1;
    try {
      e(m, null, n);
    } catch (x) {
      if (x !== o)
        return !1;
    }
    return !f(m) && i(m);
  } : function(m) {
    if (w(m))
      return !0;
    if (!m || typeof m != "function" && typeof m != "object")
      return !1;
    if (S)
      return i(m);
    if (f(m))
      return !1;
    var x = a.call(m);
    return x !== c && x !== v && !/^\[object HTML/.test(x) ? !1 : i(m);
  }, Mt;
}
var jt, Ua;
function Wi() {
  if (Ua) return jt;
  Ua = 1;
  var r = _i(), e = Object.prototype.toString, n = Object.prototype.hasOwnProperty, o = function(u, c, v) {
    for (var p = 0, y = u.length; p < y; p++)
      n.call(u, p) && (v == null ? c(u[p], p, u) : c.call(v, u[p], p, u));
  }, t = function(u, c, v) {
    for (var p = 0, y = u.length; p < y; p++)
      v == null ? c(u.charAt(p), p, u) : c.call(v, u.charAt(p), p, u);
  }, f = function(u, c, v) {
    for (var p in u)
      n.call(u, p) && (v == null ? c(u[p], p, u) : c.call(v, u[p], p, u));
  };
  function i(a) {
    return e.call(a) === "[object Array]";
  }
  return jt = function(u, c, v) {
    if (!r(c))
      throw new TypeError("iterator must be a function");
    var p;
    arguments.length >= 3 && (p = v), i(u) ? o(u, c, p) : typeof u == "string" ? t(u, c, p) : f(u, c, p);
  }, jt;
}
var Ct, Ha;
function ki() {
  return Ha || (Ha = 1, Ct = [
    "Float16Array",
    "Float32Array",
    "Float64Array",
    "Int8Array",
    "Int16Array",
    "Int32Array",
    "Uint8Array",
    "Uint8ClampedArray",
    "Uint16Array",
    "Uint32Array",
    "BigInt64Array",
    "BigUint64Array"
  ]), Ct;
}
var Dt, Ga;
function Li() {
  if (Ga) return Dt;
  Ga = 1;
  var r = /* @__PURE__ */ ki(), e = typeof globalThis > "u" ? Gr : globalThis;
  return Dt = function() {
    for (var o = [], t = 0; t < r.length; t++)
      typeof e[r[t]] == "function" && (o[o.length] = r[t]);
    return o;
  }, Dt;
}
var Ft, za;
function Ui() {
  if (za) return Ft;
  za = 1;
  var r = Wi(), e = /* @__PURE__ */ Li(), n = xr(), o = /* @__PURE__ */ Q(), t = /* @__PURE__ */ Or(), f = Jt(), i = o("Object.prototype.toString"), a = qr()(), u = typeof globalThis > "u" ? Gr : globalThis, c = e(), v = o("String.prototype.slice"), p = o("Array.prototype.indexOf", !0) || function(w, q) {
    for (var A = 0; A < w.length; A += 1)
      if (w[A] === q)
        return A;
    return -1;
  }, y = { __proto__: null };
  a && t && f ? r(c, function(R) {
    var w = new u[R]();
    if (Symbol.toStringTag in w && f) {
      var q = f(w), A = t(q, Symbol.toStringTag);
      if (!A && q) {
        var m = f(q);
        A = t(m, Symbol.toStringTag);
      }
      if (A && A.get) {
        var x = n(A.get);
        y[
          /** @type {`$${import('.').TypedArrayName}`} */
          "$" + R
        ] = x;
      }
    }
  }) : r(c, function(R) {
    var w = new u[R](), q = w.slice || w.set;
    if (q) {
      var A = (
        /** @type {import('./types').BoundSlice | import('./types').BoundSet} */
        // @ts-expect-error TODO FIXME
        n(q)
      );
      y[
        /** @type {`$${import('.').TypedArrayName}`} */
        "$" + R
      ] = A;
    }
  });
  var h = function(w) {
    var q = !1;
    return r(
      /** @type {Record<`\$${import('.').TypedArrayName}`, Getter>} */
      y,
      /** @type {(getter: Getter, name: `\$${import('.').TypedArrayName}`) => void} */
      function(A, m) {
        if (!q)
          try {
            "$" + A(w) === m && (q = /** @type {import('.').TypedArrayName} */
            v(m, 1));
          } catch {
          }
      }
    ), q;
  }, S = function(w) {
    var q = !1;
    return r(
      /** @type {Record<`\$${import('.').TypedArrayName}`, Getter>} */
      y,
      /** @type {(getter: Getter, name: `\$${import('.').TypedArrayName}`) => void} */
      function(A, m) {
        if (!q)
          try {
            A(w), q = /** @type {import('.').TypedArrayName} */
            v(m, 1);
          } catch {
          }
      }
    ), q;
  };
  return Ft = function(w) {
    if (!w || typeof w != "object")
      return !1;
    if (!a) {
      var q = v(i(w), 8, -1);
      return p(c, q) > -1 ? q : q !== "Object" ? !1 : S(w);
    }
    return t ? h(w) : null;
  }, Ft;
}
var Nt, Va;
function Hi() {
  if (Va) return Nt;
  Va = 1;
  var r = /* @__PURE__ */ Q(), e = r("ArrayBuffer.prototype.byteLength", !0), n = /* @__PURE__ */ mo();
  return Nt = function(t) {
    return n(t) ? e ? e(t) : t.byteLength : NaN;
  }, Nt;
}
var _t, Ka;
function Gi() {
  if (Ka) return _t;
  Ka = 1;
  var r = pi(), e = oo(), n = hi(), o = /* @__PURE__ */ pr(), t = qi(), f = lo(), i = Pi(), a = /* @__PURE__ */ fo(), u = co(), c = /* @__PURE__ */ mo(), v = /* @__PURE__ */ Ri(), p = So(), y = /* @__PURE__ */ $i(), h = Wt(), S = Ci(), R = /* @__PURE__ */ Ni(), w = /* @__PURE__ */ Ui(), q = /* @__PURE__ */ Hi(), A = e("SharedArrayBuffer.prototype.byteLength", !0), m = e("Date.prototype.getTime"), x = Object.getPrototypeOf, X = e("Object.prototype.toString"), H = o("%Set%", !0), F = e("Map.prototype.has", !0), D = e("Map.prototype.get", !0), I = e("Map.prototype.size", !0), N = e("Set.prototype.add", !0), L = e("Set.prototype.delete", !0), _ = e("Set.prototype.has", !0), U = e("Set.prototype.size", !0);
  function V(g, d, E, $) {
    for (var M = t(g), j; (j = M.next()) && !j.done; )
      if (G(d, j.value, E, $))
        return L(g, j.value), !0;
    return !1;
  }
  function ur(g) {
    if (typeof g > "u")
      return null;
    if (typeof g != "object")
      return typeof g == "symbol" ? !1 : typeof g == "string" || typeof g == "number" ? +g == +g : !0;
  }
  function Ir(g, d, E, $, M, j) {
    var O = ur(E);
    if (O != null)
      return O;
    var P = D(d, O), b = r({}, M, { strict: !1 });
    return typeof P > "u" && !F(d, O) || !G($, P, b, j) ? !1 : !F(g, O) && G($, P, b, j);
  }
  function mr(g, d, E) {
    var $ = ur(E);
    return $ ?? (_(d, $) && !_(g, $));
  }
  function ar(g, d, E, $, M, j) {
    for (var O = t(g), P, b; (P = O.next()) && !P.done; )
      if (b = P.value, // eslint-disable-next-line no-use-before-define
      G(E, b, M, j) && G($, D(d, b), M, j))
        return L(g, b), !0;
    return !1;
  }
  function G(g, d, E, $) {
    var M = E || {};
    if (M.strict ? i(g, d) : g === d)
      return !0;
    var j = S(g), O = S(d);
    if (j !== O)
      return !1;
    if (!g || !d || typeof g != "object" && typeof d != "object")
      return M.strict ? i(g, d) : g == d;
    var P = $.has(g), b = $.has(d), B;
    if (P && b) {
      if ($.get(g) === $.get(d))
        return !0;
    } else
      B = {};
    return P || $.set(g, B), b || $.set(d, B), yr(g, d, M, $);
  }
  function Sr(g) {
    return !g || typeof g != "object" || typeof g.length != "number" || typeof g.copy != "function" || typeof g.slice != "function" || g.length > 0 && typeof g[0] != "number" ? !1 : !!(g.constructor && g.constructor.isBuffer && g.constructor.isBuffer(g));
  }
  function Pr(g, d, E, $) {
    if (U(g) !== U(d))
      return !1;
    for (var M = t(g), j = t(d), O, P, b; (O = M.next()) && !O.done; )
      if (O.value && typeof O.value == "object")
        b || (b = new H()), N(b, O.value);
      else if (!_(d, O.value)) {
        if (E.strict || !mr(g, d, O.value))
          return !1;
        b || (b = new H()), N(b, O.value);
      }
    if (b) {
      for (; (P = j.next()) && !P.done; )
        if (P.value && typeof P.value == "object") {
          if (!V(b, P.value, E.strict, $))
            return !1;
        } else if (!E.strict && !_(g, P.value) && !V(b, P.value, E.strict, $))
          return !1;
      return U(b) === 0;
    }
    return !0;
  }
  function br(g, d, E, $) {
    if (I(g) !== I(d))
      return !1;
    for (var M = t(g), j = t(d), O, P, b, B, z, K; (O = M.next()) && !O.done; )
      if (B = O.value[0], z = O.value[1], B && typeof B == "object")
        b || (b = new H()), N(b, B);
      else if (K = D(d, B), typeof K > "u" && !F(d, B) || !G(z, K, E, $)) {
        if (E.strict || !Ir(g, d, B, z, E, $))
          return !1;
        b || (b = new H()), N(b, B);
      }
    if (b) {
      for (; (P = j.next()) && !P.done; )
        if (B = P.value[0], K = P.value[1], B && typeof B == "object") {
          if (!ar(b, g, B, K, E, $))
            return !1;
        } else if (!E.strict && (!g.has(B) || !G(D(g, B), K, E, $)) && !ar(b, g, B, K, r({}, E, { strict: !1 }), $))
          return !1;
      return U(b) === 0;
    }
    return !0;
  }
  function yr(g, d, E, $) {
    var M, j;
    if (typeof g != typeof d || g == null || d == null || X(g) !== X(d) || a(g) !== a(d))
      return !1;
    var O = u(g), P = u(d);
    if (O !== P)
      return !1;
    var b = g instanceof Error, B = d instanceof Error;
    if (b !== B || (b || B) && (g.name !== d.name || g.message !== d.message))
      return !1;
    var z = p(g), K = p(d);
    if (z !== K || (z || K) && (g.source !== d.source || n(g) !== n(d)))
      return !1;
    var J = v(g), Y = v(d);
    if (J !== Y || (J || Y) && m(g) !== m(d) || E.strict && x && x(g) !== x(d))
      return !1;
    var Z = w(g), fr = w(d);
    if (Z !== fr)
      return !1;
    if (Z || fr) {
      if (g.length !== d.length)
        return !1;
      for (M = 0; M < g.length; M++)
        if (g[M] !== d[M])
          return !1;
      return !0;
    }
    var or = Sr(g), rr = Sr(d);
    if (or !== rr)
      return !1;
    if (or || rr) {
      if (g.length !== d.length)
        return !1;
      for (M = 0; M < g.length; M++)
        if (g[M] !== d[M])
          return !1;
      return !0;
    }
    var sr = c(g), lr = c(d);
    if (sr !== lr)
      return !1;
    if (sr || lr)
      return q(g) !== q(d) ? !1 : typeof Uint8Array == "function" && G(new Uint8Array(g), new Uint8Array(d), E, $);
    var cr = y(g), Cr = y(d);
    if (cr !== Cr)
      return !1;
    if (cr || Cr)
      return A(g) !== A(d) ? !1 : typeof Uint8Array == "function" && G(new Uint8Array(g), new Uint8Array(d), E, $);
    if (typeof g != typeof d)
      return !1;
    var vr = h(g), Rr = h(d);
    if (vr.length !== Rr.length)
      return !1;
    for (vr.sort(), Rr.sort(), M = vr.length - 1; M >= 0; M--)
      if (vr[M] != Rr[M])
        return !1;
    for (M = vr.length - 1; M >= 0; M--)
      if (j = vr[M], !G(g[j], d[j], E, $))
        return !1;
    var Br = R(g), gr = R(d);
    return Br !== gr ? !1 : Br === "Set" || gr === "Set" ? Pr(g, d, E, $) : Br === "Map" ? br(g, d, E, $) : !0;
  }
  return _t = function(d, E, $) {
    return G(d, E, $, f());
  }, _t;
}
var zi = Gi();
const Vi = /* @__PURE__ */ Ho(zi);
function Ki(r) {
  return r;
}
function Ji(r, e, n, o) {
  const t = Ki, [f, i] = Ur(t(n)), a = xo(f);
  return Bo(() => {
    const u = () => {
      const c = {
        ...n,
        ...r.getAttributes(e)
      };
      Object.keys(c).forEach((p) => {
        (c[p] === null || c[p] === void 0) && (c[p] = n ? n[p] : null);
      });
      const v = t(c);
      Vi(a.current, v) || (i(v), a.current = v);
    };
    return r.on("selectionUpdate", u), r.on("transaction", u), () => {
      r.off("selectionUpdate", u), r.off("transaction", u);
    };
  }, [r, n, e, t]), f;
}
function bo(r, e = {}) {
  if (typeof r == "object") return r;
  try {
    return JSON.parse(r);
  } catch {
    return e;
  }
}
function Yi({ node: r }) {
  const { text: e, macros: n } = r.attrs, o = Hr(() => {
    try {
      return Xa.renderToString(decodeURIComponent(e || ""), {
        macros: bo(decodeURIComponent(n || ""))
      });
    } catch {
      return e;
    }
  }, [e, n]), t = Hr(
    () => e.trim() ? /* @__PURE__ */ er("span", { contentEditable: !1, dangerouslySetInnerHTML: { __html: o } }) : /* @__PURE__ */ er("span", { contentEditable: !1, children: "Not enter a formula" }),
    [e, o]
  );
  return /* @__PURE__ */ er(
    $o,
    {
      as: "span",
      style: {
        display: "inline-block"
      },
      children: t
    }
  );
}
function iu() {
  const { t: r } = Go(), [e, n] = Ur(!1), o = Mo(Ya.name), {
    icon: t = void 0,
    tooltip: f = void 0,
    tooltipOptions: i = {},
    isActive: a = void 0
  } = (o == null ? void 0 : o.componentProps) ?? {}, { editorDisabled: u } = jo(a), c = Co(), v = Ji(c, Ya.name, {
    text: "",
    macros: ""
  }), { text: p, macros: y } = v, [h, S] = Ur(decodeURIComponent(p || "")), [R, w] = Ur(decodeURIComponent(y || "")), q = To(() => {
    c.chain().focus().setKatex({
      text: encodeURIComponent(h),
      macros: encodeURIComponent(R)
    }).run(), S(""), w(""), n(!1);
  }, [c, h, R]), A = Hr(() => {
    try {
      return Xa.renderToString(h, {
        macros: bo(R)
      });
    } catch {
      return h;
    }
  }, [R, h]), m = Hr(() => `${h}`.trim() ? A : null, [h, A]);
  return /* @__PURE__ */ _r(Do, { onOpenChange: n, open: e, children: [
    /* @__PURE__ */ er(Fo, { asChild: !0, disabled: u, children: /* @__PURE__ */ er(
      No,
      {
        disabled: u,
        icon: t,
        tooltip: f,
        tooltipOptions: i,
        action: () => {
          u || n(!0);
        }
      }
    ) }),
    /* @__PURE__ */ _r(_o, { className: "richtext-z-[99999] !richtext-max-w-[1300px]", children: [
      /* @__PURE__ */ er(Wo, { children: r("editor.formula.dialog.text") }),
      /* @__PURE__ */ er("div", { style: { height: "100%", border: "1px solid hsl(var(--border))" }, children: /* @__PURE__ */ _r("div", { className: "richtext-flex richtext-gap-[10px] richtext-rounded-[10px] richtext-p-[10px]", children: [
        /* @__PURE__ */ _r("div", { className: "richtext-flex-1", children: [
          /* @__PURE__ */ er(un, { className: "mb-[6px]", children: "Expression" }),
          /* @__PURE__ */ er(
            fn,
            {
              autoFocus: !0,
              className: "richtext-mb-[10px]",
              onChange: (x) => S(x.target.value),
              placeholder: "Text",
              required: !0,
              rows: 10,
              value: h,
              style: {
                color: "hsl(var(--foreground))"
              }
            }
          ),
          /* @__PURE__ */ er(un, { className: "mb-[6px]", children: "Macros" }),
          /* @__PURE__ */ er(
            fn,
            {
              className: "richtext-flex-1",
              placeholder: "Macros",
              rows: 10,
              value: R,
              onChange: (x) => {
                w(x.target.value);
              },
              style: {
                color: "hsl(var(--foreground))"
              }
            }
          )
        ] }),
        /* @__PURE__ */ er(
          "div",
          {
            className: "richtext-flex richtext-flex-1 richtext-items-center richtext-justify-center richtext-rounded-[10px] richtext-p-[10px]",
            dangerouslySetInnerHTML: { __html: m || "" },
            style: {
              height: "100%",
              borderWidth: 1,
              minHeight: 500,
              background: "#fff"
            }
          }
        )
      ] }) }),
      /* @__PURE__ */ er(ko, { children: /* @__PURE__ */ er(Lo, { onClick: q, type: "button", children: "Save changes" }) })
    ] })
  ] });
}
function Ja(r) {
  return (e) => e.getAttribute(r);
}
const Ya = /* @__PURE__ */ Io.create({
  name: "katex",
  group: "inline",
  inline: !0,
  defining: !0,
  draggable: !0,
  selectable: !0,
  addOptions() {
    return {
      HTMLAttributes: {
        class: "katex"
      },
      button: ({ t: r }) => ({
        componentProps: {
          action: () => !0,
          isActive: () => !1,
          disabled: !1,
          icon: "KatexIcon",
          tooltip: r("editor.katex.tooltip")
        }
      })
    };
  },
  addAttributes() {
    return {
      text: {
        default: "",
        parseHTML: Ja("text")
      },
      macros: {
        default: "",
        parseHTML: Ja("macros")
      }
    };
  },
  parseHTML() {
    return [{ tag: "span.katex" }];
  },
  renderHTML({ HTMLAttributes: r }) {
    return [
      "span",
      Ro(this.options && this.options.HTMLAttributes || {}, r)
    ];
  },
  addCommands() {
    return {
      setKatex: (r) => ({ commands: e }) => e.insertContent({
        type: this.name,
        attrs: r
      })
    };
  },
  addInputRules() {
    return [
      Po({
        find: /^\$katex\$$/,
        type: this.type
      })
    ];
  },
  addNodeView() {
    return Eo(Yi);
  }
});
export {
  Ya as K,
  iu as R,
  bo as s,
  Ji as u
};
