// reactjs-tiptap-editor@1.0.2/lib/Mermaid.js
import { s as $, m as U } from "./index-Dx-Dh7KS.js";
import { E as q, F as X, h as Z, u as J, d as Y, D as K, n as Q, A as tt, o as et, p as rt, G as it, v as at, B as st, H as nt, J as ot } from "./index-C52cuQ_Y.js";
import { N as lt, a as ct } from "./clsx-_bq4MQQ7.js";
import { jsx as n, jsxs as C, Fragment as dt } from "react/jsx-runtime";
import { useState as m, useMemo as F, useCallback as A, useEffect as R, useRef as ut } from "react";
import { I as V, j as W, k as ht } from "./index-DtyAGMbq.js";
import { s as mt, i as gt } from "./shortId-WJVkrvml.js";
const z = {
  TOP_LEFT: "tl",
  TOP_RIGHT: "tr",
  BOTTOM_LEFT: "bl",
  BOTTOM_RIGHT: "br"
};
function pt({ editor: e, node: t, updateAttributes: i, getPos: s, selected: w }) {
  const [T, c] = m({
    width: V,
    height: V
  }), [S, D] = m({
    width: 0,
    height: 0
  }), [y] = m([
    z.TOP_LEFT,
    z.TOP_RIGHT,
    z.BOTTOM_LEFT,
    z.BOTTOM_RIGHT
  ]), [o, _] = m(!1), [L, d] = m({
    x: 0,
    y: 0,
    w: 0,
    h: 0,
    dir: ""
  }), { align: k } = t == null ? void 0 : t.attrs, g = F(() => {
    const { src: r, alt: p, width: f, height: b } = t == null ? void 0 : t.attrs, x = $(f) ? `${f}px` : f, a = $(b) ? `${b}px` : b;
    return {
      src: r || void 0,
      alt: p || void 0,
      style: {
        width: x || void 0,
        height: a || void 0
      }
    };
  }, [t == null ? void 0 : t.attrs]), E = F(() => {
    const {
      style: { width: r }
    } = g;
    return { width: r === "100%" ? r : void 0 };
  }, [g]);
  function B(r) {
    D({
      width: r.target.width,
      height: r.target.height
    });
  }
  function I() {
    e.commands.setNodeSelection(s());
  }
  const O = A(
    q(() => {
      const { width: r } = getComputedStyle(e.view.dom);
      c((p) => ({
        ...p,
        width: Number.parseInt(r, 10)
      }));
    }, W),
    [e]
  );
  function P(r, p) {
    r.preventDefault(), r.stopPropagation();
    const f = S.width, b = S.height, x = f / b;
    let a = Number(t.attrs.width), v = Number(t.attrs.height);
    const h = T.width;
    a && !v ? (a = a > h ? h : a, v = Math.round(a / x)) : v && !a ? (a = Math.round(v * x), a = a > h ? h : a) : !a && !v ? (a = f > h ? h : f, v = Math.round(a / x)) : a = a > h ? h : a, _(!0), d({
      x: r.clientX,
      y: r.clientY,
      w: a,
      h: v,
      dir: p
    });
  }
  const l = A(
    q((r) => {
      if (r.preventDefault(), r.stopPropagation(), !o)
        return;
      const { x: p, w: f, dir: b } = L, x = (r.clientX - p) * (/l/.test(b) ? -1 : 1), { width: a, height: v } = t == null ? void 0 : t.attrs, h = a / v, G = X(f + x, ht, T.width), j = Math.round(G / h);
      i({
        width: G,
        height: j
      });
    }, W),
    [o, L, T, i, t == null ? void 0 : t.attrs]
  ), u = A(
    (r) => {
      r.preventDefault(), r.stopPropagation(), o && (d({
        x: 0,
        y: 0,
        w: 0,
        h: 0,
        dir: ""
      }), _(!1), I());
    },
    [o, I]
  ), H = A(() => {
    document == null || document.addEventListener("mousemove", l, !0), document == null || document.addEventListener("mouseup", u, !0);
  }, [l, u]), N = A(() => {
    document == null || document.removeEventListener("mousemove", l, !0), document == null || document.removeEventListener("mouseup", u, !0);
  }, [l, u]);
  R(() => (o ? H() : N(), () => {
    N();
  }), [o, H, N]);
  const M = F(() => new ResizeObserver(() => O()), [O]);
  return R(() => (M.observe(e.view.dom), () => {
    M.disconnect();
  }), [e.view.dom, M]), /* @__PURE__ */ n(
    lt,
    {
      className: "image-view",
      style: { ...E, width: "100%", textAlign: k },
      children: /* @__PURE__ */ C(
        "div",
        {
          "data-drag-handle": !0,
          draggable: "true",
          style: { ...E, background: "#fff" },
          className: `image-view__body ${w ? "image-view__body--focused" : ""} ${o ? "image-view__body--resizing" : ""}`,
          children: [
            /* @__PURE__ */ n(
              "img",
              {
                alt: g.alt,
                className: "image-view__body__image block",
                height: "auto",
                onClick: I,
                onLoad: B,
                src: g.src,
                style: g.style
              }
            ),
            e.view.editable && (w || o) && /* @__PURE__ */ n("div", { className: "image-resizer", children: y == null ? void 0 : y.map((r) => /* @__PURE__ */ n(
              "span",
              {
                className: `image-resizer__handler image-resizer__handler--${r}`,
                onMouseDown: (p) => P(p, r)
              },
              `image-dir-${r}`
            )) })
          ]
        }
      )
    }
  );
}
const ft = `graph TB
a-->b`;
function It() {
  const e = Z(), t = J(vt.name), { tooltipOptions: i = {}, isActive: s = void 0, upload: w } = (t == null ? void 0 : t.componentProps) ?? {}, { editorDisabled: T } = Y(s), [c, S] = m(ft), [D, y] = m(""), [o, _] = m(!1), L = ut(null), [d, k] = m(null), [g, E] = m(!0), B = A((l) => {
    l && import("mermaid").then((u) => {
      k(u.default);
    }).finally(() => E(!1));
  }, []), I = async (l) => {
    try {
      const { svg: u } = await d.render("mermaid-svg", l);
      y(u);
    } catch {
      y("");
    }
  }, O = () => {
    d.initialize({
      darkMode: !1,
      startOnLoad: !1,
      // fontFamily:'',
      fontSize: 12,
      theme: "base"
    }), I(c);
  };
  return R(() => {
    !g && d && o && O();
  }, [d, o]), R(() => {
    !g && d && o && I(c);
  }, [d && c]), /* @__PURE__ */ C(K, { onOpenChange: _, open: o, children: [
    /* @__PURE__ */ n(Q, { asChild: !0, children: /* @__PURE__ */ n(
      tt,
      {
        disabled: T,
        icon: "Mermaid",
        tooltip: "Mermaid",
        tooltipOptions: i,
        action: () => {
          T || _(!0);
        }
      }
    ) }),
    /* @__PURE__ */ C(et, { className: "richtext-z-[99999] !richtext-max-w-[1300px]", children: [
      /* @__PURE__ */ n(rt, { children: "Mermaid" }),
      /* @__PURE__ */ n("div", { ref: B, style: { height: "100%", border: "1px solid hsl(var(--border))" }, children: g ? /* @__PURE__ */ n("p", { children: "Loading..." }) : /* @__PURE__ */ n(dt, { children: /* @__PURE__ */ C("div", { className: "richtext-flex richtext-gap-[10px] richtext-rounded-[10px] richtext-p-[10px]", children: [
        /* @__PURE__ */ n(
          it,
          {
            autoFocus: !0,
            className: "richtext-flex-1",
            onChange: (l) => S(l.target.value),
            placeholder: "Text",
            required: !0,
            rows: 10,
            value: c,
            style: {
              color: "hsl(var(--foreground))"
            }
          }
        ),
        /* @__PURE__ */ n(
          "div",
          {
            className: "richtext-flex richtext-flex-1 richtext-items-center richtext-justify-center richtext-rounded-[10px] richtext-p-[10px]",
            dangerouslySetInnerHTML: { __html: D },
            ref: L,
            style: {
              height: "100%",
              borderWidth: 1,
              minHeight: 500,
              background: "#fff"
            }
          }
        )
      ] }) }) }),
      /* @__PURE__ */ n(at, { children: /* @__PURE__ */ n(st, { disabled: !d, onClick: async () => {
        if (c !== "") {
          if (c) {
            const l = L.current.querySelector("svg"), { width: u, height: H } = l.getBoundingClientRect(), N = `mermaid-${mt()}.svg`;
            let M = gt(l.outerHTML);
            if (w) {
              const r = nt(M, N);
              M = await w(r);
            }
            e == null || e.chain().focus().setMermaid(
              {
                type: "mermaid",
                src: M,
                alt: encodeURIComponent(c),
                width: u,
                height: H
              },
              !!c
            ).run();
          }
          _(!1);
        }
      }, type: "button", children: "Save changes" }) })
    ] })
  ] });
}
const vt = /* @__PURE__ */ ot.extend({
  name: "mermaid",
  //@ts-expect-error
  addOptions() {
    var e;
    return {
      ...(e = this.parent) == null ? void 0 : e.call(this),
      inline: !1,
      content: "",
      marks: "",
      group: "block",
      draggable: !1,
      selectable: !0,
      atom: !0,
      HTMLAttributes: {
        class: "mermaid"
      },
      button: ({ editor: t, t: i, extension: s }) => {
        var w;
        return {
          componentProps: {
            action: () => !0,
            isActive: () => !1,
            disabled: !1,
            editor: t,
            icon: "Mermaid",
            tooltip: i("editor.mermaid.tooltip"),
            upload: (w = s == null ? void 0 : s.options) == null ? void 0 : w.upload
          }
        };
      }
    };
  },
  addAttributes() {
    var e;
    return {
      ...(e = this.parent) == null ? void 0 : e.call(this),
      width: {
        default: null,
        parseHTML: (t) => {
          const i = t.querySelector("img"), s = i == null ? void 0 : i.getAttribute("width");
          return s ? Number.parseInt(s, 10) : 320;
        },
        renderHTML: (t) => ({
          width: t.width
        })
      },
      height: {
        default: null,
        parseHTML: (t) => {
          const i = t.querySelector("img"), s = i == null ? void 0 : i.getAttribute("height");
          return s ? Number.parseInt(s, 10) : 212;
        },
        renderHTML: (t) => ({
          height: t.height
        })
      },
      align: {
        default: "center",
        parseHTML: (t) => t.getAttribute("align"),
        renderHTML: (t) => ({
          align: t.align
        })
      }
    };
  },
  addNodeView() {
    return ct(pt);
  },
  // @ts-ignore
  addCommands() {
    return {
      setMermaid: (e, t) => ({ commands: i, editor: s }) => t ? i.insertContent({
        type: this.name,
        attrs: e
      }) : i.insertContentAt(s.state.selection.anchor, {
        type: this.name,
        attrs: e
      }),
      setAlignImageMermaid: (e) => ({ commands: t }) => t.updateAttributes(this.name, { align: e })
    };
  },
  renderHTML({ HTMLAttributes: e }) {
    const { align: t } = e;
    return [
      "div",
      // Parent element
      {
        style: t ? `text-align: ${t};` : "",
        class: "imageMermaid"
      },
      [
        "img",
        U(
          // @ts-ignore
          this.options.HTMLAttributes,
          e
        )
      ]
    ];
  },
  parseHTML() {
    return [
      {
        tag: "div[class=imageMermaid]",
        getAttrs: (e) => {
          const t = e.querySelector("img"), i = t == null ? void 0 : t.getAttribute("width"), s = t == null ? void 0 : t.getAttribute("height");
          return {
            src: t == null ? void 0 : t.getAttribute("src"),
            alt: t == null ? void 0 : t.getAttribute("alt"),
            width: i ? Number.parseInt(i, 10) : null,
            height: s ? Number.parseInt(s, 10) : null,
            align: (t == null ? void 0 : t.getAttribute("align")) || e.style.textAlign || null
          };
        }
      }
    ];
  }
});
export {
  vt as Mermaid,
  It as RichTextMermaid
};
