// reactjs-tiptap-editor@1.0.2/lib/Table-DwseaFSD.js
import { Q as ue, S as z, Y as Xe, T as W, F as v, v as L, O as Ee, u as Ue, Z as Ge, N as Y, m as I, U as qe, V as Ze, E as ze, _ as Je } from "./index-Dx-Dh7KS.js";
import { TextSelection as Ye } from "@tiptap/pm/state";
import { k as Qe, D as oe, a as le } from "./index-DQy9iBzV.js";
import { jsxs as te, jsx as E, Fragment as et } from "react/jsx-runtime";
import { h as tt, u as nt, d as ot, A as lt } from "./index-C52cuQ_Y.js";
import { useState as O } from "react";
import "./theme.js";
import { P as rt, a as st, b as it } from "./popover-BB-ZDnqk.js";
import { T as F, m as j, n as V } from "./index-DtyAGMbq.js";
let re, se;
if (typeof WeakMap < "u") {
  let t = /* @__PURE__ */ new WeakMap();
  re = (e) => t.get(e), se = (e, n) => (t.set(e, n), n);
} else {
  const t = [];
  let n = 0;
  re = (o) => {
    for (let l = 0; l < t.length; l += 2) if (t[l] == o) return t[l + 1];
  }, se = (o, l) => (n == 10 && (n = 0), t[n++] = o, t[n++] = l);
}
var C = class {
  constructor(t, e, n, o) {
    this.width = t, this.height = e, this.map = n, this.problems = o;
  }
  findCell(t) {
    for (let e = 0; e < this.map.length; e++) {
      const n = this.map[e];
      if (n != t) continue;
      const o = e % this.width, l = e / this.width | 0;
      let r = o + 1, s = l + 1;
      for (let i = 1; r < this.width && this.map[e + i] == n; i++) r++;
      for (let i = 1; s < this.height && this.map[e + this.width * i] == n; i++) s++;
      return {
        left: o,
        top: l,
        right: r,
        bottom: s
      };
    }
    throw new RangeError(`No cell with offset ${t} found`);
  }
  colCount(t) {
    for (let e = 0; e < this.map.length; e++) if (this.map[e] == t) return e % this.width;
    throw new RangeError(`No cell with offset ${t} found`);
  }
  nextCell(t, e, n) {
    const { left: o, right: l, top: r, bottom: s } = this.findCell(t);
    return e == "horiz" ? (n < 0 ? o == 0 : l == this.width) ? null : this.map[r * this.width + (n < 0 ? o - 1 : l)] : (n < 0 ? r == 0 : s == this.height) ? null : this.map[o + this.width * (n < 0 ? r - 1 : s)];
  }
  rectBetween(t, e) {
    const { left: n, right: o, top: l, bottom: r } = this.findCell(t), { left: s, right: i, top: c, bottom: a } = this.findCell(e);
    return {
      left: Math.min(n, s),
      top: Math.min(l, c),
      right: Math.max(o, i),
      bottom: Math.max(r, a)
    };
  }
  cellsInRect(t) {
    const e = [], n = {};
    for (let o = t.top; o < t.bottom; o++) for (let l = t.left; l < t.right; l++) {
      const r = o * this.width + l, s = this.map[r];
      n[s] || (n[s] = !0, !(l == t.left && l && this.map[r - 1] == s || o == t.top && o && this.map[r - this.width] == s) && e.push(s));
    }
    return e;
  }
  positionAt(t, e, n) {
    for (let o = 0, l = 0; ; o++) {
      const r = l + n.child(o).nodeSize;
      if (o == t) {
        let s = e + t * this.width;
        const i = (t + 1) * this.width;
        for (; s < i && this.map[s] < l; ) s++;
        return s == i ? r - 1 : this.map[s];
      }
      l = r;
    }
  }
  static get(t) {
    return re(t) || se(t, ct(t));
  }
};
function ct(t) {
  if (t.type.spec.tableRole != "table") throw new RangeError("Not a table node: " + t.type.name);
  const e = at(t), n = t.childCount, o = [];
  let l = 0, r = null;
  const s = [];
  for (let a = 0, d = e * n; a < d; a++) o[a] = 0;
  for (let a = 0, d = 0; a < n; a++) {
    const u = t.child(a);
    d++;
    for (let p = 0; ; p++) {
      for (; l < o.length && o[l] != 0; ) l++;
      if (p == u.childCount) break;
      const m = u.child(p), { colspan: b, rowspan: g, colwidth: y } = m.attrs;
      for (let S = 0; S < g; S++) {
        if (S + a >= n) {
          (r || (r = [])).push({
            type: "overlong_rowspan",
            pos: d,
            n: g - S
          });
          break;
        }
        const N = l + S * e;
        for (let T = 0; T < b; T++) {
          o[N + T] == 0 ? o[N + T] = d : (r || (r = [])).push({
            type: "collision",
            row: a,
            pos: d,
            n: b - T
          });
          const _ = y && y[T];
          if (_) {
            const $ = (N + T) % e * 2, ee = s[$];
            ee == null || ee != _ && s[$ + 1] == 1 ? (s[$] = _, s[$ + 1] = 1) : ee == _ && s[$ + 1]++;
          }
        }
      }
      l += b, d += m.nodeSize;
    }
    const f = (a + 1) * e;
    let h = 0;
    for (; l < f; ) o[l++] == 0 && h++;
    h && (r || (r = [])).push({
      type: "missing",
      row: a,
      n: h
    }), d++;
  }
  (e === 0 || n === 0) && (r || (r = [])).push({ type: "zero_sized" });
  const i = new C(e, n, o, r);
  let c = !1;
  for (let a = 0; !c && a < s.length; a += 2) s[a] != null && s[a + 1] < n && (c = !0);
  return c && dt(i, s, t), i;
}
function at(t) {
  let e = -1, n = !1;
  for (let o = 0; o < t.childCount; o++) {
    const l = t.child(o);
    let r = 0;
    if (n) for (let s = 0; s < o; s++) {
      const i = t.child(s);
      for (let c = 0; c < i.childCount; c++) {
        const a = i.child(c);
        s + a.attrs.rowspan > o && (r += a.attrs.colspan);
      }
    }
    for (let s = 0; s < l.childCount; s++) {
      const i = l.child(s);
      r += i.attrs.colspan, i.attrs.rowspan > 1 && (n = !0);
    }
    e == -1 ? e = r : e != r && (e = Math.max(e, r));
  }
  return e;
}
function dt(t, e, n) {
  t.problems || (t.problems = []);
  const o = {};
  for (let l = 0; l < t.map.length; l++) {
    const r = t.map[l];
    if (o[r]) continue;
    o[r] = !0;
    const s = n.nodeAt(r);
    if (!s) throw new RangeError(`No cell with offset ${r} found`);
    let i = null;
    const c = s.attrs;
    for (let a = 0; a < c.colspan; a++) {
      const d = e[(l + a) % t.width * 2];
      d != null && (!c.colwidth || c.colwidth[a] != d) && ((i || (i = ut(c)))[a] = d);
    }
    i && t.problems.unshift({
      type: "colwidth mismatch",
      pos: r,
      colwidth: i
    });
  }
}
function ut(t) {
  if (t.colwidth) return t.colwidth.slice();
  const e = [];
  for (let n = 0; n < t.colspan; n++) e.push(0);
  return e;
}
function A(t) {
  let e = t.cached.tableNodeTypes;
  if (!e) {
    e = t.cached.tableNodeTypes = {};
    for (const n in t.nodes) {
      const o = t.nodes[n], l = o.spec.tableRole;
      l && (e[l] = o);
    }
  }
  return e;
}
const H = new ue("selectingCells");
function D(t) {
  for (let e = t.depth - 1; e > 0; e--) if (t.node(e).type.spec.tableRole == "row") return t.node(0).resolve(t.before(e + 1));
  return null;
}
function ft(t) {
  for (let e = t.depth; e > 0; e--) {
    const n = t.node(e).type.spec.tableRole;
    if (n === "cell" || n === "header_cell") return t.node(e);
  }
  return null;
}
function R(t) {
  const e = t.selection.$head;
  for (let n = e.depth; n > 0; n--) if (e.node(n).type.spec.tableRole == "row") return !0;
  return !1;
}
function Q(t) {
  const e = t.selection;
  if ("$anchorCell" in e && e.$anchorCell) return e.$anchorCell.pos > e.$headCell.pos ? e.$anchorCell : e.$headCell;
  if ("node" in e && e.node && e.node.type.spec.tableRole == "cell") return e.$anchor;
  const n = D(e.$head) || ht(e.$head);
  if (n) return n;
  throw new RangeError(`No cell found around position ${e.head}`);
}
function ht(t) {
  for (let e = t.nodeAfter, n = t.pos; e; e = e.firstChild, n++) {
    const o = e.type.spec.tableRole;
    if (o == "cell" || o == "header_cell") return t.doc.resolve(n);
  }
  for (let e = t.nodeBefore, n = t.pos; e; e = e.lastChild, n--) {
    const o = e.type.spec.tableRole;
    if (o == "cell" || o == "header_cell") return t.doc.resolve(n - e.nodeSize);
  }
}
function ie(t) {
  return t.parent.type.spec.tableRole == "row" && !!t.nodeAfter;
}
function pt(t) {
  return t.node(0).resolve(t.pos + t.nodeAfter.nodeSize);
}
function fe(t, e) {
  return t.depth == e.depth && t.pos >= e.start(-1) && t.pos <= e.end(-1);
}
function He(t, e, n) {
  const o = t.node(-1), l = C.get(o), r = t.start(-1), s = l.nextCell(t.pos - r, e, n);
  return s == null ? null : t.node(0).resolve(r + s);
}
function B(t, e, n = 1) {
  const o = {
    ...t,
    colspan: t.colspan - n
  };
  return o.colwidth && (o.colwidth = o.colwidth.slice(), o.colwidth.splice(e, n), o.colwidth.some((l) => l > 0) || (o.colwidth = null)), o;
}
function Le(t, e, n = 1) {
  const o = {
    ...t,
    colspan: t.colspan + n
  };
  if (o.colwidth) {
    o.colwidth = o.colwidth.slice();
    for (let l = 0; l < n; l++) o.colwidth.splice(e, 0, 0);
  }
  return o;
}
function mt(t, e, n) {
  const o = A(e.type.schema).header_cell;
  for (let l = 0; l < t.height; l++) if (e.nodeAt(t.map[n + l * t.width]).type != o) return !1;
  return !0;
}
var w = class k extends z {
  constructor(e, n = e) {
    const o = e.node(-1), l = C.get(o), r = e.start(-1), s = l.rectBetween(e.pos - r, n.pos - r), i = e.node(0), c = l.cellsInRect(s).filter((d) => d != n.pos - r);
    c.unshift(n.pos - r);
    const a = c.map((d) => {
      const u = o.nodeAt(d);
      if (!u) throw new RangeError(`No cell with offset ${d} found`);
      const f = r + d + 1;
      return new Xe(i.resolve(f), i.resolve(f + u.content.size));
    });
    super(a[0].$from, a[0].$to, a), this.$anchorCell = e, this.$headCell = n;
  }
  map(e, n) {
    const o = e.resolve(n.map(this.$anchorCell.pos)), l = e.resolve(n.map(this.$headCell.pos));
    if (ie(o) && ie(l) && fe(o, l)) {
      const r = this.$anchorCell.node(-1) != o.node(-1);
      return r && this.isRowSelection() ? k.rowSelection(o, l) : r && this.isColSelection() ? k.colSelection(o, l) : new k(o, l);
    }
    return W.between(o, l);
  }
  content() {
    const e = this.$anchorCell.node(-1), n = C.get(e), o = this.$anchorCell.start(-1), l = n.rectBetween(this.$anchorCell.pos - o, this.$headCell.pos - o), r = {}, s = [];
    for (let c = l.top; c < l.bottom; c++) {
      const a = [];
      for (let d = c * n.width + l.left, u = l.left; u < l.right; u++, d++) {
        const f = n.map[d];
        if (r[f]) continue;
        r[f] = !0;
        const h = n.findCell(f);
        let p = e.nodeAt(f);
        if (!p) throw new RangeError(`No cell with offset ${f} found`);
        const m = l.left - h.left, b = h.right - l.right;
        if (m > 0 || b > 0) {
          let g = p.attrs;
          if (m > 0 && (g = B(g, 0, m)), b > 0 && (g = B(g, g.colspan - b, b)), h.left < l.left) {
            if (p = p.type.createAndFill(g), !p) throw new RangeError(`Could not create cell with attrs ${JSON.stringify(g)}`);
          } else p = p.type.create(g, p.content);
        }
        if (h.top < l.top || h.bottom > l.bottom) {
          const g = {
            ...p.attrs,
            rowspan: Math.min(h.bottom, l.bottom) - Math.max(h.top, l.top)
          };
          h.top < l.top ? p = p.type.createAndFill(g) : p = p.type.create(g, p.content);
        }
        a.push(p);
      }
      s.push(e.child(c).copy(v.from(a)));
    }
    const i = this.isColSelection() && this.isRowSelection() ? e : s;
    return new L(v.from(i), 1, 1);
  }
  replace(e, n = L.empty) {
    const o = e.steps.length, l = this.ranges;
    for (let s = 0; s < l.length; s++) {
      const { $from: i, $to: c } = l[s], a = e.mapping.slice(o);
      e.replace(a.map(i.pos), a.map(c.pos), s ? L.empty : n);
    }
    const r = z.findFrom(e.doc.resolve(e.mapping.slice(o).map(this.to)), -1);
    r && e.setSelection(r);
  }
  replaceWith(e, n) {
    this.replace(e, new L(v.from(n), 0, 0));
  }
  forEachCell(e) {
    const n = this.$anchorCell.node(-1), o = C.get(n), l = this.$anchorCell.start(-1), r = o.cellsInRect(o.rectBetween(this.$anchorCell.pos - l, this.$headCell.pos - l));
    for (let s = 0; s < r.length; s++) e(n.nodeAt(r[s]), l + r[s]);
  }
  isColSelection() {
    const e = this.$anchorCell.index(-1), n = this.$headCell.index(-1);
    if (Math.min(e, n) > 0) return !1;
    const o = e + this.$anchorCell.nodeAfter.attrs.rowspan, l = n + this.$headCell.nodeAfter.attrs.rowspan;
    return Math.max(o, l) == this.$headCell.node(-1).childCount;
  }
  static colSelection(e, n = e) {
    const o = e.node(-1), l = C.get(o), r = e.start(-1), s = l.findCell(e.pos - r), i = l.findCell(n.pos - r), c = e.node(0);
    return s.top <= i.top ? (s.top > 0 && (e = c.resolve(r + l.map[s.left])), i.bottom < l.height && (n = c.resolve(r + l.map[l.width * (l.height - 1) + i.right - 1]))) : (i.top > 0 && (n = c.resolve(r + l.map[i.left])), s.bottom < l.height && (e = c.resolve(r + l.map[l.width * (l.height - 1) + s.right - 1]))), new k(e, n);
  }
  isRowSelection() {
    const e = this.$anchorCell.node(-1), n = C.get(e), o = this.$anchorCell.start(-1), l = n.colCount(this.$anchorCell.pos - o), r = n.colCount(this.$headCell.pos - o);
    if (Math.min(l, r) > 0) return !1;
    const s = l + this.$anchorCell.nodeAfter.attrs.colspan, i = r + this.$headCell.nodeAfter.attrs.colspan;
    return Math.max(s, i) == n.width;
  }
  eq(e) {
    return e instanceof k && e.$anchorCell.pos == this.$anchorCell.pos && e.$headCell.pos == this.$headCell.pos;
  }
  static rowSelection(e, n = e) {
    const o = e.node(-1), l = C.get(o), r = e.start(-1), s = l.findCell(e.pos - r), i = l.findCell(n.pos - r), c = e.node(0);
    return s.left <= i.left ? (s.left > 0 && (e = c.resolve(r + l.map[s.top * l.width])), i.right < l.width && (n = c.resolve(r + l.map[l.width * (i.top + 1) - 1]))) : (i.left > 0 && (n = c.resolve(r + l.map[i.top * l.width])), s.right < l.width && (e = c.resolve(r + l.map[l.width * (s.top + 1) - 1]))), new k(e, n);
  }
  toJSON() {
    return {
      type: "cell",
      anchor: this.$anchorCell.pos,
      head: this.$headCell.pos
    };
  }
  static fromJSON(e, n) {
    return new k(e.resolve(n.anchor), e.resolve(n.head));
  }
  static create(e, n, o = n) {
    return new k(e.resolve(n), e.resolve(o));
  }
  getBookmark() {
    return new gt(this.$anchorCell.pos, this.$headCell.pos);
  }
};
w.prototype.visible = !1;
z.jsonID("cell", w);
var gt = class $e {
  constructor(e, n) {
    this.anchor = e, this.head = n;
  }
  map(e) {
    return new $e(e.map(this.anchor), e.map(this.head));
  }
  resolve(e) {
    const n = e.resolve(this.anchor), o = e.resolve(this.head);
    return n.parent.type.spec.tableRole == "row" && o.parent.type.spec.tableRole == "row" && n.index() < n.parent.childCount && o.index() < o.parent.childCount && fe(n, o) ? new w(n, o) : z.near(o, 1);
  }
};
function bt(t) {
  if (!(t.selection instanceof w)) return null;
  const e = [];
  return t.selection.forEachCell((n, o) => {
    e.push(le.node(o, o + n.nodeSize, { class: "selectedCell" }));
  }), oe.create(t.doc, e);
}
function wt({ $from: t, $to: e }) {
  if (t.pos == e.pos || t.pos < e.pos - 6) return !1;
  let n = t.pos, o = e.pos, l = t.depth;
  for (; l >= 0 && !(t.after(l + 1) < t.end(l)); l--, n++) ;
  for (let r = e.depth; r >= 0 && !(e.before(r + 1) > e.start(r)); r--, o--) ;
  return n == o && /row|table/.test(t.node(l).type.spec.tableRole);
}
function Ct({ $from: t, $to: e }) {
  let n, o;
  for (let l = t.depth; l > 0; l--) {
    const r = t.node(l);
    if (r.type.spec.tableRole === "cell" || r.type.spec.tableRole === "header_cell") {
      n = r;
      break;
    }
  }
  for (let l = e.depth; l > 0; l--) {
    const r = e.node(l);
    if (r.type.spec.tableRole === "cell" || r.type.spec.tableRole === "header_cell") {
      o = r;
      break;
    }
  }
  return n !== o && e.parentOffset === 0;
}
function yt(t, e, n) {
  const o = (e || t).selection, l = (e || t).doc;
  let r, s;
  if (o instanceof Ue && (s = o.node.type.spec.tableRole)) {
    if (s == "cell" || s == "header_cell") r = w.create(l, o.from);
    else if (s == "row") {
      const i = l.resolve(o.from + 1);
      r = w.rowSelection(i, i);
    } else if (!n) {
      const i = C.get(o.node), c = o.from + 1, a = c + i.map[i.width * i.height - 1];
      r = w.create(l, c + 1, a);
    }
  } else o instanceof W && wt(o) ? r = W.create(l, o.from) : o instanceof W && Ct(o) && (r = W.create(l, o.$from.start(), o.$from.end()));
  return r && (e || (e = t.tr)).setSelection(r), e;
}
const At = new ue("fix-tables");
function We(t, e, n, o) {
  const l = t.childCount, r = e.childCount;
  e: for (let s = 0, i = 0; s < r; s++) {
    const c = e.child(s);
    for (let a = i, d = Math.min(l, s + 3); a < d; a++) if (t.child(a) == c) {
      i = a + 1, n += c.nodeSize;
      continue e;
    }
    o(c, n), i < l && t.child(i).sameMarkup(c) ? We(t.child(i), c, n + 1, o) : c.nodesBetween(0, c.content.size, o, n + 1), n += c.nodeSize;
  }
}
function De(t, e) {
  let n;
  const o = (l, r) => {
    l.type.spec.tableRole == "table" && (n = St(t, l, r, n));
  };
  return e ? e.doc != t.doc && We(e.doc, t.doc, 0, o) : t.doc.descendants(o), n;
}
function St(t, e, n, o) {
  const l = C.get(e);
  if (!l.problems) return o;
  o || (o = t.tr);
  const r = [];
  for (let c = 0; c < l.height; c++) r.push(0);
  for (let c = 0; c < l.problems.length; c++) {
    const a = l.problems[c];
    if (a.type == "collision") {
      const d = e.nodeAt(a.pos);
      if (!d) continue;
      const u = d.attrs;
      for (let f = 0; f < u.rowspan; f++) r[a.row + f] += a.n;
      o.setNodeMarkup(o.mapping.map(n + 1 + a.pos), null, B(u, u.colspan - a.n, a.n));
    } else if (a.type == "missing") r[a.row] += a.n;
    else if (a.type == "overlong_rowspan") {
      const d = e.nodeAt(a.pos);
      if (!d) continue;
      o.setNodeMarkup(o.mapping.map(n + 1 + a.pos), null, {
        ...d.attrs,
        rowspan: d.attrs.rowspan - a.n
      });
    } else if (a.type == "colwidth mismatch") {
      const d = e.nodeAt(a.pos);
      if (!d) continue;
      o.setNodeMarkup(o.mapping.map(n + 1 + a.pos), null, {
        ...d.attrs,
        colwidth: a.colwidth
      });
    } else if (a.type == "zero_sized") {
      const d = o.mapping.map(n);
      o.delete(d, d + e.nodeSize);
    }
  }
  let s, i;
  for (let c = 0; c < r.length; c++) r[c] && (s == null && (s = c), i = c);
  for (let c = 0, a = n + 1; c < l.height; c++) {
    const d = e.child(c), u = a + d.nodeSize, f = r[c];
    if (f > 0) {
      let h = "cell";
      d.firstChild && (h = d.firstChild.type.spec.tableRole);
      const p = [];
      for (let b = 0; b < f; b++) {
        const g = A(t.schema)[h].createAndFill();
        g && p.push(g);
      }
      const m = (c == 0 || s == c - 1) && i == c ? a + 1 : u - 1;
      o.insert(o.mapping.map(m), p);
    }
    a = u;
  }
  return o.setMeta(At, { fixTables: !0 });
}
function M(t) {
  const e = t.selection, n = Q(t), o = n.node(-1), l = n.start(-1), r = C.get(o);
  return {
    ...e instanceof w ? r.rectBetween(e.$anchorCell.pos - l, e.$headCell.pos - l) : r.findCell(n.pos - l),
    tableStart: l,
    map: r,
    table: o
  };
}
function Be(t, { map: e, tableStart: n, table: o }, l) {
  let r = l > 0 ? -1 : 0;
  mt(e, o, l + r) && (r = l == 0 || l == e.width ? null : 0);
  for (let s = 0; s < e.height; s++) {
    const i = s * e.width + l;
    if (l > 0 && l < e.width && e.map[i - 1] == e.map[i]) {
      const c = e.map[i], a = o.nodeAt(c);
      t.setNodeMarkup(t.mapping.map(n + c), null, Le(a.attrs, l - e.colCount(c))), s += a.attrs.rowspan - 1;
    } else {
      const c = r == null ? A(o.type.schema).cell : o.nodeAt(e.map[i + r]).type, a = e.positionAt(s, l, o);
      t.insert(t.mapping.map(n + a), c.createAndFill());
    }
  }
  return t;
}
function xt(t, e) {
  if (!R(t)) return !1;
  if (e) {
    const n = M(t);
    e(Be(t.tr, n, n.left));
  }
  return !0;
}
function Tt(t, e) {
  if (!R(t)) return !1;
  if (e) {
    const n = M(t);
    e(Be(t.tr, n, n.right));
  }
  return !0;
}
function Rt(t, { map: e, table: n, tableStart: o }, l) {
  const r = t.mapping.maps.length;
  for (let s = 0; s < e.height; ) {
    const i = s * e.width + l, c = e.map[i], a = n.nodeAt(c), d = a.attrs;
    if (l > 0 && e.map[i - 1] == c || l < e.width - 1 && e.map[i + 1] == c) t.setNodeMarkup(t.mapping.slice(r).map(o + c), null, B(d, l - e.colCount(c)));
    else {
      const u = t.mapping.slice(r).map(o + c);
      t.delete(u, u + a.nodeSize);
    }
    s += d.rowspan;
  }
}
function vt(t, e) {
  if (!R(t)) return !1;
  if (e) {
    const n = M(t), o = t.tr;
    if (n.left == 0 && n.right == n.map.width) return !1;
    for (let l = n.right - 1; Rt(o, n, l), l != n.left; l--) {
      const r = n.tableStart ? o.doc.nodeAt(n.tableStart - 1) : o.doc;
      if (!r) throw new RangeError("No table found");
      n.table = r, n.map = C.get(r);
    }
    e(o);
  }
  return !0;
}
function Mt(t, e, n) {
  var o;
  const l = A(e.type.schema).header_cell;
  for (let r = 0; r < t.width; r++) if (((o = e.nodeAt(t.map[r + n * t.width])) === null || o === void 0 ? void 0 : o.type) != l) return !1;
  return !0;
}
function _e(t, { map: e, tableStart: n, table: o }, l) {
  let r = n;
  for (let a = 0; a < l; a++) r += o.child(a).nodeSize;
  const s = [];
  let i = l > 0 ? -1 : 0;
  Mt(e, o, l + i) && (i = l == 0 || l == e.height ? null : 0);
  for (let a = 0, d = e.width * l; a < e.width; a++, d++) if (l > 0 && l < e.height && e.map[d] == e.map[d - e.width]) {
    const u = e.map[d], f = o.nodeAt(u).attrs;
    t.setNodeMarkup(n + u, null, {
      ...f,
      rowspan: f.rowspan + 1
    }), a += f.colspan - 1;
  } else {
    var c;
    const u = i == null ? A(o.type.schema).cell : (c = o.nodeAt(e.map[d + i * e.width])) === null || c === void 0 ? void 0 : c.type, f = u == null ? void 0 : u.createAndFill();
    f && s.push(f);
  }
  return t.insert(r, A(o.type.schema).row.create(null, s)), t;
}
function Nt(t, e) {
  if (!R(t)) return !1;
  if (e) {
    const n = M(t);
    e(_e(t.tr, n, n.top));
  }
  return !0;
}
function kt(t, e) {
  if (!R(t)) return !1;
  if (e) {
    const n = M(t);
    e(_e(t.tr, n, n.bottom));
  }
  return !0;
}
function Et(t, { map: e, table: n, tableStart: o }, l) {
  let r = 0;
  for (let a = 0; a < l; a++) r += n.child(a).nodeSize;
  const s = r + n.child(l).nodeSize, i = t.mapping.maps.length;
  t.delete(r + o, s + o);
  const c = /* @__PURE__ */ new Set();
  for (let a = 0, d = l * e.width; a < e.width; a++, d++) {
    const u = e.map[d];
    if (!c.has(u)) {
      if (c.add(u), l > 0 && u == e.map[d - e.width]) {
        const f = n.nodeAt(u).attrs;
        t.setNodeMarkup(t.mapping.slice(i).map(u + o), null, {
          ...f,
          rowspan: f.rowspan - 1
        }), a += f.colspan - 1;
      } else if (l < e.height && u == e.map[d + e.width]) {
        const f = n.nodeAt(u), h = f.attrs, p = f.type.create({
          ...h,
          rowspan: f.attrs.rowspan - 1
        }, f.content), m = e.positionAt(l + 1, a, n);
        t.insert(t.mapping.slice(i).map(o + m), p), a += h.colspan - 1;
      }
    }
  }
}
function zt(t, e) {
  if (!R(t)) return !1;
  if (e) {
    const n = M(t), o = t.tr;
    if (n.top == 0 && n.bottom == n.map.height) return !1;
    for (let l = n.bottom - 1; Et(o, n, l), l != n.top; l--) {
      const r = n.tableStart ? o.doc.nodeAt(n.tableStart - 1) : o.doc;
      if (!r) throw new RangeError("No table found");
      n.table = r, n.map = C.get(n.table);
    }
    e(o);
  }
  return !0;
}
function he(t) {
  const e = t.content;
  return e.childCount == 1 && e.child(0).isTextblock && e.child(0).childCount == 0;
}
function Ht({ width: t, height: e, map: n }, o) {
  let l = o.top * t + o.left, r = l, s = (o.bottom - 1) * t + o.left, i = l + (o.right - o.left - 1);
  for (let c = o.top; c < o.bottom; c++) {
    if (o.left > 0 && n[r] == n[r - 1] || o.right < t && n[i] == n[i + 1]) return !0;
    r += t, i += t;
  }
  for (let c = o.left; c < o.right; c++) {
    if (o.top > 0 && n[l] == n[l - t] || o.bottom < e && n[s] == n[s + t]) return !0;
    l++, s++;
  }
  return !1;
}
function pe(t, e) {
  const n = t.selection;
  if (!(n instanceof w) || n.$anchorCell.pos == n.$headCell.pos) return !1;
  const o = M(t), { map: l } = o;
  if (Ht(l, o)) return !1;
  if (e) {
    const r = t.tr, s = {};
    let i = v.empty, c, a;
    for (let d = o.top; d < o.bottom; d++) for (let u = o.left; u < o.right; u++) {
      const f = l.map[d * l.width + u], h = o.table.nodeAt(f);
      if (!(s[f] || !h))
        if (s[f] = !0, c == null)
          c = f, a = h;
        else {
          he(h) || (i = i.append(h.content));
          const p = r.mapping.map(f + o.tableStart);
          r.delete(p, p + h.nodeSize);
        }
    }
    if (c == null || a == null) return !0;
    if (r.setNodeMarkup(c + o.tableStart, null, {
      ...Le(a.attrs, a.attrs.colspan, o.right - o.left - a.attrs.colspan),
      rowspan: o.bottom - o.top
    }), i.size > 0) {
      const d = c + 1 + a.content.size, u = he(a) ? c + 1 : d;
      r.replaceWith(u + o.tableStart, d + o.tableStart, i);
    }
    r.setSelection(new w(r.doc.resolve(c + o.tableStart))), e(r);
  }
  return !0;
}
function me(t, e) {
  const n = A(t.schema);
  return Lt(({ node: o }) => n[o.type.spec.tableRole])(t, e);
}
function Lt(t) {
  return (e, n) => {
    const o = e.selection;
    let l, r;
    if (o instanceof w) {
      if (o.$anchorCell.pos != o.$headCell.pos) return !1;
      l = o.$anchorCell.nodeAfter, r = o.$anchorCell.pos;
    } else {
      var s;
      if (l = ft(o.$from), !l) return !1;
      r = (s = D(o.$from)) === null || s === void 0 ? void 0 : s.pos;
    }
    if (l == null || r == null || l.attrs.colspan == 1 && l.attrs.rowspan == 1) return !1;
    if (n) {
      let i = l.attrs;
      const c = [], a = i.colwidth;
      i.rowspan > 1 && (i = {
        ...i,
        rowspan: 1
      }), i.colspan > 1 && (i = {
        ...i,
        colspan: 1
      });
      const d = M(e), u = e.tr;
      for (let h = 0; h < d.right - d.left; h++) c.push(a ? {
        ...i,
        colwidth: a && a[h] ? [a[h]] : null
      } : i);
      let f;
      for (let h = d.top; h < d.bottom; h++) {
        let p = d.map.positionAt(h, d.left, d.table);
        h == d.top && (p += l.nodeSize);
        for (let m = d.left, b = 0; m < d.right; m++, b++)
          m == d.left && h == d.top || u.insert(f = u.mapping.map(p + d.tableStart, 1), t({
            node: l,
            row: h,
            col: m
          }).createAndFill(c[b]));
      }
      u.setNodeMarkup(r, t({
        node: l,
        row: d.top,
        col: d.left
      }), c[0]), o instanceof w && u.setSelection(new w(u.doc.resolve(o.$anchorCell.pos), f ? u.doc.resolve(f) : void 0)), n(u);
    }
    return !0;
  };
}
function $t(t, e) {
  return function(n, o) {
    if (!R(n)) return !1;
    const l = Q(n);
    if (l.nodeAfter.attrs[t] === e) return !1;
    if (o) {
      const r = n.tr;
      n.selection instanceof w ? n.selection.forEachCell((s, i) => {
        s.attrs[t] !== e && r.setNodeMarkup(i, null, {
          ...s.attrs,
          [t]: e
        });
      }) : r.setNodeMarkup(l.pos, null, {
        ...l.nodeAfter.attrs,
        [t]: e
      }), o(r);
    }
    return !0;
  };
}
function Wt(t) {
  return function(e, n) {
    if (!R(e)) return !1;
    if (n) {
      const o = A(e.schema), l = M(e), r = e.tr, s = l.map.cellsInRect(t == "column" ? {
        left: l.left,
        top: 0,
        right: l.right,
        bottom: l.map.height
      } : t == "row" ? {
        left: 0,
        top: l.top,
        right: l.map.width,
        bottom: l.bottom
      } : l), i = s.map((c) => l.table.nodeAt(c));
      for (let c = 0; c < s.length; c++) i[c].type == o.header_cell && r.setNodeMarkup(l.tableStart + s[c], o.cell, i[c].attrs);
      if (r.steps.length === 0) for (let c = 0; c < s.length; c++) r.setNodeMarkup(l.tableStart + s[c], o.header_cell, i[c].attrs);
      n(r);
    }
    return !0;
  };
}
function ge(t, e, n) {
  const o = e.map.cellsInRect({
    left: 0,
    top: 0,
    right: t == "row" ? e.map.width : 1,
    bottom: t == "column" ? e.map.height : 1
  });
  for (let l = 0; l < o.length; l++) {
    const r = e.table.nodeAt(o[l]);
    if (r && r.type !== n.header_cell) return !1;
  }
  return !0;
}
function P(t, e) {
  return e = e || { useDeprecatedLogic: !1 }, e.useDeprecatedLogic ? Wt(t) : function(n, o) {
    if (!R(n)) return !1;
    if (o) {
      const l = A(n.schema), r = M(n), s = n.tr, i = ge("row", r, l), c = ge("column", r, l), a = (t === "column" ? i : t === "row" && c) ? 1 : 0, d = t == "column" ? {
        left: 0,
        top: a,
        right: 1,
        bottom: r.map.height
      } : t == "row" ? {
        left: a,
        top: 0,
        right: r.map.width,
        bottom: 1
      } : r, u = t == "column" ? c ? l.cell : l.header_cell : t == "row" ? i ? l.cell : l.header_cell : l.cell;
      r.map.cellsInRect(d).forEach((f) => {
        const h = f + r.tableStart, p = s.doc.nodeAt(h);
        p && s.setNodeMarkup(h, u, p.attrs);
      }), o(s);
    }
    return !0;
  };
}
P("row", { useDeprecatedLogic: !0 });
P("column", { useDeprecatedLogic: !0 });
const Dt = P("cell", { useDeprecatedLogic: !0 });
function Bt(t, e) {
  if (e < 0) {
    const n = t.nodeBefore;
    if (n) return t.pos - n.nodeSize;
    for (let o = t.index(-1) - 1, l = t.before(); o >= 0; o--) {
      const r = t.node(-1).child(o), s = r.lastChild;
      if (s) return l - 1 - s.nodeSize;
      l -= r.nodeSize;
    }
  } else {
    if (t.index() < t.parent.childCount - 1) return t.pos + t.nodeAfter.nodeSize;
    const n = t.node(-1);
    for (let o = t.indexAfter(-1), l = t.after(); o < n.childCount; o++) {
      const r = n.child(o);
      if (r.childCount) return l + 1;
      l += r.nodeSize;
    }
  }
  return null;
}
function be(t) {
  return function(e, n) {
    if (!R(e)) return !1;
    const o = Bt(Q(e), t);
    if (o == null) return !1;
    if (n) {
      const l = e.doc.resolve(o);
      n(e.tr.setSelection(W.between(l, pt(l))).scrollIntoView());
    }
    return !0;
  };
}
function _t(t, e) {
  const n = t.selection.$anchor;
  for (let o = n.depth; o > 0; o--) if (n.node(o).type.spec.tableRole == "table")
    return e && e(t.tr.delete(n.before(o), n.after(o)).scrollIntoView()), !0;
  return !1;
}
function K(t, e) {
  const n = t.selection;
  if (!(n instanceof w)) return !1;
  if (e) {
    const o = t.tr, l = A(t.schema).cell.createAndFill().content;
    n.forEachCell((r, s) => {
      r.content.eq(l) || o.replace(o.mapping.map(s + 1), o.mapping.map(s + r.nodeSize - 1), new L(l, 0, 0));
    }), o.docChanged && e(o);
  }
  return !0;
}
function It(t) {
  if (t.size === 0) return null;
  let { content: e, openStart: n, openEnd: o } = t;
  for (; e.childCount == 1 && (n > 0 && o > 0 || e.child(0).type.spec.tableRole == "table"); )
    n--, o--, e = e.child(0).content;
  const l = e.child(0), r = l.type.spec.tableRole, s = l.type.schema, i = [];
  if (r == "row") for (let c = 0; c < e.childCount; c++) {
    let a = e.child(c).content;
    const d = c ? 0 : Math.max(0, n - 1), u = c < e.childCount - 1 ? 0 : Math.max(0, o - 1);
    (d || u) && (a = ce(A(s).row, new L(a, d, u)).content), i.push(a);
  }
  else if (r == "cell" || r == "header_cell") i.push(n || o ? ce(A(s).row, new L(e, n, o)).content : e);
  else return null;
  return Pt(s, i);
}
function Pt(t, e) {
  const n = [];
  for (let l = 0; l < e.length; l++) {
    const r = e[l];
    for (let s = r.childCount - 1; s >= 0; s--) {
      const { rowspan: i, colspan: c } = r.child(s).attrs;
      for (let a = l; a < l + i; a++) n[a] = (n[a] || 0) + c;
    }
  }
  let o = 0;
  for (let l = 0; l < n.length; l++) o = Math.max(o, n[l]);
  for (let l = 0; l < n.length; l++)
    if (l >= e.length && e.push(v.empty), n[l] < o) {
      const r = A(t).cell.createAndFill(), s = [];
      for (let i = n[l]; i < o; i++) s.push(r);
      e[l] = e[l].append(v.from(s));
    }
  return {
    height: e.length,
    width: o,
    rows: e
  };
}
function ce(t, e) {
  const n = t.createAndFill();
  return new Ge(n).replace(0, n.content.size, e).doc;
}
function Ot({ width: t, height: e, rows: n }, o, l) {
  if (t != o) {
    const r = [], s = [];
    for (let i = 0; i < n.length; i++) {
      const c = n[i], a = [];
      for (let d = r[i] || 0, u = 0; d < o; u++) {
        let f = c.child(u % c.childCount);
        d + f.attrs.colspan > o && (f = f.type.createChecked(B(f.attrs, f.attrs.colspan, d + f.attrs.colspan - o), f.content)), a.push(f), d += f.attrs.colspan;
        for (let h = 1; h < f.attrs.rowspan; h++) r[i + h] = (r[i + h] || 0) + f.attrs.colspan;
      }
      s.push(v.from(a));
    }
    n = s, t = o;
  }
  if (e != l) {
    const r = [];
    for (let s = 0, i = 0; s < l; s++, i++) {
      const c = [], a = n[i % e];
      for (let d = 0; d < a.childCount; d++) {
        let u = a.child(d);
        s + u.attrs.rowspan > l && (u = u.type.create({
          ...u.attrs,
          rowspan: Math.max(1, l - u.attrs.rowspan)
        }, u.content)), c.push(u);
      }
      r.push(v.from(c));
    }
    n = r, e = l;
  }
  return {
    width: t,
    height: e,
    rows: n
  };
}
function Ft(t, e, n, o, l, r, s) {
  const i = t.doc.type.schema, c = A(i);
  let a, d;
  if (l > e.width) for (let u = 0, f = 0; u < e.height; u++) {
    const h = n.child(u);
    f += h.nodeSize;
    const p = [];
    let m;
    h.lastChild == null || h.lastChild.type == c.cell ? m = a || (a = c.cell.createAndFill()) : m = d || (d = c.header_cell.createAndFill());
    for (let b = e.width; b < l; b++) p.push(m);
    t.insert(t.mapping.slice(s).map(f - 1 + o), p);
  }
  if (r > e.height) {
    const u = [];
    for (let p = 0, m = (e.height - 1) * e.width; p < Math.max(e.width, l); p++) {
      const b = p >= e.width ? !1 : n.nodeAt(e.map[m + p]).type == c.header_cell;
      u.push(b ? d || (d = c.header_cell.createAndFill()) : a || (a = c.cell.createAndFill()));
    }
    const f = c.row.create(null, v.from(u)), h = [];
    for (let p = e.height; p < r; p++) h.push(f);
    t.insert(t.mapping.slice(s).map(o + n.nodeSize - 2), h);
  }
  return !!(a || d);
}
function we(t, e, n, o, l, r, s, i) {
  if (s == 0 || s == e.height) return !1;
  let c = !1;
  for (let a = l; a < r; a++) {
    const d = s * e.width + a, u = e.map[d];
    if (e.map[d - e.width] == u) {
      c = !0;
      const f = n.nodeAt(u), { top: h, left: p } = e.findCell(u);
      t.setNodeMarkup(t.mapping.slice(i).map(u + o), null, {
        ...f.attrs,
        rowspan: s - h
      }), t.insert(t.mapping.slice(i).map(e.positionAt(s, p, n)), f.type.createAndFill({
        ...f.attrs,
        rowspan: h + f.attrs.rowspan - s
      })), a += f.attrs.colspan - 1;
    }
  }
  return c;
}
function Ce(t, e, n, o, l, r, s, i) {
  if (s == 0 || s == e.width) return !1;
  let c = !1;
  for (let a = l; a < r; a++) {
    const d = a * e.width + s, u = e.map[d];
    if (e.map[d - 1] == u) {
      c = !0;
      const f = n.nodeAt(u), h = e.colCount(u), p = t.mapping.slice(i).map(u + o);
      t.setNodeMarkup(p, null, B(f.attrs, s - h, f.attrs.colspan - (s - h))), t.insert(p + f.nodeSize, f.type.createAndFill(B(f.attrs, 0, s - h))), a += f.attrs.rowspan - 1;
    }
  }
  return c;
}
function ye(t, e, n, o, l) {
  let r = n ? t.doc.nodeAt(n - 1) : t.doc;
  if (!r) throw new Error("No table found");
  let s = C.get(r);
  const { top: i, left: c } = o, a = c + l.width, d = i + l.height, u = t.tr;
  let f = 0;
  function h() {
    if (r = n ? u.doc.nodeAt(n - 1) : u.doc, !r) throw new Error("No table found");
    s = C.get(r), f = u.mapping.maps.length;
  }
  Ft(u, s, r, n, a, d, f) && h(), we(u, s, r, n, c, a, i, f) && h(), we(u, s, r, n, c, a, d, f) && h(), Ce(u, s, r, n, i, d, c, f) && h(), Ce(u, s, r, n, i, d, a, f) && h();
  for (let p = i; p < d; p++) {
    const m = s.positionAt(p, c, r), b = s.positionAt(p, a, r);
    u.replace(u.mapping.slice(f).map(m + n), u.mapping.slice(f).map(b + n), new L(l.rows[p - i], 0, 0));
  }
  h(), u.setSelection(new w(u.doc.resolve(n + s.positionAt(i, c, r)), u.doc.resolve(n + s.positionAt(d - 1, a - 1, r)))), e(u);
}
const jt = Qe({
  ArrowLeft: X("horiz", -1),
  ArrowRight: X("horiz", 1),
  ArrowUp: X("vert", -1),
  ArrowDown: X("vert", 1),
  "Shift-ArrowLeft": U("horiz", -1),
  "Shift-ArrowRight": U("horiz", 1),
  "Shift-ArrowUp": U("vert", -1),
  "Shift-ArrowDown": U("vert", 1),
  Backspace: K,
  "Mod-Backspace": K,
  Delete: K,
  "Mod-Delete": K
});
function q(t, e, n) {
  return n.eq(t.selection) ? !1 : (e && e(t.tr.setSelection(n).scrollIntoView()), !0);
}
function X(t, e) {
  return (n, o, l) => {
    if (!l) return !1;
    const r = n.selection;
    if (r instanceof w) return q(n, o, z.near(r.$headCell, e));
    if (t != "horiz" && !r.empty) return !1;
    const s = Ie(l, t, e);
    if (s == null) return !1;
    if (t == "horiz") return q(n, o, z.near(n.doc.resolve(r.head + e), e));
    {
      const i = n.doc.resolve(s), c = He(i, t, e);
      let a;
      return c ? a = z.near(c, 1) : e < 0 ? a = z.near(n.doc.resolve(i.before(-1)), -1) : a = z.near(n.doc.resolve(i.after(-1)), 1), q(n, o, a);
    }
  };
}
function U(t, e) {
  return (n, o, l) => {
    if (!l) return !1;
    const r = n.selection;
    let s;
    if (r instanceof w) s = r;
    else {
      const c = Ie(l, t, e);
      if (c == null) return !1;
      s = new w(n.doc.resolve(c));
    }
    const i = He(s.$headCell, t, e);
    return i ? q(n, o, new w(s.$anchorCell, i)) : !1;
  };
}
function Vt(t, e) {
  const n = t.state.doc, o = D(n.resolve(e));
  return o ? (t.dispatch(t.state.tr.setSelection(new w(o))), !0) : !1;
}
function Kt(t, e, n) {
  if (!R(t.state)) return !1;
  let o = It(n);
  const l = t.state.selection;
  if (l instanceof w) {
    o || (o = {
      width: 1,
      height: 1,
      rows: [v.from(ce(A(t.state.schema).cell, n))]
    });
    const r = l.$anchorCell.node(-1), s = l.$anchorCell.start(-1), i = C.get(r).rectBetween(l.$anchorCell.pos - s, l.$headCell.pos - s);
    return o = Ot(o, i.right - i.left, i.bottom - i.top), ye(t.state, t.dispatch, s, i, o), !0;
  } else if (o) {
    const r = Q(t.state), s = r.start(-1);
    return ye(t.state, t.dispatch, s, C.get(r.node(-1)).findCell(r.pos - s), o), !0;
  } else return !1;
}
function Xt(t, e) {
  var n;
  if (e.button != 0 || e.ctrlKey || e.metaKey) return;
  const o = Ae(t, e.target);
  let l;
  if (e.shiftKey && t.state.selection instanceof w)
    r(t.state.selection.$anchorCell, e), e.preventDefault();
  else if (e.shiftKey && o && (l = D(t.state.selection.$anchor)) != null && ((n = ne(t, e)) === null || n === void 0 ? void 0 : n.pos) != l.pos)
    r(l, e), e.preventDefault();
  else if (!o) return;
  function r(c, a) {
    let d = ne(t, a);
    const u = H.getState(t.state) == null;
    if (!d || !fe(c, d)) if (u) d = c;
    else return;
    const f = new w(c, d);
    if (u || !t.state.selection.eq(f)) {
      const h = t.state.tr.setSelection(f);
      u && h.setMeta(H, c.pos), t.dispatch(h);
    }
  }
  function s() {
    t.root.removeEventListener("mouseup", s), t.root.removeEventListener("dragstart", s), t.root.removeEventListener("mousemove", i), H.getState(t.state) != null && t.dispatch(t.state.tr.setMeta(H, -1));
  }
  function i(c) {
    const a = c, d = H.getState(t.state);
    let u;
    if (d != null) u = t.state.doc.resolve(d);
    else if (Ae(t, a.target) != o && (u = ne(t, e), !u))
      return s();
    u && r(u, a);
  }
  t.root.addEventListener("mouseup", s), t.root.addEventListener("dragstart", s), t.root.addEventListener("mousemove", i);
}
function Ie(t, e, n) {
  if (!(t.state.selection instanceof W)) return null;
  const { $head: o } = t.state.selection;
  for (let l = o.depth - 1; l >= 0; l--) {
    const r = o.node(l);
    if ((n < 0 ? o.index(l) : o.indexAfter(l)) != (n < 0 ? 0 : r.childCount)) return null;
    if (r.type.spec.tableRole == "cell" || r.type.spec.tableRole == "header_cell") {
      const s = o.before(l), i = e == "vert" ? n > 0 ? "down" : "up" : n > 0 ? "right" : "left";
      return t.endOfTextblock(i) ? s : null;
    }
  }
  return null;
}
function Ae(t, e) {
  for (; e && e != t.dom; e = e.parentNode) if (e.nodeName == "TD" || e.nodeName == "TH") return e;
  return null;
}
function ne(t, e) {
  const n = t.posAtCoords({
    left: e.clientX,
    top: e.clientY
  });
  if (!n) return null;
  let { inside: o, pos: l } = n;
  return o >= 0 && D(t.state.doc.resolve(o)) || D(t.state.doc.resolve(l));
}
var Ut = class {
  constructor(e, n) {
    this.node = e, this.defaultCellMinWidth = n, this.dom = document.createElement("div"), this.dom.className = "tableWrapper", this.table = this.dom.appendChild(document.createElement("table")), this.table.style.setProperty("--default-cell-min-width", `${n}px`), this.colgroup = this.table.appendChild(document.createElement("colgroup")), ae(e, this.colgroup, this.table, n), this.contentDOM = this.table.appendChild(document.createElement("tbody"));
  }
  update(e) {
    return e.type != this.node.type ? !1 : (this.node = e, ae(e, this.colgroup, this.table, this.defaultCellMinWidth), !0);
  }
  ignoreMutation(e) {
    return e.type == "attributes" && (e.target == this.table || this.colgroup.contains(e.target));
  }
};
function ae(t, e, n, o, l, r) {
  let s = 0, i = !0, c = e.firstChild;
  const a = t.firstChild;
  if (a) {
    for (let u = 0, f = 0; u < a.childCount; u++) {
      const { colspan: h, colwidth: p } = a.child(u).attrs;
      for (let m = 0; m < h; m++, f++) {
        const b = l == f ? r : p && p[m], g = b ? b + "px" : "";
        if (s += b || o, b || (i = !1), c)
          c.style.width != g && (c.style.width = g), c = c.nextSibling;
        else {
          const y = document.createElement("col");
          y.style.width = g, e.appendChild(y);
        }
      }
    }
    for (; c; ) {
      var d;
      const u = c.nextSibling;
      (d = c.parentNode) === null || d === void 0 || d.removeChild(c), c = u;
    }
    i ? (n.style.width = s + "px", n.style.minWidth = "") : (n.style.width = "", n.style.minWidth = s + "px");
  }
}
const x = new ue("tableColumnResizing");
function Gt({ handleWidth: t = 5, cellMinWidth: e = 25, defaultCellMinWidth: n = 100, View: o = Ut, lastColumnResizable: l = !0 } = {}) {
  const r = new Ee({
    key: x,
    state: {
      init(s, i) {
        var c;
        const a = (c = r.spec) === null || c === void 0 || (c = c.props) === null || c === void 0 ? void 0 : c.nodeViews, d = A(i.schema).table.name;
        return o && a && (a[d] = (u, f) => new o(u, n, f)), new qt(-1, !1);
      },
      apply(s, i) {
        return i.apply(s);
      }
    },
    props: {
      attributes: (s) => {
        const i = x.getState(s);
        return i && i.activeHandle > -1 ? { class: "resize-cursor" } : {};
      },
      handleDOMEvents: {
        mousemove: (s, i) => {
          Zt(s, i, t, l);
        },
        mouseleave: (s) => {
          Jt(s);
        },
        mousedown: (s, i) => {
          Yt(s, i, e, n);
        }
      },
      decorations: (s) => {
        const i = x.getState(s);
        if (i && i.activeHandle > -1) return on(s, i.activeHandle);
      },
      nodeViews: {}
    }
  });
  return r;
}
var qt = class Z {
  constructor(e, n) {
    this.activeHandle = e, this.dragging = n;
  }
  apply(e) {
    const n = this, o = e.getMeta(x);
    if (o && o.setHandle != null) return new Z(o.setHandle, !1);
    if (o && o.setDragging !== void 0) return new Z(n.activeHandle, o.setDragging);
    if (n.activeHandle > -1 && e.docChanged) {
      let l = e.mapping.map(n.activeHandle, -1);
      return ie(e.doc.resolve(l)) || (l = -1), new Z(l, n.dragging);
    }
    return n;
  }
};
function Zt(t, e, n, o) {
  if (!t.editable) return;
  const l = x.getState(t.state);
  if (l && !l.dragging) {
    const r = en(e.target);
    let s = -1;
    if (r) {
      const { left: i, right: c } = r.getBoundingClientRect();
      e.clientX - i <= n ? s = Se(t, e, "left", n) : c - e.clientX <= n && (s = Se(t, e, "right", n));
    }
    if (s != l.activeHandle) {
      if (!o && s !== -1) {
        const i = t.state.doc.resolve(s), c = i.node(-1), a = C.get(c), d = i.start(-1);
        if (a.colCount(i.pos - d) + i.nodeAfter.attrs.colspan - 1 == a.width - 1) return;
      }
      Pe(t, s);
    }
  }
}
function Jt(t) {
  if (!t.editable) return;
  const e = x.getState(t.state);
  e && e.activeHandle > -1 && !e.dragging && Pe(t, -1);
}
function Yt(t, e, n, o) {
  var l;
  if (!t.editable) return !1;
  const r = (l = t.dom.ownerDocument.defaultView) !== null && l !== void 0 ? l : window, s = x.getState(t.state);
  if (!s || s.activeHandle == -1 || s.dragging) return !1;
  const i = t.state.doc.nodeAt(s.activeHandle), c = Qt(t, s.activeHandle, i.attrs);
  t.dispatch(t.state.tr.setMeta(x, { setDragging: {
    startX: e.clientX,
    startWidth: c
  } }));
  function a(u) {
    r.removeEventListener("mouseup", a), r.removeEventListener("mousemove", d);
    const f = x.getState(t.state);
    f != null && f.dragging && (tn(t, f.activeHandle, xe(f.dragging, u, n)), t.dispatch(t.state.tr.setMeta(x, { setDragging: null })));
  }
  function d(u) {
    if (!u.which) return a(u);
    const f = x.getState(t.state);
    if (f && f.dragging) {
      const h = xe(f.dragging, u, n);
      Te(t, f.activeHandle, h, o);
    }
  }
  return Te(t, s.activeHandle, c, o), r.addEventListener("mouseup", a), r.addEventListener("mousemove", d), e.preventDefault(), !0;
}
function Qt(t, e, { colspan: n, colwidth: o }) {
  const l = o && o[o.length - 1];
  if (l) return l;
  const r = t.domAtPos(e);
  let s = r.node.childNodes[r.offset].offsetWidth, i = n;
  if (o)
    for (let c = 0; c < n; c++) o[c] && (s -= o[c], i--);
  return s / i;
}
function en(t) {
  for (; t && t.nodeName != "TD" && t.nodeName != "TH"; ) t = t.classList && t.classList.contains("ProseMirror") ? null : t.parentNode;
  return t;
}
function Se(t, e, n, o) {
  const l = n == "right" ? -o : o, r = t.posAtCoords({
    left: e.clientX + l,
    top: e.clientY
  });
  if (!r) return -1;
  const { pos: s } = r, i = D(t.state.doc.resolve(s));
  if (!i) return -1;
  if (n == "right") return i.pos;
  const c = C.get(i.node(-1)), a = i.start(-1), d = c.map.indexOf(i.pos - a);
  return d % c.width == 0 ? -1 : a + c.map[d - 1];
}
function xe(t, e, n) {
  const o = e.clientX - t.startX;
  return Math.max(n, t.startWidth + o);
}
function Pe(t, e) {
  t.dispatch(t.state.tr.setMeta(x, { setHandle: e }));
}
function tn(t, e, n) {
  const o = t.state.doc.resolve(e), l = o.node(-1), r = C.get(l), s = o.start(-1), i = r.colCount(o.pos - s) + o.nodeAfter.attrs.colspan - 1, c = t.state.tr;
  for (let a = 0; a < r.height; a++) {
    const d = a * r.width + i;
    if (a && r.map[d] == r.map[d - r.width]) continue;
    const u = r.map[d], f = l.nodeAt(u).attrs, h = f.colspan == 1 ? 0 : i - r.colCount(u);
    if (f.colwidth && f.colwidth[h] == n) continue;
    const p = f.colwidth ? f.colwidth.slice() : nn(f.colspan);
    p[h] = n, c.setNodeMarkup(s + u, null, {
      ...f,
      colwidth: p
    });
  }
  c.docChanged && t.dispatch(c);
}
function Te(t, e, n, o) {
  const l = t.state.doc.resolve(e), r = l.node(-1), s = l.start(-1), i = C.get(r).colCount(l.pos - s) + l.nodeAfter.attrs.colspan - 1;
  let c = t.domAtPos(l.start(-1)).node;
  for (; c && c.nodeName != "TABLE"; ) c = c.parentNode;
  c && ae(r, c.firstChild, c, o, i, n);
}
function nn(t) {
  return Array(t).fill(0);
}
function on(t, e) {
  const n = [], o = t.doc.resolve(e), l = o.node(-1);
  if (!l) return oe.empty;
  const r = C.get(l), s = o.start(-1), i = r.colCount(o.pos - s) + o.nodeAfter.attrs.colspan - 1;
  for (let a = 0; a < r.height; a++) {
    const d = i + a * r.width;
    if ((i == r.width - 1 || r.map[d] != r.map[d + 1]) && (a == 0 || r.map[d] != r.map[d - r.width])) {
      var c;
      const u = r.map[d], f = s + u + l.nodeAt(u).nodeSize - 1, h = document.createElement("div");
      h.className = "column-resize-handle", !((c = x.getState(t)) === null || c === void 0) && c.dragging && n.push(le.node(s + u, s + u + l.nodeAt(u).nodeSize, { class: "column-resize-dragging" })), n.push(le.widget(f, h));
    }
  }
  return oe.create(t.doc, n);
}
function ln({ allowTableNodeSelection: t = !1 } = {}) {
  return new Ee({
    key: H,
    state: {
      init() {
        return null;
      },
      apply(e, n) {
        const o = e.getMeta(H);
        if (o != null) return o == -1 ? null : o;
        if (n == null || !e.docChanged) return n;
        const { deleted: l, pos: r } = e.mapping.mapResult(n);
        return l ? null : r;
      }
    },
    props: {
      decorations: bt,
      handleDOMEvents: { mousedown: Xt },
      createSelectionBetween(e) {
        return H.getState(e.state) != null ? e.state.selection : null;
      },
      handleTripleClick: Vt,
      handleKeyDown: jt,
      handlePaste: Kt
    },
    appendTransaction(e, n, o) {
      return yt(o, De(o, n), t);
    }
  });
}
function J(t) {
  return t === "left" || t === "right" || t === "center" ? t : null;
}
function rn(t) {
  const e = (t.style.textAlign || "").trim().toLowerCase(), n = (t.getAttribute("align") || "").trim().toLowerCase();
  return J(e || n);
}
function sn(t) {
  return J(t == null ? void 0 : t.align);
}
function Oe() {
  return {
    default: null,
    parseHTML: (t) => rn(t),
    renderHTML: (t) => t.align ? {
      style: `text-align: ${t.align}`
    } : {}
  };
}
var Fe = Y.create({
  name: "tableCell",
  addOptions() {
    return {
      HTMLAttributes: {}
    };
  },
  content: "block+",
  addAttributes() {
    return {
      colspan: {
        default: 1
      },
      rowspan: {
        default: 1
      },
      colwidth: {
        default: null,
        parseHTML: (t) => {
          var e, n;
          const o = t.getAttribute("colwidth"), l = o ? o.split(",").map((r) => parseInt(r, 10)) : null;
          if (!l) {
            const r = (e = t.closest("table")) == null ? void 0 : e.querySelectorAll("colgroup > col"), s = Array.from(((n = t.parentElement) == null ? void 0 : n.children) || []).indexOf(t);
            if (s && s > -1 && r && r[s]) {
              const i = r[s].getAttribute("width");
              return i ? [parseInt(i, 10)] : null;
            }
          }
          return l;
        }
      },
      align: Oe()
    };
  },
  tableRole: "cell",
  isolating: !0,
  parseHTML() {
    return [{ tag: "td" }];
  },
  renderHTML({ HTMLAttributes: t }) {
    return ["td", I(this.options.HTMLAttributes, t), 0];
  }
}), je = Y.create({
  name: "tableHeader",
  addOptions() {
    return {
      HTMLAttributes: {}
    };
  },
  content: "block+",
  addAttributes() {
    return {
      colspan: {
        default: 1
      },
      rowspan: {
        default: 1
      },
      colwidth: {
        default: null,
        parseHTML: (t) => {
          const e = t.getAttribute("colwidth");
          return e ? e.split(",").map((o) => parseInt(o, 10)) : null;
        }
      },
      align: Oe()
    };
  },
  tableRole: "header_cell",
  isolating: !0,
  parseHTML() {
    return [{ tag: "th" }];
  },
  renderHTML({ HTMLAttributes: t }) {
    return ["th", I(this.options.HTMLAttributes, t), 0];
  }
}), Ve = Y.create({
  name: "tableRow",
  addOptions() {
    return {
      HTMLAttributes: {}
    };
  },
  content: "(tableCell | tableHeader)*",
  tableRole: "row",
  parseHTML() {
    return [{ tag: "tr" }];
  },
  renderHTML({ HTMLAttributes: t }) {
    return ["tr", I(this.options.HTMLAttributes, t), 0];
  }
});
function de(t, e) {
  return e ? ["width", `${Math.max(e, t)}px`] : ["min-width", `${t}px`];
}
function Re(t, e, n, o, l, r) {
  var s;
  let i = 0, c = !0, a = e.firstChild;
  const d = t.firstChild;
  if (d !== null)
    for (let f = 0, h = 0; f < d.childCount; f += 1) {
      const { colspan: p, colwidth: m } = d.child(f).attrs;
      for (let b = 0; b < p; b += 1, h += 1) {
        const g = l === h ? r : m && m[b], y = g ? `${g}px` : "";
        if (i += g || o, g || (c = !1), a) {
          if (a.style.width !== y) {
            const [S, N] = de(o, g);
            a.style.setProperty(S, N);
          }
          a = a.nextSibling;
        } else {
          const S = document.createElement("col"), [N, T] = de(o, g);
          S.style.setProperty(N, T), e.appendChild(S);
        }
      }
    }
  for (; a; ) {
    const f = a.nextSibling;
    (s = a.parentNode) == null || s.removeChild(a), a = f;
  }
  const u = t.attrs.style && typeof t.attrs.style == "string" && /\bwidth\s*:/i.test(t.attrs.style);
  c && !u ? (n.style.width = `${i}px`, n.style.minWidth = "") : (n.style.width = "", n.style.minWidth = `${i}px`);
}
var cn = class {
  constructor(t, e, n, o = {}) {
    this.node = t, this.cellMinWidth = e, this.dom = document.createElement("div"), this.dom.className = "tableWrapper", this.table = this.dom.appendChild(document.createElement("table"));
    for (const [l, r] of Object.entries(o))
      r != null && (l === "style" ? this.table.style.cssText = String(r) : this.table.setAttribute(l, String(r)));
    t.attrs.style && (this.table.style.cssText = t.attrs.style), this.colgroup = this.table.appendChild(document.createElement("colgroup")), Re(t, this.colgroup, this.table, e), this.contentDOM = this.table.appendChild(document.createElement("tbody"));
  }
  update(t) {
    return t.type !== this.node.type ? !1 : (this.node = t, Re(t, this.colgroup, this.table, this.cellMinWidth), !0);
  }
  ignoreMutation(t) {
    const e = t.target, n = this.dom.contains(e), o = this.contentDOM.contains(e);
    return !!(n && !o && (t.type === "attributes" || t.type === "childList" || t.type === "characterData"));
  }
};
function an(t, e, n, o) {
  let l = 0, r = !0;
  const s = [], i = t.firstChild;
  if (!i)
    return {};
  for (let u = 0, f = 0; u < i.childCount; u += 1) {
    const { colspan: h, colwidth: p } = i.child(u).attrs;
    for (let m = 0; m < h; m += 1, f += 1) {
      const b = n === f ? o : p && p[m];
      l += b || e, b || (r = !1);
      const [g, y] = de(e, b);
      s.push(["col", { style: `${g}: ${y}` }]);
    }
  }
  const c = r ? `${l}px` : "", a = r ? "" : `${l}px`;
  return { colgroup: ["colgroup", {}, ...s], tableWidth: c, tableMinWidth: a };
}
function ve(t, e) {
  return t.createAndFill();
}
function dn(t) {
  if (t.cached.tableNodeTypes)
    return t.cached.tableNodeTypes;
  const e = {};
  return Object.keys(t.nodes).forEach((n) => {
    const o = t.nodes[n];
    o.spec.tableRole && (e[o.spec.tableRole] = o);
  }), t.cached.tableNodeTypes = e, e;
}
function un(t, e, n, o, l) {
  const r = dn(t), s = [], i = [];
  for (let a = 0; a < n; a += 1) {
    const d = ve(r.cell);
    if (d && i.push(d), o) {
      const u = ve(r.header_cell);
      u && s.push(u);
    }
  }
  const c = [];
  for (let a = 0; a < e; a += 1)
    c.push(r.row.createChecked(null, o && a === 0 ? s : i));
  return r.table.createChecked(null, c);
}
function fn(t) {
  return t instanceof w;
}
var G = ({ editor: t }) => {
  const { selection: e } = t.state;
  if (!fn(e))
    return !1;
  let n = 0;
  const o = Je(e.ranges[0].$from, (r) => r.type.name === "table");
  return o == null || o.node.descendants((r) => {
    if (r.type.name === "table")
      return !1;
    ["tableCell", "tableHeader"].includes(r.type.name) && (n += 1);
  }), n === e.ranges.length ? (t.commands.deleteTable(), !0) : !1;
}, hn = "";
function pn(t) {
  return (t || "").replace(/\s+/g, " ").trim();
}
function mn(t, e, n = {}) {
  var o;
  const l = (o = n.cellLineSeparator) != null ? o : hn;
  if (!t || !t.content || t.content.length === 0)
    return "";
  const r = [];
  t.content.forEach((m) => {
    const b = [];
    m.content && m.content.forEach((g) => {
      let y = "";
      g.content && Array.isArray(g.content) && g.content.length > 1 ? y = g.content.map(
        ($) => e.renderChildren($)
      ).join(l) : y = g.content ? e.renderChildren(g.content) : "";
      const S = pn(y), N = g.type === "tableHeader", T = sn(g.attrs);
      b.push({ text: S, isHeader: N, align: T });
    }), r.push(b);
  });
  const s = r.reduce((m, b) => Math.max(m, b.length), 0);
  if (s === 0)
    return "";
  const i = Array.from({ length: s }).fill(0);
  r.forEach((m) => {
    var b;
    for (let g = 0; g < s; g += 1) {
      const S = (((b = m[g]) == null ? void 0 : b.text) || "").length;
      S > i[g] && (i[g] = S), i[g] < 3 && (i[g] = 3);
    }
  });
  const c = (m, b) => m + " ".repeat(Math.max(0, b - m.length)), a = r[0], d = a.some((m) => m.isHeader), u = Array.from({
    length: s
  }).fill(null);
  r.forEach((m) => {
    var b;
    for (let g = 0; g < s; g += 1)
      !u[g] && ((b = m[g]) != null && b.align) && (u[g] = m[g].align);
  });
  let f = `
`;
  const h = Array.from({ length: s }).map(
    (m, b) => d && a[b] && a[b].text || ""
  );
  return f += `| ${h.map((m, b) => c(m, i[b])).join(" | ")} |
`, f += `| ${i.map((m, b) => {
    const g = Math.max(3, m), y = u[b];
    return y === "left" ? `:${"-".repeat(g)}` : y === "right" ? `${"-".repeat(g)}:` : y === "center" ? `:${"-".repeat(g)}:` : "-".repeat(g);
  }).join(" | ")} |
`, (d ? r.slice(1) : r).forEach((m) => {
    f += `| ${Array.from({ length: s }).fill(0).map((b, g) => c(m[g] && m[g].text || "", i[g])).join(" | ")} |
`;
  }), f;
}
var gn = mn, Ke = Y.create({
  name: "table",
  // @ts-ignore
  addOptions() {
    return {
      HTMLAttributes: {},
      resizable: !1,
      renderWrapper: !1,
      handleWidth: 5,
      cellMinWidth: 25,
      // TODO: fix
      View: cn,
      lastColumnResizable: !0,
      allowTableNodeSelection: !1
    };
  },
  content: "tableRow+",
  tableRole: "table",
  isolating: !0,
  group: "block",
  parseHTML() {
    return [{ tag: "table" }];
  },
  renderHTML({ node: t, HTMLAttributes: e }) {
    const { colgroup: n, tableWidth: o, tableMinWidth: l } = an(t, this.options.cellMinWidth), r = e.style;
    function s() {
      return r || (o ? `width: ${o}` : `min-width: ${l}`);
    }
    const i = [
      "table",
      I(this.options.HTMLAttributes, e, {
        style: s()
      }),
      n,
      ["tbody", 0]
    ];
    return this.options.renderWrapper ? ["div", { class: "tableWrapper" }, i] : i;
  },
  parseMarkdown: (t, e) => {
    const n = [], o = Array.isArray(t.align) ? t.align : [];
    if (t.header) {
      const l = [];
      t.header.forEach((r, s) => {
        var i;
        const c = J((i = o[s]) != null ? i : r.align), a = c ? { align: c } : {};
        l.push(
          e.createNode("tableHeader", a, [
            { type: "paragraph", content: e.parseInline(r.tokens) }
          ])
        );
      }), n.push(e.createNode("tableRow", {}, l));
    }
    return t.rows && t.rows.forEach((l) => {
      const r = [];
      l.forEach((s, i) => {
        var c;
        const a = J((c = o[i]) != null ? c : s.align), d = a ? { align: a } : {};
        r.push(
          e.createNode("tableCell", d, [
            { type: "paragraph", content: e.parseInline(s.tokens) }
          ])
        );
      }), n.push(e.createNode("tableRow", {}, r));
    }), e.createNode("table", void 0, n);
  },
  renderMarkdown: (t, e) => gn(t, e),
  addCommands() {
    return {
      insertTable: ({ rows: t = 3, cols: e = 3, withHeaderRow: n = !0 } = {}) => ({ tr: o, dispatch: l, editor: r }) => {
        const s = un(r.schema, t, e, n);
        if (l) {
          const i = o.selection.from + 1;
          o.replaceSelectionWith(s).scrollIntoView().setSelection(Ye.near(o.doc.resolve(i)));
        }
        return !0;
      },
      addColumnBefore: () => ({ state: t, dispatch: e }) => xt(t, e),
      addColumnAfter: () => ({ state: t, dispatch: e }) => Tt(t, e),
      deleteColumn: () => ({ state: t, dispatch: e }) => vt(t, e),
      addRowBefore: () => ({ state: t, dispatch: e }) => Nt(t, e),
      addRowAfter: () => ({ state: t, dispatch: e }) => kt(t, e),
      deleteRow: () => ({ state: t, dispatch: e }) => zt(t, e),
      deleteTable: () => ({ state: t, dispatch: e }) => _t(t, e),
      mergeCells: () => ({ state: t, dispatch: e }) => pe(t, e),
      splitCell: () => ({ state: t, dispatch: e }) => me(t, e),
      toggleHeaderColumn: () => ({ state: t, dispatch: e }) => P("column")(t, e),
      toggleHeaderRow: () => ({ state: t, dispatch: e }) => P("row")(t, e),
      toggleHeaderCell: () => ({ state: t, dispatch: e }) => Dt(t, e),
      mergeOrSplit: () => ({ state: t, dispatch: e }) => pe(t, e) ? !0 : me(t, e),
      setCellAttribute: (t, e) => ({ state: n, dispatch: o }) => $t(t, e)(n, o),
      goToNextCell: () => ({ state: t, dispatch: e }) => be(1)(t, e),
      goToPreviousCell: () => ({ state: t, dispatch: e }) => be(-1)(t, e),
      fixTables: () => ({ state: t, dispatch: e }) => (e && De(t), !0),
      setCellSelection: (t) => ({ tr: e, dispatch: n }) => {
        if (n) {
          const o = w.create(e.doc, t.anchorCell, t.headCell);
          e.setSelection(o);
        }
        return !0;
      }
    };
  },
  addKeyboardShortcuts() {
    return {
      Tab: () => this.editor.commands.goToNextCell() ? !0 : this.editor.can().addRowAfter() ? this.editor.chain().addRowAfter().goToNextCell().run() : !1,
      "Shift-Tab": () => this.editor.commands.goToPreviousCell(),
      Backspace: G,
      "Mod-Backspace": G,
      Delete: G,
      "Mod-Delete": G
    };
  },
  addProseMirrorPlugins() {
    return [
      ...this.options.resizable && this.editor.isEditable ? [
        Gt({
          handleWidth: this.options.handleWidth,
          cellMinWidth: this.options.cellMinWidth,
          defaultCellMinWidth: this.options.cellMinWidth,
          View: this.options.View,
          lastColumnResizable: this.options.lastColumnResizable
        })
      ] : [],
      ln({
        allowTableNodeSelection: this.options.allowTableNodeSelection
      })
    ];
  },
  addNodeView() {
    const t = this.options.resizable && this.editor.isEditable, e = this.options.View;
    return t || !e ? null : ({ node: n, view: o, HTMLAttributes: l }) => {
      const r = I(this.options.HTMLAttributes, l);
      return new e(n, this.options.cellMinWidth, o, r);
    };
  },
  extendNodeSchema(t) {
    const e = {
      name: t.name,
      options: t.options,
      storage: t.storage
    };
    return {
      tableRole: qe(Ze(t, "tableRole", e))
    };
  }
});
ze.create({
  name: "tableKit",
  addExtensions() {
    const t = [];
    return this.options.table !== !1 && t.push(Ke.configure(this.options.table)), this.options.tableCell !== !1 && t.push(Fe.configure(this.options.tableCell)), this.options.tableHeader !== !1 && t.push(je.configure(this.options.tableHeader)), this.options.tableRow !== !1 && t.push(Ve.configure(this.options.tableRow)), t;
  }
});
function bn(t, e, n) {
  if (!t.doc)
    return t;
  const o = t.doc.nodeAt(e);
  if (!o || n === o.attrs.backgroundColor)
    return t;
  const l = {
    ...o.attrs,
    backgroundColor: n
  };
  return t.setNodeMarkup(e, o.type, l, o.marks);
}
function wn(t, e, n) {
  const { doc: o, selection: l } = t;
  return !o || !l || !(l instanceof w) || l.forEachCell((r, s) => {
    t = bn(t, s, n);
  }), t;
}
function Me(t, e) {
  return ({ tr: n, state: o, dispatch: l }) => {
    const { selection: r } = o;
    return n = n.setSelection(r), n = wn(n, e, t), n.docChanged ? (l == null || l(n), !0) : !1;
  };
}
const Cn = ze.create({
  name: "tableCellBackground",
  addOptions() {
    return {
      types: ["tableCell"],
      HTMLAttributes: {}
    };
  },
  addGlobalAttributes() {
    return [
      {
        types: this.options.types,
        attributes: {
          backgroundColor: {
            parseHTML: (t) => t.style.backgroundColor || "",
            renderHTML: (t) => !t.backgroundColor || t.backgroundColor === "" ? {} : {
              style: `background-color: ${t.backgroundColor}`
            }
          }
        }
      }
    ];
  },
  addCommands() {
    return {
      setTableCellBackground: (t) => Me(t, this.options),
      unsetTableCellBackground: () => Me("", this.options)
    };
  }
}), yn = /(android|bb\d+|meego).+mobile|armv7l|avantgo|bada\/|blackberry|blazer|compal|elaine|fennec|hiptop|iemobile|ip(hone|od)|iris|kindle|lge |maemo|midp|mmp|mobile.+firefox|netfront|opera m(ob|in)i|palm( os)?|phone|p(ixi|re)\/|plucker|pocket|psp|series[46]0|samsungbrowser.*mobile|symbian|treo|up\.(browser|link)|vodafone|wap|windows (ce|phone)|xda|xiino/i, An = /CrOS/, Sn = /android|ipad|playbook|silk/i;
function Ne(t = {}) {
  let e = t.ua || typeof navigator < "u" && navigator.userAgent;
  return e && typeof e == "object" && e.headers && typeof e.headers["user-agent"] == "string" && (e = e.headers["user-agent"]), typeof e != "string" ? !1 : !!(yn.test(e) && !An.test(e) || t.tablet && Sn.test(e) || t.tablet && t.featureDetect && navigator && navigator.maxTouchPoints > 1 && e.includes("Macintosh") && e.includes("Safari"));
}
const ke = (t) => Array.from({ length: t }).map((e, n) => n + 1);
function xn(t) {
  var f;
  const [e, n] = O(!1), [o, l] = O(!0), [r, s] = O({
    rows: Ne() ? F : j,
    cols: Ne() ? F : j
  }), [i, c] = O({
    rows: V,
    cols: V
  });
  function a(h, p) {
    h === r.rows && s((m) => ({
      ...m,
      rows: Math.min(h + 1, F)
    })), p === r.cols && s((m) => ({
      ...m,
      cols: Math.min(p + 1, F)
    })), c({
      rows: h,
      cols: p
    });
  }
  function d(h, p) {
    t == null || t.createTable({ rows: h, cols: p, withHeaderRow: o }), u(), n(!1);
  }
  function u() {
    l(!1), s({
      rows: j,
      cols: j
    }), c({
      rows: V,
      cols: V
    });
  }
  return /* @__PURE__ */ te(rt, { modal: !0, onOpenChange: n, open: e, children: [
    /* @__PURE__ */ E(st, { asChild: !0, "data-state": t != null && t.dataState ? "on" : "off", children: t == null ? void 0 : t.children }),
    /* @__PURE__ */ E(it, { align: "start", className: "richtext-w-full !richtext-p-2", side: "bottom", children: /* @__PURE__ */ te("div", { className: "table-grid-size-editor richtext-p-0", children: [
      /* @__PURE__ */ E("div", { className: "richtext-flex richtext-flex-col richtext-flex-wrap richtext-justify-between richtext-gap-1", children: (f = ke(r == null ? void 0 : r.rows)) == null ? void 0 : f.map((h) => {
        var p;
        return /* @__PURE__ */ E("div", { className: "richtext-flex richtext-gap-1", children: (p = ke(r == null ? void 0 : r.cols)) == null ? void 0 : p.map((m) => /* @__PURE__ */ E(
          "div",
          {
            onMouseDown: () => d(h, m),
            onMouseOver: () => a(h, m),
            className: `richtext-cursor-pointer richtext-border-border ${m <= i.cols && h <= i.rows && "tableCellActive !richtext-bg-foreground"}`,
            children: /* @__PURE__ */ E("div", { className: "richtext-box-border richtext-size-4 richtext-rounded-[2px] !richtext-border richtext-border-solid !richtext-border-border richtext-p-1" })
          },
          `richtext-table-col-${m}`
        )) }, `richtext-table-row-${h}`);
      }) }),
      /* @__PURE__ */ te("div", { className: "richtext-mt-2 richtext-text-center richtext-text-sm richtext-text-foreground", children: [
        i.rows,
        "x",
        i.cols
      ] })
    ] }) })
  ] });
}
function Wn() {
  const t = tt(), e = nt(Tn.name), {
    icon: n = void 0,
    tooltip: o = void 0,
    action: l = void 0,
    isActive: r = void 0,
    color: s
  } = (e == null ? void 0 : e.componentProps) ?? {}, { dataState: i, disabled: c } = ot(r);
  if (!e)
    return /* @__PURE__ */ E(et, {});
  function a(d) {
    t.chain().focus().insertTable({ ...d, withHeaderRow: !1 }).run();
  }
  return /* @__PURE__ */ E(xn, { createTable: a, dataState: i, children: /* @__PURE__ */ E(
    lt,
    {
      action: l,
      color: s,
      dataState: i,
      icon: n,
      isActive: r,
      tooltip: o,
      disabled: c
    }
  ) });
}
const Tn = /* @__PURE__ */ Ke.extend({
  //@ts-expect-error
  addOptions() {
    var t;
    return {
      ...(t = this.parent) == null ? void 0 : t.call(this),
      HTMLAttributes: {
        style: `
          border: 1px solid #000;
          border-collapse: collapse;
          width: 100%;
        `
      },
      resizable: !0,
      lastColumnResizable: !0,
      allowTableNodeSelection: !1,
      button: ({ editor: e, t: n }) => ({
        componentProps: {
          isActive: () => e.isActive("table"),
          icon: "Table",
          tooltip: n("editor.table.tooltip")
        }
      })
    };
  },
  addExtensions() {
    return [
      Ve.configure(this.options.tableRow),
      je.configure(this.options.tableHeader),
      Fe.configure(this.options.tableCell),
      Cn.configure(this.options.tableCellBackground)
    ];
  }
});
export {
  w as C,
  Wn as R,
  Tn as T
};
