// reactjs-tiptap-editor@1.0.2/lib/Twitter-DP26Lkn8.js
import { N as jt, m as Wt, a1 as Bt } from "./index-Dx-Dh7KS.js";
import { c as U, s as ot, N as Ft, a as Pt } from "./clsx-_bq4MQQ7.js";
import { jsx as r, jsxs as h, Fragment as Ut } from "react/jsx-runtime";
import Ue, { useState as se, Fragment as zt, useEffect as ze, useMemo as He, createContext as Ht, useLayoutEffect as Gt, useContext as Xt, useRef as z, useCallback as Re, useDebugValue as Kt } from "react";
import { L as Qt, k as Jt, B as Yt, h as Zt, u as en, d as tn, A as nn, I as rn } from "./index-C52cuQ_Y.js";
import "./theme.js";
import { P as on, a as an, b as sn } from "./popover-BB-ZDnqk.js";
import { u as cn } from "./index-DtyAGMbq.js";
const ln = "_verified_lvyh0_1", Ge = {
  verified: ln
}, dn = () => /* @__PURE__ */ r("svg", {
  viewBox: "0 0 24 24",
  "aria-label": "Verified account",
  role: "img",
  className: Ge.verified,
  children: /* @__PURE__ */ r("g", {
    children: /* @__PURE__ */ r("path", {
      d: "M22.25 12c0-1.43-.88-2.67-2.19-3.34.46-1.39.2-2.9-.81-3.91s-2.52-1.27-3.91-.81c-.66-1.31-1.91-2.19-3.34-2.19s-2.67.88-3.33 2.19c-1.4-.46-2.91-.2-3.92.81s-1.26 2.52-.8 3.91c-1.31.67-2.2 1.91-2.2 3.34s.89 2.67 2.2 3.34c-.46 1.39-.21 2.9.8 3.91s2.52 1.26 3.91.81c.67 1.31 1.91 2.19 3.34 2.19s2.68-.88 3.34-2.19c1.39.45 2.9.2 3.91-.81s1.27-2.52.81-3.91c1.31-.67 2.19-1.91 2.19-3.34zm-11.71 4.2L6.8 12.46l1.41-1.42 2.26 2.26 4.8-5.23 1.47 1.36-6.2 6.77z"
    })
  })
}), un = () => /* @__PURE__ */ r("svg", {
  viewBox: "0 0 22 22",
  "aria-label": "Verified account",
  role: "img",
  className: Ge.verified,
  children: /* @__PURE__ */ h("g", {
    children: [
      /* @__PURE__ */ h("linearGradient", {
        gradientUnits: "userSpaceOnUse",
        id: "0-a",
        x1: "4.411",
        x2: "18.083",
        y1: "2.495",
        y2: "21.508",
        children: [
          /* @__PURE__ */ r("stop", {
            offset: "0",
            stopColor: "#f4e72a"
          }),
          /* @__PURE__ */ r("stop", {
            offset: ".539",
            stopColor: "#cd8105"
          }),
          /* @__PURE__ */ r("stop", {
            offset: ".68",
            stopColor: "#cb7b00"
          }),
          /* @__PURE__ */ r("stop", {
            offset: "1",
            stopColor: "#f4ec26"
          }),
          /* @__PURE__ */ r("stop", {
            offset: "1",
            stopColor: "#f4e72a"
          })
        ]
      }),
      /* @__PURE__ */ h("linearGradient", {
        gradientUnits: "userSpaceOnUse",
        id: "0-b",
        x1: "5.355",
        x2: "16.361",
        y1: "3.395",
        y2: "19.133",
        children: [
          /* @__PURE__ */ r("stop", {
            offset: "0",
            stopColor: "#f9e87f"
          }),
          /* @__PURE__ */ r("stop", {
            offset: ".406",
            stopColor: "#e2b719"
          }),
          /* @__PURE__ */ r("stop", {
            offset: ".989",
            stopColor: "#e2b719"
          })
        ]
      }),
      /* @__PURE__ */ h("g", {
        clipRule: "evenodd",
        fillRule: "evenodd",
        children: [
          /* @__PURE__ */ r("path", {
            d: "M13.324 3.848L11 1.6 8.676 3.848l-3.201-.453-.559 3.184L2.06 8.095 3.48 11l-1.42 2.904 2.856 1.516.559 3.184 3.201-.452L11 20.4l2.324-2.248 3.201.452.559-3.184 2.856-1.516L18.52 11l1.42-2.905-2.856-1.516-.559-3.184zm-7.09 7.575l3.428 3.428 5.683-6.206-1.347-1.247-4.4 4.795-2.072-2.072z",
            fill: "url(#0-a)"
          }),
          /* @__PURE__ */ r("path", {
            d: "M13.101 4.533L11 2.5 8.899 4.533l-2.895-.41-.505 2.88-2.583 1.37L4.2 11l-1.284 2.627 2.583 1.37.505 2.88 2.895-.41L11 19.5l2.101-2.033 2.895.41.505-2.88 2.583-1.37L17.8 11l1.284-2.627-2.583-1.37-.505-2.88zm-6.868 6.89l3.429 3.428 5.683-6.206-1.347-1.247-4.4 4.795-2.072-2.072z",
            fill: "url(#0-b)"
          }),
          /* @__PURE__ */ r("path", {
            d: "M6.233 11.423l3.429 3.428 5.65-6.17.038-.033-.005 1.398-5.683 6.206-3.429-3.429-.003-1.405.005.003z",
            fill: "#d18800"
          })
        ]
      })
    ]
  })
}), fn = () => /* @__PURE__ */ r("svg", {
  viewBox: "0 0 22 22",
  "aria-label": "Verified account",
  role: "img",
  className: Ge.verified,
  children: /* @__PURE__ */ r("g", {
    children: /* @__PURE__ */ r("path", {
      clipRule: "evenodd",
      d: "M12.05 2.056c-.568-.608-1.532-.608-2.1 0l-1.393 1.49c-.284.303-.685.47-1.1.455L5.42 3.932c-.832-.028-1.514.654-1.486 1.486l.069 2.039c.014.415-.152.816-.456 1.1l-1.49 1.392c-.608.568-.608 1.533 0 2.101l1.49 1.393c.304.284.47.684.456 1.1l-.07 2.038c-.027.832.655 1.514 1.487 1.486l2.038-.069c.415-.014.816.152 1.1.455l1.392 1.49c.569.609 1.533.609 2.102 0l1.393-1.49c.283-.303.684-.47 1.099-.455l2.038.069c.832.028 1.515-.654 1.486-1.486L18 14.542c-.015-.415.152-.815.455-1.099l1.49-1.393c.608-.568.608-1.533 0-2.101l-1.49-1.393c-.303-.283-.47-.684-.455-1.1l.068-2.038c.029-.832-.654-1.514-1.486-1.486l-2.038.07c-.415.013-.816-.153-1.1-.456zm-5.817 9.367l3.429 3.428 5.683-6.206-1.347-1.247-4.4 4.795-2.072-2.072z",
      fillRule: "evenodd"
    })
  })
}), hn = "_root_98iqw_1", _n = "_article_98iqw_21", at = {
  root: hn,
  article: _n
}, Xe = ({ className: e, children: t }) => /* @__PURE__ */ r("div", {
  className: U("react-tweet-theme", at.root, e),
  children: /* @__PURE__ */ r("article", {
    className: at.article,
    children: t
  })
}), gt = (e) => /* @__PURE__ */ r("img", {
  ...e
}), mn = "_verifiedOld_lcna5_1", pn = "_verifiedBlue_lcna5_4", vn = "_verifiedGovernment_lcna5_7", Se = {
  verifiedOld: mn,
  verifiedBlue: pn,
  verifiedGovernment: vn
}, wt = ({ user: e, className: t }) => {
  const n = e.verified || e.is_blue_verified || e.verified_type;
  let o = /* @__PURE__ */ r(dn, {}), a = Se.verifiedBlue;
  if (n)
    switch (e.is_blue_verified || (a = Se.verifiedOld), e.verified_type) {
      case "Government":
        o = /* @__PURE__ */ r(fn, {}), a = Se.verifiedGovernment;
        break;
      case "Business":
        o = /* @__PURE__ */ r(un, {}), a = null;
        break;
    }
  return n ? /* @__PURE__ */ r("div", {
    className: U(t, a),
    children: o
  }) : null;
}, gn = "_label_36ucz_1", wn = {
  label: gn
}, yt = ({ user: e, className: t }) => {
  var n;
  const o = e.highlighted_label;
  if (!o) return null;
  const a = (n = o.badge) == null ? void 0 : n.url;
  return a ? /* @__PURE__ */ r("div", {
    className: U(wn.label, t),
    children: /* @__PURE__ */ r("img", {
      src: a,
      alt: o.description
    })
  }) : null;
}, yn = "_header_nqq4j_1", bn = "_avatar_nqq4j_11", Tn = "_avatarOverflow_nqq4j_16", xn = "_avatarSquare_nqq4j_23", Nn = "_avatarShadow_nqq4j_26", Rn = "_author_nqq4j_37", Sn = "_authorLink_nqq4j_44", In = "_authorVerified_nqq4j_53", kn = "_authorLinkText_nqq4j_56", En = "_authorMeta_nqq4j_63", Cn = "_authorFollow_nqq4j_66", On = "_username_nqq4j_69", Ln = "_follow_nqq4j_74", qn = "_separator_nqq4j_82", $n = "_brand_nqq4j_86", Dn = "_twitterIcon_nqq4j_90", N = {
  header: yn,
  avatar: bn,
  avatarOverflow: Tn,
  avatarSquare: xn,
  avatarShadow: Nn,
  author: Rn,
  authorLink: Sn,
  authorVerified: In,
  authorLinkText: kn,
  authorMeta: En,
  authorFollow: Cn,
  username: On,
  follow: Ln,
  separator: qn,
  brand: $n,
  twitterIcon: Dn
}, An = ({ tweet: e, components: t }) => {
  var n;
  const o = (n = t == null ? void 0 : t.AvatarImg) != null ? n : gt, { user: a } = e;
  return /* @__PURE__ */ h("div", {
    className: N.header,
    children: [
      /* @__PURE__ */ h("a", {
        href: e.url,
        className: N.avatar,
        target: "_blank",
        rel: "noopener noreferrer",
        children: [
          /* @__PURE__ */ r("div", {
            className: U(N.avatarOverflow, a.profile_image_shape === "Square" && N.avatarSquare),
            children: /* @__PURE__ */ r(o, {
              src: a.profile_image_url_https,
              alt: a.name,
              width: 48,
              height: 48
            })
          }),
          /* @__PURE__ */ r("div", {
            className: N.avatarOverflow,
            children: /* @__PURE__ */ r("div", {
              className: N.avatarShadow
            })
          })
        ]
      }),
      /* @__PURE__ */ h("div", {
        className: N.author,
        children: [
          /* @__PURE__ */ h("a", {
            href: e.url,
            className: N.authorLink,
            target: "_blank",
            rel: "noopener noreferrer",
            children: [
              /* @__PURE__ */ r("div", {
                className: N.authorLinkText,
                children: /* @__PURE__ */ r("span", {
                  title: a.name,
                  children: a.name
                })
              }),
              /* @__PURE__ */ r(wt, {
                user: a,
                className: N.authorVerified
              }),
              /* @__PURE__ */ r(yt, {
                user: a
              })
            ]
          }),
          /* @__PURE__ */ h("div", {
            className: N.authorMeta,
            children: [
              /* @__PURE__ */ r("a", {
                href: e.url,
                className: N.username,
                target: "_blank",
                rel: "noopener noreferrer",
                children: /* @__PURE__ */ h("span", {
                  title: `@${a.screen_name}`,
                  children: [
                    "@",
                    a.screen_name
                  ]
                })
              }),
              /* @__PURE__ */ h("div", {
                className: N.authorFollow,
                children: [
                  /* @__PURE__ */ r("span", {
                    className: N.separator,
                    children: "·"
                  }),
                  /* @__PURE__ */ r("a", {
                    href: a.follow_url,
                    className: N.follow,
                    target: "_blank",
                    rel: "noopener noreferrer",
                    children: "Follow"
                  })
                ]
              })
            ]
          })
        ]
      }),
      /* @__PURE__ */ r("a", {
        href: e.url,
        className: N.brand,
        target: "_blank",
        rel: "noopener noreferrer",
        "aria-label": "View on Twitter",
        children: /* @__PURE__ */ r("svg", {
          viewBox: "0 0 24 24",
          "aria-hidden": "true",
          className: N.twitterIcon,
          children: /* @__PURE__ */ r("g", {
            children: /* @__PURE__ */ r("path", {
              d: "M18.244 2.25h3.308l-7.227 8.26 8.502 11.24H16.17l-5.214-6.817L4.99 21.75H1.68l7.73-8.835L1.254 2.25H8.08l4.713 6.231zm-1.161 17.52h1.833L7.084 4.126H5.117z"
            })
          })
        })
      })
    ]
  });
}, Vn = "_root_1ec15_1", Mn = {
  root: Vn
}, jn = ({ tweet: e }) => /* @__PURE__ */ h("a", {
  href: e.in_reply_to_url,
  className: Mn.root,
  target: "_blank",
  rel: "noopener noreferrer",
  children: [
    "Replying to @",
    e.in_reply_to_screen_name
  ]
}), Wn = "_root_eihfa_1", Bn = {
  root: Wn
}, bt = ({ href: e, children: t }) => /* @__PURE__ */ r("a", {
  href: e,
  className: Bn.root,
  target: "_blank",
  rel: "noopener noreferrer nofollow",
  children: t
}), Fn = "_root_1xq52_1", Pn = {
  root: Fn
}, Un = ({ tweet: e }) => /* @__PURE__ */ h("p", {
  className: Pn.root,
  lang: e.lang,
  dir: "auto",
  children: [
    e.entities.map((t, n) => {
      switch (t.type) {
        case "hashtag":
        case "mention":
        case "url":
        case "symbol":
          return /* @__PURE__ */ r(bt, {
            href: t.href,
            children: t.text
          }, n);
        case "media":
          return;
        default:
          return /* @__PURE__ */ r("span", {
            dangerouslySetInnerHTML: {
              __html: t.text
            }
          }, n);
      }
    }),
    e.note_tweet ? /* @__PURE__ */ r(zn, {
      tweet: e
    }) : null
  ]
});
function zn({ tweet: e }) {
  return /* @__PURE__ */ h(bt, {
    href: e.url,
    children: [
      /* @__PURE__ */ r("span", {
        children: " "
      }),
      "Show more"
    ]
  });
}
const Hn = {
  hour: "numeric",
  minute: "2-digit",
  hour12: !0,
  weekday: "short",
  month: "short",
  day: "numeric",
  year: "numeric"
}, Gn = new Intl.DateTimeFormat("en-US", Hn), Xn = (e) => {
  const t = {};
  for (const n of e)
    t[n.type] = n.value;
  return t;
}, Kn = (e) => {
  const t = Xn(Gn.formatToParts(e)), n = `${t.hour}:${t.minute} ${t.dayPeriod}`, o = `${t.month} ${t.day}, ${t.year}`;
  return `${n} · ${o}`;
}, st = (e) => `https://x.com/${e.user.screen_name}/status/${e.id_str}`, Tt = (e) => `https://x.com/${typeof e == "string" ? e : e.user.screen_name}`, Qn = (e) => `https://x.com/intent/like?tweet_id=${e.id_str}`, Jn = (e) => `https://x.com/intent/tweet?in_reply_to=${e.id_str}`, Yn = (e) => `https://x.com/intent/follow?screen_name=${e.user.screen_name}`, Zn = (e) => `https://x.com/hashtag/${e.text}`, er = (e) => `https://x.com/search?q=%24${e.text}`, tr = (e) => `https://x.com/${e.in_reply_to_screen_name}/status/${e.in_reply_to_status_id_str}`, xt = (e, t) => {
  const n = new URL(e.media_url_https), o = n.pathname.split(".").pop();
  return o ? (n.pathname = n.pathname.replace(`.${o}`, ""), n.searchParams.set("format", o), n.searchParams.set("name", t), n.toString()) : e.media_url_https;
}, nr = (e) => {
  const { variants: t } = e.video_info;
  return t.filter((o) => o.content_type === "video/mp4").sort((o, a) => {
    var s, i;
    return ((s = a.bitrate) != null ? s : 0) - ((i = o.bitrate) != null ? i : 0);
  });
}, rr = (e) => {
  const t = nr(e);
  return t.length > 1 ? t[1] : t[0];
}, De = (e) => e > 999999 ? `${(e / 1e6).toFixed(1)}M` : e > 999 ? `${(e / 1e3).toFixed(1)}K` : e.toString();
function it(e) {
  const t = Array.from(e.text), n = [
    {
      indices: e.display_text_range,
      type: "text"
    }
  ], { entities: o } = e;
  return o && (he(n, "hashtag", o.hashtags), he(n, "mention", o.user_mentions), he(n, "url", o.urls), he(n, "symbol", o.symbols), he(n, "media", o.media)), or(e, n), n.map((a) => {
    const s = t.slice(a.indices[0], a.indices[1]).join("");
    switch (a.type) {
      case "hashtag":
        return Object.assign(a, {
          href: Zn(a),
          text: s
        });
      case "mention":
        return Object.assign(a, {
          href: Tt(a.screen_name),
          text: s
        });
      case "url":
      case "media":
        return Object.assign(a, {
          href: a.expanded_url,
          text: a.display_url
        });
      case "symbol":
        return Object.assign(a, {
          href: er(a),
          text: s
        });
      default:
        return Object.assign(a, {
          text: s
        });
    }
  });
}
function he(e, t, n) {
  if (n != null && n.length)
    for (const o of n)
      for (const [a, s] of e.entries()) {
        if (s.indices[0] > o.indices[0] || s.indices[1] < o.indices[1])
          continue;
        const i = [
          {
            ...o,
            type: t
          }
        ];
        s.indices[0] < o.indices[0] && i.unshift({
          indices: [
            s.indices[0],
            o.indices[0]
          ],
          type: "text"
        }), s.indices[1] > o.indices[1] && i.push({
          indices: [
            o.indices[1],
            s.indices[1]
          ],
          type: "text"
        }), e.splice(a, 1, ...i);
        break;
      }
}
function or(e, t) {
  var n;
  const o = (n = e.entities) == null ? void 0 : n.media;
  o != null && o.length && o[0].indices[0] < e.display_text_range[1] && (e.display_text_range[1] = o[0].indices[0]);
  const a = t.at(-1);
  a && a.indices[1] > e.display_text_range[1] && (a.indices[1] = e.display_text_range[1]);
}
const ar = (e) => ({
  ...e,
  url: st(e),
  user: {
    ...e.user,
    url: Tt(e),
    follow_url: Yn(e)
  },
  like_url: Qn(e),
  reply_url: Jn(e),
  in_reply_to_url: e.in_reply_to_screen_name ? tr(e) : void 0,
  entities: it(e),
  quoted_tweet: e.quoted_tweet ? {
    ...e.quoted_tweet,
    url: st(e.quoted_tweet),
    entities: it(e.quoted_tweet)
  } : void 0
}), sr = "_root_g9tkv_1", ir = "_rounded_g9tkv_6", cr = "_mediaWrapper_g9tkv_10", lr = "_grid2Columns_g9tkv_17", dr = "_grid3_g9tkv_20", ur = "_grid2x2_g9tkv_23", fr = "_mediaContainer_g9tkv_26", hr = "_mediaLink_g9tkv_34", _r = "_skeleton_g9tkv_38", mr = "_image_g9tkv_43", E = {
  root: sr,
  rounded: ir,
  mediaWrapper: cr,
  grid2Columns: lr,
  grid3: dr,
  grid2x2: ur,
  mediaContainer: fr,
  mediaLink: hr,
  skeleton: _r,
  image: mr
}, pr = "_anchor_fp31d_1", vr = "_videoButton_fp31d_17", gr = "_videoButtonIcon_fp31d_35", wr = "_watchOnTwitter_fp31d_44", yr = "_viewReplies_fp31d_60", ae = {
  anchor: pr,
  videoButton: vr,
  videoButtonIcon: gr,
  watchOnTwitter: wr,
  viewReplies: yr
}, br = ({ tweet: e, media: t }) => {
  const [n, o] = se(!0), [a, s] = se(!1), [i, c] = se(!1), f = rr(t);
  let u = 0;
  return /* @__PURE__ */ h(Ut, {
    children: [
      /* @__PURE__ */ r("video", {
        className: E.image,
        poster: xt(t, "small"),
        controls: !n,
        playsInline: !0,
        preload: "none",
        tabIndex: n ? -1 : 0,
        onPlay: () => {
          u && window.clearTimeout(u), a || s(!0), i && c(!1);
        },
        onPause: () => {
          u && window.clearTimeout(u), u = window.setTimeout(() => {
            a && s(!1), u = 0;
          }, 100);
        },
        onEnded: () => {
          c(!0);
        },
        children: /* @__PURE__ */ r("source", {
          src: f.url,
          type: f.content_type
        })
      }),
      n && /* @__PURE__ */ r("button", {
        type: "button",
        className: ae.videoButton,
        "aria-label": "View video on X",
        onClick: (v) => {
          const m = v.currentTarget.previousSibling;
          v.preventDefault(), o(!1), m.load(), m.play().then(() => {
            s(!0), m.focus();
          }).catch((T) => {
            console.error("Error playing video:", T), o(!0), s(!1);
          });
        },
        children: /* @__PURE__ */ r("svg", {
          viewBox: "0 0 24 24",
          className: ae.videoButtonIcon,
          "aria-hidden": "true",
          children: /* @__PURE__ */ r("g", {
            children: /* @__PURE__ */ r("path", {
              d: "M21 12L4 2v20l17-10z"
            })
          })
        })
      }),
      !a && !i && /* @__PURE__ */ r("div", {
        className: ae.watchOnTwitter,
        children: /* @__PURE__ */ r("a", {
          href: e.url,
          className: ae.anchor,
          target: "_blank",
          rel: "noopener noreferrer",
          children: n ? "Watch on X" : "Continue watching on X"
        })
      }),
      i && /* @__PURE__ */ r("a", {
        href: e.url,
        className: U(ae.anchor, ae.viewReplies),
        target: "_blank",
        rel: "noopener noreferrer",
        children: "View replies"
      })
    ]
  });
}, Tr = (e) => /* @__PURE__ */ r("img", {
  ...e
}), ct = (e, t) => {
  let n = 56.25;
  return t === 1 && (n = 100 / e.original_info.width * e.original_info.height), t === 2 && (n = n * 2), {
    width: e.type === "photo" ? void 0 : "unset",
    paddingBottom: `${n}%`
  };
}, Nt = ({ tweet: e, components: t, quoted: n }) => {
  var o, a, s, i;
  const c = (o = (s = e.mediaDetails) == null ? void 0 : s.length) != null ? o : 0, f = (a = t == null ? void 0 : t.MediaImg) != null ? a : Tr;
  return /* @__PURE__ */ r("div", {
    className: U(E.root, !n && E.rounded),
    children: /* @__PURE__ */ r("div", {
      className: U(E.mediaWrapper, c > 1 && E.grid2Columns, c === 3 && E.grid3, c > 4 && E.grid2x2),
      children: (i = e.mediaDetails) == null ? void 0 : i.map((u) => /* @__PURE__ */ r(zt, {
        children: u.type === "photo" ? /* @__PURE__ */ h("a", {
          href: e.url,
          className: U(E.mediaContainer, E.mediaLink),
          target: "_blank",
          rel: "noopener noreferrer",
          children: [
            /* @__PURE__ */ r("div", {
              className: E.skeleton,
              style: ct(u, c)
            }),
            /* @__PURE__ */ r(f, {
              src: xt(u, "small"),
              alt: u.ext_alt_text || "Image",
              className: E.image,
              draggable: !0
            })
          ]
        }, u.media_url_https) : /* @__PURE__ */ h("div", {
          className: E.mediaContainer,
          children: [
            /* @__PURE__ */ r("div", {
              className: E.skeleton,
              style: ct(u, c)
            }),
            /* @__PURE__ */ r(br, {
              tweet: e,
              media: u
            })
          ]
        }, u.media_url_https)
      }, u.media_url_https))
    })
  });
}, xr = "_root_i24yn_1", Nr = {
  root: xr
}, Rr = ({ tweet: e }) => {
  const t = new Date(e.created_at), n = Kn(t);
  return /* @__PURE__ */ r("a", {
    className: Nr.root,
    href: e.url,
    target: "_blank",
    rel: "noopener noreferrer",
    "aria-label": n,
    children: /* @__PURE__ */ r("time", {
      dateTime: t.toISOString(),
      children: n
    })
  });
}, Sr = "_info_19qcj_1", Ir = "_infoLink_19qcj_10", kr = "_infoIcon_19qcj_30", Ie = {
  info: Sr,
  infoLink: Ir,
  infoIcon: kr
}, Er = ({ tweet: e }) => /* @__PURE__ */ h("div", {
  className: Ie.info,
  children: [
    /* @__PURE__ */ r(Rr, {
      tweet: e
    }),
    /* @__PURE__ */ r("a", {
      className: Ie.infoLink,
      href: "https://help.x.com/en/x-for-websites-ads-info-and-privacy",
      target: "_blank",
      rel: "noopener noreferrer",
      "aria-label": "Twitter for Websites, Ads Information and Privacy",
      children: /* @__PURE__ */ r("svg", {
        viewBox: "0 0 24 24",
        "aria-hidden": "true",
        className: Ie.infoIcon,
        children: /* @__PURE__ */ r("g", {
          children: /* @__PURE__ */ r("path", {
            d: "M13.5 8.5c0 .83-.67 1.5-1.5 1.5s-1.5-.67-1.5-1.5S11.17 7 12 7s1.5.67 1.5 1.5zM13 17v-5h-2v5h2zm-1 5.25c5.66 0 10.25-4.59 10.25-10.25S17.66 1.75 12 1.75 1.75 6.34 1.75 12 6.34 22.25 12 22.25zM20.25 12c0 4.56-3.69 8.25-8.25 8.25S3.75 16.56 3.75 12 7.44 3.75 12 3.75s8.25 3.69 8.25 8.25z"
          })
        })
      })
    })
  ]
}), Cr = "_actions_n2anf_1", Or = "_like_n2anf_13", Lr = "_reply_n2anf_14", qr = "_copy_n2anf_15", $r = "_likeIconWrapper_n2anf_27", Dr = "_likeCount_n2anf_30", Ar = "_replyIconWrapper_n2anf_35", Vr = "_copyIconWrapper_n2anf_36", Mr = "_likeIcon_n2anf_27", jr = "_replyIcon_n2anf_35", Wr = "_copyIcon_n2anf_36", Br = "_replyText_n2anf_56", Fr = "_copyText_n2anf_57", I = {
  actions: Cr,
  like: Or,
  reply: Lr,
  copy: qr,
  likeIconWrapper: $r,
  likeCount: Dr,
  replyIconWrapper: Ar,
  copyIconWrapper: Vr,
  likeIcon: Mr,
  replyIcon: jr,
  copyIcon: Wr,
  replyText: Br,
  copyText: Fr
}, Pr = ({ tweet: e }) => {
  const [t, n] = se(!1), o = () => {
    navigator.clipboard.writeText(e.url), n(!0);
  };
  return ze(() => {
    if (t) {
      const a = setTimeout(() => {
        n(!1);
      }, 6e3);
      return () => clearTimeout(a);
    }
  }, [
    t
  ]), /* @__PURE__ */ h("button", {
    type: "button",
    className: I.copy,
    "aria-label": "Copy link",
    onClick: o,
    children: [
      /* @__PURE__ */ r("div", {
        className: I.copyIconWrapper,
        children: t ? /* @__PURE__ */ r("svg", {
          viewBox: "0 0 24 24",
          className: I.copyIcon,
          "aria-hidden": "true",
          children: /* @__PURE__ */ r("g", {
            children: /* @__PURE__ */ r("path", {
              d: "M9.64 18.952l-5.55-4.861 1.317-1.504 3.951 3.459 8.459-10.948L19.4 6.32 9.64 18.952z"
            })
          })
        }) : /* @__PURE__ */ r("svg", {
          viewBox: "0 0 24 24",
          className: I.copyIcon,
          "aria-hidden": "true",
          children: /* @__PURE__ */ r("g", {
            children: /* @__PURE__ */ r("path", {
              d: "M18.36 5.64c-1.95-1.96-5.11-1.96-7.07 0L9.88 7.05 8.46 5.64l1.42-1.42c2.73-2.73 7.16-2.73 9.9 0 2.73 2.74 2.73 7.17 0 9.9l-1.42 1.42-1.41-1.42 1.41-1.41c1.96-1.96 1.96-5.12 0-7.07zm-2.12 3.53l-7.07 7.07-1.41-1.41 7.07-7.07 1.41 1.41zm-12.02.71l1.42-1.42 1.41 1.42-1.41 1.41c-1.96 1.96-1.96 5.12 0 7.07 1.95 1.96 5.11 1.96 7.07 0l1.41-1.41 1.42 1.41-1.42 1.42c-2.73 2.73-7.16 2.73-9.9 0-2.73-2.74-2.73-7.17 0-9.9z"
            })
          })
        })
      }),
      /* @__PURE__ */ r("span", {
        className: I.copyText,
        children: t ? "Copied!" : "Copy link"
      })
    ]
  });
}, Ur = ({ tweet: e }) => {
  const t = De(e.favorite_count);
  return /* @__PURE__ */ h("div", {
    className: I.actions,
    children: [
      /* @__PURE__ */ h("a", {
        className: I.like,
        href: e.like_url,
        target: "_blank",
        rel: "noopener noreferrer",
        "aria-label": `Like. This Tweet has ${t} likes`,
        children: [
          /* @__PURE__ */ r("div", {
            className: I.likeIconWrapper,
            children: /* @__PURE__ */ r("svg", {
              viewBox: "0 0 24 24",
              className: I.likeIcon,
              "aria-hidden": "true",
              children: /* @__PURE__ */ r("g", {
                children: /* @__PURE__ */ r("path", {
                  d: "M20.884 13.19c-1.351 2.48-4.001 5.12-8.379 7.67l-.503.3-.504-.3c-4.379-2.55-7.029-5.19-8.382-7.67-1.36-2.5-1.41-4.86-.514-6.67.887-1.79 2.647-2.91 4.601-3.01 1.651-.09 3.368.56 4.798 2.01 1.429-1.45 3.146-2.1 4.796-2.01 1.954.1 3.714 1.22 4.601 3.01.896 1.81.846 4.17-.514 6.67z"
                })
              })
            })
          }),
          /* @__PURE__ */ r("span", {
            className: I.likeCount,
            children: t
          })
        ]
      }),
      /* @__PURE__ */ h("a", {
        className: I.reply,
        href: e.reply_url,
        target: "_blank",
        rel: "noopener noreferrer",
        "aria-label": "Reply to this Tweet on Twitter",
        children: [
          /* @__PURE__ */ r("div", {
            className: I.replyIconWrapper,
            children: /* @__PURE__ */ r("svg", {
              viewBox: "0 0 24 24",
              className: I.replyIcon,
              "aria-hidden": "true",
              children: /* @__PURE__ */ r("g", {
                children: /* @__PURE__ */ r("path", {
                  d: "M1.751 10c0-4.42 3.584-8 8.005-8h4.366c4.49 0 8.129 3.64 8.129 8.13 0 2.96-1.607 5.68-4.196 7.11l-8.054 4.46v-3.69h-.067c-4.49.1-8.183-3.51-8.183-8.01z"
                })
              })
            })
          }),
          /* @__PURE__ */ r("span", {
            className: I.replyText,
            children: "Reply"
          })
        ]
      }),
      /* @__PURE__ */ r(Pr, {
        tweet: e
      })
    ]
  });
}, zr = "_replies_1cutb_1", Hr = "_link_1cutb_4", Gr = "_text_1cutb_23", ke = {
  replies: zr,
  link: Hr,
  text: Gr
}, Xr = ({ tweet: e }) => /* @__PURE__ */ r("div", {
  className: ke.replies,
  children: /* @__PURE__ */ r("a", {
    className: ke.link,
    href: e.url,
    target: "_blank",
    rel: "noopener noreferrer",
    children: /* @__PURE__ */ r("span", {
      className: ke.text,
      children: e.conversation_count === 0 ? "Read more on X" : e.conversation_count === 1 ? `Read ${De(e.conversation_count)} reply` : `Read ${De(e.conversation_count)} replies`
    })
  })
}), Kr = "_root_q5799_1", Qr = "_article_q5799_16", lt = {
  root: Kr,
  article: Qr
}, Jr = ({ tweet: e, children: t }) => /* @__PURE__ */ r("div", {
  className: lt.root,
  onClick: (n) => {
    n.preventDefault(), window.open(e.url, "_blank");
  },
  children: /* @__PURE__ */ r("article", {
    className: lt.article,
    children: t
  })
}), Yr = "_header_f00ab_1", Zr = "_avatar_f00ab_11", eo = "_avatarOverflow_f00ab_16", to = "_avatarSquare_f00ab_23", no = "_author_f00ab_27", ro = "_authorText_f00ab_33", oo = "_username_f00ab_40", Q = {
  header: Yr,
  avatar: Zr,
  avatarOverflow: eo,
  avatarSquare: to,
  author: no,
  authorText: ro,
  username: oo
}, ao = ({ tweet: e, components: t }) => {
  var n;
  const { user: o } = e, a = (n = t == null ? void 0 : t.AvatarImg) != null ? n : gt;
  return /* @__PURE__ */ h("div", {
    className: Q.header,
    children: [
      /* @__PURE__ */ r("a", {
        href: e.url,
        className: Q.avatar,
        target: "_blank",
        rel: "noopener noreferrer",
        children: /* @__PURE__ */ r("div", {
          className: U(Q.avatarOverflow, o.profile_image_shape === "Square" && Q.avatarSquare),
          children: /* @__PURE__ */ r(a, {
            src: o.profile_image_url_https,
            alt: o.name,
            width: 20,
            height: 20
          })
        })
      }),
      /* @__PURE__ */ h("div", {
        className: Q.author,
        children: [
          /* @__PURE__ */ r("div", {
            className: Q.authorText,
            children: /* @__PURE__ */ r("span", {
              title: o.name,
              children: o.name
            })
          }),
          /* @__PURE__ */ r(wt, {
            user: o
          }),
          /* @__PURE__ */ r(yt, {
            user: o
          }),
          /* @__PURE__ */ r("div", {
            className: Q.username,
            children: /* @__PURE__ */ h("span", {
              title: `@${o.screen_name}`,
              children: [
                "@",
                o.screen_name
              ]
            })
          })
        ]
      })
    ]
  });
}, so = "_root_175ot_1", io = {
  root: so
}, co = ({ tweet: e }) => /* @__PURE__ */ r("p", {
  className: io.root,
  lang: e.lang,
  dir: "auto",
  children: e.entities.map((t, n) => /* @__PURE__ */ r("span", {
    dangerouslySetInnerHTML: {
      __html: t.text
    }
  }, n))
}), lo = ({ tweet: e, components: t }) => {
  var n;
  return /* @__PURE__ */ h(Jr, {
    tweet: e,
    children: [
      /* @__PURE__ */ r(ao, {
        tweet: e,
        components: t
      }),
      /* @__PURE__ */ r(co, {
        tweet: e
      }),
      (n = e.mediaDetails) != null && n.length ? /* @__PURE__ */ r(Nt, {
        quoted: !0,
        tweet: e
      }) : null
    ]
  });
}, uo = ({ tweet: e, components: t }) => {
  var n;
  const o = He(() => ar(e), [
    e
  ]);
  return /* @__PURE__ */ h(Xe, {
    children: [
      /* @__PURE__ */ r(An, {
        tweet: o,
        components: t
      }),
      o.in_reply_to_status_id_str && /* @__PURE__ */ r(jn, {
        tweet: o
      }),
      /* @__PURE__ */ r(Un, {
        tweet: o
      }),
      (n = o.mediaDetails) != null && n.length ? /* @__PURE__ */ r(Nt, {
        tweet: o,
        components: t
      }) : null,
      o.quoted_tweet && /* @__PURE__ */ r(lo, {
        tweet: o.quoted_tweet,
        components: t
      }),
      /* @__PURE__ */ r(Er, {
        tweet: o
      }),
      /* @__PURE__ */ r(Ur, {
        tweet: o
      }),
      /* @__PURE__ */ r(Xr, {
        tweet: o
      })
    ]
  });
}, fo = "_root_16yxa_1", ho = {
  root: fo
}, _o = (e) => /* @__PURE__ */ r(Xe, {
  children: /* @__PURE__ */ h("div", {
    className: ho.root,
    children: [
      /* @__PURE__ */ r("h3", {
        children: "Tweet not found"
      }),
      /* @__PURE__ */ r("p", {
        children: "The embedded tweet could not be found…"
      })
    ]
  })
}), mo = "_skeleton_oi7hr_1", po = {
  skeleton: mo
}, ge = ({ style: e }) => /* @__PURE__ */ r("span", {
  className: po.skeleton,
  style: e
}), vo = "_root_17qqv_1", go = {
  root: vo
}, wo = () => /* @__PURE__ */ h(Xe, {
  className: go.root,
  children: [
    /* @__PURE__ */ r(ge, {
      style: {
        height: "3rem",
        marginBottom: "0.75rem"
      }
    }),
    /* @__PURE__ */ r(ge, {
      style: {
        height: "6rem",
        margin: "0.5rem 0"
      }
    }),
    /* @__PURE__ */ r("div", {
      style: {
        borderTop: "var(--tweet-border)",
        margin: "0.5rem 0"
      }
    }),
    /* @__PURE__ */ r(ge, {
      style: {
        height: "2rem"
      }
    }),
    /* @__PURE__ */ r(ge, {
      style: {
        height: "2rem",
        borderRadius: "9999px",
        marginTop: "0.5rem"
      }
    })
  ]
}), Rt = 0, St = 1, It = 2, dt = 3;
var ut = Object.prototype.hasOwnProperty;
function Ae(e, t) {
  var n, o;
  if (e === t) return !0;
  if (e && t && (n = e.constructor) === t.constructor) {
    if (n === Date) return e.getTime() === t.getTime();
    if (n === RegExp) return e.toString() === t.toString();
    if (n === Array) {
      if ((o = e.length) === t.length)
        for (; o-- && Ae(e[o], t[o]); ) ;
      return o === -1;
    }
    if (!n || typeof e == "object") {
      o = 0;
      for (n in e)
        if (ut.call(e, n) && ++o && !ut.call(t, n) || !(n in t) || !Ae(e[n], t[n])) return !1;
      return Object.keys(t).length === o;
    }
  }
  return e !== e && t !== t;
}
const F = /* @__PURE__ */ new WeakMap(), P = () => {
}, S = (
  /*#__NOINLINE__*/
  P()
), Ve = Object, d = (e) => e === S, W = (e) => typeof e == "function", H = (e, t) => ({
  ...e,
  ...t
}), kt = (e) => W(e.then), Ee = {}, we = {}, Ke = "undefined", _e = typeof window != Ke, Me = typeof document != Ke, yo = _e && "Deno" in window, bo = () => _e && typeof window.requestAnimationFrame != Ke, Et = (e, t) => {
  const n = F.get(e);
  return [
    // Getter
    () => !d(t) && e.get(t) || Ee,
    // Setter
    (o) => {
      if (!d(t)) {
        const a = e.get(t);
        t in we || (we[t] = a), n[5](t, H(a, o), a || Ee);
      }
    },
    // Subscriber
    n[6],
    // Get server cache snapshot
    () => !d(t) && t in we ? we[t] : !d(t) && e.get(t) || Ee
  ];
};
let je = !0;
const To = () => je, [We, Be] = _e && window.addEventListener ? [
  window.addEventListener.bind(window),
  window.removeEventListener.bind(window)
] : [
  P,
  P
], xo = () => {
  const e = Me && document.visibilityState;
  return d(e) || e !== "hidden";
}, No = (e) => (Me && document.addEventListener("visibilitychange", e), We("focus", e), () => {
  Me && document.removeEventListener("visibilitychange", e), Be("focus", e);
}), Ro = (e) => {
  const t = () => {
    je = !0, e();
  }, n = () => {
    je = !1;
  };
  return We("online", t), We("offline", n), () => {
    Be("online", t), Be("offline", n);
  };
}, So = {
  isOnline: To,
  isVisible: xo
}, Io = {
  initFocus: No,
  initReconnect: Ro
}, ft = !Ue.useId, ie = !_e || yo, ko = (e) => bo() ? window.requestAnimationFrame(e) : setTimeout(e, 1), Ce = ie ? ze : Gt, Oe = typeof navigator < "u" && navigator.connection, ht = !ie && Oe && ([
  "slow-2g",
  "2g"
].includes(Oe.effectiveType) || Oe.saveData), ye = /* @__PURE__ */ new WeakMap(), Eo = (e) => Ve.prototype.toString.call(e), Le = (e, t) => e === `[object ${t}]`;
let Co = 0;
const Fe = (e) => {
  const t = typeof e, n = Eo(e), o = Le(n, "Date"), a = Le(n, "RegExp"), s = Le(n, "Object");
  let i, c;
  if (Ve(e) === e && !o && !a) {
    if (i = ye.get(e), i) return i;
    if (i = ++Co + "~", ye.set(e, i), Array.isArray(e)) {
      for (i = "@", c = 0; c < e.length; c++)
        i += Fe(e[c]) + ",";
      ye.set(e, i);
    }
    if (s) {
      i = "#";
      const f = Ve.keys(e).sort();
      for (; !d(c = f.pop()); )
        d(e[c]) || (i += c + ":" + Fe(e[c]) + ",");
      ye.set(e, i);
    }
  } else
    i = o ? e.toJSON() : t == "symbol" ? e.toString() : t == "string" ? JSON.stringify(e) : "" + e;
  return i;
}, Qe = (e) => {
  if (W(e))
    try {
      e = e();
    } catch {
      e = "";
    }
  const t = e;
  return e = typeof e == "string" ? e : (Array.isArray(e) ? e.length : e) ? Fe(e) : "", [
    e,
    t
  ];
};
let Oo = 0;
const Pe = () => ++Oo;
async function Ct(...e) {
  const [t, n, o, a] = e, s = H({
    populateCache: !0,
    throwOnError: !0
  }, typeof a == "boolean" ? {
    revalidate: a
  } : a || {});
  let i = s.populateCache;
  const c = s.rollbackOnError;
  let f = s.optimisticData;
  const u = (T) => typeof c == "function" ? c(T) : c !== !1, v = s.throwOnError;
  if (W(n)) {
    const T = n, R = [], k = t.keys();
    for (const L of k)
      // Skip the special useSWRInfinite and useSWRSubscription keys.
      !/^\$(inf|sub)\$/.test(L) && T(t.get(L)._k) && R.push(L);
    return Promise.all(R.map(m));
  }
  return m(n);
  async function m(T) {
    const [R] = Qe(T);
    if (!R) return;
    const [k, L] = Et(t, R), [D, ce, l, me] = F.get(t), J = () => {
      const q = D[R];
      return (W(s.revalidate) ? s.revalidate(k().data, T) : s.revalidate !== !1) && (delete l[R], delete me[R], q && q[0]) ? q[0](It).then(() => k().data) : k().data;
    };
    if (e.length < 3)
      return J();
    let x = o, A, V = !1;
    const Y = Pe();
    ce[R] = [
      Y,
      0
    ];
    const p = !d(f), Z = k(), M = Z.data, ee = Z._c, G = d(ee) ? M : ee;
    if (p && (f = W(f) ? f(G, M) : f, L({
      data: f,
      _c: G
    })), W(x))
      try {
        x = x(G);
      } catch (q) {
        A = q, V = !0;
      }
    if (x && kt(x))
      if (x = await x.catch((q) => {
        A = q, V = !0;
      }), Y !== ce[R][0]) {
        if (V) throw A;
        return x;
      } else V && p && u(A) && (i = !0, L({
        data: G,
        _c: S
      }));
    if (i && !V)
      if (W(i)) {
        const q = i(x, G);
        L({
          data: q,
          error: S,
          _c: S
        });
      } else
        L({
          data: x,
          error: S,
          _c: S
        });
    if (ce[R][1] = Pe(), Promise.resolve(J()).then(() => {
      L({
        _c: S
      });
    }), V) {
      if (v) throw A;
      return;
    }
    return x;
  }
}
const _t = (e, t) => {
  for (const n in e)
    e[n][0] && e[n][0](t);
}, Lo = (e, t) => {
  if (!F.has(e)) {
    const n = H(Io, t), o = /* @__PURE__ */ Object.create(null), a = Ct.bind(S, e);
    let s = P;
    const i = /* @__PURE__ */ Object.create(null), c = (v, m) => {
      const T = i[v] || [];
      return i[v] = T, T.push(m), () => T.splice(T.indexOf(m), 1);
    }, f = (v, m, T) => {
      e.set(v, m);
      const R = i[v];
      if (R)
        for (const k of R)
          k(m, T);
    }, u = () => {
      if (!F.has(e) && (F.set(e, [
        o,
        /* @__PURE__ */ Object.create(null),
        /* @__PURE__ */ Object.create(null),
        /* @__PURE__ */ Object.create(null),
        a,
        f,
        c
      ]), !ie)) {
        const v = n.initFocus(setTimeout.bind(S, _t.bind(S, o, Rt))), m = n.initReconnect(setTimeout.bind(S, _t.bind(S, o, St)));
        s = () => {
          v && v(), m && m(), F.delete(e);
        };
      }
    };
    return u(), [
      e,
      a,
      u,
      s
    ];
  }
  return [
    e,
    F.get(e)[4]
  ];
}, qo = (e, t, n, o, a) => {
  const s = n.errorRetryCount, i = a.retryCount, c = ~~((Math.random() + 0.5) * (1 << (i < 8 ? i : 8))) * n.errorRetryInterval;
  !d(s) && i > s || setTimeout(o, c, a);
}, $o = Ae, [Ot, Do] = Lo(/* @__PURE__ */ new Map()), Ao = H(
  {
    // events
    onLoadingSlow: P,
    onSuccess: P,
    onError: P,
    onErrorRetry: qo,
    onDiscarded: P,
    // switches
    revalidateOnFocus: !0,
    revalidateOnReconnect: !0,
    revalidateIfStale: !0,
    shouldRetryOnError: !0,
    // timeouts
    errorRetryInterval: ht ? 1e4 : 5e3,
    focusThrottleInterval: 5 * 1e3,
    dedupingInterval: 2 * 1e3,
    loadingTimeout: ht ? 5e3 : 3e3,
    // providers
    compare: $o,
    isPaused: () => !1,
    cache: Ot,
    mutate: Do,
    fallback: {}
  },
  // use web preset by default
  So
), Vo = (e, t) => {
  const n = H(e, t);
  if (t) {
    const { use: o, fallback: a } = e, { use: s, fallback: i } = t;
    o && s && (n.use = o.concat(s)), a && i && (n.fallback = H(a, i));
  }
  return n;
}, Mo = Ht({}), jo = "$inf$", Lt = _e && window.__SWR_DEVTOOLS_USE__, Wo = Lt ? window.__SWR_DEVTOOLS_USE__ : [], Bo = () => {
  Lt && (window.__SWR_DEVTOOLS_REACT__ = Ue);
}, Fo = (e) => W(e[1]) ? [
  e[0],
  e[1],
  e[2] || {}
] : [
  e[0],
  null,
  (e[1] === null ? e[2] : e[1]) || {}
], Po = () => {
  const e = Xt(Mo);
  return He(() => H(Ao, e), [
    e
  ]);
}, Uo = (e) => (t, n, o) => e(t, n && ((...s) => {
  const [i] = Qe(t), [, , , c] = F.get(Ot);
  if (i.startsWith(jo))
    return n(...s);
  const f = c[i];
  return d(f) ? n(...s) : (delete c[i], f);
}), o), zo = Wo.concat(Uo), Ho = (e) => function(...n) {
  const o = Po(), [a, s, i] = Fo(n), c = Vo(o, i);
  let f = e;
  const { use: u } = c, v = (u || []).concat(zo);
  for (let m = v.length; m--; )
    f = v[m](f);
  return f(a, s || c.fetcher || null, c);
}, Go = (e, t, n) => {
  const o = t[e] || (t[e] = []);
  return o.push(n), () => {
    const a = o.indexOf(n);
    a >= 0 && (o[a] = o[o.length - 1], o.pop());
  };
};
Bo();
const qe = Ue.use || // This extra generic is to avoid TypeScript mixing up the generic and JSX sytax
// and emitting an error.
// We assume that this is only for the `use(thenable)` case, not `use(context)`.
// https://github.com/facebook/react/blob/aed00dacfb79d17c53218404c52b1c7aa59c4a89/packages/react-server/src/ReactFizzThenable.js#L45
((e) => {
  switch (e.status) {
    case "pending":
      throw e;
    case "fulfilled":
      return e.value;
    case "rejected":
      throw e.reason;
    default:
      throw e.status = "pending", e.then((t) => {
        e.status = "fulfilled", e.value = t;
      }, (t) => {
        e.status = "rejected", e.reason = t;
      }), e;
  }
}), $e = {
  dedupe: !0
}, mt = Promise.resolve(S), Xo = () => P, Ko = (e, t, n) => {
  const { cache: o, compare: a, suspense: s, fallbackData: i, revalidateOnMount: c, revalidateIfStale: f, refreshInterval: u, refreshWhenHidden: v, refreshWhenOffline: m, keepPreviousData: T, strictServerPrefetchWarning: R } = n, [k, L, D, ce] = F.get(o), [l, me] = Qe(e), J = z(!1), x = z(!1), A = z(l), V = z(t), Y = z(n), p = () => Y.current, Z = () => p().isVisible() && p().isOnline(), [M, ee, G, q] = Et(o, l), te = z({}).current, ne = d(i) ? d(n.fallback) ? S : n.fallback[l] : i, Je = (_, g) => {
    for (const w in te) {
      const y = w;
      if (y === "data") {
        if (!a(_[y], g[y]) && (!d(_[y]) || !a(ve, g[y])))
          return !1;
      } else if (g[y] !== _[y])
        return !1;
    }
    return !0;
  }, be = !J.current, Ye = He(() => {
    const _ = M(), g = q(), w = (b) => {
      const $ = H(b);
      return delete $._k, (() => {
        if (!l || !t || p().isPaused()) return !1;
        if (be && !d(c)) return c;
        const fe = d(ne) ? $.data : ne;
        return d(fe) || f;
      })() ? {
        isValidating: !0,
        isLoading: !0,
        ...$
      } : $;
    }, y = w(_), j = _ === g ? y : w(g);
    let C = y;
    return [
      () => {
        const b = w(M());
        return Je(b, C) ? (C.data = b.data, C.isLoading = b.isLoading, C.isValidating = b.isValidating, C.error = b.error, C) : (C = b, b);
      },
      () => j
    ];
  }, [
    o,
    l
  ]), re = ot.useSyncExternalStore(Re(
    (_) => G(l, (g, w) => {
      Je(w, g) || _();
    }),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [
      o,
      l
    ]
  ), Ye[0], Ye[1]), $t = k[l] && k[l].length > 0, oe = re.data, X = d(oe) ? ne && kt(ne) ? qe(ne) : ne : oe, pe = re.error, Te = z(X), ve = T ? d(oe) ? d(Te.current) ? X : Te.current : oe : X, K = l && d(X), le = z(null);
  !ie && // getServerSnapshot is only called during hydration
  // eslint-disable-next-line react-hooks/rules-of-hooks
  ot.useSyncExternalStore(Xo, () => (le.current = !1, le), () => (le.current = !0, le));
  const Dt = le.current;
  R && Dt && !s && K && console.warn(`Missing pre-initiated data for serialized key "${l}" during server-side rendering. Data fetching should be initiated on the server and provided to SWR via fallback data. You can set "strictServerPrefetchWarning: false" to disable this warning.`);
  const Ze = !l || !t || p().isPaused() || $t && !d(pe) ? !1 : be && !d(c) ? c : s ? d(X) ? !1 : f : d(X) || f, et = be && Ze, At = d(re.isValidating) ? et : re.isValidating, Vt = d(re.isLoading) ? et : re.isLoading, de = Re(
    async (_) => {
      const g = V.current;
      if (!l || !g || x.current || p().isPaused())
        return !1;
      let w, y, j = !0;
      const C = _ || {}, b = !D[l] || !C.dedupe, $ = () => ft ? !x.current && l === A.current && J.current : l === A.current, ue = {
        isValidating: !1,
        isLoading: !1
      }, fe = () => {
        ee(ue);
      }, nt = () => {
        const O = D[l];
        O && O[1] === y && delete D[l];
      }, rt = {
        isValidating: !0
      };
      d(M().data) && (rt.isLoading = !0);
      try {
        if (b && (ee(rt), n.loadingTimeout && d(M().data) && setTimeout(() => {
          j && $() && p().onLoadingSlow(l, n);
        }, n.loadingTimeout), D[l] = [
          g(me),
          Pe()
        ]), [w, y] = D[l], w = await w, b && setTimeout(nt, n.dedupingInterval), !D[l] || D[l][1] !== y)
          return b && $() && p().onDiscarded(l), !1;
        ue.error = S;
        const O = L[l];
        if (!d(O) && // case 1
        (y <= O[0] || // case 2
        y <= O[1] || // case 3
        O[1] === 0))
          return fe(), b && $() && p().onDiscarded(l), !1;
        const B = M().data;
        ue.data = a(B, w) ? B : w, b && $() && p().onSuccess(w, l, n);
      } catch (O) {
        nt();
        const B = p(), { shouldRetryOnError: xe } = B;
        B.isPaused() || (ue.error = O, b && $() && (B.onError(O, l, B), (xe === !0 || W(xe) && xe(O)) && (!p().revalidateOnFocus || !p().revalidateOnReconnect || Z()) && B.onErrorRetry(O, l, B, (Mt) => {
          const Ne = k[l];
          Ne && Ne[0] && Ne[0](dt, Mt);
        }, {
          retryCount: (C.retryCount || 0) + 1,
          dedupe: !0
        })));
      }
      return j = !1, fe(), !0;
    },
    // `setState` is immutable, and `eventsCallback`, `fnArg`, and
    // `keyValidating` are depending on `key`, so we can exclude them from
    // the deps array.
    //
    // FIXME:
    // `fn` and `config` might be changed during the lifecycle,
    // but they might be changed every render like this.
    // `useSWR('key', () => fetch('/api/'), { suspense: true })`
    // So we omit the values from the deps array
    // even though it might cause unexpected behaviors.
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [
      l,
      o
    ]
  ), tt = Re(
    // Use callback to make sure `keyRef.current` returns latest result every time
    (..._) => Ct(o, A.current, ..._),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    []
  );
  if (Ce(() => {
    V.current = t, Y.current = n, d(oe) || (Te.current = oe);
  }), Ce(() => {
    if (!l) return;
    const _ = de.bind(S, $e);
    let g = 0;
    p().revalidateOnFocus && (g = Date.now() + p().focusThrottleInterval);
    const y = Go(l, k, (j, C = {}) => {
      if (j == Rt) {
        const b = Date.now();
        p().revalidateOnFocus && b > g && Z() && (g = b + p().focusThrottleInterval, _());
      } else if (j == St)
        p().revalidateOnReconnect && Z() && _();
      else {
        if (j == It)
          return de();
        if (j == dt)
          return de(C);
      }
    });
    return x.current = !1, A.current = l, J.current = !0, ee({
      _k: me
    }), Ze && (D[l] || (d(X) || ie ? _() : ko(_))), () => {
      x.current = !0, y();
    };
  }, [
    l
  ]), Ce(() => {
    let _;
    function g() {
      const y = W(u) ? u(M().data) : u;
      y && _ !== -1 && (_ = setTimeout(w, y));
    }
    function w() {
      !M().error && (v || p().isVisible()) && (m || p().isOnline()) ? de($e).then(g) : g();
    }
    return g(), () => {
      _ && (clearTimeout(_), _ = -1);
    };
  }, [
    u,
    v,
    m,
    l
  ]), Kt(ve), s) {
    if (!ft && ie && K)
      throw new Error("Fallback data is required when using Suspense in SSR.");
    K && (V.current = t, Y.current = n, x.current = !1);
    const _ = ce[l], g = !d(_) && K ? tt(_) : mt;
    if (qe(g), !d(pe) && K)
      throw pe;
    const w = K ? de($e) : mt;
    !d(ve) && K && (w.status = "fulfilled", w.value = !0), qe(w);
  }
  return {
    mutate: tt,
    get data() {
      return te.data = !0, ve;
    },
    get error() {
      return te.error = !0, pe;
    },
    get isValidating() {
      return te.isValidating = !0, At;
    },
    get isLoading() {
      return te.isLoading = !0, Vt;
    }
  };
}, pt = Ho(Ko);
class Qo extends Error {
  constructor({ message: t, status: n, data: o }) {
    super(t), this.name = "TwitterApiError", this.status = n, this.data = o;
  }
}
const Jo = pt.default || pt, Yo = "https://react-tweet.vercel.app";
async function Zo([e, t]) {
  const n = await fetch(e, t), o = await n.json();
  if (n.ok) return o.data || null;
  throw new Qo({
    message: `Failed to fetch tweet at "${e}" with "${n.status}".`,
    data: o,
    status: n.status
  });
}
const ea = (e, t, n) => {
  const { isLoading: o, data: a, error: s } = Jo(() => t || e ? [
    t || e && `${Yo}/api/tweet/${e}`,
    n
  ] : null, Zo, {
    revalidateIfStale: !1,
    revalidateOnFocus: !1,
    shouldRetryOnError: !1
  });
  return {
    // If data is `undefined` then it might be the first render where SWR hasn't started doing
    // any work, so we set `isLoading` to `true`.
    isLoading: !!(o || a === void 0 && !s),
    data: a,
    error: s
  };
}, ta = ({ id: e, apiUrl: t, fallback: n = /* @__PURE__ */ r(wo, {}), components: o, fetchOptions: a, onError: s }) => {
  const { data: i, error: c, isLoading: f } = ea(e, t, a);
  if (f) return n;
  if (c || !i) {
    const u = (o == null ? void 0 : o.TweetNotFound) || _o;
    return /* @__PURE__ */ r(u, {
      error: s ? s(c) : c
    });
  }
  return /* @__PURE__ */ r(uo, {
    tweet: i,
    components: o
  });
};
function na({ node: e }) {
  var o;
  const t = ((o = e == null ? void 0 : e.attrs) == null ? void 0 : o.src) || "", n = t == null ? void 0 : t.split("/").pop();
  return n ? /* @__PURE__ */ r(Ft, { children: /* @__PURE__ */ r("div", { "data-twitter": "", children: /* @__PURE__ */ r(ta, { id: n }) }) }) : null;
}
function ra(e) {
  const { t } = cn(), [n, o] = se("");
  ze(() => {
    var s;
    if (e != null && e.editor) {
      const { src: i } = (s = e.editor) == null ? void 0 : s.getAttributes(qt.name);
      i && o(i);
    }
  }, [e == null ? void 0 : e.editor]);
  function a(s) {
    s.preventDefault(), s.stopPropagation(), e == null || e.onSetLink(n);
  }
  return /* @__PURE__ */ h("form", { className: "richtext-flex richtext-flex-col richtext-gap-2", onSubmit: a, children: [
    /* @__PURE__ */ r(Qt, { className: "mb-[6px]", children: t("editor.link.dialog.text") }),
    /* @__PURE__ */ r("div", { className: "richtext-flex richtext-w-full richtext-max-w-sm richtext-items-center richtext-gap-1.5", children: /* @__PURE__ */ r("div", { className: "richtext-relative richtext-w-full richtext-max-w-sm richtext-items-center", children: /* @__PURE__ */ r(
      Jt,
      {
        className: "richtext-w-80",
        onChange: (s) => o(s.target.value),
        placeholder: "Text",
        required: !0,
        type: "text",
        value: n
      }
    ) }) }),
    /* @__PURE__ */ r(Yt, { className: "richtext-mt-2 richtext-self-end", type: "submit", children: t("editor.link.dialog.button.apply") })
  ] });
}
function ma() {
  const [e, t] = se(!1), n = Zt(), o = en(qt.name), {
    icon: a = void 0,
    tooltip: s = void 0,
    tooltipOptions: i = {},
    action: c = void 0,
    isActive: f = void 0
  } = (o == null ? void 0 : o.componentProps) ?? {}, { editorDisabled: u } = tn(f);
  function v(m) {
    u || c && (c(m), t(!1));
  }
  return /* @__PURE__ */ h(on, { modal: !0, onOpenChange: t, open: e, children: [
    /* @__PURE__ */ r(an, { asChild: !0, children: /* @__PURE__ */ r(
      nn,
      {
        disabled: u,
        isActive: f,
        tooltip: s,
        tooltipOptions: i,
        children: /* @__PURE__ */ r(rn, { name: a })
      }
    ) }),
    /* @__PURE__ */ r(sn, { align: "start", className: "richtext-w-full", hideWhenDetached: !0, side: "bottom", children: /* @__PURE__ */ r(ra, { editor: n, onSetLink: v }) })
  ] });
}
const oa = /(https?:\/\/)?(www\.)?x\.com\/(\w{1,15})(\/status\/(\d+))?(\/\S*)?/g, aa = /^https?:\/\/(www\.)?x\.com\/(\w{1,15})(\/status\/(\d+))?(\/\S*)?$/;
function vt(e) {
  return e.match(aa);
}
const qt = /* @__PURE__ */ jt.create({
  name: "twitter",
  draggable: !0,
  selectable: !0,
  addOptions() {
    return {
      addPasteHandler: !0,
      HTMLAttributes: {},
      inline: !1,
      origin: "",
      button: ({ editor: e, t }) => ({
        componentProps: {
          action: (n) => {
            e.commands.setTweet({ src: n });
          },
          isActive: () => !1,
          disabled: !1,
          icon: "Twitter",
          tooltip: t("editor.twitter.tooltip"),
          editor: e
        }
      })
    };
  },
  addNodeView() {
    return Pt(na, {
      attrs: this.options.HTMLAttributes
    });
  },
  inline() {
    return this.options.inline;
  },
  group() {
    return this.options.inline ? "inline" : "block";
  },
  addAttributes() {
    return {
      src: {
        default: null
      }
    };
  },
  parseHTML() {
    return [
      {
        tag: "div[data-twitter]"
      }
    ];
  },
  addCommands() {
    return {
      setTweet: (e) => ({ commands: t }) => vt(e.src) ? t.insertContent({
        type: this.name,
        attrs: e
      }) : !1,
      updateTweet: (e) => ({ commands: t }) => vt(e.src) ? t.updateAttributes(this.name, { src: e.src }) : !1
    };
  },
  addPasteRules() {
    return this.options.addPasteHandler ? [
      Bt({
        find: oa,
        type: this.type,
        getAttributes: (e) => ({ src: e.input })
      })
    ] : [];
  },
  renderHTML({ HTMLAttributes: e }) {
    return ["div", Wt({ "data-twitter": "" }, e)];
  }
});
export {
  ra as F,
  ma as R,
  qt as T
};
